--- comm-miramar/calendar/providers/wcap/calWcapCalendarItems.js.wcap	2011-07-20 17:12:33.263995206 +0200
+++ comm-miramar/calendar/providers/wcap/calWcapCalendarItems.js	2011-07-20 17:12:56.322068020 +0200
@@ -973,8 +973,6 @@
                 items.push(item);
             }
         }
-
-        cal.processPendingEvent();
     }
 
     // tag "exceptions", i.e. items with rid:
@@ -1009,8 +1007,6 @@
         }
 
         parent.recurrenceInfo.modifyException(item, true);
-
-        cal.processPendingEvent();
     }
 
     if (itemFilter & calICalendar.ITEM_FILTER_CLASS_OCCURRENCES) {
@@ -1057,8 +1053,6 @@
             }
             // only proxies returned:
             items = items.concat(occurrences);
-
-            cal.processPendingEvent();
         }
 
     } else {
