--- dbunit.php.rpm	2015-08-07 08:17:06.899086351 +0200
+++ dbunit.php	2015-08-07 08:17:37.693235622 +0200
@@ -9,11 +9,7 @@
  * file that was distributed with this source code.
  */
 
-if (strpos('@php_bin@', '@php_bin') === 0) {
-    set_include_path(dirname(__FILE__) . PATH_SEPARATOR . get_include_path());
-}
-
-require_once 'PHPUnit/Autoload.php';
+require_once '/usr/share/php/PHPUnit/Autoload.php';
 
 $command = new PHPUnit_Extensions_Database_UI_Command(
   new PHPUnit_Extensions_Database_UI_ModeFactory()
