From 1111dab1baced14aaf05509c41ee416c3fc6a961 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@famillecollet.com>
Date: Thu, 7 Jul 2016 08:20:40 +0200
Subject: [PATCH] Fix for PHP 7.1

---
 library/SimplePie/Parse/Date.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/library/SimplePie/Parse/Date.php b/library/SimplePie/Parse/Date.php
index a415ba6..226f453 100644
--- a/library/SimplePie/Parse/Date.php
+++ b/library/SimplePie/Parse/Date.php
@@ -690,7 +690,7 @@ public function date_w3cdtf($date)
 			}
 
 			// Convert the number of seconds to an integer, taking decimals into account
-			$second = round($match[6] + $match[7] / pow(10, strlen($match[7])));
+			$second = round((int)$match[6] + (int)$match[7] / pow(10, strlen($match[7])));
 
 			return gmmktime($match[4], $match[5], $second, $match[2], $match[3], $match[1]) - $timezone;
 		}
