From b12ee80b2198c8964b72790fe27b2754c7e86708 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sat, 2 Jan 2016 07:55:43 +0100
Subject: [PATCH] fix PHP 7 dtor, fix #3

---
 env.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/env.c b/env.c
index 075139e..9263445 100644
--- a/env.c
+++ b/env.c
@@ -41,18 +41,10 @@ PHP_INI_END()
 /* }}} */
 
 
-#if PHP_VERSION_ID < 70000
 void char_ptr_dtor(char **str)
 {
 	free(*str);
 }
-#else
-void char_ptr_dtor(zval **str)
-{
-	free(Z_STRVAL_PP(str));
-	free(*str);
-}
-#endif
 
 /* {{{ PHP_GINIT_FUNCTION
  */
