diff -up libircclient-1.6/src/Makefile.in.orig libircclient-1.6/src/Makefile.in
--- libircclient-1.6/src/Makefile.in.orig	2012-04-28 16:32:56.176139347 -0300
+++ libircclient-1.6/src/Makefile.in	2012-04-28 16:36:10.406147192 -0300
@@ -14,16 +14,16 @@ static:	libircclient.a
 shared: libircclient.so
 shared_mingw: libircclient.dll
 
-install: lib
-	-mkdir -p $(DESTDIR)@prefix@/include
-	-mkdir -p $(DESTDIR)@prefix@/lib
-	cp ../include/libircclient.h $(DESTDIR)@prefix@/include/libircclient.h
-	cp ../include/libirc_errors.h $(DESTDIR)@prefix@/include/libirc_errors.h
-	cp ../include/libirc_events.h  $(DESTDIR)@prefix@/include/libirc_events.h 
-	cp ../include/libirc_options.h  $(DESTDIR)@prefix@/include/libirc_options.h 
-	cp ../include/libirc_rfcnumeric.h  $(DESTDIR)@prefix@/include/libirc_rfcnumeric.h 
-	cp libircclient.so $(DESTDIR)@prefix@/lib/
-	cp libircclient.a $(DESTDIR)@prefix@/lib/
+install: @TARGET@
+	-mkdir -p $(DESTDIR)@includedir@
+	-mkdir -p $(DESTDIR)@libdir@
+	cp ../include/libircclient.h $(DESTDIR)@includedir@/libircclient.h
+	cp ../include/libirc_errors.h $(DESTDIR)@includedir@/libirc_errors.h
+	cp ../include/libirc_events.h  $(DESTDIR)@includedir@/libirc_events.h 
+	cp ../include/libirc_options.h  $(DESTDIR)@includedir@/libirc_options.h 
+	cp ../include/libirc_rfcnumeric.h  $(DESTDIR)@includedir@/libirc_rfcnumeric.h 
+	if [ -f libircclient.so ]; then cp libircclient.so $(DESTDIR)@libdir@; else :; fi
+	if [ -f libircclient.a ]; then cp libircclient.a $(DESTDIR)@libdir@; else :; fi
 
 libircclient.so: libircclient.o
 	$(CC) -shared -s -o libircclient.so libircclient.o @LIBS@
