diff -up libmemcached-0.44/libmemcached/include.am.sasl libmemcached-0.44/libmemcached/include.am
--- libmemcached-0.44/libmemcached/include.am.sasl	2010-10-02 16:34:33.000000000 +0200
+++ libmemcached-0.44/libmemcached/include.am	2010-10-02 16:35:27.000000000 +0200
@@ -149,7 +149,7 @@ libmemcached_libmemcachedprotocol_la_DEP
 endif
 
 if HAVE_SASL
-libmemcached_libmemcached_la_LDFLAGS+= $(LIBSASL)
+libmemcached_libmemcached_la_LIBADD += $(LTLIBSASL2)
 libmemcached_libmemcached_la_SOURCES += libmemcached/sasl.c
 endif
 
diff -up libmemcached-0.44/support/libmemcached.pc.in.sasl libmemcached-0.44/support/libmemcached.pc.in
--- libmemcached-0.44/support/libmemcached.pc.in.sasl	2010-10-02 08:20:58.000000000 +0200
+++ libmemcached-0.44/support/libmemcached.pc.in	2010-10-02 08:24:07.000000000 +0200
@@ -6,5 +6,5 @@ includedir=@includedir@
 Name: libmemcached
 Description: libmemcached C library.
 Version: @VERSION@
-Libs: -L${libdir} -lmemcached
+Libs: @LTLIBSASL2@ -L${libdir} -lmemcached
 Cflags: -I${includedir}
