diff -up Horde_Idna-1.0.1/lib/Horde/Idna.php.rpm Horde_Idna-1.0.1/lib/Horde/Idna.php
--- Horde_Idna-1.0.1/lib/Horde/Idna.php.rpm	2015-01-07 15:51:42.711626839 +0100
+++ Horde_Idna-1.0.1/lib/Horde/Idna.php	2015-01-07 15:51:49.339651276 +0100
@@ -62,11 +62,7 @@ class Horde_Idna
     {
         if (!isset(self::$_backend)) {
             if (extension_loaded('mbstring')) {
-                if (file_exists(__DIR__ . '/Idna/vendor/autoload.php')) {
-                    require_once __DIR__ . '/Idna/vendor/autoload.php';
-                } else {
-                    require_once __DIR__ . '/../../bundle/vendor/autoload.php';
-                }
+                require_once 'True/Punycode.php';
                 self::$_backend = new True\Punycode();
                 mb_internal_encoding('UTF-8');
             } elseif (class_exists('Net_IDNA2')) {
