diff --git a/msgpack_unpack.h b/msgpack_unpack.h
index 5dc9524..0851625 100644
--- a/msgpack_unpack.h
+++ b/msgpack_unpack.h
@@ -16,6 +16,16 @@ typedef enum
     MSGPACK_UNPACK_PARSE_ERROR = -1,
 } msgpack_unpack_return;
 
+#if PHP_VERSION_ID >= 70100
+typedef struct php_unserialize_data {
+	void *first;
+	void *last;
+	void *first_dtor;
+	void *last_dtor;
+	HashTable *allowed_classes;
+};
+#endif
+
 typedef struct php_unserialize_data msgpack_unserialize_data_t;
 
 typedef struct {
