diff -up memcached-1.0.2-zts/php_memcached.c.orig memcached-1.0.2-zts/php_memcached.c
--- memcached-1.0.2-zts/php_memcached.c.orig	2011-10-04 17:40:03.702545377 +0200
+++ memcached-1.0.2-zts/php_memcached.c	2011-10-04 17:40:21.264586400 +0200
@@ -2022,7 +2022,7 @@ static char *php_memc_zval_to_payload(zv
 			switch (serializer) {
 #if HAVE_MEMCACHED_IGBINARY
 				case SERIALIZER_IGBINARY:
-					igbinary_serialize((uint8_t **) &buf.c, &buf.len, value);
+					igbinary_serialize((uint8_t **) &buf.c, &buf.len, value TSRMLS_CC);
 					MEMC_VAL_SET_TYPE(*flags, MEMC_VAL_IS_IGBINARY);
 					break;
 #endif
@@ -2177,7 +2177,7 @@ static int php_memc_zval_from_payload(zv
 
 		case MEMC_VAL_IS_IGBINARY:
 #if HAVE_MEMCACHED_IGBINARY
-			if (igbinary_unserialize((uint8_t *)payload, payload_len, &value)) {
+			if (igbinary_unserialize((uint8_t *)payload, payload_len, &value TSRMLS_CC)) {
 				ZVAL_FALSE(value);
 
 				if (flags & MEMC_VAL_COMPRESSED) {
