From 650b67c64ca8ac8bd140492fd1078015cb8eaa0b Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@famillecollet.com>
Date: Sat, 27 Jun 2015 18:28:31 +0200
Subject: [PATCH] Update php_memcached.c

---
 php_memcached.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/php_memcached.c b/php_memcached.c
index 1063b22..9195bfc 100644
--- a/php_memcached.c
+++ b/php_memcached.c
@@ -3119,7 +3119,7 @@ zend_bool s_serialize_value (enum memcached_serializer serializer, zval *value,
 #ifdef HAVE_MEMCACHED_MSGPACK
 		case SERIALIZER_MSGPACK:
 			php_msgpack_serialize(buf, value);
-			if (!buf->c) {
+			if (!buf->s) {
 				php_error_docref(NULL, E_WARNING, "could not serialize value with msgpack");
 				return 0;
 			}
