diff -up memcached-2.1.0/php_memcached.c.info memcached-2.1.0/php_memcached.c
--- memcached-2.1.0/php_memcached.c.info	2012-11-17 09:52:02.576769778 +0100
+++ memcached-2.1.0/php_memcached.c	2012-11-17 09:54:23.335280240 +0100
@@ -3877,6 +3877,12 @@ PHP_MINFO_FUNCTION(memcached)
 	php_info_print_table_row(2, "Version", PHP_MEMCACHED_VERSION);
 	php_info_print_table_row(2, "libmemcached version", memcached_lib_version());
 
+#if HAVE_MEMCACHED_SASL
+	php_info_print_table_row(2, "SASL support", "yes");
+#else
+	php_info_print_table_row(2, "SASL support", "no");
+#endif
+
 #ifdef HAVE_MEMCACHED_SESSION
 	php_info_print_table_row(2, "Session support", "yes");
 #else
