diff -up firefox-6.0/mozilla-beta/browser/build.mk.stub firefox-6.0/mozilla-beta/browser/build.mk
--- firefox-6.0/mozilla-beta/browser/build.mk.stub	2011-07-22 00:18:32.000000000 +0200
+++ firefox-6.0/mozilla-beta/browser/build.mk	2011-07-24 13:53:54.133240968 +0200
@@ -53,7 +53,7 @@ ifdef MOZ_SERVICES_SYNC
 tier_app_dirs += services
 endif
 
-tier_app_dirs += browser
+tier_app_dirs += browser xulrunner
 # Never add other tier_app_dirs after browser. They won't get packaged
 # properly on mac.
 
diff -up firefox-6.0/mozilla-beta/xulrunner/Makefile.in.stub firefox-6.0/mozilla-beta/xulrunner/Makefile.in
--- firefox-6.0/mozilla-beta/xulrunner/Makefile.in.stub	2011-07-22 00:18:57.000000000 +0200
+++ firefox-6.0/mozilla-beta/xulrunner/Makefile.in	2011-07-24 13:53:54.133240968 +0200
@@ -44,10 +44,7 @@ VPATH     = @srcdir@
 include $(DEPTH)/config/autoconf.mk
 
 PARALLEL_DIRS = \
-  app \
-  setup \
   stub \
-  examples \
   $(NULL)
 
 ifeq ($(OS_ARCH),WINNT)
diff -up firefox-6.0/mozilla-beta/xulrunner/stub/Makefile.in.stub firefox-6.0/mozilla-beta/xulrunner/stub/Makefile.in
--- firefox-6.0/mozilla-beta/xulrunner/stub/Makefile.in.stub	2011-07-22 00:18:57.000000000 +0200
+++ firefox-6.0/mozilla-beta/xulrunner/stub/Makefile.in	2011-07-24 13:56:29.244616384 +0200
@@ -97,7 +97,7 @@ ifdef _MSC_VER
 WIN32_EXE_LDFLAGS += -ENTRY:wmainCRTStartup
 endif
 
-LIBS += $(JEMALLOC_LIBS)
+#LIBS += $(JEMALLOC_LIBS)
 
 include $(topsrcdir)/config/rules.mk
 
