diff -up cups-1.5.0/cups/http.c.driverd-timeout cups-1.5.0/cups/http.c
--- cups-1.5.0/cups/http.c.driverd-timeout	2011-10-10 17:03:54.181458460 +0100
+++ cups-1.5.0/cups/http.c	2011-10-10 17:04:01.452321912 +0100
@@ -1314,7 +1314,7 @@ httpGets(char   *line,			/* I - Line to
       * No newline; see if there is more data to be read...
       */
 
-      while (!_httpWait(http, http->blocking ? 30000 : 10000, 1))
+      while (!_httpWait(http, http->blocking ? 70000 : 10000, 1))
       {
 	if (http->timeout_cb && (*http->timeout_cb)(http, http->timeout_data))
 	  continue;
diff -up cups-1.5.0/scheduler/ipp.c.driverd-timeout cups-1.5.0/scheduler/ipp.c
--- cups-1.5.0/scheduler/ipp.c.driverd-timeout	2011-10-10 17:03:41.801690962 +0100
+++ cups-1.5.0/scheduler/ipp.c	2011-10-10 17:03:41.861689834 +0100
@@ -5723,7 +5723,7 @@ copy_model(cupsd_client_t *con,		/* I -
   close(temppipe[1]);
 
  /*
-  * Wait up to 30 seconds for the PPD file to be copied...
+  * Wait up to 70 seconds for the PPD file to be copied...
   */
 
   total = 0;
@@ -5743,7 +5743,7 @@ copy_model(cupsd_client_t *con,		/* I -
     FD_SET(temppipe[0], &input);
     FD_SET(CGIPipes[0], &input);
 
-    timeout.tv_sec  = 30;
+    timeout.tv_sec  = 70;
     timeout.tv_usec = 0;
 
     if ((i = select(maxfd, &input, NULL, NULL, &timeout)) < 0)
