diff -up cups-1.5.0/scheduler/cups-driverd.cxx.str3921 cups-1.5.0/scheduler/cups-driverd.cxx
--- cups-1.5.0/scheduler/cups-driverd.cxx.str3921	2011-06-08 22:19:11.000000000 +0100
+++ cups-1.5.0/scheduler/cups-driverd.cxx	2011-10-11 12:07:34.979538544 +0100
@@ -1411,9 +1411,7 @@ load_drv(const char  *filename,		/* I -
   * Add a dummy entry for the file...
   */
 
-  httpAssembleURIf(HTTP_URI_CODING_ALL, uri, sizeof(uri), "drv", "", "", 0,
-		   "/%s", name);
-  add_ppd(name, uri, "", "", "", "", "", "", mtime, size, 0,
+  add_ppd(name, name, "", "", "", "", "", "", mtime, size, 0,
           PPD_TYPE_DRV, "drv");
   ChangedPPD = 1;
 
