From ac916159a08ccb135a73891dc4e25b7be18223d2 Mon Sep 17 00:00:00 2001
From: Magnus Edenhill <magnus@edenhill.se>
Date: Mon, 11 Jan 2016 10:22:29 +0100
Subject: [PATCH] Snappy endian swapping for older glibc 2.9 (issue #505, #407)

---
 src/snappy_compat.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/snappy_compat.h b/src/snappy_compat.h
index 5167a72..2c2375b 100644
--- a/src/snappy_compat.h
+++ b/src/snappy_compat.h
@@ -136,10 +136,14 @@ typedef unsigned long long u64;
 #define __LITTLE_ENDIAN__ 1
 #endif
 
-#if __LITTLE_ENDIAN__ == 1 && (defined(__LSB_VERSION__) || defined(__WIN32__))
+#if __LITTLE_ENDIAN__ == 1 || defined(__WIN32__)
+#ifndef htole16
 #define htole16(x) (x)
+#endif
+#ifndef le32toh
 #define le32toh(x) (x)
 #endif
+#endif
 
 
 #if defined(_MSC_VER)
