--- eaccelerator.ini.upstream	2012-09-09 12:38:47.000000000 +0200
+++ eaccelerator.ini	2012-09-09 16:20:49.000000000 +0200
@@ -12,13 +12,8 @@
 
 ; You must uncomment one (and only one) line from the following to load
 ; eAccelerator extension.
-extension="eaccelerator.so"
-;zend_extension="/usr/lib/php/modules/eaccelerator.so"
-;zend_extension_ts="/usr/lib/php/modules/eaccelerator.so"
-
-;extension="eaccelerator.dll"
-;zend_extension_ts="c:\php4\eaccelerator.dll"
-;zend_extension="c:\php4\eaccelerator.dll"
+;extension="eaccelerator.so"
+zend_extension="@EXTDIR@/eaccelerator.so"
 
 ; The amount of shared memory (in megabytes) that eAccelerator will use.
 ; "0" means OS default. Default value is "0".
@@ -45,7 +40,7 @@
 
 ; Set the log file for eaccelerator. When this option isn't set then the data
 ; will be logged to stderr
-eaccelerator.log_file = "/var/log/httpd/eaccelerator_log"
+eaccelerator.log_file = ""
 
 ; A string that's prepended to all keys. This allows two applications that use the
 ; same key names to run on the same host by setting this in .htaccess or in the main
@@ -79,9 +74,9 @@
 ; Enables or disables caching of compiled scripts on disk. It has no effect
 ; on session data and content caching.
 ; Default value is "0" that means - use disk and shared memory for caching.
-eaccelerator.shm_only = "0"
+eaccelerator.shm_only = "1"
 
 ; The script paths that are allowed to get admin information and do admin
 ; controls
-eaccelerator.allowed_admin_path = ""
+eaccelerator.allowed_admin_path = "/usr/share/eaccelerator"
 
