--- unitTests/Classes/PHPExcel/Reader/XEEValidatorTest.php.ORIG	2015-10-08 14:24:46.586539987 -0400
+++ unitTests/Classes/PHPExcel/Reader/XEEValidatorTest.php	2015-10-08 14:25:19.824561274 -0400
@@ -26,9 +26,9 @@
 
     public function providerInvalidXML()
     {
-        $tests = [];
+        $tests = array();
         foreach(glob('rawTestData/Reader/XEETestInvalid*.xml') as $file) {
-            $tests[] = [realpath($file), true];
+            $tests[] = array(realpath($file), true);
         }
         return $tests;
 	}
@@ -45,9 +45,9 @@
 
     public function providerValidXML()
     {
-        $tests = [];
+        $tests = array();
         foreach(glob('rawTestData/Reader/XEETestValid*.xml') as $file) {
-            $tests[] = [realpath($file), file_get_contents($file)];
+            $tests[] = array(realpath($file), file_get_contents($file));
         }
         return $tests;
 	}
