--- curl-7.14.1/configure.ac.orig	2005-09-01 22:29:20.000000000 +0200
+++ curl-7.14.1/configure.ac	2005-10-13 09:55:19.395539032 +0200
@@ -813,7 +813,7 @@
     PKGTEST="no"
     PREFIX_OPENSSL=$OPT_SSL
     LIB_OPENSSL="$PREFIX_OPENSSL/lib$libsuff"
-    LDFLAGS="$LDFLAGS -L$LIB_OPENSSL"
+    #LDFLAGS="$LDFLAGS -L$LIB_OPENSSL"
     CPPFLAGS="$CPPFLAGS -I$PREFIX_OPENSSL/include/openssl -I$PREFIX_OPENSSL/include"
     ;;
   esac
@@ -867,7 +867,7 @@
      HAVECRYPTO="yes"
      LIBS="-lcrypto $LIBS"
      ],[
-     LDFLAGS="$CLEANLDFLAGS -L$LIB_OPENSSL"
+     LDFLAGS="$CLEANLDFLAGS"
      CPPFLAGS="$CLEANCPPFLAGS -I$PREFIX_OPENSSL/include/openssl -I$PREFIX_OPENSSL/include"
      AC_CHECK_LIB(crypto, CRYPTO_add_lock,[
        HAVECRYPTO="yes"
--- curl-7.14.1/lib/Makefile.am.orig	2005-03-31 09:02:03.000000000 +0200
+++ curl-7.14.1/lib/Makefile.am	2005-10-13 09:51:55.244574696 +0200
@@ -107,8 +107,8 @@
 
 install-data-hook:
 	@if test -n "@CURL_CA_BUNDLE@"; then \
-	  $(mkinstalldirs) `dirname $(DESTDIR)@CURL_CA_BUNDLE@`; \
-	  @INSTALL_DATA@ $(srcdir)/ca-bundle.crt $(DESTDIR)@CURL_CA_BUNDLE@; \
+	  $(mkinstalldirs) `dirname $(prefix)/../@CURL_CA_BUNDLE@`; \
+	  @INSTALL_DATA@ $(srcdir)/ca-bundle.crt $(prefix)/../@CURL_CA_BUNDLE@; \
         fi
 
 # this hook is mainly for non-unix systems to build even if configure
