diff -up Horde_Mongo-1.1.0/lib/Horde/Mongo/Client.php.pr194 Horde_Mongo-1.1.0/lib/Horde/Mongo/Client.php
--- Horde_Mongo-1.1.0/lib/Horde/Mongo/Client.php.pr194	2016-07-13 07:08:15.255526238 +0200
+++ Horde_Mongo-1.1.0/lib/Horde/Mongo/Client.php	2016-07-13 07:09:37.264976432 +0200
@@ -13,10 +13,8 @@
 
 if (!extension_loaded('mongo')) {
     // use the compatibility layer and mongodb extension
-    if (file_exists(__DIR__ . '/vendor/autoload.php')) {
-        require_once __DIR__ . '/vendor/autoload.php'; // PEAR installation
-    } else {
-        require_once __DIR__ . '/../../../bundle/vendor/autoload.php'; // GIT
+    if (file_exists('/usr/share/php/Alcaeus/MongoDbAdapter/autoload.php')) {
+       require_once '/usr/share/php/Alcaeus/MongoDbAdapter/autoload.php';
     }
 }
 
