diff -up FusionInventory-Agent-2.1.8/Changes.orig FusionInventory-Agent-2.1.8/Changes
--- FusionInventory-Agent-2.1.8/Changes.orig	2011-03-31 05:48:38.000000000 +0200
+++ FusionInventory-Agent-2.1.8/Changes	2011-03-31 05:48:59.000000000 +0200
@@ -10,8 +10,6 @@ Revision history for FusionInventory::Ag
        * Win32/AntiVirus: avoid pointless OLE error
        * Add BNQ screen manufacturer support
          https://bugs.launchpad.net/bugs/736095
-       * scan-homedirs and rpc-trust-localhost don't works in command line
-         http://forge.fusioninventory.org/issues/636
        * Update the Inventory XML documentation
        * HPUX: use NAME instead of TYPE to set the CPU name (Yanick Durant)
          http://forum.fusioninventory.org/viewtopic.php?id=278
diff -up FusionInventory-Agent-2.1.8/lib/FusionInventory/Agent/Config.pm.orig FusionInventory-Agent-2.1.8/lib/FusionInventory/Agent/Config.pm
--- FusionInventory-Agent-2.1.8/lib/FusionInventory/Agent/Config.pm.orig	2011-03-31 05:49:17.000000000 +0200
+++ FusionInventory-Agent-2.1.8/lib/FusionInventory/Agent/Config.pm	2011-03-31 05:50:17.000000000 +0200
@@ -243,9 +243,9 @@ sub loadUserParams {
         'realm|r=s',
         'rpc-ip=s',
         'rpc-port=s',
-        'rpc-trust-localhost=s',
+        'rpc-trust-localhost',
         'remotedir|R=s',
-        'scan-homedirs=s',
+        'scan-homedirs',
         'share-dir=s',
         'server|s=s',
         'stdout',
@@ -331,8 +331,8 @@ Extra options:
     --logger                    Logger you want to use, can be Stderr,File or Syslog ($config->{logger})
     --rpc-ip=IP                 ip of the interface to use for peer to peer exchange ($config->{'rpc-ip'})
     --rpc-port=PORT     port use for RPC
-    --rpc-trust-localhost=X     allow local users to force an inventory from http://127.0.0.1:62354/now (0/1) ($config->{'rpc-trust-localhost'})
-    --scan-homedirs=X           permit to scan home user directories (0/1) ($config->{'scan-homedirs'})
+    --rpc-trust-localhost       allow local users to force an inventory from http://127.0.0.1:62354/now  ($config->{'rpc-trust-localhost'})
+    --scan-homedirs             permit to scan home user directories ($config->{'scan-homedirs'})
     --share-dir=DIR             path to the directory where the shared files are stored ($config->{'share-dir'})
     --stdout                    do not write or post the inventory but print it on STDOUT
     -t --tag=TAG                use TAG as tag ($config->{tag}) Will be ignored by server if a value already exists.
