diff -up php5.3-200812131330/sapi/apache2handler/config.m4.install php5.3-200812131330/sapi/apache2handler/config.m4
--- php5.3-200812131330/sapi/apache2handler/config.m4.install	2008-03-12 00:31:53.000000000 +0100
+++ php5.3-200812131330/sapi/apache2handler/config.m4	2008-12-13 16:20:06.000000000 +0100
@@ -68,7 +68,7 @@ if test "$PHP_APXS2" != "no"; then
   fi
 
   APXS_LIBEXECDIR='$(INSTALL_ROOT)'`$APXS -q LIBEXECDIR`
-  if test -z `$APXS -q SYSCONFDIR`; then
+  if true; then
     INSTALL_IT="\$(mkinstalldirs) '$APXS_LIBEXECDIR' && \
                  $APXS -S LIBEXECDIR='$APXS_LIBEXECDIR' \
                        -i -n php5"
