diff -ru icu.orig/source/configure.in icu/source/configure.in
--- icu.orig/source/configure.in	2009-09-01 08:56:46.000000000 +0100
+++ icu/source/configure.in	2009-09-01 10:38:25.000000000 +0100
@@ -462,7 +462,7 @@
 # Check to see if genccode can generate simple assembly.
 GENCCODE_ASSEMBLY=
 case "${host}" in
-i*86-*-linux*|x86_64-*-linux*|powerpc*-*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
+*-linux*|i*86-*-*bsd*|i*86-pc-gnu)
     if test "$GCC" = yes; then
         # We're using gcc, and the simple -a gcc command line works for genccode
         GENCCODE_ASSEMBLY="-a gcc"
