Index: php_krb5_kadm.h
===================================================================
--- php_krb5_kadm.h	(révision 333125)
+++ php_krb5_kadm.h	(copie de travail)
@@ -25,15 +25,13 @@
 
 #include "config.h"
 
-#ifdef SUCCESS
-#define OLD_SUCC SUCCESS
-#endif
+/* will be used in gssrpc/rpc_msg.h enum accept_stat */
+#undef  SUCCESS
 #define SUCCESS KRB5_SUCCESS
 #include <kadm5/admin.h>
-#undef SUCCESS
-#ifdef OLD_SUCC
-#define SUCCESS OLD_SUCC
-#endif
+#undef  SUCCESS
+/* default value in PHP and Kerberos */
+#define SUCCESS 0
 
 /* KADM5 Object */
 	zend_class_entry *krb5_ce_kadm5;
