Index: config.m4
===================================================================
--- config.m4	(révision 333463)
+++ config.m4	(copie de travail)
@@ -43,7 +43,7 @@
 
   PHP_CHECK_LIBRARY($LIBNAME,$LIBSYMBOL,
   [
-    PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $GEOIP_DIR/lib, GEOIP_SHARED_LIBADD)
+    PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $GEOIP_DIR/$PHP_LIBDIR, GEOIP_SHARED_LIBADD)
     AC_DEFINE(HAVE_GEOIPLIB,1,[ ])
   ],[
     AC_MSG_ERROR([wrong geoip lib version or lib not found])
@@ -95,8 +95,7 @@
   AC_CHECK_DECL(GEOIP_NETSPEED_EDITION_REV1,
   [
     MIN_GEOIP_VERSION=1004008
-  ],[
-  ],[
+  ],[],[
     #include <GeoIP.h>
   ])
   
@@ -104,8 +103,7 @@
   AC_CHECK_DECL(GEOIP_ACCURACYRADIUS_EDITION,
   [
     MIN_GEOIP_VERSION=1005000
-  ],[
-  ],[
+  ],[],[
     #include <GeoIP.h>
   ])
   
