Index: tests/014.phpt
===================================================================
--- tests/014.phpt	(revision 311865)
+++ tests/014.phpt	(working copy)
@@ -7,17 +7,15 @@
 --FILE--
 <?php
 
-var_dump(geoip_time_zone_by_country_and_region('CA',''));
-var_dump(geoip_time_zone_by_country_and_region('CA',NULL));
-var_dump(geoip_time_zone_by_country_and_region('CA'));
+var_dump(geoip_time_zone_by_country_and_region('CA','AB'));
+var_dump(geoip_time_zone_by_country_and_region('CM',NULL));
 var_dump(geoip_time_zone_by_country_and_region(NULL,''));
 var_dump(geoip_time_zone_by_country_and_region(NULL,NULL));
 
 ?>
 --EXPECTF--
 string(%d) "America/%s"
-string(%d) "America/%s"
-string(%d) "America/%s"
+string(%d) "Africa/Lagos"
 
 Warning: geoip_time_zone_by_country_and_region(): You need to specify at least the country code. in %s on line %d
 bool(false)
