diff -up facedetect/facedetect.c.php54 facedetect/facedetect.c
--- facedetect/facedetect.c.php54	2011-12-28 16:07:45.167341320 +0100
+++ facedetect/facedetect.c	2011-12-28 16:08:01.979341725 +0100
@@ -28,7 +28,7 @@
  *
  * Every user visible function must have an entry in facedetect_functions[].
  */
-static function_entry facedetect_functions[] = {
+static zend_function_entry facedetect_functions[] = {
     PHP_FE(face_detect, NULL)
     PHP_FE(face_count, NULL)
     {NULL, NULL, NULL}
