--- mod_nss-1.0.6/nss.conf.in.orig	2006-10-20 11:08:42.000000000 -0400
+++ mod_nss-1.0.6/nss.conf.in	2006-10-23 15:27:22.000000000 -0400
@@ -8,14 +8,16 @@
 # consult the online docs. You have been warned.  
 #
 
+LoadModule nss_module modules/libmodnss.so
+
 #
 # When we also provide SSL we have to listen to the 
 # standard HTTP port (see above) and to the HTTPS port
 #
 # Note: Configurations that use IPv6 but not IPv4-mapped addresses need two
-#       Listen directives: "Listen [::]:443" and "Listen 0.0.0.0:443"
+#       Listen directives: "Listen [::]:8443" and "Listen 0.0.0.0:443"
 #
-Listen 443
+Listen 8443
 
 ##
 ##  SSL Global Context
@@ -40,7 +42,7 @@
 #   Pass Phrase Helper:
 #   This helper program stores the token password pins between
 #   restarts of Apache.
-NSSPassPhraseHelper @apache_bin@/nss_pcache
+NSSPassPhraseHelper /usr/sbin/nss_pcache
 
 #   Configure the SSL Session Cache. 
 #   NSSSessionCacheSize is the number of entries in the cache.
@@ -68,17 +70,17 @@
 ## SSL Virtual Host Context
 ##
 
-<VirtualHost _default_:443>
+<VirtualHost _default_:8443>
 
 #   General setup for the virtual host
 #DocumentRoot "@apache_prefix@/htdocs"
-#ServerName www.example.com:443
+#ServerName www.example.com:8443
 #ServerAdmin you@example.com
 
 # mod_nss can log to separate log files, you can choose to do that if you'd like
 # LogLevel is not inherited from httpd.conf.
-#ErrorLog @apache_prefix@/logs/error_log
-#TransferLog @apache_prefix@/logs/access_log
+ErrorLog @apache_prefix@/logs/error_log
+TransferLog @apache_prefix@/logs/access_log
 LogLevel warn
 
 #   SSL Engine Switch:
@@ -113,7 +115,7 @@
 #   The NSS security database directory that holds the certificates and
 #   keys. The database consists of 3 files: cert8.db, key3.db and secmod.db.
 #   Provide the directory that these files exist.
-NSSCertificateDatabase @apache_conf@
+NSSCertificateDatabase /etc/httpd/alias
 
 #   Database Prefix:
 #   In order to be able to store multiple NSS databases in one directory
@@ -189,7 +191,7 @@
 <Files ~ "\.(cgi|shtml|phtml|php3?)$">
     NSSOptions +StdEnvVars
 </Files>
-<Directory "@apache_prefix@/cgi-bin">
+<Directory "/var/www/cgi-bin">
     NSSOptions +StdEnvVars
 </Directory>
 
