diff -up ./class.phpmailer.php.rpm ./class.phpmailer.php
--- ./class.phpmailer.php.rpm	2014-09-25 18:36:22.000000000 +0200
+++ ./class.phpmailer.php	2014-09-26 12:58:08.790951530 +0200
@@ -1191,6 +1191,7 @@ class PHPMailer
     public function getSMTPInstance()
     {
         if (!is_object($this->smtp)) {
+            require_once 'class.smtp.php';
             $this->smtp = new SMTP;
         }
         return $this->smtp;
@@ -1415,7 +1416,7 @@ class PHPMailer
         );
         if (empty($lang_path)) {
             // Calculate an absolute path so it can work if CWD is not here
-            $lang_path = dirname(__FILE__). DIRECTORY_SEPARATOR . 'language'. DIRECTORY_SEPARATOR;
+            $lang_path = '/usr/share/PHPMailer/language/';
         }
         $foundlang = true;
         $lang_file = $lang_path . 'phpmailer.lang-' . $langcode . '.php';
