diff -up ./class.phpmailer.php.rpm ./class.phpmailer.php
--- ./class.phpmailer.php.rpm	2016-12-09 11:03:48.000000000 +0100
+++ ./class.phpmailer.php	2016-12-09 15:03:58.489947280 +0100
@@ -1475,6 +1475,7 @@ class PHPMailer
     public function getSMTPInstance()
     {
         if (!is_object($this->smtp)) {
+            require_once 'class.smtp.php';
             $this->smtp = new SMTP;
         }
         return $this->smtp;
@@ -1730,7 +1731,7 @@ class PHPMailer
         );
         if (empty($lang_path)) {
             // Calculate an absolute path so it can work if CWD is not here
-            $lang_path = dirname(__FILE__). DIRECTORY_SEPARATOR . 'language'. DIRECTORY_SEPARATOR;
+            $lang_path = '/usr/share/PHPMailer/language/';
         }
         //Validate $langcode
         if (!preg_match('/^[a-z]{2}(?:_[a-zA-Z]{2})?$/', $langcode)) {
