diff -up xulrunner-1.9.3.0/mozilla-central/memory/mozalloc/mozalloc.h.old xulrunner-1.9.3.0/mozilla-central/memory/mozalloc/mozalloc.h
--- xulrunner-1.9.3.0/mozilla-central/memory/mozalloc/mozalloc.h.old	2010-08-06 03:09:14.000000000 +0200
+++ xulrunner-1.9.3.0/mozilla-central/memory/mozalloc/mozalloc.h	2010-08-17 11:15:40.000000000 +0200
@@ -214,7 +214,7 @@ MOZALLOC_EXPORT void* moz_valloc(size_t 
 #define MOZALLOC_THROW_IF_HAS_EXCEPTIONS throw()
 #endif
 
-#ifdef MOZ_CPP_EXCEPTIONS
+#if 1
 #define MOZALLOC_THROW_BAD_ALLOC throw(std::bad_alloc)
 #else
 #define MOZALLOC_THROW_BAD_ALLOC MOZALLOC_THROW_IF_HAS_EXCEPTIONS
