From d1290dfde07e7d33aae9017dd47e9a46d9c889f4 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Thu, 17 Nov 2016 15:24:54 +0100
Subject: [PATCH 2/2] fix test for ArgumentCountError exception with PHP 7.1

---
 tests/Predis/Command/RawCommandTest.php | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/tests/Predis/Command/RawCommandTest.php b/tests/Predis/Command/RawCommandTest.php
index 7be62c8..8d6cc14 100644
--- a/tests/Predis/Command/RawCommandTest.php
+++ b/tests/Predis/Command/RawCommandTest.php
@@ -74,10 +74,29 @@ public function testExceptionOnMissingCommandID()
      * argument is missing, PHP emits an E_WARNING.
      *
      * @group disconnected
-     * @expectedException \PHPUnit_Framework_Error_Warning
      */
     public function testPHPWarningOnMissingCommandIDWithStaticCreate()
     {
+        if (version_compare(PHP_VERSION, "7.1", '>')) {
+            $this->markTestSkipped('only for PHP < 7.1');
+        }
+        $this->setExpectedException('PHPUnit_Framework_Error_Warning');
+        RawCommand::create();
+    }
+
+    /**
+     * The signature of RawCommand::create() requires one argument which is the
+     * ID of the command (other arguments are fetched dinamically). If the first
+     * argument is missing, PHP 7.1 throw an exception
+     *
+     * @group disconnected
+     */
+    public function testPHPWarningOnMissingCommandIDWithStaticCreate71()
+    {
+        if (version_compare(PHP_VERSION, "7.1", '<')) {
+            $this->markTestSkipped('only for PHP > 7.1');
+        }
+        $this->setExpectedException('ArgumentCountError');
         RawCommand::create();
     }
 
