diff --git a/mod_bw.c b/mod_bw.c
index a49caf8..8b75973 100644
--- a/mod_bw.c
+++ b/mod_bw.c
@@ -519,7 +519,7 @@ static long get_bw_rate(request_rec * r, apr_array_header_t * a)
             return e[i].rate;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
+            if (apr_ipsubnet_test(e[i].x.ip, r->useragent_addr)) {
                 return e[i].rate;
             }
             break;
@@ -608,7 +608,7 @@ static int get_maxconn(request_rec * r, apr_array_header_t * a)
             return e[i].max;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
+            if (apr_ipsubnet_test(e[i].x.ip, r->useragent_addr)) {
                 return e[i].max;
             }
             break;
@@ -659,7 +659,7 @@ static int get_sid(request_rec * r, apr_array_header_t * a)
             return e[i].sid;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
+            if (apr_ipsubnet_test(e[i].x.ip, r->useragent_addr)) {
                 return e[i].sid;
             }
             break;
