diff -up blenc-1.1.4b/blenc.c.rpm blenc-1.1.4b/blenc.c
--- blenc-1.1.4b/blenc.c.rpm	2014-04-28 08:13:45.000000000 +0200
+++ blenc-1.1.4b/blenc.c	2014-04-28 08:14:02.000000000 +0200
@@ -80,7 +80,7 @@ ZEND_GET_MODULE(blenc)
 
 /* {{{ PHP_INI */
 PHP_INI_BEGIN()
-    STD_PHP_INI_ENTRY("blenc.key_file", "/usr/local/etc/blenckeys", PHP_INI_ALL, OnUpdateString, key_file, zend_blenc_globals, blenc_globals)
+    STD_PHP_INI_ENTRY("blenc.key_file", "/etc/blenckeys", PHP_INI_ALL, OnUpdateString, key_file, zend_blenc_globals, blenc_globals)
 PHP_INI_END()
 /* }}} */
 
diff -up blenc-1.1.4b/tools/blencode.php.rpm blenc-1.1.4b/tools/blencode.php
--- blenc-1.1.4b/tools/blencode.php.rpm	2014-04-28 08:14:21.000000000 +0200
+++ blenc-1.1.4b/tools/blencode.php	2014-04-28 08:17:43.000000000 +0200
@@ -1,3 +1,4 @@
+#!/usr/bin/php
 <?php
 /*
   +----------------------------------------------------------------------+
