diff -up firefox-7.0/mozilla-beta/browser/installer/Makefile.in.cache firefox-7.0/mozilla-beta/browser/installer/Makefile.in
--- firefox-7.0/mozilla-beta/browser/installer/Makefile.in.cache	2011-09-06 10:43:04.421293791 +0200
+++ firefox-7.0/mozilla-beta/browser/installer/Makefile.in	2011-09-06 10:43:04.422293805 +0200
@@ -114,19 +114,6 @@ ifndef LIBXUL_SDK
 INSTALL_SDK = 1
 endif
 
-ifneq (1_,$(if $(CROSS_COMPILE),1,0)_$(UNIVERSAL_BINARY))
-ifdef RUN_TEST_PROGRAM
-_ABS_RUN_TEST_PROGRAM = $(call core_abspath,$(RUN_TEST_PROGRAM))
-endif
-
-GENERATE_CACHE = \
-  $(_ABS_RUN_TEST_PROGRAM) $(LIBXUL_DIST)/bin/xpcshell$(BIN_SUFFIX) -g "$$PWD" -a "$$PWD" -f $(topsrcdir)/browser/installer/precompile_cache.js -e 'populate_startupcache("omni.jar", "startupCache.zip");' && \
-  rm -rf jsloader && \
-  $(UNZIP) startupCache.zip && \
-  rm startupCache.zip && \
-  $(ZIP) -r9m omni.jar jsloader
-endif
-
 include $(topsrcdir)/toolkit/mozapps/installer/packager.mk
 
 ifeq (bundle, $(MOZ_FS_LAYOUT))
diff -up firefox-7.0/mozilla-beta/browser/installer/package-manifest.in.cache firefox-7.0/mozilla-beta/browser/installer/package-manifest.in
--- firefox-7.0/mozilla-beta/browser/installer/package-manifest.in.cache	2011-08-31 21:16:05.000000000 +0200
+++ firefox-7.0/mozilla-beta/browser/installer/package-manifest.in	2011-09-06 10:43:04.423293819 +0200
@@ -91,6 +91,7 @@
 
 [browser]
 ; [Base Browser Files]
+@BINPATH@/jsloader/
 #ifndef XP_UNIX
 @BINPATH@/@MOZ_APP_NAME@.exe
 #else
diff -up firefox-7.0/mozilla-beta/toolkit/mozapps/installer/packager.mk.cache firefox-7.0/mozilla-beta/toolkit/mozapps/installer/packager.mk
--- firefox-7.0/mozilla-beta/toolkit/mozapps/installer/packager.mk.cache	2011-08-31 21:16:34.000000000 +0200
+++ firefox-7.0/mozilla-beta/toolkit/mozapps/installer/packager.mk	2011-09-06 10:43:52.861924899 +0200
@@ -383,7 +383,6 @@ MAKE_SDK = $(CREATE_FINAL_TAR) - $(MOZ_A
 endif
 
 ifdef MOZ_OMNIJAR
-GENERATE_CACHE ?= true
 
 OMNIJAR_FILES	= \
   chrome \
@@ -413,7 +412,6 @@ PACK_OMNIJAR	= \
     mv tmp.manifest $$m; \
   done; \
   $(ZIP) -r9m omni.jar $(OMNIJAR_FILES) -x $(NON_OMNIJAR_FILES) && \
-  $(GENERATE_CACHE) && \
   $(OPTIMIZE_JARS_CMD) --optimize $(JARLOG_DIR_AB_CD) ./ ./ && \
   mv binary.manifest components && \
   printf "manifest components/binary.manifest\n" > chrome.manifest
