--- pecl/mysqlnd_qc/trunk/mysqlnd_qc.c	2014/03/22 06:36:04	333055
+++ pecl/mysqlnd_qc/trunk/mysqlnd_qc.c	2014/03/22 08:20:55	333056
@@ -613,7 +613,11 @@
 
 /* {{{ MYSQLND_METHOD(mysqlnd_qc, store_result) */
 static MYSQLND_RES *
+#if PHP_VERSION_ID < 50600
 MYSQLND_METHOD(mysqlnd_qc, store_result)(MYSQLND_CONN_DATA * const conn TSRMLS_DC)
+#else
+MYSQLND_METHOD(mysqlnd_qc, store_result)(MYSQLND_CONN_DATA * const conn, const unsigned int flags TSRMLS_DC)
+#endif
 {
 	MYSQLND_RES * result;
 	uint64_t store_time;
@@ -621,7 +625,11 @@
 	DBG_ENTER("mysqlnd_qc::store_result");
 
 	STATS_TIME_SET(store_time);
+#if PHP_VERSION_ID < 50600
 		result = QC_CALL_ORIGINAL_CONN_DATA_METHOD(store_result)(conn TSRMLS_CC);
+#else
+		result = QC_CALL_ORIGINAL_CONN_DATA_METHOD(store_result)(conn, flags TSRMLS_CC);
+#endif
 	STATS_TIME_DIFF(store_time);
 	MYSQLND_QC_INC_STATISTIC_W_VALUE(QC_STAT_QUERY_AGGR_STORE_TIME_TOTAL, (uint64_t)store_time);
 
@@ -777,13 +785,21 @@
 
 /* {{{ MYSQLND_METHOD(mysqlnd_qc, use_result) */
 static MYSQLND_RES *
+#if PHP_VERSION_ID < 50600
 MYSQLND_METHOD(mysqlnd_qc, use_result)(MYSQLND_CONN_DATA * const conn TSRMLS_DC)
+#else
+MYSQLND_METHOD(mysqlnd_qc, use_result)(MYSQLND_CONN_DATA * const conn, const unsigned int flags TSRMLS_DC)
+#endif
 {
 	MYSQLND_RES * ret;
 	DBG_ENTER("mysqlnd_qc::use_result");
 	MYSQLND_QC_INC_STATISTIC_W_VALUE2(QC_STAT_MISS, 1, QC_STAT_QUERY_UNCACHED_USE_RESULT, 1);
 
+#if PHP_VERSION_ID < 50600
 	ret = QC_CALL_ORIGINAL_CONN_DATA_METHOD(use_result)(conn TSRMLS_CC);
+#else
+	ret = QC_CALL_ORIGINAL_CONN_DATA_METHOD(use_result)(conn, flags TSRMLS_CC);
+#endif
 	if (ret) {
 		QC_DECLARE_AND_LOAD_CONN_DATA(conn_data_pp, conn);
 
--- pecl/mysqlnd_qc/trunk/web/mysqlnd_qc_monitor_core_functions_inc.php	2014/03/22 08:20:55	333056
+++ pecl/mysqlnd_qc/trunk/web/mysqlnd_qc_monitor_core_functions_inc.php	2014/03/22 09:06:50	333057
@@ -84,7 +84,7 @@
 		<td align="right"><?php print phpversion(); ?></td>
 	</tr>
 	<?php
-	$handler = mysqlnd_qc_get_handler();
+	$handler = mysqlnd_qc_get_available_handlers();
 	foreach ($handler as $name => $details) {
 		?>
 		<tr bgcolor="<?php print ($color_idx++ %2 == 0) ? QCM_TD_LIST_COLOR0 : QCM_TD_LIST_COLOR1; ?>">
@@ -126,7 +126,7 @@
 *
 */
 function qc_print_box_ini_settings_summary() {
-	$handler = mysqlnd_qc_get_handler();
+	$handler = mysqlnd_qc_get_available_handlers();
 	$color_idx = 0;
 
 	?>
@@ -1148,4 +1148,4 @@
 	<?php
 	}
 } // end func qc_print_main_auto_append_details
-?>
\ No newline at end of file
+?>
