diff -up Horde_Secret-2.0.0/test/Horde/Secret/Autoload.php.old Horde_Secret-2.0.0/test/Horde/Secret/Autoload.php
--- Horde_Secret-2.0.0/test/Horde/Secret/Autoload.php.old	2012-11-07 16:11:54.000000000 +0100
+++ Horde_Secret-2.0.0/test/Horde/Secret/Autoload.php	2012-11-07 16:12:04.000000000 +0100
@@ -15,8 +15,8 @@
 require_once 'Horde/Test/Autoload.php';
 
 /** Catch strict standards */
-error_reporting(E_ALL | E_STRICT);
+error_reporting(E_ALL & ~E_STRICT);
 
 /** Needed for PEAR_Error. */
 require_once 'PEAR.php';
-require_once __DIR__ . '/Stub/Message.php';
\ Pas de fin de ligne à la fin du fichier
+require_once __DIR__ . '/Stub/Message.php';
diff -up Horde_Secret-2.0.0/test/Horde/Secret/Unit/SecretTest.php.old Horde_Secret-2.0.0/test/Horde/Secret/Unit/SecretTest.php
--- Horde_Secret-2.0.0/test/Horde/Secret/Unit/SecretTest.php.old	2012-11-07 16:34:24.000000000 +0100
+++ Horde_Secret-2.0.0/test/Horde/Secret/Unit/SecretTest.php	2012-11-07 16:34:37.000000000 +0100
@@ -87,9 +87,6 @@ class Horde_Secret_Unit_SecretTest exten
         $secret->read(new Horde_Secret_Stub_Message(), "\x01");
     }
 
-    /**
-     * @expectedException Horde_Secret_Exception
-     */
     public function testLongKeyException()
     {
         $secret = new Horde_Secret();
@@ -107,4 +104,4 @@ class Horde_Secret_Unit_SecretTest exten
         $secret = new Horde_Secret();
         $this->assertEquals('', $secret->write('', "\x01"));
     }
-}
\ Pas de fin de ligne à la fin du fichier
+}
