--- program/include/iniset.php~	2012-08-20 10:19:53.000000000 -0500
+++ program/include/iniset.php	2012-08-20 10:22:57.372357067 -0500
@@ -24,7 +24,7 @@
 */
 
 $config = array(
-    'error_reporting'         => E_ALL &~ (E_NOTICE | E_STRICT),
+    'error_reporting'         => E_ALL & ~E_NOTICE & ~E_STRICT,
     // Some users are not using Installer, so we'll check some
     // critical PHP settings here. Only these, which doesn't provide
     // an error/warning in the logs later. See (#1486307).
