From 96f8436d77af05cf3c2d4f87970868221ddc6891 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 29 Oct 2012 09:22:33 +0100
Subject: [PATCH] fix call to MDB2::isError for MDB2 2.5.0b4

---
 program/include/rcube_mdb2.php |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/program/include/rcube_mdb2.php b/program/include/rcube_mdb2.php
index 3d8891d..876ee02 100644
--- a/program/include/rcube_mdb2.php
+++ b/program/include/rcube_mdb2.php
@@ -288,7 +288,7 @@ private function _query($query, $offset, $numrows, $params)
         else {
             $params = (array)$params;
             $q = $this->db_handle->prepare($query, null, $mode=='w' ? MDB2_PREPARE_MANIP : null);
-            if ($this->db_handle->isError($q)) {
+            if (MDB2::isError($q)) {
                 $this->db_error = true;
                 $this->db_error_msg = $q->userinfo;
 
@@ -369,7 +369,7 @@ function insert_id($table = '')
 
         $id = $this->db_handle->lastInsertID($table);
 
-        return $this->db_handle->isError($id) ? null : $id;
+        return MDB2::isError($id) ? null : $id;
     }
 
 
-- 
1.7.10

