diff -up gosa-combined-2.6.10/html/helpviewer.php.04 gosa-combined-2.6.10/html/helpviewer.php
--- gosa-combined-2.6.10/html/helpviewer.php.04	2010-01-26 09:27:39.000000000 +0100
+++ gosa-combined-2.6.10/html/helpviewer.php	2010-05-24 08:23:40.308670455 +0200
@@ -104,7 +104,7 @@ $helpdir ="";
 if(session::global_is_set('current_class_for_help')){
 
   /* Create new XML parser with the path to the Xml file */
-  $xml = new parseXml("../doc/guide.xml");
+  $xml = new parseXml("/etc/gosa/guide.xml");
   
   /* Generate help array */
   $str = $xml->parse();
diff -up gosa-combined-2.6.10/update-gosa.04 gosa-combined-2.6.10/update-gosa
--- gosa-combined-2.6.10/update-gosa.04	2010-05-24 08:23:14.269545335 +0200
+++ gosa-combined-2.6.10/update-gosa	2010-05-24 08:23:40.308670455 +0200
@@ -195,7 +195,7 @@ function rescan_i18n()
 
 function rescan_guide()
 {
-	$master_guide= "doc/guide.xml";
+	$master_guide= "/etc/gosa/guide.xml";
 	echo "Updating Online Help Index...\n";
 	$master_guide_content="<?xml version=\"1.0\"?>\n".
 		"<!--\n".
