From 0100e1a89bc7ba3f7e94960437f69294638322a6 Mon Sep 17 00:00:00 2001
From: Lubos Kardos <lkardos@redhat.com>
Date: Tue, 7 Jul 2015 10:25:40 +0200
Subject: [PATCH] Honor CFLAGS passed to cmake resovles: #1239997

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4cfeef3..d0f96fe 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -4,7 +4,7 @@ SET(MODULE_CMD "/usr/bin/modulecmd" )
 SET(CONF_DIR "/etc/scl/conf/" )
 CONFIGURE_FILE( config.h.cmake config.h )
 
-SET( CMAKE_C_FLAGS "-Wall -pedantic --std=gnu99 -D_GNU_SOURCE -g" )
+SET( CMAKE_C_FLAGS "-Wall -pedantic --std=gnu99 -D_GNU_SOURCE -g ${CMAKE_C_FLAGS}" )
 INCLUDE_DIRECTORIES ("${PROJECT_BINARY_DIR}/src")
 list(APPEND SOURCES scl.c debug.c scllib.c lib_common.c args.c sclmalloc.c fallback.c)
 ADD_EXECUTABLE (scl ${SOURCES})
-- 
1.9.3

