From c041733a012ec179950c5113a8875a6d4b8c2867 Mon Sep 17 00:00:00 2001
From: Lubos Kardos <lkardos@redhat.com>
Date: Thu, 27 Aug 2015 13:46:06 +0200
Subject: [PATCH] Fix core dumps with large input on stdin (rhbz:1257274)

---
 src/args.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/args.c b/src/args.c
index c524dbc..dda57cd 100644
--- a/src/args.c
+++ b/src/args.c
@@ -139,8 +139,8 @@ static int extract_command_stdin(struct scl_args *args)
 
     len = 0;
     while ((r = fread(command+len, 1, BUFSIZ, stdin)) == BUFSIZ) {
-        command = xrealloc(command, len+BUFSIZ+1);
         len += r;
+        command = xrealloc(command, len+BUFSIZ+1);
     }
 
     if (feof(stdin)) {
-- 
1.9.3

