diff -up rpm/macros.scl.orig rpm/macros.scl
--- rpm/macros.scl.orig	2015-06-24 07:26:39.838044392 +0200
+++ rpm/macros.scl	2015-06-24 07:34:13.762968582 +0200
@@ -69,10 +69,13 @@ package or when debugging this package.
 %global _datadir		%{_prefix}/share
 %global _sysconfdir		%{_scl_root}/etc
 %{?nfsmountable:		%global _sysconfdir %{_root_sysconfdir}%{_scl_prefix}/scls/%{scl}}
+%{?rh_layout:			%global _sysconfdir %{_root_sysconfdir}%{_scl_prefix}/%{scl}}
 %global _sharedstatedir		%{_scl_root}/var/lib
 %{?nfsmountable:		%global _sharedstatedir %{_root_localstatedir}%{_scl_prefix}/scls/%{scl}/lib}
+%{?rh_layout:			%global _sharedstatedir %{_root_localstatedir}%{_scl_prefix}/%{scl}/lib}
 %global _localstatedir		%{_scl_root}/var
 %{?nfsmountable:		%global _localstatedir %{_root_localstatedir}%{_scl_prefix}/scls/%{scl}}
+%{?rh_layout:			%global _localstatedir %{_root_localstatedir}%{_scl_prefix}/%{scl}}
 %global _libdir			%{_exec_prefix}/%{_lib}
 %global _includedir		%{_prefix}/include
 %global _infodir		%{_datadir}/info
