diff -up mozilla-beta/browser/installer/Makefile.in.BAD mozilla-beta/browser/installer/Makefile.in
--- mozilla-beta/browser/installer/Makefile.in.BAD	2011-06-02 16:17:18.194629509 -0400
+++ mozilla-beta/browser/installer/Makefile.in	2011-06-02 16:20:19.336609146 -0400
@@ -124,13 +124,13 @@ ifdef RUN_TEST_PROGRAM
 _ABS_RUN_TEST_PROGRAM = $(call core_abspath,$(RUN_TEST_PROGRAM))
 endif
 
-GENERATE_CACHE = \
-  $(_ABS_RUN_TEST_PROGRAM) $(_ABS_DIST)/bin/xpcshell$(BIN_SUFFIX) -g "$$PWD" -f $(topsrcdir)/browser/installer/precompile_cache.js -e 'populate_startupcache("omni.jar", "startupCache.zip");' && \
-  rm -rf jsloader && \
-  $(UNZIP) startupCache.zip && \
-  rm startupCache.zip && \
-  find jsloader | xargs touch -t 201001010000 && \
-  $(ZIP) -r9mX omni.jar jsloader
+# GENERATE_CACHE = \
+#   $(_ABS_RUN_TEST_PROGRAM) $(_ABS_DIST)/bin/xpcshell$(BIN_SUFFIX) -g "$$PWD" -f $(topsrcdir)/browser/installer/precompile_cache.js -e 'populate_startupcache("omni.jar", "startupCache.zip");' && \
+#   rm -rf jsloader && \
+#   $(UNZIP) startupCache.zip && \
+#   rm startupCache.zip && \
+#   find jsloader | xargs touch -t 201001010000 && \
+#   $(ZIP) -r9mX omni.jar jsloader
 endif
 
 include $(topsrcdir)/toolkit/mozapps/installer/packager.mk
