From 56d27611af6ee100db07fd07a2fbd76be4341354 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 13 Feb 2017 16:27:43 +0100
Subject: [PATCH] Fix #50 build for PHP 5

---
 backtrace.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/backtrace.c b/backtrace.c
index 5b0d57c..692a1a5 100644
--- a/backtrace.c
+++ b/backtrace.c
@@ -378,11 +378,13 @@ static void append_flat_zval_r(zval *expr TSRMLS_DC, smart_str *trace_str, char
 	}
 
 	switch (Z_TYPE_P(expr)) {
+#if PHP_VERSION_ID >= 70000
 		case IS_REFERENCE:
 			ZVAL_DEREF(expr);
 			smart_str_appendc(trace_str, '&');
 			append_flat_zval_r(expr, trace_str, depth);
 			break;
+#endif
 		case IS_ARRAY:
 			smart_str_appendc(trace_str, '[');
 #if PHP_VERSION_ID >= 70000
