diff -up ext/intl/common/common_error.c.old ext/intl/common/common_error.c
--- ext/intl/common/common_error.c.old	2011-11-26 09:31:56.566028846 +0100
+++ ext/intl/common/common_error.c	2011-11-26 09:32:19.356128073 +0100
@@ -241,7 +241,9 @@ void intl_expose_icu_error_codes( INIT_F
 	INTL_EXPOSE_CONST( U_IDNA_VERIFICATION_ERROR );
 	INTL_EXPOSE_CONST( U_IDNA_LABEL_TOO_LONG_ERROR );
 	INTL_EXPOSE_CONST( U_IDNA_ZERO_LENGTH_LABEL_ERROR );
+#if U_ICU_VERSION_MAJOR_NUM > 3 || U_ICU_VERSION_MAJOR_NUM == 3 && U_ICU_VERSION_MINOR_NUM >= 8
 	INTL_EXPOSE_CONST( U_IDNA_DOMAIN_NAME_TOO_LONG_ERROR );
+#endif
 	INTL_EXPOSE_CONST( U_IDNA_ERROR_LIMIT );
 
 	/* Aliases for StringPrep */
