diff -up php-5.4.0RC5/ext/mysqlnd/mysqlnd.c.sock php-5.4.0RC5/ext/mysqlnd/mysqlnd.c
--- php-5.4.0RC5/ext/mysqlnd/mysqlnd.c.sock	2012-01-01 14:15:04.000000000 +0100
+++ php-5.4.0RC5/ext/mysqlnd/mysqlnd.c	2012-01-13 18:25:59.507030997 +0100
@@ -695,7 +695,11 @@ MYSQLND_METHOD(mysqlnd_conn_data, connec
 		if (host_len == sizeof("localhost") - 1 && !strncasecmp(host, "localhost", host_len)) {
 			DBG_INF_FMT("socket=%s", socket_or_pipe? socket_or_pipe:"n/a");
 			if (!socket_or_pipe) {
+#ifdef PHP_MYSQL_UNIX_SOCK_ADDR
+				socket_or_pipe = PHP_MYSQL_UNIX_SOCK_ADDR;
+#else
 				socket_or_pipe = "/tmp/mysql.sock";
+#endif
 			}
 			transport_len = mnd_sprintf(&transport, 0, "unix://%s", socket_or_pipe);
 			unix_socket = TRUE;
