From 8f69e3dfafd3f6e75fa4ffb36fa0b8938a27d9ad Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 29 Jun 2016 18:02:35 +0200
Subject: [PATCH] ensure there is only 1 leading NS_SEPARATOR

---
 src/Generator/ClassGenerator.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Generator/ClassGenerator.php b/src/Generator/ClassGenerator.php
index 2d382eb..c716e3d 100644
--- a/src/Generator/ClassGenerator.php
+++ b/src/Generator/ClassGenerator.php
@@ -1027,6 +1027,6 @@ private function generateShortOrCompleteClassname($fqnClassName)
             return $className;
         }
 
-        return '\\' . $fqnClassName;
+        return '\\' . trim($fqnClassName, '\\');
     }
 }
