--- pecl/radius/trunk/radius.c	2008/03/31 10:09:40	256497
+++ pecl/radius/trunk/radius.c	2010/03/31 20:39:48	297236
@@ -62,7 +62,7 @@
  *
  * Every user visible function must have an entry in radius_functions[].
  */
-function_entry radius_functions[] = {
+zend_function_entry radius_functions[] = {
 	PHP_FE(radius_auth_open,    NULL)
 	PHP_FE(radius_acct_open,    NULL)
 	PHP_FE(radius_close,        NULL)
