diff -up php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/sapi/litespeed/lsapilib.c.dlopen php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/sapi/litespeed/lsapilib.c
--- php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/sapi/litespeed/lsapilib.c.dlopen	2015-03-25 08:54:25.529213259 +0100
+++ php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/sapi/litespeed/lsapilib.c	2015-03-25 08:54:57.937345809 +0100
@@ -582,7 +582,7 @@ static int (*fp_lve_leave)(struct liblve
 static int (*fp_lve_jail)( struct passwd *, char *) = NULL;
 static int lsapi_load_lve_lib()
 {
-    s_liblve = dlopen("liblve.so.0", RTLD_LAZY);
+    s_liblve = dlopen("liblve.so.0", RTLD_NOW);
     if (s_liblve)
     {
         fp_lve_is_available = dlsym(s_liblve, "lve_is_available");
diff -up php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/Zend/zend.h.dlopen php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/Zend/zend.h
diff -up php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/Zend/zend_portability.h.dlopen php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/Zend/zend_portability.h
--- php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/Zend/zend_portability.h.dlopen	2015-03-25 08:53:31.701993105 +0100
+++ php-src-02be73b5fac63a02313b9fe0f204c65d4828edeb/Zend/zend_portability.h	2015-03-25 08:56:09.976640451 +0100
@@ -122,11 +122,11 @@
 # endif
 
 # if defined(RTLD_GROUP) && defined(RTLD_WORLD) && defined(RTLD_PARENT)
-#  define DL_LOAD(libname)			dlopen(libname, RTLD_LAZY | RTLD_GLOBAL | RTLD_GROUP | RTLD_WORLD | RTLD_PARENT)
+#  define DL_LOAD(libname)			dlopen(libname, RTLD_NOW  | RTLD_GLOBAL | RTLD_GROUP | RTLD_WORLD | RTLD_PARENT)
 # elif defined(RTLD_DEEPBIND)
-#  define DL_LOAD(libname)			dlopen(libname, RTLD_LAZY | RTLD_GLOBAL | RTLD_DEEPBIND)
+#  define DL_LOAD(libname)			dlopen(libname, RTLD_NOW  | RTLD_GLOBAL | RTLD_DEEPBIND)
 # else
-#  define DL_LOAD(libname)			dlopen(libname, RTLD_LAZY | RTLD_GLOBAL)
+#  define DL_LOAD(libname)			dlopen(libname, RTLD_NOW  | RTLD_GLOBAL)
 # endif
 # define DL_UNLOAD					dlclose
 # if defined(DLSYM_NEEDS_UNDERSCORE)
