diff -up ./ext/pcre/tests/bug37911.phpt.pcre834 ./ext/pcre/tests/bug37911.phpt
--- ./ext/pcre/tests/bug37911.phpt.pcre834	2014-03-26 14:10:18.285452752 +0100
+++ ./ext/pcre/tests/bug37911.phpt	2014-03-26 14:10:40.028526763 +0100
@@ -37,5 +37,5 @@ array(3) {
   string(4) "blub"
 }
 
-Warning: preg_replace_callback(): Compilation failed: group name must start with a non-digit at offset %d in %sbug37911.php on line %d
+Warning: preg_replace_callback(): Numeric named subpatterns are not allowed in %sbug37911.php on line %d
 NULL
diff -up ./ext/pcre/tests/match_flags3.phpt.pcre834 ./ext/pcre/tests/match_flags3.phpt
--- ./ext/pcre/tests/match_flags3.phpt.pcre834	2014-03-26 14:06:48.792739665 +0100
+++ ./ext/pcre/tests/match_flags3.phpt	2014-03-26 14:07:31.820886128 +0100
@@ -42,5 +42,5 @@ array(1) {
   }
 }
 
-Warning: preg_match(): Compilation failed: group name must start with a non-digit at offset %d in %smatch_flags3.php on line %d
+Warning: preg_match(): Numeric named subpatterns are not allowed in %smatch_flags3.php on line %d
 bool(false)
