--- ./CMakeLists.txt.opts	2016-04-06 11:28:38.677132787 +0200
+++ ./CMakeLists.txt	2016-04-06 11:29:03.131245198 +0200
@@ -24,9 +24,7 @@
 ENABLE_TESTING()
 
 if(UNIX)
-    set(CMAKE_C_FLAGS "-Wall -Wno-unknown-pragmas -Wextra -Wshadow -Winit-self -Wmissing-prototypes -D_GNU_SOURCE")
-    set(CMAKE_C_FLAGS_DEBUG "-O0 -g")
-    set(CMAKE_C_FLAGS_RELEASE "-O3 -g")
+    set(CMAKE_C_FLAGS "-fPIC -D_GNU_SOURCE")
 endif()
 
 if(WIN32)
