From e4395722611f6ace5f9f48a4b4a848c434cbd417 Mon Sep 17 00:00:00 2001
From: Michael Barker <mikeb01@gmail.com>
Date: Wed, 6 Apr 2016 20:56:42 +1200
Subject: [PATCH] Fix #31

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e16b0b3..24536a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,9 +14,9 @@ project("hdr_histogram")
 # 3. If any interfaces have been added since the last public release, then increment age.
 # 4. If any interfaces have been removed since the last public release, then set age to 0.
 
-set(HDR_SOVERSION_CURRENT   0)
-set(HDR_SOVERSION_AGE       9)
-set(HDR_SOVERSION_REVISION  3)
+set(HDR_SOVERSION_CURRENT   1)
+set(HDR_SOVERSION_AGE       0)
+set(HDR_SOVERSION_REVISION  0)
 
 set(HDR_VERSION ${HDR_SOVERSION_CURRENT}.${HDR_SOVERSION_AGE}.${HDR_SOVERSION_REVISION})
 set(HDR_SOVERSION ${HDR_SOVERSION_CURRENT})
