diff -up src/CaBundle.php.rpm src/CaBundle.php
--- src/CaBundle.php.rpm	2016-04-30 09:44:06.624946839 +0200
+++ src/CaBundle.php	2016-04-30 09:44:37.576088602 +0200
@@ -129,7 +129,7 @@ class CaBundle
      */
     public static function getBundledCaBundlePath()
     {
-        return __DIR__.'/../res/cacert.pem';
+        return '/etc/pki/tls/certs/ca-bundle.crt'; // System CA, always
     }
 
     /**
