--- mysql-5.1.26-rc/sql/mysql_priv.h.orig	2008-07-01 00:36:19.000000000 +0200
+++ mysql-5.1.26-rc/sql/mysql_priv.h	2008-07-13 23:19:37.000000000 +0200
@@ -284,7 +284,7 @@
  Feel free to raise this by the smallest amount you can to get the
  "execution_constants" test to pass.
  */
-#define STACK_MIN_SIZE          12000   ///< Abort if less stack during eval.
+#define STACK_MIN_SIZE          16384   ///< Abort if less stack during eval.
 
 #define STACK_MIN_SIZE_FOR_OPEN 1024*80
 #define STACK_BUFF_ALLOC        352     ///< For stack overrun checks
