diff -up src/ProxyManager/Version.php.rpm src/ProxyManager/Version.php
--- src/ProxyManager/Version.php.rpm	2016-06-29 16:37:55.000000000 +0200
+++ src/ProxyManager/Version.php	2016-06-29 16:38:44.000000000 +0200
@@ -20,8 +20,6 @@ declare(strict_types=1);
 
 namespace ProxyManager;
 
-use PackageVersions\Versions;
-
 /**
  * Version class - to be adjusted when a new release is created.
  *
@@ -47,6 +45,6 @@ final class Version
      */
     public static function getVersion() : string
     {
-        return Versions::getVersion('ocramius/proxy-manager');
+        return '@VERSION@@@COMMIT@';
     }
 }
