Index: trunk/report/doublons/doublons.php
===================================================================
--- trunk/report/doublons/doublons.php	(révision 322)
+++ trunk/report/doublons/doublons.php	(révision 323)
@@ -206,7 +206,7 @@
    $SerialBlacklist = "''";
 
    $res = $DB->query("SELECT `value`
-                      FROM `glpi__blacklists`
+                      FROM `glpi_blacklists`
                       WHERE `type` = '3'");
    while ($data = $DB->fetch_array($res)) {
       $SerialBlacklist .= ",'".addslashes($data["value"])."'";
