Index: trunk/report/doublons/doublons.php
===================================================================
--- trunk/report/doublons/doublons.php	(révision 321)
+++ trunk/report/doublons/doublons.php	(révision 322)
@@ -75,11 +75,6 @@
 echo "<tr class='tab_bg_1 center'>";
 echo "<th colspan='3'>".__('Duplicate computers', 'reports')."</th></tr>\n";
 
-if (Session::haveRight("config", READ)) { // Check only read as we probably use the replicate (no 'w' in this case)
-   echo "<tr class='tab_bg_3 center'><td colspan='".(($crit > 0)?'3':'2')."'>";
-   echo "<a href='./doublons.config.php'>".__('Report configuration', 'reports')."</a></td></tr>\n";
-}
-
 echo "<tr class='tab_bg_1'><td class='right'>"._n('Criterion', 'Criteria', 2). "</td><td>";
 echo "<select name='crit'>";
 
