Index: branches/0.83-bugfixes/inc/search.class.php
===================================================================
--- branches/0.83-bugfixes/inc/search.class.php	(revision 18172)
+++ branches/0.83-bugfixes/inc/search.class.php	(revision 18173)
@@ -827,10 +827,13 @@
                $infoc = new Infocom();
                $isadmin = ($infoc->canUpdate() || $infoc->canCreate());
             }
-            $showmassiveactions = count(Dropdown::getMassiveActions($itemtype,$p['is_deleted']));
-            if ($showmassiveactions && $output_type == HTML_OUTPUT) {
-               echo "<form method='post' name='massiveaction_form' id='massiveaction_form' action=\"".
-                     $CFG_GLPI["root_doc"]."/front/massiveaction.php\">";
+            $showmassiveactions = false;
+            if ($itemtype!='States') {
+               $showmassiveactions = count(Dropdown::getMassiveActions($itemtype, $p['is_deleted']));
+               if ($showmassiveactions && $output_type == HTML_OUTPUT) {
+                  echo "<form method='post' name='massiveaction_form' id='massiveaction_form' action=\"".
+                        $CFG_GLPI["root_doc"]."/front/massiveaction.php\">";
+               }
             }
 
             // Compute number of columns to display
