diff -up glpi/install/install.php.cron glpi/install/install.php
--- glpi/install/install.php.cron	2011-06-13 09:24:03.000000000 +0200
+++ glpi/install/install.php	2011-06-13 09:26:15.000000000 +0200
@@ -361,6 +361,11 @@ function step4 ($host, $user, $password,
       $query = "UPDATE `glpi_users`
                 SET `language` = NULL";
       $DB->query($query) or die("4203 ".$LANG['update'][90].$DB->error());
+      
+      // RPM provides a good system cron
+      $query = "UPDATE `glpi_crontasks` SET `mode`=2 WHERE `mode`=1";
+      $DB->query($query)
+      or die("RPM update of glpi_crontasks. " . $LANG['update'][90] . $DB->error());      
    }
 
 
diff -up glpi/install/mysql/glpi-0.80-empty.sql.cron glpi/install/mysql/glpi-0.80-empty.sql
diff -up glpi/install/update_0782_080.php.cron glpi/install/update_0782_080.php
diff -up glpi/install/update.php.cron glpi/install/update.php
--- glpi/install/update.php.cron	2011-06-13 09:21:52.000000000 +0200
+++ glpi/install/update.php	2011-06-13 09:26:06.000000000 +0200
@@ -760,6 +760,11 @@ function updateDbUpTo031() {
    $plugin = new Plugin();
    $plugin->unactivateAll();
 
+   // RPM provides a good system cron
+   $query = "UPDATE `glpi_crontasks` SET `mode`=2 WHERE `mode`=1";
+   $DB->query($query)
+   or die("RPM update of glpi_crontasks. " . $LANG['update'][90] . $DB->error());
+   
    echo "<h3>".$LANG['update'][139]."</h3>";
 
    optimize_tables($migration);
