From bffc917a976f4d2d945c6e631b38dc7e15557b63 Mon Sep 17 00:00:00 2001
From: Shawn Iwinski <shawn.iwinski@gmail.com>
Date: Sat, 7 Jun 2014 11:48:37 -0400
Subject: [PATCH] Fix Raven_Tests_ClientTest::testGetAuthHeader

---
 test/Raven/Tests/ClientTest.php | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/Raven/Tests/ClientTest.php b/test/Raven/Tests/ClientTest.php
index dbbc8dd..0a300a4 100644
--- a/test/Raven/Tests/ClientTest.php
+++ b/test/Raven/Tests/ClientTest.php
@@ -509,7 +509,8 @@ public function testGetAuthHeader() {
         $timestamp = '1234341324.340000';
 
         $expected = "Sentry sentry_timestamp={$timestamp}, sentry_client={$clientstring}, " .
-                    "sentry_version=4, sentry_key=publickey, sentry_secret=secretkey";
+                    "sentry_version=" . Dummy_Raven_Client::PROTOCOL . ", " .
+                    "sentry_key=publickey, sentry_secret=secretkey";
 
         $this->assertEquals($expected, $client->get_auth_header($timestamp, 'raven-php/test', 'publickey', 'secretkey'));
     }
-- 
1.9.3

