diff -up tests/bootstrap.php.rpm tests/bootstrap.php
--- tests/bootstrap.php.rpm	2016-06-20 12:18:26.000000000 +0200
+++ tests/bootstrap.php	2016-06-20 13:03:04.814182986 +0200
@@ -16,8 +16,6 @@ date_default_timezone_set('Europe/Prague
 
 // create temporary directory
 define('TEMP_DIR', __DIR__ . '/tmp/' . lcg_value());
-@mkdir(TEMP_DIR, 0777, TRUE); // @ - base directory may already exist
-register_shutdown_function(function () {
-	Tester\Helpers::purge(TEMP_DIR);
-	rmdir(TEMP_DIR);
-});
+@mkdir(__DIR__ . '/tmp/');
+@mkdir(TEMP_DIR); // @ - base directory may already exist
+
