diff -up ./data/handleDB.php.rpm ./data/handleDB.php
--- ./data/handleDB.php.rpm	2015-11-04 09:31:27.320112762 +0100
+++ ./data/handleDB.php	2015-11-04 09:32:05.209285133 +0100
@@ -12,7 +12,8 @@
  * @since    Release 1.0.0alpha1 of PHP_CompatInfo_Db
  */
 
-require_once dirname(__DIR__) . '/vendor/autoload.php';
+require_once '/usr/share/php/Symfony/autoload.php';
+require_once dirname(__DIR__) . '/src/Bartlett/CompatInfoDb/autoload.php';
 require_once __DIR__ . '/ReferenceCollection.php';
 
 use Bartlett\CompatInfoDb\ExtensionFactory;
diff -up ./src/Bartlett/CompatInfoDb/Environment.php.rpm ./src/Bartlett/CompatInfoDb/Environment.php
--- ./src/Bartlett/CompatInfoDb/Environment.php.rpm	2015-11-04 09:38:03.048913082 +0100
+++ ./src/Bartlett/CompatInfoDb/Environment.php	2015-11-04 09:39:39.974354034 +0100
@@ -37,6 +37,12 @@ class Environment
      */
     public static function initRefDb()
     {
+        if (($database = getenv('BARTLETT_COMPATINFO_DB')) && file_exists($database)) {
+            return new \PDO('sqlite:' . $database);
+        } else if (file_exists($database = '/usr/share/php-bartlett-php-compatinfo-db/compatinfo.sqlite')) {
+            return new \PDO('sqlite:' . $database);
+        }
+
         $database = 'compatinfo.sqlite';
         $tempDir  = sys_get_temp_dir() . '/bartlett';
 
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2015-11-04 09:34:12.089862363 +0100
+++ ./tests/bootstrap.php	2015-11-04 09:34:49.361031923 +0100
@@ -1,7 +1,7 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = 'Bartlett/CompatInfoDb'; // tests use include_path
 
 require_once $vendorDir . '/autoload.php';
 
