diff -up Horde_Autoloader-2.1.0/lib/Horde/Autoloader/Default.php.fedora Horde_Autoloader-2.1.0/lib/Horde/Autoloader/Default.php
--- Horde_Autoloader-2.1.0/lib/Horde/Autoloader/Default.php.fedora	2014-03-04 07:18:14.580352294 +0100
+++ Horde_Autoloader-2.1.0/lib/Horde/Autoloader/Default.php	2014-03-04 07:20:42.591840198 +0100
@@ -14,6 +14,7 @@
 require_once 'Horde/Autoloader.php';
 require_once 'Horde/Autoloader/ClassPathMapper.php';
 require_once 'Horde/Autoloader/ClassPathMapper/Default.php';
+require_once 'Horde/Autoloader/ClassPathMapper/Prefix.php';
 
 /**
  * Default autoloader definition that uses the include path with default
@@ -45,6 +46,14 @@ class Horde_Autoloader_Default extends H
                 new Horde_Autoloader_ClassPathMapper_Default($path)
             );
         }
+        /* Ensure Sabre is taken from /usr/share/php, as /usr/share/pear may contains old version */
+        if (is_dir($sabre = '/usr/share/php/Sabre')) {
+            $this->addClassPathMapper(new Horde_Autoloader_ClassPathMapper_Prefix('/^Sabre\\\\/', $sabre));
+        }
+        /* Keep VObject from pear (v2.1), not ready for v3.1 */
+        if (is_dir($vobject = '/usr/share/pear/Sabre/VObject')) {
+            $this->addClassPathMapper(new Horde_Autoloader_ClassPathMapper_Prefix('/^Sabre\\\\VObject\\\\/', $vobject));
+        }
     }
 
 }
