--- Horde_Editor-2.0.1/lib/Horde/Editor/Ckeditor.php.old	2013-01-12 09:31:49.000000000 +0100
+++ Horde_Editor-2.0.1/lib/Horde/Editor/Ckeditor.php	2013-01-12 09:32:25.000000000 +0100
@@ -33,8 +33,8 @@
         }
 
         $ck_file = empty($params['basic'])
-            ? 'ckeditor.js'
-            : 'ckeditor_basic.js';
+            ? 'ckeditor/ckeditor.js'
+            : 'ckeditor/ckeditor_basic.js';
 
         if (isset($params['config'])) {
             if (is_array($params['config'])) {
