diff -up PHPUnit-3.7.35/PHPUnit/TextUI/TestRunner.php.old PHPUnit-3.7.35/PHPUnit/TextUI/TestRunner.php
--- PHPUnit-3.7.35/PHPUnit/TextUI/TestRunner.php.old	2014-04-22 17:24:35.000000000 +0200
+++ PHPUnit-3.7.35/PHPUnit/TextUI/TestRunner.php	2014-04-22 17:25:46.000000000 +0200
@@ -488,16 +488,6 @@ class PHPUnit_TextUI_TestRunner extends
         if (!self::$versionStringPrinted) {
             self::write(PHPUnit_Runner_Version::getVersionString() . "\n\n");
 
-            if (strpos('3.7.35', '@version') !== 0) {
-                self::write(
-                  "You have installed PHPUnit via PEAR. This installation method is no longer\n" .
-                  "supported and http://pear.phpunit.de/ will be shut down no later than\n" .
-                  "December, 31 2014.\n\n" .
-                  "Please read http://phpunit.de/manual/current/en/installation.html and\n" .
-                  "learn how to use PHPUnit from a PHAR or install it via Composer.\n\n"
-                );
-            }
-
             self::$versionStringPrinted = TRUE;
         }
     }
