diff -up ./phpunit.rpm ./phpunit
--- ./phpunit.rpm	2014-12-28 08:57:05.000000000 +0100
+++ ./phpunit	2014-12-28 09:30:07.294714996 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 /* PHPUnit
  *
@@ -39,24 +39,7 @@ if (!ini_get('date.timezone')) {
     ini_set('date.timezone', 'UTC');
 }
 
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/../vendor/autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        define('PHPUNIT_COMPOSER_INSTALL', $file);
-        break;
-    }
-}
-
-unset($file);
-
-if (!defined('PHPUNIT_COMPOSER_INSTALL')) {
-    fwrite(STDERR,
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-    die(1);
-}
-
+define ('PHPUNIT_COMPOSER_INSTALL', stream_resolve_include_path('PHPUnit/Autoload.php'));
 require PHPUNIT_COMPOSER_INSTALL;
 
 PHPUnit_TextUI_Command::main();
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2014-12-28 09:28:16.402309771 +0100
+++ ./tests/bootstrap.php	2014-12-28 09:28:45.638416606 +0100
@@ -1,6 +1,9 @@
 <?php
 // Needed for isolated tests
-require __DIR__ . '/../vendor/autoload.php';
+require __DIR__ . '/../src/Framework/Assert/Functions.php';
+require __DIR__ . '/_files/CoveredFunction.php';
+require __DIR__ . '/../src/Autoload.php';
+require __DIR__ . '/autoload.php';
 
 ini_set('precision', 14);
 ini_set('serialize_precision', 14);
