diff -up ./phpunit.rpm ./phpunit
--- ./phpunit.rpm	2014-08-08 07:13:30.000000000 +0200
+++ ./phpunit	2014-08-11 11:12:26.435360930 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 /* PHPUnit
  *
@@ -35,20 +35,7 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        define('PHPUNIT_COMPOSER_INSTALL', $file);
-        break;
-    }
-}
-
-if (!defined('PHPUNIT_COMPOSER_INSTALL')) {
-    echo 'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL;
-    die(1);
-}
-
+define ('PHPUNIT_COMPOSER_INSTALL', stream_resolve_include_path('PHPUnit/Autoload.php'));
 require PHPUNIT_COMPOSER_INSTALL;
 
 PHPUnit_TextUI_Command::main();
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2014-08-17 17:36:46.325953778 +0200
+++ ./tests/bootstrap.php	2014-08-17 17:37:54.634043416 +0200
@@ -1,5 +1,8 @@
 <?php
-require __DIR__ . '/../vendor/autoload.php';
+require __DIR__ . '/../src/Framework/Assert/Functions.php';
+require __DIR__ . '/_files/CoveredFunction.php';
+require __DIR__ . '/../src/Autoload.php';
+require __DIR__ . '/autoload.php';
 
 if (!ini_get('date.timezone') && !defined('HHVM_VERSION')) {
     echo PHP_EOL . 'Error: PHPUnit\'s test suite requires the "date.timezone" runtime configuration to be set. Please check your php.ini.' . PHP_EOL;
