diff -up ffmpeg-php-0.6.0/config.m4.orig ffmpeg-php-0.6.0/config.m4
--- ffmpeg-php-0.6.0/config.m4.orig	2010-03-21 09:05:15.000000000 +0100
+++ ffmpeg-php-0.6.0/config.m4	2010-03-21 09:12:32.000000000 +0100
@@ -41,6 +41,18 @@ if test "$PHP_FFMPEG" != "no"; then
       PHP_ADD_INCLUDE($i/include/libavdevice/)
       FFMPEG_INC_FOUND=$i/include/libavcodec
       break
+    elif test -f $i/libavcodec/avcodec.h; then
+      dnl ffmpeg svn revision 12194 and newer put each header in its own dir
+      dnl ffmpeg-devel RPM install in /usr/include/ffmpeg/*
+      PHP_ADD_INCLUDE($i)
+      PHP_ADD_INCLUDE($i/libavcodec/)
+      PHP_ADD_INCLUDE($i/libavformat/)
+      PHP_ADD_INCLUDE($i/libavutil/)
+      PHP_ADD_INCLUDE($i/libswscale/)
+      PHP_ADD_INCLUDE($i/libavfilter/)
+      PHP_ADD_INCLUDE($i/libavdevice/)
+      FFMPEG_INC_FOUND=$i
+      break
     fi
   done
 

