--- pecl/apc/trunk/tests/apc54_004.phpt	2012/09/03 17:19:06	327448
+++ pecl/apc/trunk/tests/apc54_004.phpt	2012/09/03 17:31:18	327449
@@ -31,63 +31,63 @@
 
 --EXPECTF--
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 <br />
-<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>/home/weltling/dws/src/apc_trunk/tests/index.php</b> on line <b>3</b><br />
+<b>Notice</b>:  Undefined property: stdClass::$qwerty in <b>%s/tests/index.php</b> on line <b>3</b><br />
 done
--- pecl/apc/trunk/tests/apc54_018.phpt	2012/09/03 17:31:18	327449
+++ pecl/apc/trunk/tests/apc54_018.phpt	2012/09/03 17:37:18	327450
@@ -42,7 +42,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -67,7 +67,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -92,7 +92,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -117,7 +117,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -142,7 +142,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -167,7 +167,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -192,7 +192,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -217,7 +217,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
@@ -242,7 +242,7 @@
   ["type"]=>
   string(4) "user"
   ["key"]=>
-  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf5675"
+  string(42) "test-niko-asdfasdfasdfkjasdflkasjdfasf%d"
   ["value"]=>
   string(%d) "%s"
   ["num_hits"]=>
