Index: apc_compile.c
===================================================================
--- apc_compile.c	(révision 328824)
+++ apc_compile.c	(copie de travail)
@@ -2469,9 +2469,11 @@
         dst->alias_len = src->alias_len;
     }
 
+#ifndef ZEND_ENGINE_2_5
     if (src->function) {
         CHECK(dst->function = my_copy_function(NULL, src->function, ctxt TSRMLS_CC));
     }
+#endif
 
     APC_COPY_TRAIT_METHOD(dst->trait_method, src->trait_method);
 
@@ -2506,9 +2508,11 @@
     }
     memcpy(dst, src, sizeof(zend_trait_precedence));
 
+#ifndef ZEND_ENGINE_2_5
     if (src->function) {
         CHECK(dst->function = my_copy_function(NULL, src->function, ctxt TSRMLS_CC));
     } 
+#endif
 
     if (src->exclude_from_classes && *src->exclude_from_classes) {
         int i = 0, num_classes = 0;
Index: tests/apc54_error_010.phpt
===================================================================
--- tests/apc54_error_010.phpt	(révision 328955)
+++ tests/apc54_error_010.phpt	(copie de travail)
@@ -23,8 +23,8 @@
 }
 
 class bar {
-	use foo, c { c::test insteadof foo, b; }
-	use foo, c { c::test insteadof foo, b; }
+	use foo, c { c::test insteadof foo; }
+	use foo, c { c::test insteadof foo; }
 }
 
 \$x = new bar;
Index: tests/apc54_error_010_2.phpt
===================================================================
--- tests/apc54_error_010_2.phpt	(révision 328955)
+++ tests/apc54_error_010_2.phpt	(copie de travail)
@@ -23,7 +23,7 @@
 }
 
 class bar {
-	use foo, c { c::test insteadof foo, b; }
+	use foo, c { c::test insteadof foo; }
 }
 
 \$x = new bar;
