From 518c35e15f7034a26e63beb4e049b6576013014d Mon Sep 17 00:00:00 2001
From: Xinchen Hui <laruence@php.net>
Date: Sun, 10 Mar 2013 15:58:26 +0000
Subject: [PATCH 1/8] default on

---
 php_apc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/php_apc.c b/php_apc.c
index 1a6fdcf..6d18914 100644
--- a/php_apc.c
+++ b/php_apc.c
@@ -279,7 +279,7 @@ STD_PHP_INI_BOOLEAN("apc.use_request_time", "1", PHP_INI_ALL, OnUpdateBool, use_
 STD_PHP_INI_BOOLEAN("apc.lazy_functions", "0", PHP_INI_SYSTEM, OnUpdateBool, lazy_functions, zend_apc_globals, apc_globals)
 STD_PHP_INI_BOOLEAN("apc.lazy_classes", "0", PHP_INI_SYSTEM, OnUpdateBool, lazy_classes, zend_apc_globals, apc_globals)
 STD_PHP_INI_ENTRY("apc.serializer", "default", PHP_INI_SYSTEM, OnUpdateStringUnempty, serializer_name, zend_apc_globals, apc_globals)
-STD_PHP_INI_BOOLEAN("apc.enable_opcode_cache", "0", PHP_INI_SYSTEM, OnUpdateBool, enable_opcode_cache, zend_apc_globals, apc_globals)
+STD_PHP_INI_BOOLEAN("apc.enable_opcode_cache", "1", PHP_INI_SYSTEM, OnUpdateBool, enable_opcode_cache, zend_apc_globals, apc_globals)
 PHP_INI_END()
 
 /* }}} */
-- 
1.8.4.2

