From 63a5d09b77a59b2c51b32eb3cead274efdccb6af Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 13 Jun 2016 16:56:47 +0200
Subject: [PATCH] fix typo in test

---
 tests/EasyMockTest.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/EasyMockTest.php b/tests/EasyMockTest.php
index cfddc9e..68f285a 100644
--- a/tests/EasyMockTest.php
+++ b/tests/EasyMockTest.php
@@ -138,7 +138,7 @@ public function should_allow_to_spy_method_calls()
             $mock->__phpunit_verify();
             $this->fail('Exception not thrown');
         } catch (\PHPUnit_Framework_ExpectationFailedException $e) {
-            $this->assertContains('Expected invocation at least once but it never occured', $e->getMessage());
+            $this->assertContains('Expected invocation at least once but it never occur', $e->getMessage());
         }
 
         // Invoke the mock: the test should now pass
