diff -up src/SpdxLicenses.php.rpm src/SpdxLicenses.php
--- src/SpdxLicenses.php.rpm	2015-09-08 09:17:14.000000000 +0200
+++ src/SpdxLicenses.php	2015-09-08 09:19:33.416111340 +0200
@@ -178,7 +178,7 @@ class SpdxLicenses
      */
     public static function getResourcesDir()
     {
-        return dirname(__DIR__) . '/res';
+        return (getenv('BUILDROOT_SPDX')?:'') . '/usr/share/php-composer-spdx-licenses';
     }
 
     private function loadLicenses()
diff -up tests/SpdxLicensesTest.php.rpm tests/SpdxLicensesTest.php
--- tests/SpdxLicensesTest.php.rpm	2015-09-08 09:20:20.453342020 +0200
+++ tests/SpdxLicensesTest.php	2015-09-08 09:21:51.442788249 +0200
@@ -64,12 +64,6 @@ class SpdxLicensesTest extends \PHPUnit_
             is_dir($dir),
             'Expected resources directory to exist.'
         );
-
-        $this->assertEquals(
-            realpath($dir),
-            realpath(__DIR__ . '/../res'),
-            'Expected resources directory to be "res" (relative to project root).'
-        );
     }
 
     /**
