diff -up rrd-0.9.0/config.m4.v14 rrd-0.9.0/config.m4
--- rrd-0.9.0/config.m4.v14	2011-01-04 18:18:43.051373965 +0100
+++ rrd-0.9.0/config.m4	2011-01-04 18:25:35.044351885 +0100
@@ -79,6 +79,9 @@ dnl Finish the setup
     ac_cv_rrdversion=no
     ])
 
+  dnl New function in version 1.4.x
+  PHP_CHECK_FUNC(rrd_lastupdate_r, rrd)
+
   if test "$ac_cv_rrdversion" = yes; then
     AC_DEFINE(HAVE_RRD_12X, 1, [Whether you have rrd_version])
   fi
diff -up rrd-0.9.0/rrd.c.v14 rrd-0.9.0/rrd.c
--- rrd-0.9.0/rrd.c.v14	2011-01-04 18:18:49.152540190 +0100
+++ rrd-0.9.0/rrd.c	2011-01-04 18:27:52.320923086 +0100
@@ -244,8 +244,13 @@ PHP_FUNCTION(rrd_lastupdate)
 
 	if (rrd_test_error()) rrd_clear_error();
 
+#ifdef HAVE_RRD_LASTUPDATE_R
+	if (rrd_lastupdate_r(argv[2], &last_update, &ds_cnt, &ds_namv,
+		&last_ds) == -1) {
+#else
 	if (rrd_lastupdate(2, &argv[1], &last_update, &ds_cnt, &ds_namv,
 		&last_ds) == -1) {
+#endif
 		efree(argv[2]); efree(argv[1]);
 		RETURN_FALSE;
 	}
