--- rrd-0.10.0/tests/rrd_012.phpt.build	2011-03-06 08:42:14.445560763 +0100
+++ rrd-0.10.0/tests/rrd_012.phpt	2011-03-06 08:42:20.897395396 +0100
@@ -5,20 +5,24 @@
 --FILE--
 <?php
 $rrdFile = dirname(__FILE__) . "/testData/speed.rrd";
-var_dump(rrd_info($rrdFile));
+$info=rrd_info($rrdFile);
+# remove system / version / arch specific values
+$info['filename'] = basename($info['filename']);
+unset($info['header_size']);
+unset($info['rra[0].cur_row']);
+unset($info['rra[1].cur_row']);
+var_dump($info);
 ?>
 --EXPECTF--
-array(27) {
+array(24) {
   ["filename"]=>
-  string(61) %s
+  string(9) "speed.rrd"
   ["rrd_version"]=>
   string(4) "0003"
   ["step"]=>
   int(300)
   ["last_update"]=>
   int(920808900)
-  ["header_size"]=>
-  int(740)
   ["ds[speed].index"]=>
   int(0)
   ["ds[speed].type"]=>
@@ -39,8 +43,6 @@
   string(7) "AVERAGE"
   ["rra[0].rows"]=>
   int(24)
-  ["rra[0].cur_row"]=>
-  int(9)
   ["rra[0].pdp_per_row"]=>
   int(1)
   ["rra[0].xff"]=>
@@ -53,8 +55,6 @@
   string(7) "AVERAGE"
   ["rra[1].rows"]=>
   int(10)
-  ["rra[1].cur_row"]=>
-  int(0)
   ["rra[1].pdp_per_row"]=>
   int(6)
   ["rra[1].xff"]=>
