diff -up rrd-0.9.0/rrd.c.orig rrd-0.9.0/rrd.c
--- rrd-0.9.0/rrd.c.orig	2011-01-03 16:16:45.452775867 +0100
+++ rrd-0.9.0/rrd.c	2011-01-03 16:21:42.853690719 +0100
@@ -23,6 +23,7 @@
 #include "php.h"
 #include "ext/standard/php_smart_str.h"
 #include "ext/standard/php_array.h"
+#include "ext/standard/info.h"
 
 #include <rrd.h>
 
diff -up rrd-0.9.0/rrd_create.c.orig rrd-0.9.0/rrd_create.c
--- rrd-0.9.0/rrd_create.c.orig	2011-01-03 16:28:23.116703802 +0100
+++ rrd-0.9.0/rrd_create.c	2011-01-03 16:28:35.260188909 +0100
@@ -25,6 +25,7 @@
 #include "rrd_create.h"
 #include "php_rrd.h"
 #include "ext/standard/php_array.h"
+#include "Zend/zend_exceptions.h"
 
 /* declare class entry */
 static zend_class_entry *ce_rrd_create;
diff -up rrd-0.9.0/rrd_graph.c.orig rrd-0.9.0/rrd_graph.c
--- rrd-0.9.0/rrd_graph.c.orig	2011-01-03 16:27:35.826814625 +0100
+++ rrd-0.9.0/rrd_graph.c	2011-01-03 16:27:50.712409292 +0100
@@ -22,6 +22,7 @@
 
 #include "php.h"
 #include "ext/standard/php_smart_str.h"
+#include "Zend/zend_exceptions.h"
 
 #include <rrd.h>
 
diff -up rrd-0.9.0/rrd_update.c.orig rrd-0.9.0/rrd_update.c
--- rrd-0.9.0/rrd_update.c.orig	2011-01-03 16:17:25.720388022 +0100
+++ rrd-0.9.0/rrd_update.c	2011-01-03 16:29:48.748124529 +0100
@@ -25,6 +25,7 @@
 #include "php_rrd.h"
 #include "rrd_update.h"
 #include "ext/standard/php_smart_str.h"
+#include "Zend/zend_exceptions.h"
 
 /* declare class entry */
 static zend_class_entry *ce_rrd_update;
