From 798621dda8328228a6cccfc09a7b7cce77ba3424 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 30 Mar 2016 11:45:21 +0200
Subject: [PATCH] fix https connection via proxy

---
 PEAR/REST.php | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/PEAR/REST.php b/PEAR/REST.php
index 1e472b7..00129ba 100644
--- a/PEAR/REST.php
+++ b/PEAR/REST.php
@@ -360,11 +360,7 @@ function downloadHttp($url, $lastmodified = null, $accept = false, $channel = fa
         if ($this->config->get('http_proxy')&&
               $proxy = parse_url($this->config->get('http_proxy'))
         ) {
-            $proxy_host = isset($proxy['host']) ? $proxy['host'] : null;
-            if ($schema === 'https') {
-                $proxy_host = 'ssl://' . $proxy_host;
-            }
-
+            $proxy_host   = isset($proxy['host']) ? $proxy['host'] : null;
             $proxy_port   = isset($proxy['port']) ? $proxy['port'] : 8080;
             $proxy_user   = isset($proxy['user']) ? urldecode($proxy['user']) : null;
             $proxy_pass   = isset($proxy['pass']) ? urldecode($proxy['pass']) : null;
