diff -up mysqlnd_qc-1.0.1/config.m4.lib64 mysqlnd_qc-1.0.1/config.m4
--- mysqlnd_qc-1.0.1/config.m4.lib64	2011-09-18 09:58:24.722809410 +0200
+++ mysqlnd_qc-1.0.1/config.m4	2011-09-18 10:01:46.002411077 +0200
@@ -43,7 +43,14 @@ if test "$PHP_MYSQLND_QC" && test "$PHP_
       AC_MSG_RESULT([$PHP_LIBMEMCACHED_DIR])
       PHP_LIBMEMCACHED_INCDIR="$PHP_LIBMEMCACHED_DIR/include"
       PHP_ADD_INCLUDE($PHP_LIBMEMCACHED_INCDIR)
-      PHP_ADD_LIBRARY_WITH_PATH(memcached, $PHP_LIBMEMCACHED_DIR/lib, MEMCACHED_SHARED_LIBADD)
+      PHP_CHECK_LIBRARY(memcached,memcached_free,
+      [
+         PHP_ADD_LIBRARY_WITH_PATH(memcached, $PHP_LIBMEMCACHED_DIR/$PHP_LIBDIR, MYSQLND_QC_SHARED_LIBADD)
+      ],[
+        AC_MSG_ERROR([wrong memcached lib not found in $PHP_LIBMEMCACHED_DIR/$PHP_LIBDIR])
+      ],[
+        -L$PHP_LIBMEMCACHED_DIR/$PHP_LIBDIR
+      ])
 
       AC_DEFINE([MYSQLND_QC_HAVE_MEMCACHE], 1, [Whether memcache support is enabled])
       mysqlnd_qc_sources="$mysqlnd_qc_sources mysqlnd_qc_memcache_handler.c"
@@ -73,7 +80,7 @@ if test "$PHP_MYSQLND_QC" && test "$PHP_
       AC_MSG_RESULT([$PHP_SQLITE_DIR])
       if test -r "$PHP_SQLITE_DIR/include/sqlite3.h"; then
         MYSQLND_QC_SQLITE_DIR="$PHP_SQLITE_DIR/include"
-        MYSQLND_QC_SQLITE_LIB_DIR="$PHP_SQLITE_DIR/lib"
+        MYSQLND_QC_SQLITE_LIB_DIR="$PHP_SQLITE_DIR/$PHP_LIBDIR"
 
         PHP_CHECK_LIBRARY(sqlite3,sqlite3_open,
         [
@@ -81,7 +88,7 @@ if test "$PHP_MYSQLND_QC" && test "$PHP_
         ],[
           AC_MSG_ERROR([wrong SQLite3 lib not found])
         ],[
-          -L$MYSQLND_QC_SQLITE_LIB_DIR/lib -lm
+          -L$MYSQLND_QC_SQLITE_LIB_DIR/$PHP_LIBDIR -lm
         ])
       else
         AC_MSG_ERROR([SQLite3 header not found in $PHP_SQLITE_DIR.])
