--- mysqlnd_qc-1.1.0/config.m4.build	2012-01-27 22:36:46.000000000 +0100
+++ mysqlnd_qc-1.1.0/config.m4	2012-01-30 10:06:58.656134426 +0100
@@ -43,7 +43,14 @@
       AC_MSG_RESULT([$PHP_LIBMEMCACHED_DIR])
       PHP_LIBMEMCACHED_INCDIR="$PHP_LIBMEMCACHED_DIR/include"
       PHP_ADD_INCLUDE($PHP_LIBMEMCACHED_INCDIR)
-      PHP_ADD_LIBRARY_WITH_PATH(memcached, $PHP_LIBMEMCACHED_DIR/lib, MEMCACHED_SHARED_LIBADD)
+      PHP_CHECK_LIBRARY(memcached,memcached_free,
+      [
+         PHP_ADD_LIBRARY_WITH_PATH(memcached, $PHP_LIBMEMCACHED_DIR/$PHP_LIBDIR, MYSQLND_QC_SHARED_LIBADD)
+      ],[
+        AC_MSG_ERROR([wrong memcached lib not found in $PHP_LIBMEMCACHED_DIR/$PHP_LIBDIR])
+      ],[
+        -L$PHP_LIBMEMCACHED_DIR/$PHP_LIBDIR
+      ])
 
       AC_DEFINE([MYSQLND_QC_HAVE_MEMCACHE], 1, [Whether memcache support is enabled])
       PHP_CHECK_LIBRARY(memcached, memcache_exists, [AC_DEFINE(HAVE_MEMCACHE_EXISTS, 1, [ ])], [ ], [ ])
@@ -74,7 +81,7 @@
       AC_MSG_RESULT([$PHP_SQLITE_DIR])
       if test -r "$PHP_SQLITE_DIR/include/sqlite3.h"; then
         MYSQLND_QC_SQLITE_DIR="$PHP_SQLITE_DIR/include"
-        MYSQLND_QC_SQLITE_LIB_DIR="$PHP_SQLITE_DIR/lib"
+        MYSQLND_QC_SQLITE_LIB_DIR="$PHP_SQLITE_DIR/$PHP_LIBDIR"
 
         PHP_CHECK_LIBRARY(sqlite3,sqlite3_open,
         [
@@ -82,7 +89,7 @@
         ],[
           AC_MSG_ERROR([wrong SQLite3 lib not found])
         ],[
-          -L$MYSQLND_QC_SQLITE_LIB_DIR/lib -lm
+          -L$MYSQLND_QC_SQLITE_LIB_DIR/$PHP_LIBDIR -lm
         ])
       else
         AC_MSG_ERROR([SQLite3 header not found in $PHP_SQLITE_DIR.])
