From f275dff446a2c80dedc941e50e330d9eaf427615 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 24 Jun 2015 09:05:56 +0200
Subject: [PATCH]  use zend_ce_throwable introduce in 7.0.0alpha2, fix #9

---
 hprose_service.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hprose_service.c b/hprose_service.c
index b7e5924..e6a62d8 100644
--- a/hprose_service.c
+++ b/hprose_service.c
@@ -30,7 +30,7 @@
 #if PHP_MAJOR_VERSION < 7
 #define zend_exception_ce zend_exception_get_default(TSRMLS_C)
 #else
-#define zend_exception_ce zend_exception_get_base()
+#define zend_exception_ce zend_ce_throwable
 #endif
 
 static zend_bool is_magic_method(char *name, int32_t len) {
