From 909f803f5a9d0af114ab86ce47dbcdcd3f387c8b Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 17 Jun 2016 12:34:11 +0200
Subject: [PATCH] use str_efree to fix #742

---
 redis_commands.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redis_commands.c b/redis_commands.c
index a84f760..2f84577 100644
--- a/redis_commands.c
+++ b/redis_commands.c
@@ -1298,7 +1298,7 @@ int redis_set_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
     CMD_SET_SLOT(slot,key,key_len);
 
     if(key_free) efree(key);
-    if(val_free) efree(val);
+    if(val_free) str_efree(val);
 
     return SUCCESS;
 }
