diff -up facedetect/config.m4.BAD facedetect/config.m4
--- facedetect/config.m4.BAD	2011-05-10 15:20:16.295313177 -0400
+++ facedetect/config.m4	2011-05-10 15:20:36.292054317 -0400
@@ -27,10 +27,7 @@ if test "$PHP_FACEDETECT" != "no"; then
   AC_CHECK_HEADER([$FACEDETECT_DIR/include/opencv/cv.h], [], AC_MSG_ERROR('opencv/cv.h' header not found))
   AC_CHECK_HEADER([$FACEDETECT_DIR/include/opencv/highgui.h], [], AC_MSG_ERROR('opencv/highgui.h' header not found))
 
-  PHP_ADD_LIBRARY_WITH_PATH(cv, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-  PHP_ADD_LIBRARY_WITH_PATH(cvaux, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-  PHP_ADD_LIBRARY_WITH_PATH(highgui, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-  PHP_ADD_LIBRARY_WITH_PATH(cxcore, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(opencv_objdetect, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
 
   PHP_SUBST(FACEDETECT_SHARED_LIBADD)
   AC_DEFINE(HAVE_FACEDETECT, 1, [ ])
diff -up facedetect/facedetect.c.BAD facedetect/facedetect.c
--- facedetect/facedetect.c.BAD	2011-05-10 15:15:04.603349547 -0400
+++ facedetect/facedetect.c	2011-05-10 15:19:31.982886849 -0400
@@ -18,6 +18,7 @@
 #include "config.h"
 #endif
 #include "php.h"
+#include <standard/info.h>
 #include "php_facedetect.h"
 
 #include <opencv/cv.h>
@@ -102,7 +103,7 @@ static void php_facedetect(INTERNAL_FUNC
 
 	storage = cvCreateMemStorage(0);
 
-	faces = cvHaarDetectObjects(gray, cascade, storage, 1.1, 2, CV_HAAR_DO_CANNY_PRUNING, cvSize(0, 0));
+	faces = cvHaarDetectObjects(gray, cascade, storage, 1.1, 2, CV_HAAR_DO_CANNY_PRUNING, cvSize(0, 0), cvSize(0, 0));
 
 	if(return_type) {
 
