diff -ruN facedetect.orig/config.m4 facedetect/config.m4
--- facedetect.orig/config.m4	2010-03-15 02:07:12.000000000 +0200
+++ facedetect/config.m4	2011-01-11 13:03:47.794020008 +0200
@@ -25,20 +25,12 @@
   PHP_ADD_INCLUDE($FACEDETECT_DIR/include)
 
   AC_CHECK_HEADER([$FACEDETECT_DIR/include/opencv/cv.h], [], AC_MSG_ERROR('opencv/cv.h' header not found))
-  AC_CHECK_HEADER([$FACEDETECT_DIR/include/opencv/cvver.h], [], AC_MSG_ERROR('opencv/cvver.h' header not found))
   AC_CHECK_HEADER([$FACEDETECT_DIR/include/opencv/highgui.h], [], AC_MSG_ERROR('opencv/highgui.h' header not found))
 
-  PHP_CHECK_LIBRARY(cv, cvLoad,
-  [
-    PHP_ADD_LIBRARY_WITH_PATH(cv, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-    PHP_ADD_LIBRARY_WITH_PATH(cvaux, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-    PHP_ADD_LIBRARY_WITH_PATH(highgui, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-    PHP_ADD_LIBRARY_WITH_PATH(cxcore, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
-    AC_DEFINE(HAVE_FACEDETECT, 1, [ ])
-  ],[
-    AC_MSG_ERROR([wrong OpenCV version or OpenCV not found])
-  ],[
-  ])
+  PHP_ADD_LIBRARY_WITH_PATH(cv, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(cvaux, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(highgui, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(cxcore, $FACEDETECT_DIR/lib, FACEDETECT_SHARED_LIBADD)
 
   PHP_SUBST(FACEDETECT_SHARED_LIBADD)
   AC_DEFINE(HAVE_FACEDETECT, 1, [ ])
diff -ruN facedetect.orig/facedetect.c facedetect/facedetect.c
--- facedetect.orig/facedetect.c	2010-03-15 02:07:12.000000000 +0200
+++ facedetect/facedetect.c	2011-01-11 13:03:09.466030640 +0200
@@ -21,7 +21,6 @@
 #include "php_facedetect.h"
 
 #include <opencv/cv.h>
-#include <opencv/cvver.h>
 #include <opencv/highgui.h>
 
 /* {{{ facedetect_functions[]
@@ -66,7 +65,6 @@
 	php_info_print_table_start();
 	php_info_print_table_row(2, "facedetect support", "enabled");
 	php_info_print_table_row(2, "facedetect version", PHP_FACEDETECT_VERSION);
-	php_info_print_table_row(2, "OpenCV version", CV_VERSION);
 	php_info_print_table_end();
 }
 /* }}} */
