diff --git a/3rdparty/composer/autoload_psr4.php b/3rdparty/composer/autoload_psr4.php
index 50419c3..0c50d0a 100644
--- a/3rdparty/composer/autoload_psr4.php
+++ b/3rdparty/composer/autoload_psr4.php
@@ -6,26 +6,4 @@ $vendorDir = dirname(dirname(__FILE__));
 $baseDir = $vendorDir;
 
 return array(
-    'phpseclib\\' => array($vendorDir . '/phpseclib/phpseclib/phpseclib'),
-    'ownCloud\\TarStreamer\\' => array($vendorDir . '/deepdiver1975/tarstreamer/src'),
-    'bantu\\IniGetWrapper\\' => array($vendorDir . '/bantu/ini-get-wrapper/src'),
-    'ZipStreamer\\' => array($vendorDir . '/mcnetic/zipstreamer/src'),
-    'SuperClosure\\' => array($vendorDir . '/jeremeamia/SuperClosure/src'),
-    'Sabre\\VObject\\' => array($vendorDir . '/sabre/vobject/lib'),
-    'Sabre\\HTTP\\' => array($vendorDir . '/sabre/http/lib'),
-    'Sabre\\Event\\' => array($vendorDir . '/sabre/event/lib'),
-    'Sabre\\DAV\\' => array($vendorDir . '/sabre/dav/lib/DAV'),
-    'Sabre\\DAVACL\\' => array($vendorDir . '/sabre/dav/lib/DAVACL'),
-    'Sabre\\CardDAV\\' => array($vendorDir . '/sabre/dav/lib/CardDAV'),
-    'Sabre\\CalDAV\\' => array($vendorDir . '/sabre/dav/lib/CalDAV'),
-    'React\\Promise\\' => array($vendorDir . '/react/promise/src'),
-    'Punic\\' => array($vendorDir . '/punic/punic/code'),
-    'Patchwork\\' => array($vendorDir . '/patchwork/jsqueeze/src'),
-    'League\\Flysystem\\' => array($vendorDir . '/league/flysystem/src'),
-    'InterfaSys\\LogNormalizer\\' => array($vendorDir . '/interfasys/lognormalizer/src'),
-    'Icewind\\Streams\\Tests\\' => array($vendorDir . '/icewind/streams/tests'),
-    'Icewind\\Streams\\' => array($vendorDir . '/icewind/streams/src'),
-    'GuzzleHttp\\Stream\\' => array($vendorDir . '/guzzlehttp/streams/src'),
-    'GuzzleHttp\\Ring\\' => array($vendorDir . '/guzzlehttp/ringphp/src'),
-    'GuzzleHttp\\' => array($vendorDir . '/guzzlehttp/guzzle/src'),
 );
