diff --git a/lib/private/integritycheck/checker.php b/lib/private/integritycheck/checker.php
index b991f66..e2111b7 100644
--- a/lib/private/integritycheck/checker.php
+++ b/lib/private/integritycheck/checker.php
@@ -107,7 +107,7 @@ class Checker {
 		 * applicable for very specific scenarios and we should not advertise it
 		 * too prominent. So please do not add it to config.sample.php.
 		 */
-		$isIntegrityCheckDisabled = $this->config->getSystemValue('integrity.check.disabled', false);
+		$isIntegrityCheckDisabled = $this->config->getSystemValue('integrity.check.disabled', true);
 		if($isIntegrityCheckDisabled === true) {
 			return false;
 		}
