diff --git a/3rdparty/composer/autoload_files.php b/3rdparty/composer/autoload_files.php
index eda4201..d87ed77 100644
--- a/3rdparty/composer/autoload_files.php
+++ b/3rdparty/composer/autoload_files.php
@@ -9,6 +9,5 @@ return array(
     'ad155f8f1cf0d418fe49e248db8c661b' => $vendorDir . '/react/promise/src/functions_include.php',
     'e40631d46120a9c38ea139981f8dab26' => $vendorDir . '/ircmaxell/password-compat/lib/password.php',
     '2c102faa651ef8ea5874edb585946bce' => $vendorDir . '/swiftmailer/swiftmailer/lib/swift_required.php',
-    'ce89ac35a6c330c55f4710717db9ff78' => $vendorDir . '/kriswallsmith/assetic/src/functions.php',
 );
 
