diff -rup owncloud.old/apps/files_external/ajax/oauth1.php owncloud/apps/files_external/ajax/oauth1.php
--- owncloud.old/apps/files_external/ajax/oauth1.php	2016-03-08 15:12:48.000000000 +0100
+++ owncloud/apps/files_external/ajax/oauth1.php	2016-03-24 07:09:08.151200875 +0100
@@ -23,7 +23,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>
  *
  */
-require_once __DIR__ . '/../3rdparty/Dropbox/autoload.php';
+require_once '/usr/share/pear/Dropbox/autoload.php';
 
 OCP\JSON::checkAppEnabled('files_external');
 OCP\JSON::checkLoggedIn();
diff -rup owncloud.old/apps/files_external/ajax/oauth2.php owncloud/apps/files_external/ajax/oauth2.php
--- owncloud.old/apps/files_external/ajax/oauth2.php	2016-03-08 15:12:48.000000000 +0100
+++ owncloud/apps/files_external/ajax/oauth2.php	2016-03-24 07:09:27.973288077 +0100
@@ -25,9 +25,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>
  *
  */
-set_include_path(get_include_path().PATH_SEPARATOR.
-	\OC_App::getAppPath('files_external').'/3rdparty/google-api-php-client/src');
-require_once 'Google/Client.php';
+require_once '/usr/share/php/Google/autoload.php';
 
 OCP\JSON::checkAppEnabled('files_external');
 OCP\JSON::checkLoggedIn();
diff -rup owncloud.old/apps/files_external/lib/amazons3.php owncloud/apps/files_external/lib/amazons3.php
--- owncloud.old/apps/files_external/lib/amazons3.php	2016-03-08 15:12:48.000000000 +0100
+++ owncloud/apps/files_external/lib/amazons3.php	2016-03-24 07:08:38.075068563 +0100
@@ -35,9 +35,7 @@
 
 namespace OC\Files\Storage;
 
-set_include_path(get_include_path() . PATH_SEPARATOR .
-	\OC_App::getAppPath('files_external') . '/3rdparty/aws-sdk-php');
-require 'aws-autoloader.php';
+require '/usr/share/php/Aws/autoload.php';
 
 use Aws\S3\S3Client;
 use Aws\S3\Exception\S3Exception;
diff -rup owncloud.old/apps/files_external/lib/dropbox.php owncloud/apps/files_external/lib/dropbox.php
--- owncloud.old/apps/files_external/lib/dropbox.php	2016-03-08 15:12:48.000000000 +0100
+++ owncloud/apps/files_external/lib/dropbox.php	2016-03-24 07:07:41.210818403 +0100
@@ -31,7 +31,7 @@ namespace OC\Files\Storage;
 
 use Icewind\Streams\IteratorDirectory;
 
-require_once __DIR__ . '/../3rdparty/Dropbox/autoload.php';
+require_once '/usr/share/pear/Dropbox/autoload.php';
 
 class Dropbox extends \OC\Files\Storage\Common {
 
diff -rup owncloud.old/apps/files_external/lib/google.php owncloud/apps/files_external/lib/google.php
--- owncloud.old/apps/files_external/lib/google.php	2016-03-08 15:12:48.000000000 +0100
+++ owncloud/apps/files_external/lib/google.php	2016-03-24 07:06:22.995474314 +0100
@@ -35,10 +35,7 @@ namespace OC\Files\Storage;
 
 use Icewind\Streams\IteratorDirectory;
 
-set_include_path(get_include_path().PATH_SEPARATOR.
-	\OC_App::getAppPath('files_external').'/3rdparty/google-api-php-client/src');
-require_once 'Google/Client.php';
-require_once 'Google/Service/Drive.php';
+require_once '/usr/share/php/Google/autoload.php';
 
 class Google extends \OC\Files\Storage\Common {
 
