diff --git a/lib/base.php b/lib/base.php
index bb60273..0f4021d 100644
--- a/lib/base.php
+++ b/lib/base.php
@@ -527,7 +527,9 @@ class OC {
 			if (!file_exists($vendorAutoLoad)) {
 				throw new \RuntimeException('Composer autoloader not found, unable to continue. Check the folder "3rdparty".');
 			}
-			require_once $vendorAutoLoad;
+			$loader = require_once $vendorAutoLoad;
+            $loader->add('Pimple', '/usr/share/php/Pimple');
+            $loader->add('Sabre', '/usr/share/php');
 
 		} catch (\RuntimeException $e) {
 			OC_Response::setStatus(OC_Response::STATUS_SERVICE_UNAVAILABLE);
@@ -656,7 +659,7 @@ class OC {
 		self::registerFilesystemHooks();
 		if (\OC::$server->getSystemConfig()->getValue('enable_previews', true)) {
 			self::registerPreviewHooks();
-		}	
+		}
 		self::registerShareHooks();
 		self::registerLogRotate();
 		self::registerLocalAddressBook();
