diff --git a/core/templates/update.admin.php b/core/templates/update.admin.php
index fbd3025..b9b00ef 100644
--- a/core/templates/update.admin.php
+++ b/core/templates/update.admin.php
@@ -37,7 +37,7 @@
 		<input class="updateButton" type="button" value="<?php p($l->t('Start update')) ?>">
 		<div class="infogroup">
 			<?php p($l->t('To avoid timeouts with larger installations, you can instead run the following command from your installation directory:')) ?>
-			<pre>./occ upgrade</pre>
+			<pre>sudo -u apache php /usr/share/owncloud/occ upgrade</pre>
 		</div>
 	</div>
 
