--- owncloud/apps/files_external/lib/google.php	2015-06-04 06:43:13.000000000 -0700
+++ owncloud/apps/files_external/lib/google.php.new	2015-07-01 15:53:19.363382205 -0700
@@ -23,8 +23,7 @@
 
 set_include_path(get_include_path().PATH_SEPARATOR.
 	\OC_App::getAppPath('files_external').'/3rdparty/google-api-php-client/src');
-require_once 'Google/Client.php';
-require_once 'Google/Service/Drive.php';
+require_once 'Google/autoload.php';
 
 class Google extends \OC\Files\Storage\Common {
 
--- owncloud/apps/files_external/ajax/google.php	2015-06-04 06:43:13.000000000 -0700
+++ owncloud/apps/files_external/ajax/google.php.new	2015-07-01 15:53:28.051505933 -0700
@@ -1,7 +1,7 @@
 <?php
 set_include_path(get_include_path().PATH_SEPARATOR.
 	\OC_App::getAppPath('files_external').'/3rdparty/google-api-php-client/src');
-require_once 'Google/Client.php';
+require_once 'Google/autoload.php';
 
 OCP\JSON::checkAppEnabled('files_external');
 OCP\JSON::checkLoggedIn();
