--- ./settings/Controller/CheckSetupController.php.old	2016-10-07 13:04:26.349194620 +0200
+++ ./settings/Controller/CheckSetupController.php	2016-10-07 13:04:56.642349253 +0200
@@ -214,10 +214,7 @@
 	private function isPhpSupported() {
 		$eol = false;
 
-		//PHP 5.4 is EOL on 14 Sep 2015
-		if (version_compare(PHP_VERSION, '5.5.0') === -1) {
-			$eol = true;
-		}
+		// Red Hat provide support for the system PHP so never eol
 
 		return ['eol' => $eol, 'version' => PHP_VERSION];
 	}
