diff --git a/settings/controller/checksetupcontroller.php b/settings/controller/checksetupcontroller.php
index bd0737a..064058a 100644
--- a/settings/controller/checksetupcontroller.php
+++ b/settings/controller/checksetupcontroller.php
@@ -205,10 +205,7 @@ class CheckSetupController extends Controller {
 	private function isPhpSupported() {
 		$eol = false;
 
-		//PHP 5.4 is EOL on 14 Sep 2015
-		if (version_compare(PHP_VERSION, '5.5.0') === -1) {
-			$eol = true;
-		}
+    // Red Hat provide support for the system PHP so never eol
 
 		return ['eol' => $eol, 'version' => PHP_VERSION];
 	}
