diff --git a/apps/files_external/ajax/oauth1.php b/apps/files_external/ajax/oauth1.php
index e363e6c..d2aefbc 100644
--- a/apps/files_external/ajax/oauth1.php
+++ b/apps/files_external/ajax/oauth1.php
@@ -23,7 +23,6 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>
  *
  */
-require_once __DIR__ . '/../3rdparty/Dropbox/autoload.php';
 
 OCP\JSON::checkAppEnabled('files_external');
 OCP\JSON::checkLoggedIn();
diff --git a/apps/files_external/lib/Lib/Storage/Dropbox.php b/apps/files_external/lib/Lib/Storage/Dropbox.php
index f0f62f9..1b20a62 100644
--- a/apps/files_external/lib/Lib/Storage/Dropbox.php
+++ b/apps/files_external/lib/Lib/Storage/Dropbox.php
@@ -34,8 +34,6 @@ use Icewind\Streams\IteratorDirectory;
 use Icewind\Streams\RetryWrapper;
 use OCP\Files\StorageNotAvailableException;
 
-require_once __DIR__ . '/../../../3rdparty/Dropbox/autoload.php';
-
 class Dropbox extends \OC\Files\Storage\Common {
 
 	private $dropbox;
