--- ./settings/Controller/CheckSetupController.php.old	2016-10-07 13:08:05.373312550 +0200
+++ ./settings/Controller/CheckSetupController.php	2016-10-07 13:11:14.959280009 +0200
@@ -94,8 +94,8 @@ class CheckSetupController extends Contr
 
 		try {
 			$client = $this->clientService->newClient();
-			$client->get('https://www.owncloud.org/');
-			$client->get('http://www.owncloud.org/');
+			$client->get('https://owncloud.org/');
+			$client->get('http://owncloud.org/');
 			return true;
 		} catch (\Exception $e) {
 			return false;
@@ -191,13 +191,13 @@ class CheckSetupController extends Contr
 		if(strpos($versionString, 'NSS/') === 0) {
 			try {
 				$firstClient = $this->clientService->newClient();
-				$firstClient->get('https://www.owncloud.org/');
+				$firstClient->get('https://api.owncloud.com/');
 
 				$secondClient = $this->clientService->newClient();
-				$secondClient->get('https://owncloud.org/');
+				$secondClient->get('https://apps.owncloud.com/');
 			} catch (ClientException $e) {
 				if($e->getResponse()->getStatusCode() === 400) {
-					return (string) $this->l10n->t('cURL is using an outdated %s version (%s). Please update your operating system or features such as %s will not work reliably.', ['NSS', $versionString, $features]);
+					return (string) $this->l10n->t('cURL is using %s version (%s). There may be an issue with features such as %s, this is expected to fixed in the RHEL7.3 milestone - bz#1241172', ['NSS', $versionString, $features]);
 				}
 			}
 		}
--- ./lib/private/OCSClient.php.old	2016-10-07 13:11:45.206434361 +0200
+++ ./lib/private/OCSClient.php	2016-10-07 13:13:50.509073781 +0200
@@ -346,7 +346,12 @@ class OCSClient {
 		$tmp = $data->data->content;
 		$app = [];
 		if (isset($tmp->downloadlink)) {
-			$app['downloadlink'] = (string)$tmp->downloadlink;
+			if (preg_match("/apps\.owncloud\.com/", (string)$tmp->downloadlink)) {
+				$downloadURL = preg_replace("/apps\.owncloud\.com/", "api.owncloud.com", (string)$tmp->downloadlink);
+			} else {
+				$downloadURL = (string)$tmp->downloadlink;
+			}
+			$app['downloadlink'] = $downloadURL;
 		} else {
 			$app['downloadlink'] = '';
 		}
