diff --git a/lib/private/setup.php b/lib/private/setup.php
index 6303d0d..7c30955 100644
--- a/lib/private/setup.php
+++ b/lib/private/setup.php
@@ -373,10 +373,6 @@ class Setup {
 			// out that this is indeed an ownCloud data directory
 			file_put_contents($config->getSystemValue('datadirectory', \OC::$SERVERROOT.'/data').'/.ocdata', '');
 
-			// Update .htaccess files
-			Setup::updateHtaccess();
-			Setup::protectDataDirectory();
-
 			//try to write logtimezone
 			if (date_default_timezone_get()) {
 				$config->setSystemValue('logtimezone', date_default_timezone_get());
diff --git a/lib/private/updater.php b/lib/private/updater.php
index 0d567b8..6b0b1ad 100644
--- a/lib/private/updater.php
+++ b/lib/private/updater.php
@@ -55,10 +55,10 @@ class Updater extends BasicEmitter {
 
 	/** @var ILogger $log */
 	private $log;
-	
+
 	/** @var \OC\HTTPHelper $helper */
 	private $httpHelper;
-	
+
 	/** @var IConfig */
 	private $config;
 
@@ -294,14 +294,6 @@ class Updater extends BasicEmitter {
 			throw new \Exception('Updates between multiple major versions and downgrades are unsupported.');
 		}
 
-		// Update .htaccess files
-		try {
-			Setup::updateHtaccess();
-			Setup::protectDataDirectory();
-		} catch (\Exception $e) {
-			throw new \Exception($e->getMessage());
-		}
-
 		// create empty file in data dir, so we can later find
 		// out that this is indeed an ownCloud data directory
 		// (in case it didn't exist before)
