diff --git a/lib/base.php b/lib/base.php
index 0f4021d..431a1f3 100644
--- a/lib/base.php
+++ b/lib/base.php
@@ -531,5 +531,6 @@ class OC {
             $loader->add('Pimple', '/usr/share/php/Pimple');
             $loader->add('Sabre', '/usr/share/php');
+            $loader->setUseIncludePath(true);
 
 		} catch (\RuntimeException $e) {
 			OC_Response::setStatus(OC_Response::STATUS_SERVICE_UNAVAILABLE);
