From 4d4453b555ec50701d9f50e8d97e5ef9720a4bfe Mon Sep 17 00:00:00 2001
From: mpenick <michael.penick@datastax.com>
Date: Fri, 10 Mar 2017 11:50:09 -0700
Subject: [PATCH] Fixed cass_int64_t <--> int64_t compatability on GCC

---
 include/cassandra.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/cassandra.h b/include/cassandra.h
index d77cb47..02d4ed1 100644
--- a/include/cassandra.h
+++ b/include/cassandra.h
@@ -104,8 +104,13 @@ typedef __UINT64_TYPE__ cass_uint64_t;
 typedef __INT64_TYPE__ cass_int64_t;
 typedef unsigned __INT64_TYPE__ cass_uint64_t;
 #elif defined(__GNUC__)
+#  if  defined(__x86_64__)
+typedef long int cass_int64_t;
+typedef unsigned long int cass_uint64_t;
+#  else
 typedef long long int cass_int64_t;
 typedef unsigned long long int cass_uint64_t;
+#  endif
 #else
 typedef long long cass_int64_t;
 typedef unsigned long long cass_uint64_t;
