--- modules-3.2.9/init/csh.in	2010-11-11 19:23:18.000000000 +0100
+++ modules-3.2.9-new/init/csh.in	2013-05-14 08:43:11.686954109 +0200
@@ -34,7 +34,7 @@
   set prefix  = "$prefix""set noglob;"
   set postfix = "$postfix""unset noglob;"
 endif
-set postfix = "set _exit="'$status'"; $postfix; test 0 = "'$_exit;'
+set postfix = "set _exit="'$status'"; $postfix; @bindir@/test 0 = "'$_exit;'
 
 @VERSIONING@alias module $prefix'eval `'$exec_prefix'/bin/modulecmd '$modules_shell' '$histchar'*`; '$postfix
 @NOTVERSIONING@alias module $prefix'eval `'$exec_prefix'/modulecmd '$modules_shell' '$histchar'*`; '$postfix
