From 7d4794ab5b5097ea3b92bd432d64663488c2e66e Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Thu, 28 May 2015 19:07:10 +0200
Subject: [PATCH] fix zend_parse_parameters call

---
 PHK_Cache.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/PHK_Cache.c b/PHK_Cache.c
index 69354d3..c975e76 100644
--- a/PHK_Cache.c
+++ b/PHK_Cache.c
@@ -141,9 +141,12 @@ static PHP_METHOD(PHK_Cache, cacheID)
 
 static PHP_METHOD(PHK_Cache, setCacheMaxSize)
 {
-	if (zend_parse_parameters(ZEND_NUM_ARGS()TSRMLS_CC, "l", &cache_maxsize)
+	long tmp;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS()TSRMLS_CC, "l", &tmp)
 		== FAILURE)
 		EXCEPTION_ABORT("Cannot parse parameters");
+	cache_maxsize = (int)tmp;
 }
 
 /* }}} */
