diff -up Net_SSH2-0.3.7/SSH2.php.orig Net_SSH2-0.3.7/SSH2.php
--- Net_SSH2-0.3.7/SSH2.php.orig	2014-07-07 17:36:06.047391436 +0200
+++ Net_SSH2-0.3.7/SSH2.php	2014-07-07 17:38:56.852300410 +0200
@@ -1147,7 +1147,7 @@ class Net_SSH2
                     array('twofish128-ctr', 'twofish192-ctr', 'twofish256-ctr', 'twofish128-cbc', 'twofish192-cbc', 'twofish256-cbc', 'twofish-cbc')
                 );
             }
-            if (phpseclib_resolve_include_path('Crypt/Blowfish.php') === false) {
+            if (phpseclib_resolve_include_path('Crypt/Blowfish-phpseclib.php') === false) {
                 $encryption_algorithms = array_diff(
                     $encryption_algorithms,
                     array('blowfish-ctr', 'blowfish-cbc')
@@ -1515,14 +1515,14 @@ class Net_SSH2
                 break;
             case 'blowfish-cbc':
                 if (!class_exists('Crypt_Blowfish')) {
-                    include_once 'Crypt/Blowfish.php';
+                    include_once 'Crypt/Blowfish-phpseclib.php';
                 }
                 $this->encrypt = new Crypt_Blowfish();
                 $this->encrypt_block_size = 8;
                 break;
             case 'blowfish-ctr':
                 if (!class_exists('Crypt_Blowfish')) {
-                    include_once 'Crypt/Blowfish.php';
+                    include_once 'Crypt/Blowfish-phpseclib.php';
                 }
                 $this->encrypt = new Crypt_Blowfish(CRYPT_BLOWFISH_MODE_CTR);
                 $this->encrypt_block_size = 8;
@@ -1591,14 +1591,14 @@ class Net_SSH2
                 break;
             case 'blowfish-cbc':
                 if (!class_exists('Crypt_Blowfish')) {
-                    include_once 'Crypt/Blowfish.php';
+                    include_once 'Crypt/Blowfish-phpseclib.php';
                 }
                 $this->decrypt = new Crypt_Blowfish();
                 $this->decrypt_block_size = 8;
                 break;
             case 'blowfish-ctr':
                 if (!class_exists('Crypt_Blowfish')) {
-                    include_once 'Crypt/Blowfish.php';
+                    include_once 'Crypt/Blowfish-phpseclib.php';
                 }
                 $this->decrypt = new Crypt_Blowfish(CRYPT_BLOWFISH_MODE_CTR);
                 $this->decrypt_block_size = 8;
