--- configure.in.ppc64	2009-07-21 14:01:25.000000000 +0200
+++ configure.in	2009-07-21 14:01:51.000000000 +0200
@@ -1451,7 +1451,7 @@
     CPU_ARCH=sparc
     ;;
 
-x86_64 | ia64)
+x86_64 | ia64 | ppc64)
     CPU_ARCH="$OS_TEST"
     ;;
 esac
@@ -4148,6 +4148,9 @@
 if test -n "${CPU_ARCH}" -a -n "${TARGET_COMPILER_ABI}"; then
     TARGET_XPCOM_ABI="${CPU_ARCH}-${TARGET_COMPILER_ABI}"
 fi
+echo CPU_ARCH=${CPU_ARCH}
+echo TARGET_COMPILER_ABI=${TARGET_COMPILER_ABI}
+echo TARGET_XPCOM_ABI=$TARGET_XPCOM_ABI
 
 dnl Mozilla specific options
 dnl ========================================================
--- mozilla/configure.in.ppc64	2009-07-15 22:36:48.000000000 +0200
+++ mozilla/configure.in	2009-07-21 13:57:46.000000000 +0200
@@ -1317,7 +1317,7 @@
     CPU_ARCH=sparc
     ;;
 
-x86_64 | ia64)
+x86_64 | ia64 | ppc64)
     CPU_ARCH="$OS_TEST"
     ;;
 esac
@@ -4033,6 +4033,9 @@
 if test -n "${CPU_ARCH}" -a -n "${TARGET_COMPILER_ABI}"; then
     TARGET_XPCOM_ABI="${CPU_ARCH}-${TARGET_COMPILER_ABI}"
 fi
+echo CPU_ARCH=${CPU_ARCH}
+echo TARGET_COMPILER_ABI=${TARGET_COMPILER_ABI}
+echo TARGET_XPCOM_ABI=$TARGET_XPCOM_ABI
 
 dnl Mozilla specific options
 dnl ========================================================
