diff -up mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc.stat mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc
--- mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc.stat	2010-06-16 15:12:57.000000000 +0200
+++ mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/file_id.cc	2010-06-16 15:13:02.000000000 +0200
@@ -42,6 +42,7 @@
 #include <link.h>
 #include <string.h>
 #include <sys/mman.h>
+#include <sys/stat.h>
 #include <unistd.h>
 
 #include <algorithm>
