diff -up mozilla-central/browser/components/shell/src/nsGNOMEShellService.cpp.default mozilla-central/browser/components/shell/src/nsGNOMEShellService.cpp
--- mozilla-central/browser/components/shell/src/nsGNOMEShellService.cpp.default	2011-02-07 14:53:16.883934747 -0800
+++ mozilla-central/browser/components/shell/src/nsGNOMEShellService.cpp	2011-02-07 17:11:55.209941360 -0800
@@ -104,8 +104,6 @@ static const char kDesktopColorKey[] = D
 nsresult
 nsGNOMEShellService::Init()
 {
-  nsresult rv;
-
   // GConf _must_ be available, or we do not allow
   // CreateInstance to succeed.
 
@@ -120,19 +118,9 @@ nsGNOMEShellService::Init()
   // the locale encoding.  If it's not set, they use UTF-8.
   mUseLocaleFilenames = PR_GetEnv("G_BROKEN_FILENAMES") != nsnull;
 
-  nsCOMPtr<nsIProperties> dirSvc
-    (do_GetService("@mozilla.org/file/directory_service;1"));
-  NS_ENSURE_TRUE(dirSvc, NS_ERROR_NOT_AVAILABLE);
-
-  nsCOMPtr<nsILocalFile> appPath;
-  rv = dirSvc->Get(NS_XPCOM_CURRENT_PROCESS_DIR, NS_GET_IID(nsILocalFile),
-                   getter_AddRefs(appPath));
-  NS_ENSURE_SUCCESS(rv, rv);
+  mAppPath.Assign(NS_LITERAL_CSTRING(MOZ_APP_NAME));
 
-  rv = appPath->AppendNative(NS_LITERAL_CSTRING(MOZ_APP_NAME));
-  NS_ENSURE_SUCCESS(rv, rv);
-
-  return appPath->GetNativePath(mAppPath);
+  return NS_OK;
 }
 
 NS_IMPL_ISUPPORTS1(nsGNOMEShellService, nsIShellService)
