--- cmake/modules/FindOracle.cmake.orig	2009-09-25 16:52:32.000000000 +0200
+++ cmake/modules/FindOracle.cmake	2009-09-25 16:53:04.000000000 +0200
@@ -72,7 +72,7 @@
     FIND_PROGRAM(AWK awk)
     FIND_PROGRAM(SQLPLUS sqlplus PATHS ENV ORACLE_HOME NO_DEFAULT_PATH)
     FIND_PROGRAM(SQLPLUS sqlplus)
-    SET (sqlplus_version "${SQLPLUS} -version | ${AWK} '/Release/ {print $3}'")
+    SET (sqlplus_version "sqlplus -version 2>/dev/null || sqlplus64 -version 2>/dev/null | ${AWK} '/Release/ {print $3}'")
     EXEC_PROGRAM(${sqlplus_version} OUTPUT_VARIABLE sqlplus_out)
     MESSAGE(STATUS "found sqlplus version: ${sqlplus_out}")
 
