From 1e86632064d5e149356d39077ee8da97164424f0 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Tue, 28 Oct 2014 13:27:08 +0100
Subject: [PATCH] fix class registration

---
 swoole_http.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/swoole_http.c b/swoole_http.c
index 11a0d69..0ad632c 100644
--- a/swoole_http.c
+++ b/swoole_http.c
@@ -382,7 +382,7 @@ void swoole_http_init(int module_number TSRMLS_DC)
     INIT_CLASS_ENTRY(swoole_http_response_ce, "swoole_http_response", swoole_http_response_methods);
     swoole_http_response_class_entry_ptr = zend_register_internal_class(&swoole_http_response_ce TSRMLS_CC);
 
-    INIT_CLASS_ENTRY(swoole_http_response_ce, "swoole_http_request", NULL);
+    INIT_CLASS_ENTRY(swoole_http_request_ce, "swoole_http_request", NULL);
     swoole_http_request_class_entry_ptr = zend_register_internal_class(&swoole_http_request_ce TSRMLS_CC);
 }
 
