Index: trunk/inc/appliancepdf.class.php
===================================================================
--- trunk/inc/appliancepdf.class.php	(revision 188)
+++ trunk/inc/appliancepdf.class.php	(revision 189)
@@ -34,7 +34,7 @@
 class PluginAppliancesAppliancePDF extends PluginPdfCommon {
 
 
-   function __construct(PluginAppliancesAppliance $obj=NULL) {
+   function __construct(CommonGLPI $obj=NULL) {
 
       $this->obj = ($obj ? $obj : new PluginAppliancesAppliance());
    }
