Index: trunk/inc/appliancepdf.class.php
===================================================================
--- trunk/inc/appliancepdf.class.php	(revision 189)
+++ trunk/inc/appliancepdf.class.php	(revision 190)
@@ -40,6 +40,14 @@
    }
 
 
+   function defineAllTabs($options=array()) {
+
+      $onglets = parent::defineAllTabs($options);
+      unset($onglets['Item_Problem$1']); // TODO add method to print linked Problems
+      return $onglets;
+   }
+
+
    static function displayTabContentForPDF(PluginPdfSimplePDF $pdf, CommonGLPI $item, $tab) {
 
       switch ($tab) {
Index: trunk/inc/appliance.class.php
===================================================================
--- trunk/inc/appliance.class.php	(revision 189)
+++ trunk/inc/appliance.class.php	(revision 190)
@@ -254,6 +254,7 @@
       $this->addStandardTab('PluginAppliancesAppliance_Item', $ong, $options);
       $this->addStandardTab('PluginAppliancesOptvalue', $ong, $options);
       $this->addStandardTab('Ticket', $ong, $options);
+      $this->addStandardTab('Item_Problem', $ong, $options);
       $this->addStandardTab('Infocom', $ong, $options);
       $this->addStandardTab('Contract_Item', $ong, $options);
       $this->addStandardTab('Document', $ong, $options);
