From 9cd2576effe2185a9cbfdad4a26c336a19dc7e8b Mon Sep 17 00:00:00 2001
From: "A. Jesse Jiryu Davis" <jesse@mongodb.com>
Date: Fri, 24 Apr 2015 15:08:54 -0400
Subject: [PATCH] CDRIVER-624 fix gcc and clang version checks

---
 build/autotools/CheckCompiler.m4 | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/build/autotools/CheckCompiler.m4 b/build/autotools/CheckCompiler.m4
index b7fd118..0dac394 100644
--- a/build/autotools/CheckCompiler.m4
+++ b/build/autotools/CheckCompiler.m4
@@ -15,14 +15,27 @@ AC_PROG_CXX
 c_compiler="unknown"
 AC_LANG_PUSH([C])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
-#if !(defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER) && __GNUC__ >= 4 && __GNUC_MINOR__ >= 1)
+#if !(defined(__GNUC__) && !defined(__clang__) && !defined(__INTEL_COMPILER))
 #error Not a supported GCC compiler
 #endif
+#if defined(__GNUC__)
+#define GCC_VERSION (__GNUC__ * 10000 \
+                     + __GNUC_MINOR__ * 100 \
+                     + __GNUC_PATCHLEVEL__)
+#if GCC_VERSION < 40100
+#error Not a supported GCC compiler
+#endif
+#endif
 ])], [c_compiler="gcc"], [])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
-#if !(defined(__clang__) && (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 3)))
+#if defined(__clang__)
+#define CLANG_VERSION (__clang_major * 10000 \
+                       + __clang_minor__ * 100 \
+                       + __clang_patchlevel__)
+#if CLANG_VERSION < 30300
 #error Not a supported Clang compiler
 #endif
+#endif
 ])], [c_compiler="clang"], [])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
 #if !(defined(__SUNPRO_C))
