From 5cd61e3170cfe81e4bc9be12d4b2d86e4e1ca53e Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 31 Aug 2015 18:48:15 +0200
Subject: [PATCH] allow driver to use mongoc-log-private

---
 src/mongoc/mongoc-log-private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mongoc/mongoc-log-private.h b/src/mongoc/mongoc-log-private.h
index fa02960..49bf1cc 100644
--- a/src/mongoc/mongoc-log-private.h
+++ b/src/mongoc/mongoc-log-private.h
@@ -17,7 +17,7 @@
 #ifndef MONGOC_LOG_PRIVATE_H
 #define MONGOC_LOG_PRIVATE_H
 
-#if !defined (MONGOC_INSIDE) && !defined (MONGOC_COMPILATION)
+#if !defined (MONGOC_INSIDE) && !defined (MONGOC_COMPILATION) && !defined (MONGOC_I_AM_A_DRIVER)
 # error "Only <mongoc.h> can be included directly."
 #endif
 
From d5de834e40e157b34a6a384c5856917eb1b56cdc Mon Sep 17 00:00:00 2001
From: Hannes Magnusson <bjori@php.net>
Date: Mon, 31 Aug 2015 10:36:38 -0700
Subject: [PATCH] CDRIVER-820: error: #pragma GCC diagnostic not allowed inside
 functions

And remove the unhelpful (debug) assert that cause two tickets so far
---
 src/mongoc/mongoc-matcher-op.c | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/mongoc/mongoc-matcher-op.c b/src/mongoc/mongoc-matcher-op.c
index f098ec6..77f66e2 100644
--- a/src/mongoc/mongoc-matcher-op.c
+++ b/src/mongoc/mongoc-matcher-op.c
@@ -162,15 +162,6 @@ _mongoc_matcher_op_compare_new (mongoc_matcher_opcode_t  opcode, /* IN */
 {
    mongoc_matcher_op_t *op;
 
-#if defined(__GNUC__)
-# pragma GCC diagnostic push
-# pragma GCC diagnostic ignored "-Wtype-limits"
-#endif
-   BSON_ASSERT ((opcode >= MONGOC_MATCHER_OPCODE_EQ) &&
-                (opcode <= MONGOC_MATCHER_OPCODE_NIN));
-#if defined(__GNUC__)
-# pragma GCC diagnostic pop
-#endif
    BSON_ASSERT (path);
    BSON_ASSERT (iter);
 
