From 1675064f16399197e403563d59cf06e6f4fe216e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Fri, 10 May 2013 09:25:54 +0200
Subject: [PATCH] Fixed usage of UTF8ToUTF16BE in new TCPDF

---
 libraries/PDF.class.php | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/libraries/PDF.class.php b/libraries/PDF.class.php
index 5147eb7..7074aab 100644
--- a/libraries/PDF.class.php
+++ b/libraries/PDF.class.php
@@ -97,7 +97,12 @@ public function empty_string($str) {
      */
     function SetAlias($name, $value)
     {
-        $this->Alias[$this->UTF8ToUTF16BE($name)] = $this->UTF8ToUTF16BE($value);
+        $name = TCPDF_FONTS::UTF8ToUTF16BE(
+            $name, false, true, $this->CurrentFont
+        );
+        $this->Alias[$name] = TCPDF_FONTS::UTF8ToUTF16BE(
+            $value, false, true, $this->CurrentFont
+        );
     }
 
     /**
-- 
1.8.1.6

From 325d423abcd55856ffb38ecb6cb0f932148a2cab Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Fri, 10 May 2013 09:27:48 +0200
Subject: [PATCH] Drop compatibility with older TCPDF using empty_string

---
 libraries/PDF.class.php                          | 11 -----------
 libraries/plugins/export/PMA_ExportPdf.class.php |  2 +-
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/libraries/PDF.class.php b/libraries/PDF.class.php
index 7074aab..431111e 100644
--- a/libraries/PDF.class.php
+++ b/libraries/PDF.class.php
@@ -77,17 +77,6 @@ function Footer()
     }
 
     /**
-     * Function to test an empty string (was in tcpdf < 6.0)
-     *
-     * @param string $str to test
-     *
-     * @return boolean
-     */
-    public function empty_string($str) {
-            return (is_null($str) OR (is_string($str) AND (strlen($str) == 0)));
-    }
-
-    /**
      * Function to set alias which will be expanded on page rendering.
      *
      * @param string $name  name of the alias
diff --git a/libraries/plugins/export/PMA_ExportPdf.class.php b/libraries/plugins/export/PMA_ExportPdf.class.php
index 1053bf6..a8c88fa 100644
--- a/libraries/plugins/export/PMA_ExportPdf.class.php
+++ b/libraries/plugins/export/PMA_ExportPdf.class.php
@@ -36,7 +36,7 @@ class PMA_ExportPdf extends PMA_PDF
      */
     function checkPageBreak($h = 0, $y = '', $addpage = true)
     {
-        if ($this->empty_string($y)) {
+        if (TCPDF_STATIC::empty_string($y)) {
             $y = $this->y;
         }
         $current_page = $this->page;
-- 
1.8.1.6

