diff -up phpunit-skeleton-generator-e933d394bdfacec34b7ff4e8fc53c625e09e9721/phpunit-skelgen.rpm phpunit-skeleton-generator-e933d394bdfacec34b7ff4e8fc53c625e09e9721/phpunit-skelgen
--- phpunit-skeleton-generator-e933d394bdfacec34b7ff4e8fc53c625e09e9721/phpunit-skelgen.rpm	2014-05-13 15:03:03.000000000 +0200
+++ phpunit-skeleton-generator-e933d394bdfacec34b7ff4e8fc53c625e09e9721/phpunit-skelgen	2014-05-13 15:04:42.000000000 +0200
@@ -42,23 +42,7 @@
  * @since     File available since Release 1.0.0
  */
 
-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
-}
-
-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require '/usr/share/php/SebastianBergmann/PHPUnit/SkeletonGenerator/autoload.php';
 
 $application = new SebastianBergmann\PHPUnit\SkeletonGenerator\CLI\Application;
 $application->run();
