diff -up src/Framework/MockObject/Generator.php.orig src/Framework/MockObject/Generator.php
--- src/Framework/MockObject/Generator.php.orig	2015-09-29 15:05:06.584010593 +0200
+++ src/Framework/MockObject/Generator.php	2015-09-29 15:09:06.450975301 +0200
@@ -12,6 +12,9 @@ use Doctrine\Instantiator\Instantiator;
 use Doctrine\Instantiator\Exception\InvalidArgumentException as InstantiatorInvalidArgumentException;
 use Doctrine\Instantiator\Exception\UnexpectedValueException as InstantiatorUnexpectedValueException;
 
+# temporary workaround
+require_once 'Text/Template.php';
+
 /**
  * Mock Object Code Generator
  *
