diff -up ImageMagick-6.4.9-6/Magick++/bin/Magick++-config.in.multilib ImageMagick-6.4.9-6/Magick++/bin/Magick++-config.in
--- ImageMagick-6.4.9-6/Magick++/bin/Magick++-config.in.multilib	2008-12-08 20:00:35.000000000 +0100
+++ ImageMagick-6.4.9-6/Magick++/bin/Magick++-config.in	2009-02-24 10:55:05.000000000 +0100
@@ -7,8 +7,6 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
 
 usage='Usage: Magick++-config [--cppflags] [--cxxflags] [--exec-prefix] [--ldflags] [--libs] [--prefix] [--version]
 
@@ -43,19 +41,19 @@ while test $# -gt 0; do
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      echo "-I${includedir} @MAGICK_PCFLAGS@"
+      pkg-config --cflags Magick++
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags Magick++
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags Magick++
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs Magick++
       ;;
     --libs)
-      echo "-L${libdir} -lMagick++ -lMagickWand @MAGICK_LIBS@"
+      pkg-config --libs Magick++
       ;;
     *)
       echo "${usage}" 1>&2
diff -up ImageMagick-6.4.9-6/magick/Magick-config.in.multilib ImageMagick-6.4.9-6/magick/Magick-config.in
--- ImageMagick-6.4.9-6/magick/Magick-config.in.multilib	2008-12-08 20:00:35.000000000 +0100
+++ ImageMagick-6.4.9-6/magick/Magick-config.in	2009-02-24 10:57:33.000000000 +0100
@@ -6,8 +6,6 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@/ImageMagick
 
 usage="\
 Usage: Magick-config [--cflags] [--cppflags] [--exec-prefix] [--ldflags] [--libs] [--prefix] [--version]"
@@ -40,19 +38,19 @@ while test $# -gt 0; do
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      echo "-I${includedir} @MAGICK_PCFLAGS@"
+      pkg-config --cflags MagickCore
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags MagickCore
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags MagickCore
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs MagickCore
       ;;
     --libs)
-      echo "-L${libdir} @MAGICK_LIBS@"
+      pkg-config --libs MagickCore
       ;;
     *)
       echo "${usage}" 1>&2
diff -up ImageMagick-6.4.9-6/magick/MagickCore-config.in.multilib ImageMagick-6.4.9-6/magick/MagickCore-config.in
--- ImageMagick-6.4.9-6/magick/MagickCore-config.in.multilib	2008-12-08 20:00:35.000000000 +0100
+++ ImageMagick-6.4.9-6/magick/MagickCore-config.in	2009-02-24 11:02:23.000000000 +0100
@@ -6,8 +6,6 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@/ImageMagick
 
 usage="\
 Usage: MagickCore-config [--cflags] [--cppflags] [--exec-prefix] [--ldflags] [--libs] [--prefix] [--version]"
@@ -40,25 +38,19 @@ while test $# -gt 0; do
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      echo "-I${includedir} @MAGICK_PCFLAGS@"
+      pkg-config --cflags MagickCore
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags MagickCore
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags MagickCore
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs MagickCore
       ;;
     --libs)
-      echo "-L${libdir} @MAGICK_LIBS@"
-      ;;
-    --coder-path)
-      echo "@CODER_PATH@"
-      ;;
-    --filter-path)
-      echo "@FILTER_PATH@"
+      pkg-config --libs MagickCore
       ;;
     *)
       echo "${usage}" 1>&2
diff -up ImageMagick-6.4.9-6/wand/MagickWand-config.in.multilib ImageMagick-6.4.9-6/wand/MagickWand-config.in
--- ImageMagick-6.4.9-6/wand/MagickWand-config.in.multilib	2008-12-08 20:00:35.000000000 +0100
+++ ImageMagick-6.4.9-6/wand/MagickWand-config.in	2009-02-24 11:01:09.000000000 +0100
@@ -6,8 +6,6 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@/ImageMagick
 
 usage="\
 Usage: MagickWand-config [--cflags] [--cppflags] [--exec-prefix] [--ldflags] [--libs] [--prefix] [--version]"
@@ -40,19 +38,19 @@ while test $# -gt 0; do
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      echo "-I${includedir} @MAGICK_PCFLAGS@"
+      pkg-config --cflags MagickWand
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags MagickWand
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags MagickWand
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs MagickWand
       ;;
     --libs)
-      echo "-L${libdir} -lMagickWand @MAGICK_LIBS@"
+      pkg-config --libs MagickWand
       ;;
     *)
       echo "${usage}" 1>&2
diff -up ImageMagick-6.4.9-6/wand/Wand-config.in.multilib ImageMagick-6.4.9-6/wand/Wand-config.in
--- ImageMagick-6.4.9-6/wand/Wand-config.in.multilib	2008-12-08 20:00:35.000000000 +0100
+++ ImageMagick-6.4.9-6/wand/Wand-config.in	2009-02-24 10:54:27.000000000 +0100
@@ -6,8 +6,6 @@
 
 prefix=@prefix@
 exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@/ImageMagick
 
 usage="\
 Usage: Wand-config [--cflags] [--cppflags] [--exec-prefix] [--ldflags] [--libs] [--prefix] [--version]"
@@ -40,19 +38,19 @@ while test $# -gt 0; do
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      echo "-I${includedir} @MAGICK_PCFLAGS@"
+      pkg-config --cflags MagickWand
       ;;
     --cxxflags)
-      echo '@MAGICK_CXXFLAGS@'
+      pkg-config --cflags MagickWand
       ;;
     --cppflags)
-      echo '@MAGICK_CPPFLAGS@'
+      pkg-config --cflags MagickWand
       ;;
     --ldflags)
-      echo '@MAGICK_LDFLAGS@'
+      pkg-config --libs MagickWand
       ;;
     --libs)
-      echo "-L${libdir} -lMagickWand @MAGICK_LIBS@"
+      pkg-config --libs MagickWand
       ;;
     *)
       echo "${usage}" 1>&2
