From abea4d2fab3d8a605226775b17d9aa679266d627 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 4 Feb 2013 09:50:23 +0100
Subject: [PATCH 2/6] new filter changes in php 5.5

---
 PHP/CompatInfo/Reference/filter.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/PHP/CompatInfo/Reference/filter.php b/PHP/CompatInfo/Reference/filter.php
index ba66a48..0f4c713 100644
--- a/PHP/CompatInfo/Reference/filter.php
+++ b/PHP/CompatInfo/Reference/filter.php
@@ -249,6 +249,8 @@ class PHP_CompatInfo_Reference_Filter
             'FILTER_FLAG_STRIP_BACKTICK'            => array('5.3.2', ''),
             // 2010-03-31
             'FILTER_SANITIZE_FULL_SPECIAL_CHARS'    => array('5.3.3', ''),
+            // Soon
+            'FILTER_VALIDATE_MAC'                   => array('5.5.0-dev', ''),
         );
         $this->applyFilter($release, $items, $constants);
 
-- 
1.7.11.7

