From de9aed8e22cdd55a20e038b832cdca033eee6fd6 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sat, 1 Dec 2012 19:38:19 +0100
Subject: [PATCH 03/17] tokoniser reference for php 5.5

---
 PHP/CompatInfo/Reference/tokenizer.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/PHP/CompatInfo/Reference/tokenizer.php b/PHP/CompatInfo/Reference/tokenizer.php
index 36d4b6c..3af0bea 100644
--- a/PHP/CompatInfo/Reference/tokenizer.php
+++ b/PHP/CompatInfo/Reference/tokenizer.php
@@ -215,6 +215,7 @@ class PHP_CompatInfo_Reference_Tokenizer
             'T_CLONE'                        => array('5.0.0', ''),
             'T_DIR'                          => array('5.3.0', ''),
             'T_FINAL'                        => array('5.0.0', ''),
+            'T_FINALLY'                      => array('5.5.0-dev', ''),
             'T_GOTO'                         => array('5.3.0', ''),
             'T_HALT_COMPILER'                => array('5.1.0', ''),
             'T_IMPLEMENTS'                   => array('5.0.0', ''),
@@ -234,6 +235,7 @@ class PHP_CompatInfo_Reference_Tokenizer
             'T_THROW'                        => array('5.0.0', ''),
             'T_UNSET_CAST'                   => array('5.0.0', ''),
             'T_USE'                          => array('5.3.0', ''),
+            'T_YIELD'                        => array('5.5.0-dev', ''),
         );
         $this->applyFilter($release, $items, $constants);
 
-- 
1.7.11.7

