From 1a72a586223d64fbaa6254dbb26186e852b1a214 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 4 Feb 2013 10:01:51 +0100
Subject: [PATCH 4/6] new intl changes in php 5.5

---
 PHP/CompatInfo/Reference/intl.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PHP/CompatInfo/Reference/intl.php b/PHP/CompatInfo/Reference/intl.php
index 86c6d3d..adaa96d 100644
--- a/PHP/CompatInfo/Reference/intl.php
+++ b/PHP/CompatInfo/Reference/intl.php
@@ -132,6 +132,7 @@ class PHP_CompatInfo_Reference_Intl
             'IntlPartsIterator'             => array('5.5.0-dev', ''),
             'IntlRuleBasedBreakIterator'    => array('5.5.0-dev', ''),
             'IntlTimeZone'                  => array('5.5.0-dev', ''),
+            'UConverter'                    => array('5.5.0-dev', ''),
         );
         $this->applyFilter($release, $items, $classes);
 
-- 
1.7.11.7

