From 5a14d016dca96d296838156aa87bee03a8f8da32 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sat, 1 Dec 2012 20:24:52 +0100
Subject: [PATCH 06/17] openssl reference for php 5.5

---
 PHP/CompatInfo/Reference/openssl.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PHP/CompatInfo/Reference/openssl.php b/PHP/CompatInfo/Reference/openssl.php
index f5cf3b5..804213a 100644
--- a/PHP/CompatInfo/Reference/openssl.php
+++ b/PHP/CompatInfo/Reference/openssl.php
@@ -94,6 +94,7 @@ class PHP_CompatInfo_Reference_Openssl
             'openssl_get_privatekey'         => array('4.0.4', ''),
             'openssl_get_publickey'          => array('4.0.4', ''),
             'openssl_open'                   => array('4.0.4', ''),
+            'openssl_pbkdf2'                 => array('5.5.0-dev', ''),
             'openssl_pkcs12_export'          => array('5.2.2', ''),
             'openssl_pkcs12_export_to_file'  => array('5.2.2', ''),
             'openssl_pkcs12_read'            => array('5.2.2', ''),
-- 
1.7.11.7

