From 388eb5f2155218c497c695c7e1a3c7d49c820f15 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sun, 2 Dec 2012 08:35:53 +0100
Subject: [PATCH 10/17] fix Core reference for PHP 5.5

---
 PHP/CompatInfo/Reference/core.php | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/PHP/CompatInfo/Reference/core.php b/PHP/CompatInfo/Reference/core.php
index 1cdf216..50e159f 100644
--- a/PHP/CompatInfo/Reference/core.php
+++ b/PHP/CompatInfo/Reference/core.php
@@ -95,6 +95,12 @@ class PHP_CompatInfo_Reference_Core
         );
         $this->applyFilter($release, $items, $classes);
 
+        $release = '5.5.0';       // soon
+        $items = array(
+            'Generator'                      => array('5.5.0-dev', ''),
+        );
+        $this->applyFilter($release, $items, $classes);
+
         return $classes;
     }
 
-- 
1.7.11.7

