From 9c1c9ccb848a048b0cff7b9448e2099bae676268 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 14 Jan 2013 18:08:52 +0100
Subject: [PATCH 15/17] mysqli new constant

---
 PHP/CompatInfo/Reference/mysqli.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/PHP/CompatInfo/Reference/mysqli.php b/PHP/CompatInfo/Reference/mysqli.php
index 82e7e8f..e635150 100644
--- a/PHP/CompatInfo/Reference/mysqli.php
+++ b/PHP/CompatInfo/Reference/mysqli.php
@@ -288,6 +288,8 @@ class PHP_CompatInfo_Reference_Mysqli
             'MYSQLI_NUM'                     => array('5.0.0', ''),
             'MYSQLI_NUM_FLAG'                => array('5.0.0', ''),
             'MYSQLI_ON_UPDATE_NOW_FLAG'      => array('5.3.0', ''),
+            'MYSQLI_OPT_CAN_HANDLE_EXPIRED_PASSWORDS'
+                                             => array('5.5.0-dev', ''),
             'MYSQLI_OPT_CONNECT_TIMEOUT'     => array('5.0.0', ''),
             'MYSQLI_OPT_INT_AND_FLOAT_NATIVE'
                                              => array('5.3.0', ''),
-- 
1.7.11.7

