From 0c164aac95d7667e9c52b968722e306f4e6c0c8e Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 4 Feb 2013 10:00:27 +0100
Subject: [PATCH 3/6] new curl changes in php 5.5

---
 PHP/CompatInfo/Reference/sockets.php | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/PHP/CompatInfo/Reference/sockets.php b/PHP/CompatInfo/Reference/sockets.php
index 2c4c327..0abc9ed 100644
--- a/PHP/CompatInfo/Reference/sockets.php
+++ b/PHP/CompatInfo/Reference/sockets.php
@@ -107,6 +107,14 @@ class PHP_CompatInfo_Reference_Sockets
         );
         $this->applyFilter($release, $items, $functions);
 
+        $release = '5.5.0';       // soon
+        $items = array(
+            'socket_cmsg_space'              => array('5.5.0-dev', ''),
+            'socket_recvmsg'                 => array('5.5.0-dev', ''),
+            'socket_sendmsg'                 => array('5.5.0-dev', ''),
+        );
+        $this->applyFilter($release, $items, $functions);
+
         return $functions;
     }
 
@@ -135,9 +143,16 @@ class PHP_CompatInfo_Reference_Sockets
             'AF_UNIX'                        => array('4.1.0', ''),
             'IPPROTO_IP'                     => array('5.4.0', ''),
             'IPPROTO_IPV6'                   => array('5.4.0', ''),
+            'IPV6_HOPLIMIT'                  => array('5.5.0-dev', ''),
             'IPV6_MULTICAST_HOPS'            => array('5.4.0', ''),
             'IPV6_MULTICAST_IF'              => array('5.4.0', ''),
             'IPV6_MULTICAST_LOOP'            => array('5.4.0', ''),
+            'IPV6_PKTINFO'                   => array('5.5.0-dev', ''),
+            'IPV6_RECVTCLASS'                => array('5.5.0-dev', ''),
+            'IPV6_RECVHOPLIMIT'              => array('5.5.0-dev', ''),
+            'IPV6_RECVPKTINFO'               => array('5.5.0-dev', ''),
+            'IPV6_TCLASS'                    => array('5.5.0-dev', ''),
+            'IPV6_UNICAST_HOPS'              => array('5.5.0-dev', ''),
             'IP_MULTICAST_IF'                => array('5.4.0', ''),
             'IP_MULTICAST_LOOP'              => array('5.4.0', ''),
             'IP_MULTICAST_TTL'               => array('5.4.0', ''),
@@ -147,15 +162,25 @@ class PHP_CompatInfo_Reference_Sockets
             'MCAST_LEAVE_GROUP'              => array('5.4.0', ''),
             'MCAST_LEAVE_SOURCE_GROUP'       => array('5.4.0', ''),
             'MCAST_UNBLOCK_SOURCE'           => array('5.4.0', ''),
+            'MSG_CMSG_CLOEXEC'               => array('5.5.0-dev', ''),
+            'MSG_CTRUNC'                     => array('5.5.0-dev', ''),
+            'MSG_CONFIRM'                    => array('5.5.0-dev', ''),
             'MSG_DONTROUTE'                  => array('4.1.0', ''),
             'MSG_DONTWAIT'                   => array('5.2.10', ''),
             'MSG_EOF'                        => array('5.2.4', ''),
+            'MSG_ERRQUEUE'                   => array('5.5.0-dev', ''),
             'MSG_EOR'                        => array('5.2.4', ''),
+            'MSG_MORE'                       => array('5.5.0-dev', ''),
+            'MSG_NOSIGNAL'                   => array('5.5.0-dev', ''),
             'MSG_OOB'                        => array('4.1.0', ''),
             'MSG_PEEK'                       => array('4.1.0', ''),
+            'MSG_TRUNC'                      => array('5.5.0-dev', ''),
+            'MSG_WAITFORONE'                 => array('5.5.0-dev', ''),
             'MSG_WAITALL'                    => array('4.1.0', ''),
             'PHP_BINARY_READ'                => array('4.1.0', ''),
             'PHP_NORMAL_READ'                => array('4.1.0', ''),
+            'SCM_CREDENTIALS'                => array('5.5.0-dev', ''),
+            'SCM_RIGHTS'                     => array('5.5.0-dev', ''),
             'SOCKET_E2BIG'                   => array('4.3.0', ''),
             'SOCKET_EACCES'                  => array('4.1.0', ''),
             'SOCKET_EADDRINUSE'              => array('4.1.0', ''),
@@ -284,6 +309,7 @@ class PHP_CompatInfo_Reference_Sockets
             'SO_KEEPALIVE'                   => array('4.1.0', ''),
             'SO_LINGER'                      => array('4.1.0', ''),
             'SO_OOBINLINE'                   => array('4.1.0', ''),
+            'SO_PASSCRED'                    => array('5.5.0-dev', ''),
             'SO_RCVBUF'                      => array('4.1.0', ''),
             'SO_RCVLOWAT'                    => array('4.1.0', ''),
             'SO_RCVTIMEO'                    => array('4.1.0', ''),
-- 
1.7.11.7

