From 4b7619c705882351d12ada35be473779287a1c6e Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 13 Feb 2013 11:22:28 +0100
Subject: [PATCH 6/6] fix mysqli reference for php 5.5 (trans)

---
 PHP/CompatInfo/Reference/mysqli.php | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/PHP/CompatInfo/Reference/mysqli.php b/PHP/CompatInfo/Reference/mysqli.php
index e635150..d2215e0 100644
--- a/PHP/CompatInfo/Reference/mysqli.php
+++ b/PHP/CompatInfo/Reference/mysqli.php
@@ -113,6 +113,7 @@ class PHP_CompatInfo_Reference_Mysqli
         $items = array(
             'mysqli_affected_rows'           => array('5.0.0', ''),
             'mysqli_autocommit'              => array('5.0.0', ''),
+            'mysqli_begin_transaction'       => array('5.5.0-dev', ''),
             'mysqli_bind_param'              => array('5.0.0', self::LATEST_PHP_5_3),
             'mysqli_bind_result'             => array('5.0.0', self::LATEST_PHP_5_3),
             'mysqli_change_user'             => array('5.0.0', ''),
@@ -185,11 +186,13 @@ class PHP_CompatInfo_Reference_Mysqli
             'mysqli_real_query'              => array('5.0.0', ''),
             'mysqli_reap_async_query'        => array('5.0.0', ''),
             'mysqli_refresh'                 => array('5.3.0', ''),
+            'mysqli_release_savepoint'       => array('5.5.0-dev', ''),
             'mysqli_report'                  => array('5.0.0', ''),
             'mysqli_rollback'                => array('5.0.0', ''),
             'mysqli_rpl_parse_enabled'       => array('5.0.0', self::LATEST_PHP_5_2),
             'mysqli_rpl_probe'               => array('5.0.0', self::LATEST_PHP_5_2),
             'mysqli_rpl_query_type'          => array('5.0.0', self::LATEST_PHP_5_2),
+            'mysqli_savepoint'               => array('5.5.0-dev', ''),
             'mysqli_select_db'               => array('5.0.0', ''),
             'mysqli_send_long_data'          => array('5.0.0', self::LATEST_PHP_5_3),
             'mysqli_send_query'              => array('5.0.0', self::LATEST_PHP_5_2),
@@ -336,6 +339,14 @@ class PHP_CompatInfo_Reference_Mysqli
                                              => array('5.0.0', ''),
             'MYSQLI_STORE_RESULT'            => array('5.0.0', ''),
             'MYSQLI_TIMESTAMP_FLAG'          => array('5.0.0', ''),
+            'MYSQLI_TRANS_COR_AND_CHAIN'     => array('5.5.0-dev', ''),
+            'MYSQLI_TRANS_COR_AND_NO_CHAIN'  => array('5.5.0-dev', ''),
+            'MYSQLI_TRANS_COR_NO_RELEASE'    => array('5.5.0-dev', ''),
+            'MYSQLI_TRANS_COR_RELEASE'       => array('5.5.0-dev', ''),
+            'MYSQLI_TRANS_START_READ_ONLY'   => array('5.5.0-dev', ''),
+            'MYSQLI_TRANS_START_READ_WRITE'  => array('5.5.0-dev', ''),
+            'MYSQLI_TRANS_START_WITH_CONSISTENT_SNAPSHOT'
+                                             => array('5.5.0-dev', ''),
             'MYSQLI_TYPE_BIT'                => array('5.0.0', ''),
             'MYSQLI_TYPE_BLOB'               => array('5.0.0', ''),
             'MYSQLI_TYPE_CHAR'               => array('5.0.0', ''),
-- 
1.7.11.7

