From 94a4f11cac43454ed26d37fb71b4eafc4de61bc6 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sat, 9 Mar 2013 16:26:31 +0100
Subject: [PATCH] fix gd reference for php 5.5.0alpha6

---
 PHP/CompatInfo/Reference/gd.php | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/PHP/CompatInfo/Reference/gd.php b/PHP/CompatInfo/Reference/gd.php
index 60e16f3..d10be0f 100644
--- a/PHP/CompatInfo/Reference/gd.php
+++ b/PHP/CompatInfo/Reference/gd.php
@@ -118,6 +118,8 @@ class PHP_CompatInfo_Reference_Gd
             'imagecreatefromxpm'             => array('4.0.1', ''),
             'imagecreatetruecolor'           => array('4.0.6', ''),
             'imagecreatefromwebp'            => array('5.4.0', ''),
+            'imagecrop'                      => array('5.5.0-dev', ''),
+            'imagecropauto'                  => array('5.5.0-dev', ''),
             'imagewebp'                      => array('5.4.0', ''),
             'imagedashedline'                => array('4.0.0', ''),
             'imagedestroy'                   => array('4.0.0', ''),
@@ -129,6 +131,7 @@ class PHP_CompatInfo_Reference_Gd
             'imagefilledrectangle'           => array('4.0.0', ''),
             'imagefilltoborder'              => array('4.0.0', ''),
             'imagefilter'                    => array('5.0.0', ''),
+            'imageflip'                      => array('5.5.0-dev', ''),
             'imagefontheight'                => array('4.0.0', ''),
             'imagefontwidth'                 => array('4.0.0', ''),
             'imageftbbox'                    => array('4.0.7', ''),
@@ -217,6 +220,12 @@ class PHP_CompatInfo_Reference_Gd
             'IMG_COLOR_STYLEDBRUSHED'        => array('4.0.0', ''),
             'IMG_COLOR_TILED'                => array('4.0.0', ''),
             'IMG_COLOR_TRANSPARENT'          => array('4.0.0', ''),
+            'IMG_CROP_BLACK'                 => array('5.5.0-dev', ''),
+            'IMG_CROP_DEFAULT'               => array('5.5.0-dev', ''),
+            'IMG_CROP_SIDES'                 => array('5.5.0-dev', ''),
+            'IMG_CROP_TRANSPARENT'           => array('5.5.0-dev', ''),
+            'IMG_CROP_THRESHOLD'             => array('5.5.0-dev', ''),
+            'IMG_CROP_WHITE'                 => array('5.5.0-dev', ''),
             'IMG_EFFECT_ALPHABLEND'          => array('4.0.0', ''),
             'IMG_EFFECT_NORMAL'              => array('4.0.0', ''),
             'IMG_EFFECT_OVERLAY'             => array('4.0.0', ''),
@@ -233,6 +242,9 @@ class PHP_CompatInfo_Reference_Gd
             'IMG_FILTER_PIXELATE'            => array('5.3.0', ''),
             'IMG_FILTER_SELECTIVE_BLUR'      => array('4.0.0', ''),
             'IMG_FILTER_SMOOTH'              => array('4.0.0', ''),
+            'IMG_FLIP_HORIZONTAL'            => array('5.5.0-dev', ''),
+            'IMG_FLIP_VERTICAL'              => array('5.5.0-dev', ''),
+            'IMG_FLIP_BOTH'                  => array('5.5.0-dev', ''),
             'IMG_GD2_COMPRESSED'             => array('4.0.0', ''),
             'IMG_GD2_RAW'                    => array('4.0.0', ''),
             'IMG_GIF'                        => array('4.0.0', ''),
-- 
1.7.11.7

