From 77931f9c94a4cb639f62cf077ae7faaf701973a9 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 20 Mar 2013 14:45:23 +0100
Subject: [PATCH] new 'array_column' function

---
 PHP/CompatInfo/Reference/standard.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PHP/CompatInfo/Reference/standard.php b/PHP/CompatInfo/Reference/standard.php
index 68cbc66..5762d01 100644
--- a/PHP/CompatInfo/Reference/standard.php
+++ b/PHP/CompatInfo/Reference/standard.php
@@ -756,6 +756,7 @@ public function getFunctions($extension = null, $version = null, $condition = nu
             'password_hash'                  => array('5.5.0-dev', ''),
             'password_needs_rehash'          => array('5.5.0-dev', ''),
             'password_verify'                => array('5.5.0-dev', ''),
+            'array_column'                   => array('5.5.0-dev', ''),
         );
         $this->applyFilter($release, $items, $functions);
 
-- 
1.8.1.5

