From 7c4f30cbcaae602d81c9cff6adaa9b32f676955c Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Thu, 28 Mar 2013 13:05:00 +0100
Subject: [PATCH] fix gd reference for php 5.5.0beta2

---
 PHP/CompatInfo/Reference/gd.php | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/PHP/CompatInfo/Reference/gd.php b/PHP/CompatInfo/Reference/gd.php
index d10be0f..1ddb29f 100644
--- a/PHP/CompatInfo/Reference/gd.php
+++ b/PHP/CompatInfo/Reference/gd.php
@@ -78,6 +78,9 @@ public function getFunctions($extension = null, $version = null, $condition = nu
         $items = array(
             'gd_info'                        => array('4.3.0', ''),
             'image2wbmp'                     => array('4.0.5', ''),
+            'imageaffine'                    => array('5.5.0-dev', ''),
+            'imageaffinematrixconcat'        => array('5.5.0-dev', ''),
+            'imageaffinematrixget'           => array('5.5.0-dev', ''),
             'imagealphablending'             => array('4.0.6', ''),
             'imageantialias'                 => array('4.3.2', ''),
             'imagearc'                       => array('4.0.0', ''),
@@ -149,6 +152,7 @@ public function getFunctions($extension = null, $version = null, $condition = nu
             'imageline'                      => array('4.0.0', ''),
             'imageloadfont'                  => array('4.0.0', ''),
             'imagepalettecopy'               => array('4.0.1', ''),
+            'imagepalettetotruecolor'        => array('5.5.0-dev', ''),
             'imagepng'                       => array('4.0.0', ''),
             'imagepolygon'                   => array('4.0.0', ''),
             'imagepsbbox'                    => array('4.0.0', ''),
@@ -161,7 +165,9 @@ public function getFunctions($extension = null, $version = null, $condition = nu
             'imagerectangle'                 => array('4.0.0', ''),
             'imagerotate'                    => array('4.3.0', ''),
             'imagesavealpha'                 => array('4.3.2', ''),
+            'imagescale'                     => array('5.5.0-dev', ''),
             'imagesetbrush'                  => array('4.0.6', ''),
+            'imagesetinterpolation'          => array('5.5.0-dev', ''),
             'imagesetpixel'                  => array('4.0.0', ''),
             'imagesetstyle'                  => array('4.0.6', ''),
             'imagesetthickness'              => array('4.0.6', ''),
@@ -210,11 +216,25 @@ public function getConstants($extension = null, $version = null, $condition = nu
             'GD_MINOR_VERSION'               => array('5.2.4', ''),
             'GD_RELEASE_VERSION'             => array('5.2.4', ''),
             'GD_VERSION'                     => array('5.2.4', ''),
+            'IMG_AFFINE_ROTATE'              => array('5.5.0-dev', ''),
+            'IMG_AFFINE_SCALE'               => array('5.5.0-dev', ''),
+            'IMG_AFFINE_SHEAR_HORIZONTAL'    => array('5.5.0-dev', ''),
+            'IMG_AFFINE_SHEAR_VERTICAL'      => array('5.5.0-dev', ''),
+            'IMG_AFFINE_TRANSLATE'           => array('5.5.0-dev', ''),
             'IMG_ARC_CHORD'                  => array('4.0.0', ''),
             'IMG_ARC_EDGED'                  => array('4.0.0', ''),
             'IMG_ARC_NOFILL'                 => array('4.0.0', ''),
             'IMG_ARC_PIE'                    => array('4.0.0', ''),
             'IMG_ARC_ROUNDED'                => array('4.0.0', ''),
+            'IMG_BELL'                       => array('5.5.0-dev', ''),
+            'IMG_BESSEL'                     => array('5.5.0-dev', ''),
+            'IMG_BICUBIC'                    => array('5.5.0-dev', ''),
+            'IMG_BICUBIC_FIXED'              => array('5.5.0-dev', ''),
+            'IMG_BILINEAR_FIXED'             => array('5.5.0-dev', ''),
+            'IMG_BLACKMAN'                   => array('5.5.0-dev', ''),
+            'IMG_BOX'                        => array('5.5.0-dev', ''),
+            'IMG_BSPLINE'                    => array('5.5.0-dev', ''),
+            'IMG_CATMULLROM'                 => array('5.5.0-dev', ''),
             'IMG_COLOR_BRUSHED'              => array('4.0.0', ''),
             'IMG_COLOR_STYLED'               => array('4.0.0', ''),
             'IMG_COLOR_STYLEDBRUSHED'        => array('4.0.0', ''),
@@ -245,13 +265,25 @@ public function getConstants($extension = null, $version = null, $condition = nu
             'IMG_FLIP_HORIZONTAL'            => array('5.5.0-dev', ''),
             'IMG_FLIP_VERTICAL'              => array('5.5.0-dev', ''),
             'IMG_FLIP_BOTH'                  => array('5.5.0-dev', ''),
+            'IMG_GAUSSIAN'                   => array('5.5.0-dev', ''),
+            'IMG_GENERALIZED_CUBIC'          => array('5.5.0-dev', ''),
             'IMG_GD2_COMPRESSED'             => array('4.0.0', ''),
             'IMG_GD2_RAW'                    => array('4.0.0', ''),
             'IMG_GIF'                        => array('4.0.0', ''),
+            'IMG_HAMMING'                    => array('5.5.0-dev', ''),
+            'IMG_HANNING'                    => array('5.5.0-dev', ''),
+            'IMG_HERMITE'                    => array('5.5.0-dev', ''),
             'IMG_JPEG'                       => array('4.0.0', ''),
             'IMG_JPG'                        => array('4.0.0', ''),
+            'IMG_MITCHELL'                   => array('5.5.0-dev', ''),
+            'IMG_NEAREST_NEIGHBOUR'          => array('5.5.0-dev', ''),
             'IMG_PNG'                        => array('4.0.0', ''),
+            'IMG_POWER'                      => array('5.5.0-dev', ''),
+            'IMG_QUADRATIC'                  => array('5.5.0-dev', ''),
+            'IMG_SINC'                       => array('5.5.0-dev', ''),
+            'IMG_TRIANGLE'                   => array('5.5.0-dev', ''),
             'IMG_WBMP'                       => array('4.0.0', ''),
+            'IMG_WEIGHTED4'                  => array('5.5.0-dev', ''),
             'IMG_XPM'                        => array('4.0.0', ''),
             'PNG_ALL_FILTERS'                => array('4.0.0', ''),
             'PNG_FILTER_AVG'                 => array('4.0.0', ''),
-- 
1.8.1.5

