From d75d31cc71065a6c7f44de9f029e255dce18aab9 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 12 Apr 2013 08:53:32 +0200
Subject: [PATCH] fix sphinx reference test

---
 tests/Reference/SphinxTest.php | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/Reference/SphinxTest.php b/tests/Reference/SphinxTest.php
index d91ef30..39714bf 100644
--- a/tests/Reference/SphinxTest.php
+++ b/tests/Reference/SphinxTest.php
@@ -42,9 +42,12 @@ protected function setUp()
     {
         // Constants conditionnaly exists (according to libsphinx version)
         $this->optionalconstants = array(
+            'SPH_RANK_EXPR',
             'SPH_RANK_FIELDMASK',
             'SPH_RANK_MATCHANY',
             'SPH_RANK_PROXIMITY',
+            'SPH_RANK_SPH04',
+            'SPH_RANK_TOTAL',
         );
         $this->obj = new PHP_CompatInfo_Reference_Sphinx();
         parent::setUp();
-- 
1.8.1.5

