From b84b0440e26be71b2f9bd6ab547859add813a236 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 12 Apr 2013 08:57:25 +0200
Subject: [PATCH] add opcache_invalidate (7.0.2-dev, or PHP 5.5.0beta3)

---
 PHP/CompatInfo/Reference/zendopcache.php | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/PHP/CompatInfo/Reference/zendopcache.php b/PHP/CompatInfo/Reference/zendopcache.php
index c488024..a326bdf 100644
--- a/PHP/CompatInfo/Reference/zendopcache.php
+++ b/PHP/CompatInfo/Reference/zendopcache.php
@@ -85,6 +85,12 @@ public function getFunctions($extension = null, $version = null, $condition = nu
         );
         $this->applyFilter($release, $items, $functions);
 
+        $release = '7.0.2';       // not yet
+        $items = array(
+            'opcache_invalidate'               => array('5.2.0', ''),
+        );
+        $this->applyFilter($release, $items, $functions);
+
         return $functions;
     }
 }
-- 
1.8.1.5

