From 8df8c44ed7d416709e9c1aa1277b075216cde4be Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 10 May 2013 11:19:28 +0200
Subject: [PATCH] new SOAP_SSL_METHOD_* in php 5.5.0RC1

---
 PHP/CompatInfo/Reference/soap.php | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/PHP/CompatInfo/Reference/soap.php b/PHP/CompatInfo/Reference/soap.php
index 3c7d4c0..bd342a6 100644
--- a/PHP/CompatInfo/Reference/soap.php
+++ b/PHP/CompatInfo/Reference/soap.php
@@ -154,6 +154,10 @@ public function getConstants($extension = null, $version = null, $condition = nu
             'SOAP_ENC_OBJECT'               => array('5.0.0', ''),
             'SOAP_FUNCTIONS_ALL'            => array('5.0.0', ''),
             'SOAP_LITERAL'                  => array('5.0.0', ''),
+            'SOAP_SSL_METHOD_TLS'           => array('5.5.0-dev', ''),
+            'SOAP_SSL_METHOD_SSLv2'         => array('5.5.0-dev', ''),
+            'SOAP_SSL_METHOD_SSLv23'        => array('5.5.0-dev', ''),
+            'SOAP_SSL_METHOD_SSLv3'         => array('5.5.0-dev', ''),
             'SOAP_PERSISTENCE_REQUEST'      => array('5.0.0', ''),
             'SOAP_PERSISTENCE_SESSION'      => array('5.0.0', ''),
             'SOAP_RPC'                      => array('5.0.0', ''),
-- 
1.8.1.6

