diff -up ./bin/phpcompatinfo.json.dist.rpm ./bin/phpcompatinfo.json.dist
--- ./bin/phpcompatinfo.json.dist.rpm	2015-12-05 10:11:45.000000000 +0100
+++ ./bin/phpcompatinfo.json.dist	2015-12-07 09:08:39.785273332 +0100
@@ -6,6 +6,19 @@
         }
     ],
     "plugins": [
+        {
+            "name": "Cache",
+            "class": "Bartlett\\Reflect\\Plugin\\CachePlugin",
+            "options": {
+                "adapter": "DoctrineCacheAdapter",
+                "backend": {
+                    "class": "Doctrine\\Common\\Cache\\FilesystemCache",
+                    "args": [
+                        "%{HOME}/.cache/php-reflect"
+                    ]
+                }
+            }
+        }
     ],
     "analysers" : [
     ],
diff -up ./bin/phpcompatinfo.rpm ./bin/phpcompatinfo
--- ./bin/phpcompatinfo.rpm	2015-12-05 10:11:45.000000000 +0100
+++ ./bin/phpcompatinfo	2015-12-07 09:14:17.072783363 +0100
@@ -1,52 +1,21 @@
 #!/usr/bin/env php
 <?php
 $appName = 'phpCompatInfo';
+$vendorDir = '/usr/share/php';
 
-if (\Phar::running()) {
-    $vendorDir = 'phar://' . strtolower($appName) . '.phar/vendor';
-} else {
-    $baseDir   = dirname(__DIR__);
-    $vendorDir = $baseDir . '/vendor';
-
-    if (!file_exists($vendorDir . '/autoload.php')) {
-        $vendorDir = dirname(dirname($baseDir));
-    }
-
-    if (!getenv("BARTLETTRC")) {
-        putenv("BARTLETTRC=" . strtolower($appName) . '.json');
-    }
-}
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->setUseIncludePath(true);
-
-if (\Phar::running()) {
-    if ($home) {
-        // when optional resources are not embedded in phar distribution,
-        // avoid PHP Warnings, and lookup first in global vendor dir (COMPOSER_HOME) if exists
-        $classMapfiltered = $fallbackClassMap($loader->getClassMap(), 'Net_Growl');
-        $loader->addClassMap($classMapfiltered);
-
-        // try to find PEAR Net_Growl classes package in folder $fallbackNetGrowlDir (see phar-stub.php)
-        $loader->add(false, $fallbackNetGrowlDir);
-    }
+require_once $vendorDir . '/Bartlett/CompatInfo/autoload.php';
+if (!getenv("BARTLETTRC")) {
+     putenv("BARTLETTRC=" . strtolower($appName) . '.json');
 }
 
 if (PHP_SAPI !== 'cli') {
     return;
 }
 
-if (\Phar::running()) {
-    try {
-        Bartlett\CompatInfoDb\Environment::checkRequirements();
-    } catch (\RuntimeException $e) {
-        die ($e->getMessage());
-    }
-}
-
 use Bartlett\Reflect\Environment;
 use Bartlett\CompatInfo\Console\Application;
 
 Environment::setScanDir();
 
-$application = new Application($appName, '5.0');
+$application = new Application($appName, '@package_version@');
 $application->run();

