From bcd5f85c93653ea2ba06b7a246b0b4778217c0f9 Mon Sep 17 00:00:00 2001
From: Laurent Laville <laurent.laville@gmail.com>
Date: Wed, 7 Oct 2015 21:52:18 +0200
Subject: [PATCH] fix for issue GH-210

---
 src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php b/src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php
index a79f05e..3577ed5 100644
--- a/src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php
+++ b/src/Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php
@@ -1221,5 +1221,10 @@ private function computeInternalVersions(Node $node, $element, $context, $extra
 
         // update parent context
         $this->updateContextVersion($versions);
+
+        if ($versions['ext.name'] !== 'user') {
+            // update versions of extension's $element
+            $this->updateElementVersion('extensions', $versions['ext.name'], $versions);
+        }
     }
 }
