diff -up ./bin/phpcompatinfo.json.dist.rpm ./bin/phpcompatinfo.json.dist
--- ./bin/phpcompatinfo.json.dist.rpm	2014-08-22 08:17:40.045281815 +0200
+++ ./bin/phpcompatinfo.json.dist	2014-08-22 08:17:46.784308376 +0200
@@ -9,6 +9,19 @@
         {
             "name": "Analyser",
             "class": "Bartlett\\Reflect\\Plugin\\Analyser\\AnalyserPlugin"
+        },
+        {
+            "name": "Cache",
+            "class": "Bartlett\\Reflect\\Plugin\\Cache\\CachePlugin",
+            "options": {
+                "adapter": "DoctrineCacheAdapter",
+                "backend": {
+                    "class": "Doctrine\\Common\\Cache\\FilesystemCache",
+                    "args": [
+                        "%{HOME}/.cache/phpreflect"
+                    ]
+                }
+            }
         }
     ],
     "analysers" : [
diff -up ./bin/phpcompatinfo.rpm ./bin/phpcompatinfo
--- ./bin/phpcompatinfo.rpm	2014-08-21 21:09:22.000000000 +0200
+++ ./bin/phpcompatinfo	2014-08-22 08:28:20.750827392 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 ini_set('memory_limit', -1);
 
@@ -23,7 +23,27 @@ if (\Phar::running()) {
         }
     }
 }
-require_once $vendorDir . '/autoload.php';
+$vendorDir = '/usr/share/php';
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/PHP/Timer.php';
+require_once $vendorDir . '/SebastianBergmann/Version/Version.php';
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Doctrine\\Common\\Cache'             => $vendorDir,
+    'Symfony\\Component\\Finder'          => $vendorDir,
+    'Symfony\\Component\\EventDispatcher' => $vendorDir,
+    'Symfony\\Component\\Console'         => $vendorDir,
+    'Symfony\\Component\\ClassLoader'     => $vendorDir,
+    'Seld\\JsonLint'                      => $vendorDir,
+    'Bartlett'                            => $vendorDir,
+));
+$loader->register();
 
 if (PHP_SAPI !== 'cli') {
     return;
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2014-08-21 21:09:22.000000000 +0200
+++ ./tests/bootstrap.php	2014-08-22 08:30:12.203270207 +0200
@@ -1,8 +1,30 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-require_once $vendorDir . '/autoload.php';
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
 
+require_once $vendorDir . '/PHP/Timer.php';
+require_once $vendorDir . '/SebastianBergmann/Version/Version.php';
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Symfony\\Component\\Finder'          => $vendorDir,
+    'Symfony\\Component\\EventDispatcher' => $vendorDir,
+    'Symfony\\Component\\Console'         => $vendorDir,
+    'Symfony\\Component\\ClassLoader'     => $vendorDir,
+    'Bartlett\\Reflect'                   => $vendorDir,
+    'Bartlett\\Tests\\CompatInfo'         => $baseDir . '/tests',
+    'Bartlett'                            => $baseDir . '/src',
+ ));
+
+$loader->register();
+
+# Class not catched by autoloader
+require $vendorDir . '/Bartlett/Reflect.php';
 require __DIR__ . '/Reference/GenericTest.php';
