From b504dcdd1a4d1b976a9a63631237685ee886907a Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 20 Sep 2013 10:59:22 +0200
Subject: [PATCH] new constant in Jsonc extension

---
 tests/Reference/JsonTest.php | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/Reference/JsonTest.php b/tests/Reference/JsonTest.php
index ab6aeca..d6976b2 100644
--- a/tests/Reference/JsonTest.php
+++ b/tests/Reference/JsonTest.php
@@ -43,6 +43,8 @@ protected function setUp()
     {
         // New features of JSONC alternative extension
         $this->ignoredconstants = array(
+            'JSON_C_BUNDLED',
+            'JSON_C_VERSION',
             'JSON_PARSER_NOTSTRICT',
         );
         $this->ignoredclasses = array(
-- 
1.8.4

