--- PHP_CompatInfo-2.14.0/PHP/CompatInfo/Report/Extension.php	2013-03-16 10:58:51.000000000 +0100
+++ PHP_CompatInfo-2.14.0/PHP/CompatInfo/Report/Extension.php	2013-03-18 16:36:30.001438567 +0100
@@ -45,7 +45,7 @@
             $this->totalConditions = 0;
             $this->globalVersions  = array('4.0.0', '');
 
-            if ($this->isEmpty($elements)) {
+            if ($this->isEmpty($extensions)) {
                 return;
             }
             $this->printTHeader($base, false, $extensions);
