diff -up ./phpcpd.rpm ./phpcpd
--- ./phpcpd.rpm	2015-03-26 19:04:39.896182549 +0100
+++ ./phpcpd	2015-03-26 19:05:35.300399520 +0100
@@ -15,23 +15,7 @@ if (ini_get('mbstring.internal_encoding'
     ini_set('mbstring.internal_encoding', NULL);
 }
 
-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
-}
-
-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require '/usr/share/php/SebastianBergmann/PHPCPD/autoload.php';
 
 $application = new SebastianBergmann\PHPCPD\CLI\Application;
 $application->run();
