diff -Naur PHPMailer_v5.1.orig/class.phpmailer.php PHPMailer_v5.1.new/class.phpmailer.php
--- PHPMailer_v5.1.orig/class.phpmailer.php	2009-11-11 01:04:14.000000000 +0100
+++ PHPMailer_v5.1.new/class.phpmailer.php	2010-01-15 18:29:55.000000000 +0100
@@ -1467,12 +1467,12 @@
         }
       }
       if (PHP_VERSION < 6) {
-        $magic_quotes = get_magic_quotes_runtime();
-        set_magic_quotes_runtime(0);
+        $magic_quotes = @get_magic_quotes_runtime();
+        @set_magic_quotes_runtime(0);
       }
       $file_buffer  = file_get_contents($path);
       $file_buffer  = $this->EncodeString($file_buffer, $encoding);
-      if (PHP_VERSION < 6) { set_magic_quotes_runtime($magic_quotes); }
+      if (PHP_VERSION < 6) { @set_magic_quotes_runtime($magic_quotes); }
       return $file_buffer;
     } catch (Exception $e) {
       $this->SetError($e->getMessage());
@@ -2317,4 +2317,4 @@
     return $errorMsg;
   }
 }
-?>
\ No newline at end of file
+?>
diff -Naur PHPMailer_v5.1.orig/docs/use_gmail.txt PHPMailer_v5.1.new/docs/use_gmail.txt
--- PHPMailer_v5.1.orig/docs/use_gmail.txt	2009-05-24 07:48:46.000000000 +0200
+++ PHPMailer_v5.1.new/docs/use_gmail.txt	2010-01-15 18:29:14.000000000 +0100
@@ -8,7 +8,7 @@
 $mail             = new PHPMailer();
 
 $body             = $mail->getFile('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->IsSMTP();
 $mail->SMTPAuth   = true;                  // enable SMTP authentication
diff -Naur PHPMailer_v5.1.orig/examples/test_db_smtp_basic.php PHPMailer_v5.1.new/examples/test_db_smtp_basic.php
--- PHPMailer_v5.1.orig/examples/test_db_smtp_basic.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_db_smtp_basic.php	2010-01-15 18:29:14.000000000 +0100
@@ -17,7 +17,7 @@
 $mail                = new PHPMailer();
 
 $body                = file_get_contents('contents.html');
-$body                = eregi_replace("[\]",'',$body);
+$body                = preg_replace('/\\\\/','',$body);
 
 $mail->IsSMTP(); // telling the class to use SMTP
 $mail->Host          = "smtp1.site.com;smtp2.site.com";
diff -Naur PHPMailer_v5.1.orig/examples/test_mail_basic.php PHPMailer_v5.1.new/examples/test_mail_basic.php
--- PHPMailer_v5.1.orig/examples/test_mail_basic.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_mail_basic.php	2010-01-15 18:29:14.000000000 +0100
@@ -11,7 +11,7 @@
 $mail             = new PHPMailer(); // defaults to using php "mail()"
 
 $body             = file_get_contents('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->AddReplyTo("name@yourdomain.com","First Last");
 
diff -Naur PHPMailer_v5.1.orig/examples/test_pop_before_smtp_basic.php PHPMailer_v5.1.new/examples/test_pop_before_smtp_basic.php
--- PHPMailer_v5.1.orig/examples/test_pop_before_smtp_basic.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_pop_before_smtp_basic.php	2010-01-15 18:29:14.000000000 +0100
@@ -14,7 +14,7 @@
 $mail = new PHPMailer();
 
 $body             = file_get_contents('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->IsSMTP();
 $mail->SMTPDebug = 2;
diff -Naur PHPMailer_v5.1.orig/examples/test_sendmail_basic.php PHPMailer_v5.1.new/examples/test_sendmail_basic.php
--- PHPMailer_v5.1.orig/examples/test_sendmail_basic.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_sendmail_basic.php	2010-01-15 18:29:14.000000000 +0100
@@ -13,7 +13,7 @@
 $mail->IsSendmail(); // telling the class to use SendMail transport
 
 $body             = file_get_contents('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->AddReplyTo("name@yourdomain.com","First Last");
 
diff -Naur PHPMailer_v5.1.orig/examples/test_smtp_basic_no_auth.php PHPMailer_v5.1.new/examples/test_smtp_basic_no_auth.php
--- PHPMailer_v5.1.orig/examples/test_smtp_basic_no_auth.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_smtp_basic_no_auth.php	2010-01-15 18:29:14.000000000 +0100
@@ -17,7 +17,7 @@
 $mail             = new PHPMailer();
 
 $body             = file_get_contents('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->IsSMTP(); // telling the class to use SMTP
 $mail->Host       = "mail.yourdomain.com"; // SMTP server
diff -Naur PHPMailer_v5.1.orig/examples/test_smtp_basic.php PHPMailer_v5.1.new/examples/test_smtp_basic.php
--- PHPMailer_v5.1.orig/examples/test_smtp_basic.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_smtp_basic.php	2010-01-15 18:29:14.000000000 +0100
@@ -17,7 +17,7 @@
 $mail             = new PHPMailer();
 
 $body             = file_get_contents('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->IsSMTP(); // telling the class to use SMTP
 $mail->Host       = "mail.yourdomain.com"; // SMTP server
diff -Naur PHPMailer_v5.1.orig/examples/test_smtp_gmail_basic.php PHPMailer_v5.1.new/examples/test_smtp_gmail_basic.php
--- PHPMailer_v5.1.orig/examples/test_smtp_gmail_basic.php	2009-05-24 07:48:48.000000000 +0200
+++ PHPMailer_v5.1.new/examples/test_smtp_gmail_basic.php	2010-01-15 18:29:14.000000000 +0100
@@ -17,7 +17,7 @@
 $mail             = new PHPMailer();
 
 $body             = file_get_contents('contents.html');
-$body             = eregi_replace("[\]",'',$body);
+$body             = preg_replace('/\\\\/','',$body);
 
 $mail->IsSMTP(); // telling the class to use SMTP
 $mail->Host       = "mail.yourdomain.com"; // SMTP server
