diff -up src/lib/Herrera/Annotations/Convert/ToXml.php.rpm src/lib/Herrera/Annotations/Convert/ToXml.php
--- src/lib/Herrera/Annotations/Convert/ToXml.php.rpm	2015-10-27 16:58:31.834414038 +0100
+++ src/lib/Herrera/Annotations/Convert/ToXml.php	2015-10-27 16:59:44.395738935 +0100
@@ -16,7 +16,7 @@ use Herrera\Annotations\Tokens;
  */
 define(
     'HERRERA_ANNOTATIONS_SCHEMA',
-    __DIR__ . '/../../../../../res/annotations.xsd'
+    (getenv('HERRERA_ANNOTATIONS_SCHEMA') ?: '/usr/share/php-herrera-io-annotations/res/annotations.xsd')
 );
 
 /**
