From d5878063d8c654419f27d958134a1bf220e2344b Mon Sep 17 00:00:00 2001
From: Xinchen Hui <laruence@gmail.com>
Date: Sun, 1 Jun 2014 19:57:31 +0800
Subject: [PATCH] Fixed bug #67291 (Compile Error with PHP-5.6)

---
 yaf_config.c | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/yaf_config.c b/yaf_config.c
index e4a56b9..c1f1997 100644
--- a/yaf_config.c
+++ b/yaf_config.c
@@ -77,8 +77,10 @@ static void yaf_config_zval_dtor(zval **value) {
 				pefree((*value)->value.str.val, 1);
 				pefree(*value, 1);
 				break;
-			case IS_ARRAY:
-			case IS_CONSTANT_ARRAY: {
+#if ((PHP_MAJOR_VERSION == 5) && (PHP_MINOR_VERSION < 6))
+			case IS_CONSTANT_ARRAY:
+#endif
+			case IS_ARRAY: {
 				zend_hash_destroy((*value)->value.ht);
 				pefree((*value)->value.ht, 1);
 				pefree(*value, 1);
@@ -179,8 +181,10 @@ static zval * yaf_config_ini_zval_persistent(zval *zvalue TSRMLS_DC) {
 				ret->value.str.val = pestrndup(zvalue->value.str.val, zvalue->value.str.len, 1);
 				ret->value.str.len = zvalue->value.str.len;
 			break;
-		case IS_ARRAY:
-		case IS_CONSTANT_ARRAY: {
+#if ((PHP_MAJOR_VERSION == 5) && (PHP_MINOR_VERSION < 6))
+		case IS_CONSTANT_ARRAY:
+#endif
+		case IS_ARRAY: {
 				HashTable *tmp_ht, *original_ht = zvalue->value.ht;
 
 				tmp_ht = (HashTable *)pemalloc(sizeof(HashTable), 1);
@@ -218,8 +222,10 @@ static zval * yaf_config_ini_zval_losable(zval *zvalue TSRMLS_DC) {
 			CHECK_ZVAL_STRING(zvalue);
 			ZVAL_STRINGL(ret, zvalue->value.str.val, zvalue->value.str.len, 1);
 			break;
-		case IS_ARRAY:
-		case IS_CONSTANT_ARRAY: {
+#if ((PHP_MAJOR_VERSION == 5) && (PHP_MINOR_VERSION < 6))
+		case IS_CONSTANT_ARRAY:
+#endif
+		case IS_ARRAY: {
 			HashTable *original_ht = zvalue->value.ht;
 			array_init(ret);
 			yaf_config_copy_losable(Z_ARRVAL_P(ret), original_ht TSRMLS_CC);
-- 
1.9.3

