diff -up mozilla-beta/modules/libpr0n/decoders/nsJPEGDecoder.cpp.turbo mozilla-beta/modules/libpr0n/decoders/nsJPEGDecoder.cpp
--- mozilla-beta/modules/libpr0n/decoders/nsJPEGDecoder.cpp.turbo	2011-05-27 18:12:24.000000000 +0200
+++ mozilla-beta/modules/libpr0n/decoders/nsJPEGDecoder.cpp	2011-06-01 19:08:04.414802739 +0200
@@ -430,14 +430,6 @@ nsJPEGDecoder::WriteInternal(const char 
       return; /* I/O suspension */
     }
 
-    /* Force to use our YCbCr to Packed RGB converter when possible */
-    if (!mTransform && (mCMSMode != eCMSMode_All) &&
-        mInfo.jpeg_color_space == JCS_YCbCr && mInfo.out_color_space == JCS_RGB) {
-      /* Special case for the most common case: transform from YCbCr direct into packed ARGB */
-      mInfo.out_color_components = 4; /* Packed ARGB pixels are always 4 bytes...*/
-      mInfo.cconvert->color_convert = ycc_rgb_convert_argb;
-    }
-
     /* If this is a progressive JPEG ... */
     mState = mInfo.buffered_image ? JPEG_DECOMPRESS_PROGRESSIVE : JPEG_DECOMPRESS_SEQUENTIAL;
   }
@@ -582,15 +574,6 @@ nsJPEGDecoder::OutputScanlines(PRBool* s
       PRUint32 *imageRow = ((PRUint32*)mImageData) +
                            (mInfo.output_scanline * mInfo.output_width);
 
-      if (mInfo.cconvert->color_convert == ycc_rgb_convert_argb) {
-        /* Special case: scanline will be directly converted into packed ARGB */
-        if (jpeg_read_scanlines(&mInfo, (JSAMPARRAY)&imageRow, 1) != 1) {
-          *suspend = PR_TRUE; /* suspend */
-          break;
-        }
-        continue; /* all done for this row! */
-      }
-
       JSAMPROW sampleRow = (JSAMPROW)imageRow;
       if (mInfo.output_components == 3) {
         /* Put the pixels at end of row to enable in-place expansion */
