From 1959c39f0ee575794f9c6b06b89e18b0081f7564 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Thu, 2 Apr 2015 17:42:49 +0200
Subject: [PATCH] fix typo in rev ea710a2fbba928c2f8af7c9d1b7027e64fa9d4aa

---
 config.m4 | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.m4 b/config.m4
index 96ab23c..2a7fe2e 100644
--- a/config.m4
+++ b/config.m4
@@ -31,12 +31,12 @@ if test "$PHP_LIBSODIUM" != "no"; then
 
   PHP_CHECK_LIBRARY($LIBNAME,$LIBSYMBOL,
   [
-    PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $LIBSODIUM_DIR/$PHP_LIB, LIBSODIUM_SHARED_LIBADD)
+    PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $LIBSODIUM_DIR/$PHP_LIBDIR, LIBSODIUM_SHARED_LIBADD)
     AC_DEFINE(HAVE_LIBSODIUMLIB,1,[ ])
   ],[
     AC_MSG_ERROR([wrong libsodium lib version or lib not found])
   ],[
-    -L$LIBSODIUM_DIR/$PHP_LIB
+    -L$LIBSODIUM_DIR/$PHP_LIBDIR
   ])
   
   PHP_SUBST(LIBSODIUM_SHARED_LIBADD)
