--- Makefile.in.ol	2016-01-21 10:52:15.347439341 +0100
+++ Makefile.in	2016-01-21 10:52:45.601555239 +0100
@@ -1869,7 +1869,7 @@
 	    $(INSTALL_DATA) "$$file" "$(DESTDIR)$(man3dir)/$$inst" || exit $$?; \
 	  fi; \
 	done; \
-	for i in $$list; do echo "$$i"; done | $(am__base_list) | \
+	for i in $$list; do echo "$$i"; done | sort -u | $(am__base_list) | \
 	while read files; do \
 	  test -z "$$files" || { \
 	    echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(man3dir)'"; \
