From 419834a441cc65bb4e7220e1128356dc99d86211 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 22 Apr 2015 08:49:51 +0200
Subject: [PATCH] fix usage of virtual_file_ex

---
 usefulstuff.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/usefulstuff.c b/usefulstuff.c
index 83725c7..3bc4d1a 100644
--- a/usefulstuff.c
+++ b/usefulstuff.c
@@ -344,7 +344,11 @@ char *xdebug_path_to_url(const char *fileurl TSRMLS_DC)
 			cwd[0] = '\0';
 		}
 
+#if PHP_VERSION_ID < 50600
 		new_state.cwd = strdup(cwd);
+#else
+		new_state.cwd = estrdup(cwd);
+#endif
 		new_state.cwd_length = strlen(cwd);
 
 		if (!virtual_file_ex(&new_state, fileurl, NULL, 1 TSRMLS_CC)) {
@@ -352,7 +356,11 @@ char *xdebug_path_to_url(const char *fileurl TSRMLS_DC)
 			tmp = xdebug_sprintf("file://%s",s);
 			efree(s);
 		}
+#if PHP_VERSION_ID < 50600
 		free(new_state.cwd);
+#else
+		efree(new_state.cwd);
+#endif
 
 	} else if (fileurl[1] == '/' || fileurl[1] == '\\') {
 		/* convert UNC paths (eg. \\server\sharepath) */
