From 904388b5cc122a3331120b73aaa987aa8b70f84d Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 8 Sep 2014 18:54:42 +0200
Subject: [PATCH] fix #10 segfault in ZTS test suite

---
 loop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loop.c b/loop.c
index c180b7f..85c3f10 100644
--- a/loop.c
+++ b/loop.c
@@ -64,7 +64,7 @@ static void php_ev_loop_object_ctor(INTERNAL_FUNCTION_PARAMETERS, const zend_boo
 
 		/* Save return_value in MyG(default_loop) */
 		if (is_default_loop && !*default_loop_ptr_ptr) {
-			MAKE_STD_ZVAL(*default_loop_ptr_ptr);
+			ALLOC_INIT_ZVAL(*default_loop_ptr_ptr);
 			REPLACE_ZVAL_VALUE(default_loop_ptr_ptr, return_value, 1);
 		}
 	} else {
-- 
1.8.5.2

