diff -up tests/KdybyTests/bootstrap.php.old tests/KdybyTests/bootstrap.php
--- tests/KdybyTests/bootstrap.php.old	2016-08-04 16:10:59.039641276 +0200
+++ tests/KdybyTests/bootstrap.php	2016-08-04 16:11:04.206667988 +0200
@@ -33,5 +33,5 @@ function id($val) {
 }
 
 function run(Tester\TestCase $testCase) {
-	$testCase->run(isset($_SERVER['argv'][1]) ? $_SERVER['argv'][1] : NULL);
+	$testCase->run();
 }
