diff -up Security-2.3.6/Symfony/Component/Security/Core/Encoder/BCryptPasswordEncoder.php.old Security-2.3.6/Symfony/Component/Security/Core/Encoder/BCryptPasswordEncoder.php
--- Security-2.3.6/Symfony/Component/Security/Core/Encoder/BCryptPasswordEncoder.php.old	2013-10-18 18:05:13.000000000 +0200
+++ Security-2.3.6/Symfony/Component/Security/Core/Encoder/BCryptPasswordEncoder.php	2013-10-18 18:09:03.000000000 +0200
@@ -14,6 +14,8 @@ namespace Symfony\Component\Security\Cor
 use Symfony\Component\Security\Core\Encoder\BasePasswordEncoder;
 use Symfony\Component\Security\Core\Exception\BadCredentialsException;
 
+require_once "password_compat/password.php";
+
 /**
  * @author Elnur Abdurrakhimov <elnur@elnur.pro>
  * @author Terje Bråten <terje@braten.be>
