diff --git a/fusinvdeploy/front/group.form.php b/fusinvdeploy/front/group.form.php
index 755bce2..2d68640 100644
--- a/fusinvdeploy/front/group.form.php
+++ b/fusinvdeploy/front/group.form.php
@@ -157,7 +157,7 @@ if (isset($_POST["add"])) {
    Html::back();
 
 } else {
-   Html::header($LANG['plugin_fusinvdeploy']["title"][0],$_SERVER["PHP_SELF"],"plugins",
+   Html::header($LANG['plugin_fusinvdeploy']['title'][0],$_SERVER["PHP_SELF"],"plugins",
    "fusioninventory","group");
 
    PluginFusioninventoryMenu::displayMenu("mini");
diff --git a/fusinvdeploy/front/mirror.php b/fusinvdeploy/front/mirror.php
index be01ebb..9346aa7 100644
--- a/fusinvdeploy/front/mirror.php
+++ b/fusinvdeploy/front/mirror.php
@@ -47,7 +47,7 @@ if (!defined('GLPI_ROOT')) {
 include (GLPI_ROOT."/inc/includes.php");
 Session::checkLoginUser();
 
-Html::header($LANG['plugin_fusinvdeploy']['form']['mirror'][2],$_SERVER["PHP_SELF"],"plugins",
+Html::header($LANG['plugin_fusinvdeploy']['form']['mirror'][1],$_SERVER["PHP_SELF"],"plugins",
              "fusioninventory","mirror");
 
 //PluginFusioninventoryProfile::checkRight("Fusioninventory", "agents","r");
diff --git a/fusinvdeploy/front/package.form.php b/fusinvdeploy/front/package.form.php
index 867d3c6..3acde57 100644
--- a/fusinvdeploy/front/package.form.php
+++ b/fusinvdeploy/front/package.form.php
@@ -44,7 +44,7 @@ define('GLPI_ROOT', '../../..');
 include (GLPI_ROOT . "/inc/includes.php");
 Session::checkLoginUser();
 
-Html::header($LANG['plugin_fusinvdeploy']["title"][0],$_SERVER["PHP_SELF"],"plugins",
+Html::header($LANG['plugin_fusinvdeploy']['title'][0],$_SERVER["PHP_SELF"],"plugins",
    "fusioninventory","packages");
 
 //PluginFusioninventoryProfile::checkRight("Fusioninventory", "agents","r");
diff --git a/fusinvdeploy/front/task.form.php b/fusinvdeploy/front/task.form.php
index d2e7a9a..c8cf8c6 100644
--- a/fusinvdeploy/front/task.form.php
+++ b/fusinvdeploy/front/task.form.php
@@ -92,7 +92,7 @@ if (isset($_POST['forcestart'])) {
    Html::back();
 
 } else {
-   Html::header($LANG['plugin_fusinvdeploy']["title"][0],$_SERVER["PHP_SELF"],"plugins",
+   Html::header($LANG['plugin_fusinvdeploy']['title'][0],$_SERVER["PHP_SELF"],"plugins",
    "fusioninventory","task");
 
    PluginFusioninventoryMenu::displayMenu("mini");
diff --git a/fusinvdeploy/hook.php b/fusinvdeploy/hook.php
index e55a49b..1f8959c 100644
--- a/fusinvdeploy/hook.php
+++ b/fusinvdeploy/hook.php
@@ -101,7 +101,7 @@ function plugin_fusinvdeploy_MassiveActionsDisplay($options=array()) {
       case 'Computer' :
          switch ($options['action']) {
              case 'plugin_fusinvdeploy_targetDeployTask' :
-               echo "<br/>". $LANG['plugin_fusinvdeploy']['task'][5].":&nbsp;";
+               echo "<br/>". $LANG['plugin_fusinvdeploy']['task'][1].":&nbsp;";
                $rand = mt_rand();
                Dropdown::show('PluginFusinvdeployTask', array(
                      'name'      => "tasks_id",
diff --git a/fusinvdeploy/inc/action.class.php b/fusinvdeploy/inc/action.class.php
index 3b91afe..4d2c162 100644
--- a/fusinvdeploy/inc/action.class.php
+++ b/fusinvdeploy/inc/action.class.php
@@ -147,7 +147,7 @@ class PluginFusinvdeployAction extends CommonDBTM {
             $row[$render.'value'].= " <b>".$LANG['plugin_fusinvdeploy']['form']['action_message'][2].
                " : </b> ";
             $row[$render.'value'].= $action->getField('message');
-            $row[$render.'value'].= " <b>".$LANG['plugin_fusinvdeploy']['form']['action_message'][3].
+            $row[$render.'value'].= " <b>".$LANG['plugin_fusinvdeploy']['form']['label'][0].
                " : </b> ";
             $row[$render.'value'].= $action->getField('type');
 
@@ -229,7 +229,7 @@ class PluginFusinvdeployAction extends CommonDBTM {
          $row[$render.'value'].= " <b>".$LANG['plugin_fusinvdeploy']['form']['action_message'][2].
             " : </b> ";
          $row[$render.'value'].= $action->getField('message');
-         $row[$render.'value'].= " <b>".$LANG['plugin_fusinvdeploy']['form']['action_message'][3].
+         $row[$render.'value'].= " <b>".$LANG['plugin_fusinvdeploy']['form']['label'][0].
             " : </b> ";
          $row[$render.'value'].= $action->getField('type');
 
diff --git a/fusinvdeploy/inc/config.class.php b/fusinvdeploy/inc/config.class.php
index 81dd1d0..abc27d6 100644
--- a/fusinvdeploy/inc/config.class.php
+++ b/fusinvdeploy/inc/config.class.php
@@ -111,7 +111,7 @@ class PluginFusinvdeployConfig extends CommonDBTM {
       echo "<table class='tab_cadre_fixe'>";
 
 //      echo "<tr class='tab_bg_1'>";
-//      echo "<td>".$LANG['plugin_fusinvdeploy']["config"][0]."&nbsp;:</td>";
+//      echo "<td>".$LANG['plugin_fusinvdeploy']['config'][0]."&nbsp;:</td>";
 //      echo "<td>";
 //      Html::autocompletionTextField($config, 'glpi_path', array(
 //         'name'   => 'glpi_path',
diff --git a/fusinvdeploy/inc/group.class.php b/fusinvdeploy/inc/group.class.php
index aa171c4..f4e4539 100644
--- a/fusinvdeploy/inc/group.class.php
+++ b/fusinvdeploy/inc/group.class.php
@@ -52,7 +52,7 @@ class PluginFusinvdeployGroup extends CommonDBTM {
       global $LANG;
 
       if ($nb>1) {
-         return $LANG['plugin_fusinvdeploy']['task'][5];
+         return $LANG['plugin_fusinvdeploy']['task'][1];
       }
       return $LANG['plugin_fusinvdeploy']['group'][0];
    }
diff --git a/fusinvdeploy/inc/staticmisc.class.php b/fusinvdeploy/inc/staticmisc.class.php
index 352db20..e3e9151 100644
--- a/fusinvdeploy/inc/staticmisc.class.php
+++ b/fusinvdeploy/inc/staticmisc.class.php
@@ -192,7 +192,7 @@ class PluginFusinvdeployStaticmisc {
 
       $a_menu = array();
       if (PluginFusioninventoryProfile::haveRight("fusinvdeploy", "packages", "r")) {
-         $a_menu[0]['name'] = $LANG['plugin_fusinvdeploy']["package"][6];
+         $a_menu[0]['name'] = $LANG['plugin_fusinvdeploy']['package'][6];
          $a_menu[0]['pic']  = GLPI_ROOT."/plugins/fusinvdeploy/pics/menu_package.png";
          $a_menu[0]['link'] = GLPI_ROOT."/plugins/fusinvdeploy/front/package.php";
       }
@@ -215,7 +215,7 @@ class PluginFusinvdeployStaticmisc {
       return array(array('profil'  => 'packages',
                          'name'    => $LANG['plugin_fusinvdeploy']['profile'][2]),
                    array('profil'  => 'status',
-                         'name'    => $LANG['plugin_fusinvdeploy']['profile'][3]));
+                         'name'    => $LANG['plugin_fusinvdeploy']['deploystatus'][0]));
    }
 
    static function task_deploy_getParameters() {
diff --git a/fusinvdeploy/js/package_action.front.php b/fusinvdeploy/js/package_action.front.php
index 7e7d959..0118392 100644
--- a/fusinvdeploy/js/package_action.front.php
+++ b/fusinvdeploy/js/package_action.front.php
@@ -150,7 +150,7 @@ var {$render}actionColumns =  [{
 }, {
    id: '{$render}messagetype',
    hidden:true,
-   header: '{$LANG['plugin_fusinvdeploy']['form']['action_message'][3]}',
+   header: '{$LANG['plugin_fusinvdeploy']['form']['label'][0]}',
    width: {$column_width[12]},
    dataIndex: '{$render}messagetype'
 }, {
@@ -359,7 +359,7 @@ var {$render}actionGridProxyRetChecks = {
                dataIndex: 'id',
                hidden: true
             }, {
-               header: '{$LANG['plugin_fusinvdeploy']['form']['command_status'][1]}',
+               header: '{$LANG['plugin_fusinvdeploy']['form']['label'][0]}',
                dataIndex: 'type',
                width: 180,
                renderer: function(val) {
@@ -390,7 +390,7 @@ var {$render}actionGridProxyRetChecks = {
                   })
                })
             }, {
-               header: '{$LANG['plugin_fusinvdeploy']['form']['command_status'][2]}',
+               header: '{$LANG['plugin_fusinvdeploy']['form']['label'][2]}',
                dataIndex: 'value',
                allowBlank: false,
                width: 100,
@@ -509,7 +509,7 @@ var {$render}Command_fieldset_item_PluginFusinvdeployAction_Message = [{
       width: {$field_width},
       height : {$field_height}
    } , {
-      fieldLabel: '{$LANG['plugin_fusinvdeploy']['form']['action_message'][3]}',
+      fieldLabel: '{$LANG['plugin_fusinvdeploy']['form']['label'][0]}',
       name: '{$render}messagetype',
       hiddenName : '{$render}messagetype',
       xtype : 'combo',
diff --git a/fusinvdeploy/js/package_check.front.php b/fusinvdeploy/js/package_check.front.php
index 7c7b4d6..02f35ce 100644
--- a/fusinvdeploy/js/package_check.front.php
+++ b/fusinvdeploy/js/package_check.front.php
@@ -121,7 +121,7 @@ function {$render}renderType(val) {
       case '{$chkConst['FILE_EXISTS']}':
          return '{$LANG['plugin_fusinvdeploy']['form']['check'][3]}';
       case '{$chkConst['FILE_MISSING']}':
-         return '{$LANG['plugin_fusinvdeploy']['form']['check'][4]}';
+         return '{$LANG['plugin_fusinvdeploy']['form']['label'][15]}';
       case '{$chkConst['FILE_SIZEGREATER']}':
          return '{$LANG['plugin_fusinvdeploy']['form']['check'][5]}';
       case '{$chkConst['FILE_SIZEEQUAL']}':
@@ -459,7 +459,7 @@ var {$render}checkForm = new Ext.FormPanel({
             ['{$chkConst['WINKEY_MISSING']}',  '{$LANG['plugin_fusinvdeploy']['form']['check'][1]}'],
             ['{$chkConst['WINKEY_EQUAL']}',    '{$LANG['plugin_fusinvdeploy']['form']['check'][2]}'],
             ['{$chkConst['FILE_EXISTS']}',    '{$LANG['plugin_fusinvdeploy']['form']['check'][3]}'],
-            ['{$chkConst['FILE_MISSING']}',    '{$LANG['plugin_fusinvdeploy']['form']['check'][4]}'],
+            ['{$chkConst['FILE_MISSING']}',    '{$LANG['plugin_fusinvdeploy']['form']['label'][15]}'],
             ['{$chkConst['FILE_SIZEGREATER']}','{$LANG['plugin_fusinvdeploy']['form']['check'][5]}'],
             ['{$chkConst['FILE_SIZEEQUAL']}',  '{$LANG['plugin_fusinvdeploy']['form']['check'][8]}'],
             ['{$chkConst['FILE_SIZELOWER']}',  '{$LANG['plugin_fusinvdeploy']['form']['check'][9]}'],
diff --git a/fusinvdeploy/js/task_job.front.php b/fusinvdeploy/js/task_job.front.php
index 9374094..ced2636 100644
--- a/fusinvdeploy/js/task_job.front.php
+++ b/fusinvdeploy/js/task_job.front.php
@@ -236,7 +236,7 @@ var taskJobGrid = new Ext.grid.GridPanel({
       emptyGroupText: ''
    }),
    tbar: [{
-      text: '{$LANG['plugin_fusinvdeploy']['form']['title'][10]}',
+      text: '{$LANG['plugin_fusinvdeploy']['task'][15]}',
       iconCls: 'exticon-add',
       handler: function(btn,ev) {
          var u = new taskJobStore.recordType({
@@ -251,10 +251,10 @@ var taskJobGrid = new Ext.grid.GridPanel({
          });
          taskJobStore.insert(0,u);
          taskJobGrid.getSelectionModel().selectFirstRow();
-         taskJobForm.setTitle('{$LANG['plugin_fusinvdeploy']['form']['title'][10]}');
+         taskJobForm.setTitle('{$LANG['plugin_fusinvdeploy']['task'][15]}');
       }
    }, '-', {
-      text: '{$LANG['plugin_fusinvdeploy']['form']['title'][11]}',
+      text: '{$LANG['plugin_fusinvdeploy']['task'][16]}',
       iconCls: 'exticon-delete',
       handler: function(btn,ev) {
          var selection = taskJobGrid.getSelectionModel().getSelections();
@@ -293,7 +293,7 @@ var taskJobGrid = new Ext.grid.GridPanel({
          rowselect: function(g,index,ev) {
             var rec = taskJobGrid.store.getAt(index);
             taskJobForm.loadData(rec);
-            taskJobForm.setTitle('{$LANG['plugin_fusinvdeploy']['form']['title'][12]}');
+            taskJobForm.setTitle('{$LANG['plugin_fusinvdeploy']['task'][17]}');
             taskJobForm.buttons[0].setDisabled(false);
 
             taskJobForm.expand();
diff --git a/fusinvdeploy/locales/en_GB.php b/fusinvdeploy/locales/en_GB.php
index 59799f8..bfe3249 100644
--- a/fusinvdeploy/locales/en_GB.php
+++ b/fusinvdeploy/locales/en_GB.php
@@ -40,10 +40,7 @@
    ------------------------------------------------------------------------
  */
 
-$title="FusionInventory DEPLOY";
-$version="2.3.0-1";
-
-$LANG['plugin_fusinvdeploy']['title'][0]="$title";
+$LANG['plugin_fusinvdeploy']['title'][0]="FusionInventory DEPLOY";
 
 $LANG['plugin_fusinvdeploy']['massiveactions'][0]="Target a task";
 $LANG['plugin_fusinvdeploy']['massiveactions'][1]="Create a job for each computer";
@@ -57,12 +54,10 @@ $LANG['plugin_fusinvdeploy']['package'][4]="Store";
 $LANG['plugin_fusinvdeploy']['package'][5]="Packages";
 $LANG['plugin_fusinvdeploy']['package'][6]="Package management";
 $LANG['plugin_fusinvdeploy']['package'][7]="Package";
-$LANG['plugin_fusinvdeploy']['package'][8]="Package management";
 $LANG['plugin_fusinvdeploy']['package'][9]="Number of fragments";
 $LANG['plugin_fusinvdeploy']['package'][10]="Module";
 $LANG['plugin_fusinvdeploy']['package'][11]="Audits";
 $LANG['plugin_fusinvdeploy']['package'][12]="Files";
-$LANG['plugin_fusinvdeploy']['package'][13]="Actions";
 $LANG['plugin_fusinvdeploy']['package'][14]="Installation";
 $LANG['plugin_fusinvdeploy']['package'][15]="Uninstallation";
 $LANG['plugin_fusinvdeploy']['package'][16]="Package deployment";
@@ -103,17 +98,14 @@ $LANG['plugin_fusinvdeploy']['config'][0]="Address of the GLPI server (without h
 $LANG['plugin_fusinvdeploy']['config'][1]="Root folder for sending files from server";
 $LANG['plugin_fusinvdeploy']['config'][2]="Enable alerts on the size of MS Windows paths?";
 
-$LANG['plugin_fusinvdeploy']['setup'][17]="Plugin ".$title." needs FusionInventory plugin activated before activation.";
-$LANG['plugin_fusinvdeploy']['setup'][18]="Plugin ".$title." needs FusionInventory plugin activated before uninstall.";
-$LANG['plugin_fusinvdeploy']['setup'][19]="Plugin ".$title." needs Webservices plugin (>= 1.2.0) installed before activation.";
-$LANG['plugin_fusinvdeploy']['setup'][20]="Plugin ".$title." needs Webservices plugin (>= 1.2.0) installed before uninstall.";
-$LANG['plugin_fusinvdeploy']['setup'][21]="Plugin ".$title." needs FusionInventory INVENTORY plugin installed before activation.";
-$LANG['plugin_fusinvdeploy']['setup'][22]="Plugin ".$title." needs FusionInventory INVENTORY plugin installed before uninstall.";
+$LANG['plugin_fusinvdeploy']['setup'][17]="Plugin FusionInventory DEPLOY needs FusionInventory plugin activated before activation.";
+$LANG['plugin_fusinvdeploy']['setup'][18]="Plugin FusionInventory DEPLOY needs FusionInventory plugin activated before uninstall.";
+$LANG['plugin_fusinvdeploy']['setup'][19]="Plugin FusionInventory DEPLOY needs Webservices plugin (1.2.0 minimum) installed before activation.";
+$LANG['plugin_fusinvdeploy']['setup'][20]="Plugin FusionInventory DEPLOY needs Webservices plugin (1.2.0 minimum) installed before uninstall.";
+$LANG['plugin_fusinvdeploy']['setup'][21]="Plugin FusionInventory DEPLOY needs FusionInventory INVENTORY plugin installed before activation.";
+$LANG['plugin_fusinvdeploy']['setup'][22]="Plugin FusionInventory DEPLOY needs FusionInventory INVENTORY plugin installed before uninstall.";
 
-$LANG['plugin_fusinvdeploy']['profile'][1]="$title";
 $LANG['plugin_fusinvdeploy']['profile'][2]="Manage packages";
-$LANG['plugin_fusinvdeploy']['profile'][3]="Deployment status";
-
 
 $LANG['plugin_fusinvdeploy']['form']['label'][0] = "Type";
 $LANG['plugin_fusinvdeploy']['form']['label'][1] = "Name";
@@ -159,9 +151,6 @@ $LANG['plugin_fusinvdeploy']['form']['title'][6] = "Add command";
 $LANG['plugin_fusinvdeploy']['form']['title'][7] = "Edit command";
 $LANG['plugin_fusinvdeploy']['form']['title'][8] = "Actions to achieve";
 $LANG['plugin_fusinvdeploy']['form']['title'][9] = "Delete a check";
-$LANG['plugin_fusinvdeploy']['form']['title'][10] = "Add order";
-$LANG['plugin_fusinvdeploy']['form']['title'][11] = "Delete order";
-$LANG['plugin_fusinvdeploy']['form']['title'][12] = "Edit order";
 $LANG['plugin_fusinvdeploy']['form']['title'][13] = "Delete file";
 $LANG['plugin_fusinvdeploy']['form']['title'][14] = "Delete command";
 $LANG['plugin_fusinvdeploy']['form']['title'][15] = "during installation";
@@ -173,11 +162,7 @@ $LANG['plugin_fusinvdeploy']['form']['message'][0] = "Empty form";
 $LANG['plugin_fusinvdeploy']['form']['message'][1] = "Invalid form";
 $LANG['plugin_fusinvdeploy']['form']['message'][2] = "Loading...";
 $LANG['plugin_fusinvdeploy']['form']['message'][3] = "File already exist";
-$LANG['plugin_fusinvdeploy']['form']['message'][4] = "Paths on MS Windows do not accept more than 255 characters, "
-."the value you entered exceeds the limit.<br />"
-."<br /><b>Do you want to continue?</b><br />"
-."<br /><div class='message_smalltext right'>You can disable this message in the "
-."<a href='##URL##'>plugin configuration</a></div>";
+$LANG['plugin_fusinvdeploy']['form']['message'][4] = "Paths on MS Windows do not accept more than 255 characters, the value you entered exceeds the limit.<br /><br /><b>Do you want to continue?</b><br /><br /><div class='message_smalltext right'>You can disable this message in the <a href='##URL##'>plugin configuration</a></div>";
 $LANG['plugin_fusinvdeploy']['form']['message'][5] = "Attention";
 $LANG['plugin_fusinvdeploy']['form']['message'][6] = "Wish to the command to install and uninstall is automatically added for your file?";
 
@@ -185,7 +170,6 @@ $LANG['plugin_fusinvdeploy']['form']['check'][0] = "Register key exist";
 $LANG['plugin_fusinvdeploy']['form']['check'][1] = "Register key missing";
 $LANG['plugin_fusinvdeploy']['form']['check'][2] = "Register key value";
 $LANG['plugin_fusinvdeploy']['form']['check'][3] = "File exist";
-$LANG['plugin_fusinvdeploy']['form']['check'][4] = "File missing";
 $LANG['plugin_fusinvdeploy']['form']['check'][5] = "File size greater";
 $LANG['plugin_fusinvdeploy']['form']['check'][6] = "SHA-512 hash value";
 $LANG['plugin_fusinvdeploy']['form']['check'][7] = "Free space";
@@ -193,12 +177,9 @@ $LANG['plugin_fusinvdeploy']['form']['check'][8] = "Filesize equal to";
 $LANG['plugin_fusinvdeploy']['form']['check'][9] = "Filesize lower than";
 
 $LANG['plugin_fusinvdeploy']['form']['mirror'][1] = "Mirror servers";
-$LANG['plugin_fusinvdeploy']['form']['mirror'][2] = "Mirror servers";
 $LANG['plugin_fusinvdeploy']['form']['mirror'][3] = "Mirror server address";
 
 $LANG['plugin_fusinvdeploy']['form']['command_status'][0] = "Make your choice...";
-$LANG['plugin_fusinvdeploy']['form']['command_status'][1] = "Type";
-$LANG['plugin_fusinvdeploy']['form']['command_status'][2] = "Value";
 $LANG['plugin_fusinvdeploy']['form']['command_status'][3] = "Expected return code";
 $LANG['plugin_fusinvdeploy']['form']['command_status'][4] = "Invalid return code";
 $LANG['plugin_fusinvdeploy']['form']['command_status'][5] = "Expected regular expression";
@@ -208,15 +189,12 @@ $LANG['plugin_fusinvdeploy']['form']['command_envvariable'][1] = "Environment va
 
 $LANG['plugin_fusinvdeploy']['form']['action_message'][1] = "Title";
 $LANG['plugin_fusinvdeploy']['form']['action_message'][2] = "Content";
-$LANG['plugin_fusinvdeploy']['form']['action_message'][3] = "Type";
 $LANG['plugin_fusinvdeploy']['form']['action_message'][4] = "Informations";
 $LANG['plugin_fusinvdeploy']['form']['action_message'][5] = "report of the install";
 
 $LANG['plugin_fusinvdeploy']['task'][0] = "Deployment tasks";
 $LANG['plugin_fusinvdeploy']['task'][1] = "Task";
 $LANG['plugin_fusinvdeploy']['task'][3] = "Add task";
-$LANG['plugin_fusinvdeploy']['task'][5] = "Task";
-$LANG['plugin_fusinvdeploy']['task'][7] = "Actions";
 $LANG['plugin_fusinvdeploy']['task'][8] = "Actions list";
 $LANG['plugin_fusinvdeploy']['task'][11] = "Edit task";
 $LANG['plugin_fusinvdeploy']['task'][12] = "Add a task";
@@ -236,9 +214,4 @@ $LANG['plugin_fusinvdeploy']['group'][3] = "Group of computers";
 $LANG['plugin_fusinvdeploy']['group'][4] = "Add group";
 $LANG['plugin_fusinvdeploy']['group'][5] = "If no line in the list is selected, the text fields on the left will be used for search.";
 
-$LANG['plugin_fusinvdeploy']['menu'][1] = "Package management";
-$LANG['plugin_fusinvdeploy']['menu'][2] = "Mirror servers";
-$LANG['plugin_fusinvdeploy']['menu'][3] = "Deployment tasks";
-$LANG['plugin_fusinvdeploy']['menu'][4] = "Groups of computers";
-$LANG['plugin_fusinvdeploy']['menu'][5] = "Deployment status";
-?>
+?>
\ No newline at end of file
diff --git a/fusinvdeploy/setup.php b/fusinvdeploy/setup.php
index be18393..afc6672 100644
--- a/fusinvdeploy/setup.php
+++ b/fusinvdeploy/setup.php
@@ -158,27 +158,27 @@ function plugin_init_fusinvdeploy() {
 
    // Breadcrumbs
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['packages']['title'] =
-      $LANG['plugin_fusinvdeploy']['menu'][1];
+      $LANG['plugin_fusinvdeploy']['package'][6];
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['packages']['page'] =
       '/plugins/fusinvdeploy/front/package.php';
 
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['mirror']['title'] =
-      $LANG['plugin_fusinvdeploy']['menu'][2];
+      $LANG['plugin_fusinvdeploy']['form']['mirror'][1];
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['mirror']['page'] =
       '/plugins/fusinvdeploy/front/mirror.php';
 
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['task']['title'] =
-      $LANG['plugin_fusinvdeploy']['menu'][3];
+      $LANG['plugin_fusinvdeploy']['task'][0];
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['task']['page'] =
       '/plugins/fusinvdeploy/front/task.php';
 
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['group']['title'] =
-      $LANG['plugin_fusinvdeploy']['menu'][4];
+      $LANG['plugin_fusinvdeploy']['group'][0];
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['group']['page'] =
       '/plugins/fusinvdeploy/front/group.php';
 
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['deploy']['title'] =
-      $LANG['plugin_fusinvdeploy']['menu'][5];
+      $LANG['plugin_fusinvdeploy']['deploystatus'][0];
    $PLUGIN_HOOKS['submenu_entry']['fusioninventory']['options']['deploy']['page'] =
       '/plugins/fusinvdeploy/front/deploystate.php';
 
@@ -212,15 +212,15 @@ function plugin_fusinvdeploy_check_prerequisites() {
         return false;
       }
       if (!$plugin->isActivated("fusioninventory")) {
-         print $LANG['plugin_fusinvdeploy']["setup"][17]."<br />\n";
+         print $LANG['plugin_fusinvdeploy']['setup'][17]."<br />\n";
          return false;
       }
       if (!$plugin->isActivated("fusinvinventory")) {
-         print $LANG['plugin_fusinvdeploy']["setup"][21]."<br />\n";
+         print $LANG['plugin_fusinvdeploy']['setup'][21]."<br />\n";
          return false;
       }
       if (!$plugin->isInstalled("webservices")) {
-         print $LANG['plugin_fusinvdeploy']["setup"][19]."<br />\n";
+         print $LANG['plugin_fusinvdeploy']['setup'][19]."<br />\n";
          return false;
       } else {
          //cheeck version of webservice
@@ -228,13 +228,13 @@ function plugin_fusinvdeploy_check_prerequisites() {
          $tmp = $plugin->find("directory = 'webservices'");
          $webservices_plugin = array_pop($tmp);
          if (version_compare($webservices_plugin['version'], '1.2.0') < 0) {
-            print $LANG['plugin_fusinvdeploy']["setup"][19]."<br />\n";
+            print $LANG['plugin_fusinvdeploy']['setup'][19]."<br />\n";
             return false;
          }
       }
       return true;
    } else {
-      echo $LANG['plugin_fusinvdeploy']["errors"][50];
+      echo $LANG['plugin_fusinvdeploy']['errors'][50];
    }
 }
 
diff --git a/fusinvinventory/locales/de_DE.php b/fusinvinventory/locales/de_DE.php
index b5b3d4d..5394218 100644
--- a/fusinvinventory/locales/de_DE.php
+++ b/fusinvinventory/locales/de_DE.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antivirus";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Keine Antiviren-Software auf diesem Computer";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="Dieser Computer ist kein Windows Desktop";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Version";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Aktuell";
 
diff --git a/fusinvinventory/locales/es_ES.php b/fusinvinventory/locales/es_ES.php
index 3f5ad47..fbef7c9 100644
--- a/fusinvinventory/locales/es_ES.php
+++ b/fusinvinventory/locales/es_ES.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antivirus";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Sin antivirus en este ordenador";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Versión";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Actualizado";
 
diff --git a/fusinvinventory/locales/et_EE.php b/fusinvinventory/locales/et_EE.php
index f68323f..8b299d1 100644
--- a/fusinvinventory/locales/et_EE.php
+++ b/fusinvinventory/locales/et_EE.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Viirusetõrje";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="No antivirus on this computer";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Versioon";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Ajakohane";
 
diff --git a/fusinvinventory/locales/fr_FR.php b/fusinvinventory/locales/fr_FR.php
index ad687a9..c99cca3 100644
--- a/fusinvinventory/locales/fr_FR.php
+++ b/fusinvinventory/locales/fr_FR.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antivirus";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Soit cet ordinateur n'a pas d'antivirus, soit ce n'est pas un poste Windows XP “Desktop” ou supérieur";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Version";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="A jour";
 
diff --git a/fusinvinventory/locales/he_IL.php b/fusinvinventory/locales/he_IL.php
index 7f00c33..15cdce7 100644
--- a/fusinvinventory/locales/he_IL.php
+++ b/fusinvinventory/locales/he_IL.php
@@ -3,7 +3,7 @@
 /*
    ------------------------------------------------------------------------
    FusionInventory
-   Copyright (C) 2010-2011 by the FusionInventory Development Team.
+   Copyright (C) 2010-2012 by the FusionInventory Development Team.
 
    http://www.fusioninventory.org/   http://forge.fusioninventory.org/
    ------------------------------------------------------------------------
@@ -31,7 +31,7 @@
    @author    David Durieux
    @co-author 
    @comment   Not translate this file, use https://www.transifex.net/projects/p/FusionInventory/
-   @copyright Copyright (c) 2010-2011 FusionInventory team
+   @copyright Copyright (c) 2010-2012 FusionInventory team
    @license   AGPL License 3.0 or (at your option) any later version
               http://www.gnu.org/licenses/agpl-3.0-standalone.html
    @link      http://www.fusioninventory.org/
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="אנטיוירוס";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="אין אנטיוירוס במחשב";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="גירסה";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="מעודכן";
 
diff --git a/fusinvinventory/locales/it_IT.php b/fusinvinventory/locales/it_IT.php
index ac321b9..60f6b43 100644
--- a/fusinvinventory/locales/it_IT.php
+++ b/fusinvinventory/locales/it_IT.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antivirus";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Nessun antivirus nel computer";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Versione";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Aggiornato";
 
diff --git a/fusinvinventory/locales/pl_PL.php b/fusinvinventory/locales/pl_PL.php
index 9f11e1a..9afaaae 100644
--- a/fusinvinventory/locales/pl_PL.php
+++ b/fusinvinventory/locales/pl_PL.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antywirus";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Brak antywirusa na tym komputerze";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Wersja";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Zaktualizowany";
 
@@ -53,8 +53,8 @@ $LANG['plugin_fusinvinventory']['blacklist'][0]="Wartości Czarnej Listy";
 $LANG['plugin_fusinvinventory']['blacklist'][1]="Nowa wartość do Czarnej Listy";
 
 $LANG['plugin_fusinvinventory']['computer'][0]="Ostatnia inwentaryzacja";
-$LANG['plugin_fusinvinventory']['computer'][1]="Owner";
-$LANG['plugin_fusinvinventory']['computer'][2]="Company";
+$LANG['plugin_fusinvinventory']['computer'][1]="Właściciel";
+$LANG['plugin_fusinvinventory']['computer'][2]="Firma";
 
 $LANG['plugin_fusinvinventory']['importxml'][0]="Import pliku XML od Agent-a";
 $LANG['plugin_fusinvinventory']['importxml'][1]="Komputer włączony do GLPI";
diff --git a/fusinvinventory/locales/pt_BR.php b/fusinvinventory/locales/pt_BR.php
index a7f262b..f5c4321 100644
--- a/fusinvinventory/locales/pt_BR.php
+++ b/fusinvinventory/locales/pt_BR.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antivirus";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Sem antivírus no computador";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Versão";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Up to date";
 
diff --git a/fusinvinventory/locales/ru_RU.php b/fusinvinventory/locales/ru_RU.php
index 7573f95..1d66881 100644
--- a/fusinvinventory/locales/ru_RU.php
+++ b/fusinvinventory/locales/ru_RU.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Антивирус";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="Нет антивируса на этом компьютере";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Версия";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Обновление";
 
diff --git a/fusinvinventory/locales/tr_TR.php b/fusinvinventory/locales/tr_TR.php
index 12980e7..f8f8c7d 100644
--- a/fusinvinventory/locales/tr_TR.php
+++ b/fusinvinventory/locales/tr_TR.php
@@ -43,7 +43,7 @@
 
 
 $LANG['plugin_fusinvinventory']['antivirus'][0]="Antivirüs";
-$LANG['plugin_fusinvinventory']['antivirus'][1]="No antivirus on this computer";
+$LANG['plugin_fusinvinventory']['antivirus'][1]="This computer is not a Windows Desktop >";
 $LANG['plugin_fusinvinventory']['antivirus'][2]="Versiyon";
 $LANG['plugin_fusinvinventory']['antivirus'][3]="Up to date";
 
diff --git a/fusinvsnmp/locales/de_DE.php b/fusinvsnmp/locales/de_DE.php
index afaeedd..084bc4d 100644
--- a/fusinvsnmp/locales/de_DE.php
+++ b/fusinvsnmp/locales/de_DE.php
@@ -50,11 +50,12 @@ $LANG['plugin_fusinvsnmp']['agents'][27]="SNMPQuery Modul-Version";
 $LANG['plugin_fusinvsnmp']['codetasklog'][1]="inventarisierte / abgefragte Geräte";
 $LANG['plugin_fusinvsnmp']['codetasklog'][2]="entdeckte Gerät";
 $LANG['plugin_fusinvsnmp']['codetasklog'][3]="Die SNMP Gerätedefinition am Agenten ist nicht aktuell. Für den nächsten Lauf, wird er eine neue Version vom Server erhalten.";
-$LANG['plugin_fusinvsnmp']['codetasklog'][4]="Add the item";
-$LANG['plugin_fusinvsnmp']['codetasklog'][5]="Update the item";
+$LANG['plugin_fusinvsnmp']['codetasklog'][4]="Objekt hinzufügen";
+$LANG['plugin_fusinvsnmp']['codetasklog'][5]="Objekt aktualisieren";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Inventarisierung gestartet";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Detail";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Port Typen zum Import (für Netzwerk Equipment)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Netzwerk-Inventarisierung (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Netzwerkscan";
 $LANG['plugin_fusinvsnmp']['config'][8]="Nie";
@@ -166,7 +167,7 @@ $LANG['plugin_fusinvsnmp']['mapping'][186]="Trommel Magenta verbleiend";
 $LANG['plugin_fusinvsnmp']['mapping'][187]="Trommel Gelb Maximum";
 $LANG['plugin_fusinvsnmp']['mapping'][188]="Trommel Gelb verbraucht";
 $LANG['plugin_fusinvsnmp']['mapping'][189]="Trommel Gelb verbleibend";
-$LANG['plugin_fusinvsnmp']['mapping'][18]="networking > port > type";
+$LANG['plugin_fusinvsnmp']['mapping'][18]="Netzwerk > Port > Typ";
 $LANG['plugin_fusinvsnmp']['mapping'][190]="Abfalleinheit Maximum";
 $LANG['plugin_fusinvsnmp']['mapping'][191]="Abfalleinheit benutzt";
 $LANG['plugin_fusinvsnmp']['mapping'][192]="Abfalleinheit verbleibend";
@@ -197,7 +198,7 @@ $LANG['plugin_fusinvsnmp']['mapping'][37]="Drucker > Verbrauchsmaterial > Kartus
 $LANG['plugin_fusinvsnmp']['mapping'][38]="Drucker > Verbrauchsmaterial > Kartusche Magenta (%)";
 $LANG['plugin_fusinvsnmp']['mapping'][39]="Drucker > Verbrauchsmaterial > Kartusche Photo-Cyan (%)";
 $LANG['plugin_fusinvsnmp']['mapping'][3]="Netzwerk > Uptime";
-$LANG['plugin_fusinvsnmp']['mapping'][400]="printer > consumables > maintenance kit (%)";
+$LANG['plugin_fusinvsnmp']['mapping'][400]="Drucker > Verbrauchsmaterial > Wartungskit (%)";
 $LANG['plugin_fusinvsnmp']['mapping'][401]="Netzwerk > CPU Benutzer";
 $LANG['plugin_fusinvsnmp']['mapping'][402]="Netzwerk > CPU System";
 $LANG['plugin_fusinvsnmp']['mapping'][403]="Netzwerk > Kontakt";
@@ -233,8 +234,8 @@ $LANG['plugin_fusinvsnmp']['mapping'][42]="Drucker > Verbrauchsmaterial > Photol
 $LANG['plugin_fusinvsnmp']['mapping'][430]="Netzwerk > Port > VLAN";
 $LANG['plugin_fusinvsnmp']['mapping'][435]="Netzwerk > CDP remote sysdescr";
 $LANG['plugin_fusinvsnmp']['mapping'][436]="Netzwerk > CDP remote id";
-$LANG['plugin_fusinvsnmp']['mapping'][437]="networking > CDP remote model device";
-$LANG['plugin_fusinvsnmp']['mapping'][438]="networking > LLDP remote sysdescr";
+$LANG['plugin_fusinvsnmp']['mapping'][437]="Netzwerk > CDP remote model device";
+$LANG['plugin_fusinvsnmp']['mapping'][438]="Netzwerk > LLDP remote Systembeschreibung";
 $LANG['plugin_fusinvsnmp']['mapping'][439]="Netzwerk > LLDP remote id";
 $LANG['plugin_fusinvsnmp']['mapping'][43]="Drucker > Verbrauchsmaterial > Photoleiter Farbe (%)";
 $LANG['plugin_fusinvsnmp']['mapping'][440]="Netzwerk > LLDP remote port Beschreibung";
@@ -277,7 +278,7 @@ $LANG['plugin_fusinvsnmp']['mapping'][76]="Drucker > Verbrauchsmaterial > Photol
 $LANG['plugin_fusinvsnmp']['mapping'][77]="Drucker > Verbrauchsmaterial > Photoleiter Cyan (Maximum)";
 $LANG['plugin_fusinvsnmp']['mapping'][78]="Drucker > Verbrauchsmaterial > Photoleiter Cyan (verbleibend)";
 $LANG['plugin_fusinvsnmp']['mapping'][79]="Drucker > Verbrauchsmaterial > Photoleiter Yellow (Maximum)";
-$LANG['plugin_fusinvsnmp']['mapping'][7]="networking > ports > last change";
+$LANG['plugin_fusinvsnmp']['mapping'][7]="Netzwerk > Ports > Letzte Änderungen";
 $LANG['plugin_fusinvsnmp']['mapping'][80]="Drucker > Verbrauchsmaterial > Photoleiter Yellow (verbleibend)";
 $LANG['plugin_fusinvsnmp']['mapping'][81]="Drucker > Verbrauchsmaterial > Photoleiter Magenta (Maximum)";
 $LANG['plugin_fusinvsnmp']['mapping'][82]="Drucker > Verbrauchsmaterial > Photoleiter Magenta (verbleibend)";
@@ -297,8 +298,8 @@ $LANG['plugin_fusinvsnmp']['mapping'][94]="Drucker > Verbrauchsmaterial > Fixier
 $LANG['plugin_fusinvsnmp']['mapping'][95]="Drucker > Verbrauchsmaterial > Reinigungsmodul (Maximum)";
 $LANG['plugin_fusinvsnmp']['mapping'][96]="Drucker > Verbrauchsmaterial > Reinigungsmodul (verbleibend)";
 $LANG['plugin_fusinvsnmp']['mapping'][97]="Drucker > Port > Typ";
-$LANG['plugin_fusinvsnmp']['mapping'][98]="printer > consumables > maintenance kit (max)";
-$LANG['plugin_fusinvsnmp']['mapping'][99]="printer > consumables > maintenance kit (remaining)";
+$LANG['plugin_fusinvsnmp']['mapping'][98]="Drucker > Verbrauchsmaterial > Wartungskit (max.)";
+$LANG['plugin_fusinvsnmp']['mapping'][99]="Drucker > Verbrauchsmaterial > Wartungskit (verbleibend)";
 $LANG['plugin_fusinvsnmp']['mapping'][9]="Netzwerk > Port > Anzahl Bytes ausgehend";
 
 $LANG['plugin_fusinvsnmp']['massiveaction'][1]="SNMP-Modell zuordnen";
@@ -385,7 +386,7 @@ $LANG['plugin_fusinvsnmp']['snmp'][4]="Sysdescr";
 $LANG['plugin_fusinvsnmp']['snmp'][51]="Duplex";
 $LANG['plugin_fusinvsnmp']['snmp'][53]="Letzte Inventarisierung";
 $LANG['plugin_fusinvsnmp']['snmp'][54]="Keine Daten verfügbar";
-$LANG['plugin_fusinvsnmp']['snmp'][55]="Number per second";
+$LANG['plugin_fusinvsnmp']['snmp'][55]="Anzahl pro Sekunden";
 
 $LANG['plugin_fusinvsnmp']['snmpauth'][1]="SNMP Community";
 $LANG['plugin_fusinvsnmp']['snmpauth'][2]="Benutzer";
@@ -393,13 +394,13 @@ $LANG['plugin_fusinvsnmp']['snmpauth'][4]="Verschlüsseungs für die Authentifiz
 $LANG['plugin_fusinvsnmp']['snmpauth'][5]="Passwort";
 $LANG['plugin_fusinvsnmp']['snmpauth'][6]="Verschlüsselungmethode für Daten";
 
-$LANG['plugin_fusinvsnmp']['state'][10]="Devices imported";
+$LANG['plugin_fusinvsnmp']['state'][10]="Importierte Geräte";
 $LANG['plugin_fusinvsnmp']['state'][4]="Startdatum";
 $LANG['plugin_fusinvsnmp']['state'][5]="Enddatum";
 $LANG['plugin_fusinvsnmp']['state'][6]="Anzahl entdeckte Geräte";
 $LANG['plugin_fusinvsnmp']['state'][7]="Davon fehlerhaft";
-$LANG['plugin_fusinvsnmp']['state'][8]="Devices not imported";
-$LANG['plugin_fusinvsnmp']['state'][9]="Devices linked";
+$LANG['plugin_fusinvsnmp']['state'][8]="Nicht importierte Geräte";
+$LANG['plugin_fusinvsnmp']['state'][9]="Verknüpfte Geräte";
 
 $LANG['plugin_fusinvsnmp']['stats'][0]="Seitenzähler";
 $LANG['plugin_fusinvsnmp']['stats'][1]="Seiten/Tag";
diff --git a/fusinvsnmp/locales/es_ES.php b/fusinvsnmp/locales/es_ES.php
index 1b22c16..e973681 100644
--- a/fusinvsnmp/locales/es_ES.php
+++ b/fusinvsnmp/locales/es_ES.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="modificar el elemento";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Inventario iniciado";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Detalle";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Inventario de red (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Descubrimiento de red";
 $LANG['plugin_fusinvsnmp']['config'][8]="Nunca";
diff --git a/fusinvsnmp/locales/fr_FR.php b/fusinvsnmp/locales/fr_FR.php
index 89e8fb4..1b91d37 100644
--- a/fusinvsnmp/locales/fr_FR.php
+++ b/fusinvsnmp/locales/fr_FR.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="Mise à jour de l'élément";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="L'inventaire a démarré";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Détail";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Inventaire réseau (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Découverte réseau";
 $LANG['plugin_fusinvsnmp']['config'][8]="Jamais";
diff --git a/fusinvsnmp/locales/it_IT.php b/fusinvsnmp/locales/it_IT.php
index 0953b61..7c9a468 100644
--- a/fusinvsnmp/locales/it_IT.php
+++ b/fusinvsnmp/locales/it_IT.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="Update the item";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Inventory started";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Detaglio";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Inventario di rete (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Scansione rete";
 $LANG['plugin_fusinvsnmp']['config'][8]="Mai";
diff --git a/fusinvsnmp/locales/pl_PL.php b/fusinvsnmp/locales/pl_PL.php
index b9a771c..4a328b2 100644
--- a/fusinvsnmp/locales/pl_PL.php
+++ b/fusinvsnmp/locales/pl_PL.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="Aktualizacja pozycji";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Start inwentaryzacji";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Więcej";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Inwentaryzacja sieci (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Przeszukiwanie sieci";
 $LANG['plugin_fusinvsnmp']['config'][8]="Nigdy";
diff --git a/fusinvsnmp/locales/pt_BR.php b/fusinvsnmp/locales/pt_BR.php
index cb7337b..30f90aa 100644
--- a/fusinvsnmp/locales/pt_BR.php
+++ b/fusinvsnmp/locales/pt_BR.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="Update the item";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Inventory started";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Detail";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Inventário de rede (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="A descoberta de rede";
 $LANG['plugin_fusinvsnmp']['config'][8]="Nunca";
diff --git a/fusinvsnmp/locales/ru_RU.php b/fusinvsnmp/locales/ru_RU.php
index f80dac6..6a52618 100644
--- a/fusinvsnmp/locales/ru_RU.php
+++ b/fusinvsnmp/locales/ru_RU.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="Обновить элемент";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Инфенторизация запущена";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Подробно";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Сетевая инвентаризация (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Сетевое обнаружение";
 $LANG['plugin_fusinvsnmp']['config'][8]="Никогда";
diff --git a/fusinvsnmp/locales/tr_TR.php b/fusinvsnmp/locales/tr_TR.php
index 065b8db..5efe4b4 100644
--- a/fusinvsnmp/locales/tr_TR.php
+++ b/fusinvsnmp/locales/tr_TR.php
@@ -55,6 +55,7 @@ $LANG['plugin_fusinvsnmp']['codetasklog'][5]="Update the item";
 $LANG['plugin_fusinvsnmp']['codetasklog'][6]="Inventory started";
 $LANG['plugin_fusinvsnmp']['codetasklog'][7]="Detail";
 
+$LANG['plugin_fusinvsnmp']['config'][10]="Ports types to import (for network equipments)";
 $LANG['plugin_fusinvsnmp']['config'][3]="Network inventory (SNMP)";
 $LANG['plugin_fusinvsnmp']['config'][4]="Network discovery";
 $LANG['plugin_fusinvsnmp']['config'][8]="Never";
diff --git a/fusioninventory/front/wizard.form.php b/fusioninventory/front/wizard.form.php
index 6eb6685..56ee775 100644
--- a/fusioninventory/front/wizard.form.php
+++ b/fusioninventory/front/wizard.form.php
@@ -90,7 +90,9 @@ if (!isset($_GET['wizz'])) {
               break;
            
            case 'runagain':
-
+              $url = $_SERVER['HTTP_REFERER'];
+              $url = str_replace("w_tasksend", "w_tasksforcerun", $url);
+              Html::redirect($url);
               break;
             
          }         
diff --git a/fusioninventory/inc/wizard.class.php b/fusioninventory/inc/wizard.class.php
index 5ff5194..9bcfac2 100644
--- a/fusioninventory/inc/wizard.class.php
+++ b/fusioninventory/inc/wizard.class.php
@@ -1035,12 +1035,13 @@ class PluginFusioninventoryWizard {
       
       $a_credentials = $pfCredential->find("`entities_id` IN (".$_SESSION['glpiactiveentities_string'].")");
       foreach ($a_credentials as $data) {
+         $pfCredential->getFromDB($data['id']);
          echo "<tr class='tab_bg_1'>";
          echo "<td colspan='2'></td>";
          echo "<td><input type='radio' name='credential[]' value='".$data['id']."' />
-            &nbsp;".$data['name']."</td>";
+            &nbsp;".$pfCredential->getLink(1)."</td>";
          echo "<td>".$data['username']."</td>";
-         echo "<td>".$data['password']."</td>";
+         echo "<td>******</td>";
          echo "</tr>";
       }
       
@@ -1053,7 +1054,7 @@ class PluginFusioninventoryWizard {
       echo "Login : <input type='text' name='username' value=''/>";
       echo "</td>";         
       echo "<td>";
-      echo "pass : <input type='text' name='password' value=''/>";
+      echo "pass : <input type='password' name='password' value=''/>";
       echo "</td>";
       echo "</tr>";
       
diff --git a/fusioninventory/locales/de_DE.php b/fusioninventory/locales/de_DE.php
index e1aaf97..9ec6678 100644
--- a/fusioninventory/locales/de_DE.php
+++ b/fusioninventory/locales/de_DE.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Regelmässigkeit der Inventarisier
 $LANG['plugin_fusioninventory']['credential'][1]="Anmeldedaten für Remote-Server";
 $LANG['plugin_fusioninventory']['credential'][2]="Remote-Inventarisierung";
 $LANG['plugin_fusioninventory']['credential'][3]="Anmeldedaten";
+$LANG['plugin_fusioninventory']['credential'][4]="Erlaubnis";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Anzahl importierter Geräte";
 $LANG['plugin_fusioninventory']['discovery'][9]="Number of devices not imported because type non defined";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Number of devices not imported
 $LANG['plugin_fusioninventory']['errors'][1]="PHP-Option allow_url_fopen ist deaktiviert, Agents können nicht für eine Inventarisierung geweckt werden.";
 $LANG['plugin_fusioninventory']['errors'][22]="Nicht erkannte Elemente in";
 $LANG['plugin_fusioninventory']['errors'][2]="PHP-Option allow_url_fopen is deaktiviert, ist Voraussetzung für den Push-Modus";
-$LANG['plugin_fusioninventory']['errors'][50]="Ihre GLPI-Version ist nicht kompatibel, Voraussetzung ist Version 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Ihre Version von GLPI ist nicht kompatibel, benötigt wird 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Funktionen";
 $LANG['plugin_fusioninventory']['functionalities'][16]="SNMP-Authentifizierung ";
@@ -121,7 +122,7 @@ $LANG['plugin_fusioninventory']['locks'][1]="Sperrung hinzufügen";
 $LANG['plugin_fusioninventory']['menu'][1]="Agent-Verwaltung";
 $LANG['plugin_fusioninventory']['menu'][2]="IP-Adressbereich Konfiguration";
 $LANG['plugin_fusioninventory']['menu'][3]="Menü";
-$LANG['plugin_fusioninventory']['menu'][4]="Ubekanntes Gerät";
+$LANG['plugin_fusioninventory']['menu'][4]="Unbekannte Geräte";
 $LANG['plugin_fusioninventory']['menu'][5]="Authentifizierung mit Remote-Gerät (VMware)";
 $LANG['plugin_fusioninventory']['menu'][6]="Remote-Gerät inventarisieren (VMware)";
 $LANG['plugin_fusioninventory']['menu'][7]="Aktive / Laufende Aufgaben";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Seriennummer Partition";
 $LANG['plugin_fusioninventory']['rules'][15]="UUID";
 $LANG['plugin_fusioninventory']['rules'][16]="FusionInventory Tag";
 $LANG['plugin_fusioninventory']['rules'][18]="Gerät erstellt";
+$LANG['plugin_fusioninventory']['rules'][19]="Nicht importierte Geräte";
+$LANG['plugin_fusioninventory']['rules'][20]="Regel Import Protokolle";
+$LANG['plugin_fusioninventory']['rules'][21]="Informationen importieren";
 $LANG['plugin_fusioninventory']['rules'][2]="Importregeln für Geräte";
 $LANG['plugin_fusioninventory']['rules'][3]="Durchsuche GLPI nach Material mit dem Status";
 $LANG['plugin_fusioninventory']['rules'][4]="Ziel-Entität des Materials";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="Stunden";
 $LANG['plugin_fusioninventory']['task'][37]="Tage";
 $LANG['plugin_fusioninventory']['task'][38]="Monate";
 $LANG['plugin_fusioninventory']['task'][39]="Kann Task nicht ausführen, weil bereits ein anderer aktiv ist!";
-$LANG['plugin_fusioninventory']['task'][40]="Ausführung erzwingen";
+$LANG['plugin_fusioninventory']['task'][40]="Start erzwingen";
 $LANG['plugin_fusioninventory']['task'][41]="Server kontaktiert Agent (Push-Methode)";
 $LANG['plugin_fusioninventory']['task'][42]="Agent kontaktiert Server (Pull-Methode)";
 $LANG['plugin_fusioninventory']['task'][43]="Kommunikationsmodus";
 $LANG['plugin_fusioninventory']['task'][44]="Alle Informationen zu diesem Task anzeigen";
 $LANG['plugin_fusioninventory']['task'][45]="Erweiterte Optionen";
-$LANG['plugin_fusioninventory']['task'][46]="Zurücksetzen";
+$LANG['plugin_fusioninventory']['task'][46]="Neuinitialisierung ";
 $LANG['plugin_fusioninventory']['task'][47]="Eindeutige ID";
 $LANG['plugin_fusioninventory']['task'][48]="Zum Inventar";
+$LANG['plugin_fusioninventory']['task'][49]="Erweiterter Modus";
+$LANG['plugin_fusioninventory']['task'][50]="Aufgabe beenden";
+$LANG['plugin_fusioninventory']['task'][51]="Aktion nach Beendigung des laufenden Tasks";
+$LANG['plugin_fusioninventory']['task'][52]="Diese Aufgabe löschen und beenden";
+$LANG['plugin_fusioninventory']['task'][53]="Ende";
+$LANG['plugin_fusioninventory']['task'][54]="Aufgabe erneut ausführen";
+$LANG['plugin_fusioninventory']['task'][55]="Kein Agent berechtigt die ESX-Inventur auszuführen";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="gestartet";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="OK";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Fehler";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="Unbekannt";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="Aktiv";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Vorbereitet";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="Alle Ausführungen anzeigen";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="Protokoll dieser Ausführung anzeigen";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusInv";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Willkommen bei FusionInventory: K
 $LANG['plugin_fusioninventory']['wizard'][15]="Computer und Peripherie";
 $LANG['plugin_fusioninventory']['wizard'][16]="ESX Server";
 $LANG['plugin_fusioninventory']['wizard'][17]="Netzwergeräte und -Drucker";
+$LANG['plugin_fusioninventory']['wizard'][18]="Assistent";
 $LANG['plugin_fusioninventory']['wizard'][1]="Typ des zu inventarisierenden Geräts";
 $LANG['plugin_fusioninventory']['wizard'][2]="Import-Optionen";
 $LANG['plugin_fusioninventory']['wizard'][3]="Entitätsregeln";
diff --git a/fusioninventory/locales/es_ES.php b/fusioninventory/locales/es_ES.php
index e85f70d..176f47e 100644
--- a/fusioninventory/locales/es_ES.php
+++ b/fusioninventory/locales/es_ES.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Frecuencia de inventario (en horas
 $LANG['plugin_fusioninventory']['credential'][1]="Credencial para inventario remoto";
 $LANG['plugin_fusioninventory']['credential'][2]="Inventario de servidor VMware ESX";
 $LANG['plugin_fusioninventory']['credential'][3]="Credenciales";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Número de dispositivos importados";
 $LANG['plugin_fusioninventory']['discovery'][9]="Número de dispositivos no importados por no tener un tipo definido";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Número de dispositivos no impo
 $LANG['plugin_fusioninventory']['errors'][1]="La directiva 'allow_url_fopen' de PHP está desactivada. No es posible llamar al agente para hacer el inventario";
 $LANG['plugin_fusioninventory']['errors'][22]="Elemento desatendido en";
 $LANG['plugin_fusioninventory']['errors'][2]="La directiva 'allow_url_fopen' de PHP está desactivada. El modo 'push' no puede funcionar";
-$LANG['plugin_fusioninventory']['errors'][50]="Su versión de GLPI no es compatible, necesita la versión 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Características";
 $LANG['plugin_fusioninventory']['functionalities'][16]="Autenticación SNMP";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Número de serie de la partición
 $LANG['plugin_fusioninventory']['rules'][15]="UUID";
 $LANG['plugin_fusioninventory']['rules'][16]="Etiqueta FusionInventory";
 $LANG['plugin_fusioninventory']['rules'][18]="Dispositivo creado";
+$LANG['plugin_fusioninventory']['rules'][19]="Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][20]="Rule import logs";
+$LANG['plugin_fusioninventory']['rules'][21]="Import informations";
 $LANG['plugin_fusioninventory']['rules'][2]="Reglas de importación y actualización de bienes";
 $LANG['plugin_fusioninventory']['rules'][3]="Buscar equipo GLPI con estado";
 $LANG['plugin_fusioninventory']['rules'][4]="Entidad de destino del equipo";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="horas";
 $LANG['plugin_fusioninventory']['task'][37]="días";
 $LANG['plugin_fusioninventory']['task'][38]="meses";
 $LANG['plugin_fusioninventory']['task'][39]="¡Imposible lanzar la tarea porque quedan acciones en curso!";
-$LANG['plugin_fusioninventory']['task'][40]="Forzar la ejecución";
+$LANG['plugin_fusioninventory']['task'][40]="Force start";
 $LANG['plugin_fusioninventory']['task'][41]="El servidor inicia el contacto con el agente (push)";
 $LANG['plugin_fusioninventory']['task'][42]="El agente inicia el contacto con el servidor (pull)";
 $LANG['plugin_fusioninventory']['task'][43]="Modo de comunicación";
 $LANG['plugin_fusioninventory']['task'][44]="Ver toda la información sobre la tarea";
 $LANG['plugin_fusioninventory']['task'][45]="Opciones avanzadas";
-$LANG['plugin_fusioninventory']['task'][46]="Reiniciar";
+$LANG['plugin_fusioninventory']['task'][46]="Reinitialization";
 $LANG['plugin_fusioninventory']['task'][47]="Identificador";
 $LANG['plugin_fusioninventory']['task'][48]="Consultados";
+$LANG['plugin_fusioninventory']['task'][49]="Advanced mode";
+$LANG['plugin_fusioninventory']['task'][50]="End Task";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Terminado";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="Iniciado";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Correcto";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Error";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="desconocido";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="En curso";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Preparado";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="See all executions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="View logs of this execution";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusionInventory";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Bienvenido a FusionInventory. ¿Q
 $LANG['plugin_fusioninventory']['wizard'][15]="Ordenadores y periféricos";
 $LANG['plugin_fusioninventory']['wizard'][16]="Servidores VMware ESX";
 $LANG['plugin_fusioninventory']['wizard'][17]="Dispositivos e impresoras de red";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Tipo de dispositivo a inventariar";
 $LANG['plugin_fusioninventory']['wizard'][2]="Importar opciones";
 $LANG['plugin_fusioninventory']['wizard'][3]="Reglas de entidad";
diff --git a/fusioninventory/locales/fr_FR.php b/fusioninventory/locales/fr_FR.php
index 1933e19..155a613 100644
--- a/fusioninventory/locales/fr_FR.php
+++ b/fusioninventory/locales/fr_FR.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Fréquence des inventaires (en heu
 $LANG['plugin_fusioninventory']['credential'][1]="Compte d'inventaire distant";
 $LANG['plugin_fusioninventory']['credential'][2]="Matériel distant à inventorier";
 $LANG['plugin_fusioninventory']['credential'][3]="Compte";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Nombre de matériels importés";
 $LANG['plugin_fusioninventory']['discovery'][9]="Nombre de matériels non importés car type non défini";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Nombre de matériels non import
 $LANG['plugin_fusioninventory']['errors'][1]="PHP allow_url_fopen est désactivé, Réveil impossible de l'agent pour faire l'inventaire";
 $LANG['plugin_fusioninventory']['errors'][22]="Elément inattendu dans";
 $LANG['plugin_fusioninventory']['errors'][2]="PHP allow_url_fopen est désactivé, Le mode push ne peut pas fonctionner";
-$LANG['plugin_fusioninventory']['errors'][50]="La version de GLPI n'est pas compatible, vous avez besoin de la version 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="La version de GLPI n'est pas compatible, la version 0.83 est requise";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Fonctionnalités";
 $LANG['plugin_fusioninventory']['functionalities'][16]="Stockage de l'authentification SNMP";
@@ -142,7 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Numéro de série de partition dis
 $LANG['plugin_fusioninventory']['rules'][15]="uuid";
 $LANG['plugin_fusioninventory']['rules'][16]="Etiquette FusionInventory";
 $LANG['plugin_fusioninventory']['rules'][18]="Equipement créé";
-$LANG['plugin_fusioninventory']['rules'][19] = "Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][19]="Import des équipements ignoré";
+$LANG['plugin_fusioninventory']['rules'][20]="Logs des règles d'import";
+$LANG['plugin_fusioninventory']['rules'][21]="Informations d'import";
 $LANG['plugin_fusioninventory']['rules'][2]="Règles d'import et de liaison des matériels";
 $LANG['plugin_fusioninventory']['rules'][3]="Chercher les matériels GLPI ayant le statut";
 $LANG['plugin_fusioninventory']['rules'][4]="Entité de destination de la machine";
@@ -183,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="heures";
 $LANG['plugin_fusioninventory']['task'][37]="jours";
 $LANG['plugin_fusioninventory']['task'][38]="mois";
 $LANG['plugin_fusioninventory']['task'][39]="Impossible de lancer la tâche car il reste des actions en cours!";
-$LANG['plugin_fusioninventory']['task'][40]="Forcer l'exécution";
+$LANG['plugin_fusioninventory']['task'][40]="Forcer le démarrage";
 $LANG['plugin_fusioninventory']['task'][41]="Le serveur initie le contact avec l'agent (push)";
 $LANG['plugin_fusioninventory']['task'][42]="L'agent initie le contact avec le serveur (pull)";
 $LANG['plugin_fusioninventory']['task'][43]="Mode de communication";
 $LANG['plugin_fusioninventory']['task'][44]="Voir les informations complètes de la tâche";
 $LANG['plugin_fusioninventory']['task'][45]="Options avancées";
-$LANG['plugin_fusioninventory']['task'][46]="Reset";
+$LANG['plugin_fusioninventory']['task'][46]="Réinitialiser";
 $LANG['plugin_fusioninventory']['task'][47]="id unique";
 $LANG['plugin_fusioninventory']['task'][48]="A inventorier";
+$LANG['plugin_fusioninventory']['task'][49]="Mode avancé";
+$LANG['plugin_fusioninventory']['task'][50]="Terminer la tâche";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Fin";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="Démarré";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Ok";
@@ -200,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Erreur";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="Inconnu";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="En cours";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Préparé";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="Voir toutes les exécutions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="Voir les logs de cette exécution";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusInv";
@@ -217,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Bienvenue dans FusionInventory. C
 $LANG['plugin_fusioninventory']['wizard'][15]="Ordinateurs et périphériques";
 $LANG['plugin_fusioninventory']['wizard'][16]="Serveurs ESX";
 $LANG['plugin_fusioninventory']['wizard'][17]="Equipements et imprimantes réseau";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Type de matériel à inventorier";
 $LANG['plugin_fusioninventory']['wizard'][2]="Options d'import";
 $LANG['plugin_fusioninventory']['wizard'][3]="Règles d'entités";
diff --git a/fusioninventory/locales/it_IT.php b/fusioninventory/locales/it_IT.php
index 3ad1551..d658df4 100644
--- a/fusioninventory/locales/it_IT.php
+++ b/fusioninventory/locales/it_IT.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Frequenza dell'inventario (in ore)
 $LANG['plugin_fusioninventory']['credential'][1]="Credenziali per inventario remoto";
 $LANG['plugin_fusioninventory']['credential'][2]="Inventario dispositivo remoto";
 $LANG['plugin_fusioninventory']['credential'][3]="Credenziali";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Numero di dispositivi importati";
 $LANG['plugin_fusioninventory']['discovery'][9]="Numero di dispositivi non importati poiché il tipo non è definito";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Numero di dispositivi non impor
 $LANG['plugin_fusioninventory']['errors'][1]="PHP allow_url_fopen è disattivata, Sveglia dell'agente impossibile per fare l'inventario";
 $LANG['plugin_fusioninventory']['errors'][22]="Voce inaspettata in";
 $LANG['plugin_fusioninventory']['errors'][2]="PHP allow_url_fopen è disattivata, il modulo push non può funzionare";
-$LANG['plugin_fusioninventory']['errors'][50]="La versione di GLPI non é compatible, avete bisogno della versione 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Caratteristiche";
 $LANG['plugin_fusioninventory']['functionalities'][16]="Autenticazione SNMP";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Numero di serie della partitione d
 $LANG['plugin_fusioninventory']['rules'][15]="uuid";
 $LANG['plugin_fusioninventory']['rules'][16]="Etichetta FusionInventory";
 $LANG['plugin_fusioninventory']['rules'][18]="Dispositivo creato";
+$LANG['plugin_fusioninventory']['rules'][19]="Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][20]="Rule import logs";
+$LANG['plugin_fusioninventory']['rules'][21]="Import informations";
 $LANG['plugin_fusioninventory']['rules'][2]="Regole d'importo e di associazione dei materiali";
 $LANG['plugin_fusioninventory']['rules'][3]="Cerca hardware GLPI con lo stato";
 $LANG['plugin_fusioninventory']['rules'][4]="Destinazione dell'entità del computer";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="ore";
 $LANG['plugin_fusioninventory']['task'][37]="giorni";
 $LANG['plugin_fusioninventory']['task'][38]="mesi";
 $LANG['plugin_fusioninventory']['task'][39]="Impossibile avviare il compito, perché sono rimaste azioni in corso!";
-$LANG['plugin_fusioninventory']['task'][40]="Forzare l'esecuzione";
+$LANG['plugin_fusioninventory']['task'][40]="Force start";
 $LANG['plugin_fusioninventory']['task'][41]="Il server inizia un contatto con l'agente (push)";
 $LANG['plugin_fusioninventory']['task'][42]="L'agente inizia un contatto con il server (pull)";
 $LANG['plugin_fusioninventory']['task'][43]="Modo di comunicazione";
 $LANG['plugin_fusioninventory']['task'][44]="Vedere le informazioni complete del compito";
 $LANG['plugin_fusioninventory']['task'][45]="Opzioni avanzate";
-$LANG['plugin_fusioninventory']['task'][46]="Reset";
+$LANG['plugin_fusioninventory']['task'][46]="Reinitialization";
 $LANG['plugin_fusioninventory']['task'][47]="Identificativo univoco";
 $LANG['plugin_fusioninventory']['task'][48]="Da inventariare";
+$LANG['plugin_fusioninventory']['task'][49]="Advanced mode";
+$LANG['plugin_fusioninventory']['task'][50]="End Task";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Fine";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="In esecuzione";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Ok";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Errore";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="Sconosciuto";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="In corso";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Preparato";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="See all executions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="View logs of this execution";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusInv";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Benvenuto in FusionInventory. Con
 $LANG['plugin_fusioninventory']['wizard'][15]="Computer e periferiche";
 $LANG['plugin_fusioninventory']['wizard'][16]="Server ESX";
 $LANG['plugin_fusioninventory']['wizard'][17]="Stampanti e dispositivi di rete";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Typo di materiale da inventoriare";
 $LANG['plugin_fusioninventory']['wizard'][2]="Opzioni d'importo";
 $LANG['plugin_fusioninventory']['wizard'][3]="Regole entità";
diff --git a/fusioninventory/locales/pl_PL.php b/fusioninventory/locales/pl_PL.php
index 1d3d836..d0b5a1d 100644
--- a/fusioninventory/locales/pl_PL.php
+++ b/fusioninventory/locales/pl_PL.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Częstotliwość inwentaryzacji (w
 $LANG['plugin_fusioninventory']['credential'][1]="Poświadczenia do zdalnej inwentaryzacji";
 $LANG['plugin_fusioninventory']['credential'][2]="Inwentaryzacja zdalnych urządzeń";
 $LANG['plugin_fusioninventory']['credential'][3]="Referencje";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Liczba zaimportowanych urządzeń";
 $LANG['plugin_fusioninventory']['discovery'][9]="Liczba niezaimportowanych urządzeń z powodu braku określenia typu urządzenia";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Liczba niezaimportowanych urzą
 $LANG['plugin_fusioninventory']['errors'][1]="Opcja PHP allow_url_fopen jest wyłączona, nie można";
 $LANG['plugin_fusioninventory']['errors'][22]="Element bez nadzoru w";
 $LANG['plugin_fusioninventory']['errors'][2]="Opcja PHP allow_url_fopen jest wyłączona, tryb push jest niedostępny";
-$LANG['plugin_fusioninventory']['errors'][50]="Masz niekompatybilną wersję GLPI, wymagana jest wersja 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Funkcje";
 $LANG['plugin_fusioninventory']['functionalities'][16]="Uwierzytelnienie SNMP";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Numer seryjny partycji";
 $LANG['plugin_fusioninventory']['rules'][15]="UUID";
 $LANG['plugin_fusioninventory']['rules'][16]="Tag z Fusioninventory";
 $LANG['plugin_fusioninventory']['rules'][18]="Urządzenie utworzono";
+$LANG['plugin_fusioninventory']['rules'][19]="Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][20]="Rule import logs";
+$LANG['plugin_fusioninventory']['rules'][21]="Import informations";
 $LANG['plugin_fusioninventory']['rules'][2]="Reguły importu sprzętu i kojarzenia";
 $LANG['plugin_fusioninventory']['rules'][3]="Znajdź zasoby GLPI ze statusem";
 $LANG['plugin_fusioninventory']['rules'][4]="Miejsce przeznaczenia sprzętu";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="godzin";
 $LANG['plugin_fusioninventory']['task'][37]="dni";
 $LANG['plugin_fusioninventory']['task'][38]="miesięcy";
 $LANG['plugin_fusioninventory']['task'][39]="Nie można uruchomić zadania, ponieważ niektóre procesy jeszcze trwają!";
-$LANG['plugin_fusioninventory']['task'][40]="Uruchom teraz";
+$LANG['plugin_fusioninventory']['task'][40]="Force start";
 $LANG['plugin_fusioninventory']['task'][41]="Serwer kontaktuje się z agentem (push)";
 $LANG['plugin_fusioninventory']['task'][42]="Agent kontaktuje się z serwerem (push)";
 $LANG['plugin_fusioninventory']['task'][43]="Tryb komunikacji";
 $LANG['plugin_fusioninventory']['task'][44]="Obejrzyj wszystkie informacje o zadaniu";
 $LANG['plugin_fusioninventory']['task'][45]="Zaawansowane opcje";
-$LANG['plugin_fusioninventory']['task'][46]="Wyczyść";
+$LANG['plugin_fusioninventory']['task'][46]="Reinitialization";
 $LANG['plugin_fusioninventory']['task'][47]="Unikalne id";
 $LANG['plugin_fusioninventory']['task'][48]="Zinwentaryzowane";
+$LANG['plugin_fusioninventory']['task'][49]="Advanced mode";
+$LANG['plugin_fusioninventory']['task'][50]="End Task";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Ukończone";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="Uruchomiony";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Ok";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Błąd";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="nieznany";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="Uruchomiony";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Gotowy";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="See all executions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="View logs of this execution";
 
 $LANG['plugin_fusioninventory']['title'][0]="Fusioninventory";
 $LANG['plugin_fusioninventory']['title'][1]="Fusinv";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Witamy w Fusioninventory. Chcesz
 $LANG['plugin_fusioninventory']['wizard'][15]="Komputery i akcesoria";
 $LANG['plugin_fusioninventory']['wizard'][16]="Serwery ESX/ESXi";
 $LANG['plugin_fusioninventory']['wizard'][17]="Urządzenia sieciowe i drukarki";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Typ urządzenia do inwentaryzacji";
 $LANG['plugin_fusioninventory']['wizard'][2]="Opcje importu";
 $LANG['plugin_fusioninventory']['wizard'][3]="Reguły dla jednostki";
diff --git a/fusioninventory/locales/pt_BR.php b/fusioninventory/locales/pt_BR.php
index 1cbfa11..35960f5 100644
--- a/fusioninventory/locales/pt_BR.php
+++ b/fusioninventory/locales/pt_BR.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Frequência do inventário (em hor
 $LANG['plugin_fusioninventory']['credential'][1]="Credencial para inventário remoto";
 $LANG['plugin_fusioninventory']['credential'][2]="Inventário de dispositivo remoto";
 $LANG['plugin_fusioninventory']['credential'][3]="Credenciais";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Número de dispositivos importados";
 $LANG['plugin_fusioninventory']['discovery'][9]="Número de dispositivos não importados devido ao tipo não definido";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Número de dispositivos não im
 $LANG['plugin_fusioninventory']['errors'][1]="PHP allow_url_fopen esta desligado, não consigo acordar agente para fazer inventário";
 $LANG['plugin_fusioninventory']['errors'][22]="Elemento autônomo em";
 $LANG['plugin_fusioninventory']['errors'][2]="PHP allow_url_fopen esta desligado, o modo push não funciona";
-$LANG['plugin_fusioninventory']['errors'][50]="Versão do GLPI não compatível. Necessário versão 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Funções";
 $LANG['plugin_fusioninventory']['functionalities'][16]="Autenticação SNMP";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Número de série da partição";
 $LANG['plugin_fusioninventory']['rules'][15]="uuid";
 $LANG['plugin_fusioninventory']['rules'][16]="FusionInventory tag";
 $LANG['plugin_fusioninventory']['rules'][18]="Dispositivo criado";
+$LANG['plugin_fusioninventory']['rules'][19]="Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][20]="Rule import logs";
+$LANG['plugin_fusioninventory']['rules'][21]="Import informations";
 $LANG['plugin_fusioninventory']['rules'][2]="Importação de equipamentos e regras de ligação";
 $LANG['plugin_fusioninventory']['rules'][3]="Pesquisa equipamentos GLPI com o status";
 $LANG['plugin_fusioninventory']['rules'][4]="Destino da entidade do equipamento";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="horas";
 $LANG['plugin_fusioninventory']['task'][37]="dia";
 $LANG['plugin_fusioninventory']['task'][38]="mês";
 $LANG['plugin_fusioninventory']['task'][39]="Não é possível executar tarefa porque alguns processos estão em execução ainda!";
-$LANG['plugin_fusioninventory']['task'][40]="Força execução";
+$LANG['plugin_fusioninventory']['task'][40]="Force start";
 $LANG['plugin_fusioninventory']['task'][41]="O servidor contacta o agente (push)";
 $LANG['plugin_fusioninventory']['task'][42]="O agente contacta o servidor (pull)";
 $LANG['plugin_fusioninventory']['task'][43]="Modo de comunicação";
 $LANG['plugin_fusioninventory']['task'][44]="Veja todas as informações de tarefa";
 $LANG['plugin_fusioninventory']['task'][45]="Opções avançadas";
-$LANG['plugin_fusioninventory']['task'][46]="Redefinir";
+$LANG['plugin_fusioninventory']['task'][46]="Reinitialization";
 $LANG['plugin_fusioninventory']['task'][47]="Id único";
 $LANG['plugin_fusioninventory']['task'][48]="Ao inventário";
+$LANG['plugin_fusioninventory']['task'][49]="Advanced mode";
+$LANG['plugin_fusioninventory']['task'][50]="End Task";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Pronto";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="Iniciado";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Ok";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Erro";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="desconhecido";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="Executando";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Preparado";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="See all executions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="View logs of this execution";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusInv";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Bem-vindo em FusionInventory. Ini
 $LANG['plugin_fusioninventory']['wizard'][15]="Computadores e periféricos";
 $LANG['plugin_fusioninventory']['wizard'][16]="ESX servidores";
 $LANG['plugin_fusioninventory']['wizard'][17]="Dispositivos de rede e impressoras";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Tipo de dispositivo para o inventário";
 $LANG['plugin_fusioninventory']['wizard'][2]="Opções de importação";
 $LANG['plugin_fusioninventory']['wizard'][3]="Regras de entidade";
diff --git a/fusioninventory/locales/ru_RU.php b/fusioninventory/locales/ru_RU.php
index 7260df8..f8bcfee 100644
--- a/fusioninventory/locales/ru_RU.php
+++ b/fusioninventory/locales/ru_RU.php
@@ -82,6 +82,7 @@ $LANG['plugin_fusioninventory']['config'][0]="Интервал инвентар
 $LANG['plugin_fusioninventory']['credential'][1]="Права для удаленной инвентаризации";
 $LANG['plugin_fusioninventory']['credential'][2]="Инвентаризация удаленного устройства";
 $LANG['plugin_fusioninventory']['credential'][3]="Права доступа";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Количество импортированных устройств";
 $LANG['plugin_fusioninventory']['discovery'][9]="Количество устройств не импортированных по причине не определённого типа";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Количество устро
 $LANG['plugin_fusioninventory']['errors'][1]="PHP allow_url_fopen is off, can't wake agent to do inventory";
 $LANG['plugin_fusioninventory']['errors'][22]="Не подключённый элемент в";
 $LANG['plugin_fusioninventory']['errors'][2]="PHP allow_url_fopen is off, push mode can't work";
-$LANG['plugin_fusioninventory']['errors'][50]="GLPI версия не совместима, требуется версия 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Функции";
 $LANG['plugin_fusioninventory']['functionalities'][16]="SNMP Аутентификация";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Серийный номер раз
 $LANG['plugin_fusioninventory']['rules'][15]="uuid";
 $LANG['plugin_fusioninventory']['rules'][16]="FusionInventory tag";
 $LANG['plugin_fusioninventory']['rules'][18]="Устройство создано";
+$LANG['plugin_fusioninventory']['rules'][19]="Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][20]="Rule import logs";
+$LANG['plugin_fusioninventory']['rules'][21]="Import informations";
 $LANG['plugin_fusioninventory']['rules'][2]="Импорт оборудования и ссылок правил";
 $LANG['plugin_fusioninventory']['rules'][3]="Поиск GLPI оборудования со статусом";
 $LANG['plugin_fusioninventory']['rules'][4]="Назначение оборудования организации";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="часы";
 $LANG['plugin_fusioninventory']['task'][37]="дни";
 $LANG['plugin_fusioninventory']['task'][38]="месяцы";
 $LANG['plugin_fusioninventory']['task'][39]="Не возможно запустить задачу потому что некоторые задачи все еще запущены!";
-$LANG['plugin_fusioninventory']['task'][40]="Принудительный запуск";
+$LANG['plugin_fusioninventory']['task'][40]="Force start";
 $LANG['plugin_fusioninventory']['task'][41]="Сервер связывается с агентом (push)";
 $LANG['plugin_fusioninventory']['task'][42]="Агент связывается с сервером (pull)";
 $LANG['plugin_fusioninventory']['task'][43]="Режим взаимодействия";
 $LANG['plugin_fusioninventory']['task'][44]="Посмотреть всю информацию о задаче";
 $LANG['plugin_fusioninventory']['task'][45]="Дополнительные параметры";
-$LANG['plugin_fusioninventory']['task'][46]="Сброс";
+$LANG['plugin_fusioninventory']['task'][46]="Reinitialization";
 $LANG['plugin_fusioninventory']['task'][47]="Уникальный id";
 $LANG['plugin_fusioninventory']['task'][48]="В Inventory";
+$LANG['plugin_fusioninventory']['task'][49]="Advanced mode";
+$LANG['plugin_fusioninventory']['task'][50]="End Task";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Завершить";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="Начато";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Ok";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Ошибка";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="неизвестно";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="Запуск";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Подготовлен";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="See all executions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="View logs of this execution";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusInv";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Добро пожаловать 
 $LANG['plugin_fusioninventory']['wizard'][15]="Компьютеры и периферия";
 $LANG['plugin_fusioninventory']['wizard'][16]="ESX сервера";
 $LANG['plugin_fusioninventory']['wizard'][17]="Сетевые устройства и принтеры";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Тип устройства для инвентаризации";
 $LANG['plugin_fusioninventory']['wizard'][2]="Параметры импорта";
 $LANG['plugin_fusioninventory']['wizard'][3]="Правила организации";
diff --git a/fusioninventory/locales/tr_TR.php b/fusioninventory/locales/tr_TR.php
index dc56232..16b43cf 100644
--- a/fusioninventory/locales/tr_TR.php
+++ b/fusioninventory/locales/tr_TR.php
@@ -75,13 +75,14 @@ $LANG['plugin_fusioninventory']['choice'][3]="ve";
 
 $LANG['plugin_fusioninventory']['codetasklog'][1]="Bad token, impossible to start agent";
 $LANG['plugin_fusioninventory']['codetasklog'][2]="Agent stopped/crashed";
-$LANG['plugin_fusioninventory']['codetasklog'][3]="Import denied ";
+$LANG['plugin_fusioninventory']['codetasklog'][3]="Import denied";
 
 $LANG['plugin_fusioninventory']['config'][0]="Inventory frequency (in hours)";
 
 $LANG['plugin_fusioninventory']['credential'][1]="Credential for remote inventory";
 $LANG['plugin_fusioninventory']['credential'][2]="Remote device inventory";
 $LANG['plugin_fusioninventory']['credential'][3]="Credentials";
+$LANG['plugin_fusioninventory']['credential'][4]="Credential";
 
 $LANG['plugin_fusioninventory']['discovery'][5]="Number of imported devices";
 $LANG['plugin_fusioninventory']['discovery'][9]="Number of devices not imported because type not defined";
@@ -89,7 +90,7 @@ $LANG['plugin_fusioninventory']['discovery'][9]="Number of devices not imported
 $LANG['plugin_fusioninventory']['errors'][1]="PHP allow_url_fopen is off, can't wake agent to do inventory";
 $LANG['plugin_fusioninventory']['errors'][22]="Unattended element in";
 $LANG['plugin_fusioninventory']['errors'][2]="PHP allow_url_fopen is off, push mode can't work";
-$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.80";
+$LANG['plugin_fusioninventory']['errors'][50]="Your GLPI version not compatible, require 0.83";
 
 $LANG['plugin_fusioninventory']['functionalities'][0]="Features";
 $LANG['plugin_fusioninventory']['functionalities'][16]="SNMP authentication";
@@ -142,6 +143,9 @@ $LANG['plugin_fusioninventory']['rules'][14]="Partition serial number";
 $LANG['plugin_fusioninventory']['rules'][15]="UUID";
 $LANG['plugin_fusioninventory']['rules'][16]="FusionInventory tag";
 $LANG['plugin_fusioninventory']['rules'][18]="Device created";
+$LANG['plugin_fusioninventory']['rules'][19]="Ignored import devices";
+$LANG['plugin_fusioninventory']['rules'][20]="Rule import logs";
+$LANG['plugin_fusioninventory']['rules'][21]="Import informations";
 $LANG['plugin_fusioninventory']['rules'][2]="Equipment import and link rules";
 $LANG['plugin_fusioninventory']['rules'][3]="Search GLPI equipment with the status";
 $LANG['plugin_fusioninventory']['rules'][4]="Destination of equipment entity";
@@ -182,15 +186,22 @@ $LANG['plugin_fusioninventory']['task'][36]="saat";
 $LANG['plugin_fusioninventory']['task'][37]="gün";
 $LANG['plugin_fusioninventory']['task'][38]="ay";
 $LANG['plugin_fusioninventory']['task'][39]="Unable to run task because some jobs is running yet!";
-$LANG['plugin_fusioninventory']['task'][40]="Force running";
+$LANG['plugin_fusioninventory']['task'][40]="Force start";
 $LANG['plugin_fusioninventory']['task'][41]="Server contacts the agent (push)";
 $LANG['plugin_fusioninventory']['task'][42]="Agent contacts the server (pull)";
 $LANG['plugin_fusioninventory']['task'][43]="Communication mode";
 $LANG['plugin_fusioninventory']['task'][44]="See all informations of task";
 $LANG['plugin_fusioninventory']['task'][45]="Advanced options";
-$LANG['plugin_fusioninventory']['task'][46]="Reset";
+$LANG['plugin_fusioninventory']['task'][46]="Reinitialization";
 $LANG['plugin_fusioninventory']['task'][47]="Unique id";
 $LANG['plugin_fusioninventory']['task'][48]="To inventory";
+$LANG['plugin_fusioninventory']['task'][49]="Advanced mode";
+$LANG['plugin_fusioninventory']['task'][50]="End Task";
+$LANG['plugin_fusioninventory']['task'][51]="Action after finish running task";
+$LANG['plugin_fusioninventory']['task'][52]="Delete this task and finish";
+$LANG['plugin_fusioninventory']['task'][53]="Finish";
+$LANG['plugin_fusioninventory']['task'][54]="Run again this task";
+$LANG['plugin_fusioninventory']['task'][55]="No agent allowed to run ESX inventory";
 
 $LANG['plugin_fusioninventory']['taskjoblog'][1]="Started";
 $LANG['plugin_fusioninventory']['taskjoblog'][2]="Tamam";
@@ -199,6 +210,8 @@ $LANG['plugin_fusioninventory']['taskjoblog'][4]="Hata";
 $LANG['plugin_fusioninventory']['taskjoblog'][5]="bilinmeyen";
 $LANG['plugin_fusioninventory']['taskjoblog'][6]="Running";
 $LANG['plugin_fusioninventory']['taskjoblog'][7]="Prepared";
+$LANG['plugin_fusioninventory']['taskjoblog'][8]="See all executions";
+$LANG['plugin_fusioninventory']['taskjoblog'][9]="View logs of this execution";
 
 $LANG['plugin_fusioninventory']['title'][0]="FusionInventory";
 $LANG['plugin_fusioninventory']['title'][1]="FusInv";
@@ -216,6 +229,7 @@ $LANG['plugin_fusioninventory']['wizard'][14]="Welcome in FusionInventory. Begin
 $LANG['plugin_fusioninventory']['wizard'][15]="Computers and peripherals";
 $LANG['plugin_fusioninventory']['wizard'][16]="ESX servers";
 $LANG['plugin_fusioninventory']['wizard'][17]="Network devices and printers";
+$LANG['plugin_fusioninventory']['wizard'][18]="Wizard";
 $LANG['plugin_fusioninventory']['wizard'][1]="Type of device to inventory";
 $LANG['plugin_fusioninventory']['wizard'][2]="Import options";
 $LANG['plugin_fusioninventory']['wizard'][3]="Entity rules";
diff --git a/fusioninventory/setup.php b/fusioninventory/setup.php
index c8b7fc8..8fcb231 100644
--- a/fusioninventory/setup.php
+++ b/fusioninventory/setup.php
@@ -292,12 +292,18 @@ function plugin_version_fusioninventory() {
 
 // Optional : check prerequisites before install : may print errors or add to message after redirect
 function plugin_fusioninventory_check_prerequisites() {
-   global $LANG;
+   global $LANG,$DB;
    
    if (version_compare(GLPI_VERSION,'0.83','lt') || version_compare(GLPI_VERSION,'0.84','ge')) {
       echo $LANG['plugin_fusioninventory']['errors'][50];
       return false;
    }
+   if (TableExists("glpi_plugin_fusioninventory_agents")
+           AND !FieldExists("glpi_plugin_fusioninventory_agents", "tag")) {
+      $DB->query("UPDATE `glpi_plugin_fusioninventory_configs` SET `value`='0.80+1.4' WHERE `type`='version'");
+      $DB->query("UPDATE `glpi_plugins` SET `version`='0.80+1.4' WHERE `directory` LIKE 'fusi%'");
+   }
+   
    return true;
 }
 

