diff -up ./phploc.rpm ./phploc
--- ./phploc.rpm	2015-05-26 16:03:47.000000000 +0200
+++ ./phploc	2015-05-26 19:14:44.687212792 +0200
@@ -9,23 +9,16 @@
  * file that was distributed with this source code.
  */

-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
+// Ensure correct include_path for RHSCL
+$inc = get_include_path();
+$dirs = explode(':', $inc);
+if (!in_array('/usr/share/php', $dirs)) {
+    $dirs[] = '/usr/share/php';
+    set_include_path(implode(':', $dirs));
 }
+unset ($inc, $dirs);

-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require 'SebastianBergmann/PHPLOC/autoload.php';

 $application = new SebastianBergmann\PHPLOC\CLI\Application;
 $application->run();
