diff -up ./phploc.rpm ./phploc
--- ./phploc.rpm	2015-06-04 15:18:14.000000000 +0200
+++ ./phploc	2015-06-04 17:19:24.277906076 +0200
@@ -9,23 +9,7 @@
  * file that was distributed with this source code.
  */

-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
-}
-
-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require '/usr/share/php/SebastianBergmann/PHPLOC/autoload.php';

 $application = new SebastianBergmann\PHPLOC\CLI\Application;
 $application->run();
