diff -up ./phploc.rpm ./phploc
--- ./phploc.rpm	2014-05-03 18:50:56.000000000 +0200
+++ ./phploc	2014-05-03 18:51:38.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 /**
  * phploc
@@ -42,23 +42,7 @@
  * @since     File available since Release 1.0.0
  */
 
-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
-}
-
-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require 'SebastianBergmann/PHPLOC/autoload.php';
 
 $application = new SebastianBergmann\PHPLOC\CLI\Application;
 $application->run();
