diff -up xulrunner-25.0/mozilla-release/gfx/ycbcr/moz.build.arm xulrunner-25.0/mozilla-release/gfx/ycbcr/moz.build
--- xulrunner-25.0/mozilla-release/gfx/ycbcr/moz.build.arm	2013-10-23 15:52:17.056493264 +0200
+++ xulrunner-25.0/mozilla-release/gfx/ycbcr/moz.build	2013-10-23 16:03:04.496045050 +0200
@@ -61,7 +61,7 @@ else:
         'yuv_row_other.cpp',
     ]
 
-if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['HAVE_ARM_NEON']:
+if CONFIG['HAVE_ARM_NEON']:
     ASFILES += [
         'yuv_row_arm.s',
     ]
