--- mozilla-release/gfx/layers/BufferUnrotate.cpp.old	2013-12-05 17:07:31.000000000 +0100
+++ mozilla-release/gfx/layers/BufferUnrotate.cpp	2013-12-10 14:43:25.948164298 +0100
@@ -8,6 +8,7 @@
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 void BufferUnrotate(uint8_t* aBuffer, int aByteWidth, int aHeight,
                     int aByteStride, int aXBoundary, int aYBoundary)
--- mozilla-release/gfx/ycbcr/moz.build.arm	2013-10-23 15:52:17.056493264 +0200
+++ mozilla-release/gfx/ycbcr/moz.build	2013-10-23 16:03:04.496045050 +0200
@@ -61,7 +61,7 @@ else:
         'yuv_row_other.cpp',
     ]
 
-if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['HAVE_ARM_NEON']:
+if CONFIG['HAVE_ARM_NEON']:
     ASFILES += [
         'yuv_row_arm.s',
     ]
