diff -up mozilla-release/storage/src/Makefile.in.938730 mozilla-release/storage/src/Makefile.in
--- mozilla-release/storage/src/Makefile.in.938730	2013-12-09 15:15:44.708527295 +0100
+++ mozilla-release/storage/src/Makefile.in	2013-12-09 15:17:16.119682503 +0100
@@ -6,14 +6,23 @@
 # Don't use the jemalloc allocator on Android, because we can't guarantee
 # that Gecko will configure sqlite before it is first used (bug 730495).
 #
+# Don't use the jemalloc allocator when using system NSS. Linked in libraries
+# (such as NSS) might trigger an initialization of sqlite and allocation
+# of memory using the default allocator, prior to the storage service
+# registering its allocator, causing memory management failures (bug 938730).
+#
 # Note: On Windows our sqlite build assumes we use jemalloc.  If you disable
 # MOZ_STORAGE_MEMORY on Windows, you will also need to change the "ifdef
 # MOZ_MEMORY" options in db/sqlite3/src/Makefile.in.
+
+ifndef _USE_SYSTEM_NSS
 ifdef MOZ_MEMORY
 ifneq ($(OS_TARGET), Android)
 DEFINES += -DMOZ_STORAGE_MEMORY
 endif
 endif
+endif
+
 
 # For nsDependentJSString
 LOCAL_INCLUDES = \
