diff -up mozilla-aurora-7676a9a06403/security/build/Makefile.in.build mozilla-aurora-7676a9a06403/security/build/Makefile.in
--- mozilla-aurora-7676a9a06403/security/build/Makefile.in.build	2012-09-13 14:18:35.000000000 +0200
+++ mozilla-aurora-7676a9a06403/security/build/Makefile.in	2012-09-18 10:58:45.560063613 +0200
@@ -139,7 +139,6 @@ DEFAULT_GMAKE_FLAGS += NSPR_INCLUDE_DIR=
 DEFAULT_GMAKE_FLAGS += NSPR_LIB_DIR=$(NSPR_LIB_DIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
-DEFAULT_GMAKE_FLAGS += NSS_ENABLE_ECC=1
 DEFAULT_GMAKE_FLAGS += NSINSTALL="$(NSINSTALL)"
 ifndef MOZ_NATIVE_SQLITE
 DEFAULT_GMAKE_FLAGS += SQLITE_LIB_NAME=mozsqlite3
diff -up mozilla-aurora-7676a9a06403/security/manager/Makefile.in.build mozilla-aurora-7676a9a06403/security/manager/Makefile.in
diff -up mozilla-aurora-7676a9a06403/security/manager/ssl/src/Makefile.in.build mozilla-aurora-7676a9a06403/security/manager/ssl/src/Makefile.in
--- mozilla-aurora-7676a9a06403/security/manager/ssl/src/Makefile.in.build	2012-09-13 14:18:35.000000000 +0200
+++ mozilla-aurora-7676a9a06403/security/manager/ssl/src/Makefile.in	2012-09-18 10:54:16.310505794 +0200
@@ -84,7 +84,6 @@ CSRCS += md4.c
 EXTRA_DEPS = $(NSS_DEP_LIBS)
 
 DEFINES += \
-  -DNSS_ENABLE_ECC \
   -DDLL_PREFIX=\"$(DLL_PREFIX)\" \
   -DDLL_SUFFIX=\"$(DLL_SUFFIX)\" \
   $(NULL)
