diff -up mozilla-beta/security/manager/Makefile.in.build mozilla-beta/security/manager/Makefile.in
--- mozilla-beta/security/manager/Makefile.in.build	2012-03-09 01:39:56.000000000 +0100
+++ mozilla-beta/security/manager/Makefile.in	2012-03-09 13:56:28.612328083 +0100
@@ -174,7 +174,6 @@ DEFAULT_GMAKE_FLAGS += NSPR_INCLUDE_DIR=
 DEFAULT_GMAKE_FLAGS += NSPR_LIB_DIR=$(NSPR_LIB_DIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
-DEFAULT_GMAKE_FLAGS += NSS_ENABLE_ECC=1
 DEFAULT_GMAKE_FLAGS += NSINSTALL="$(NSINSTALL)"
 ifndef MOZ_NATIVE_SQLITE
 DEFAULT_GMAKE_FLAGS += SQLITE_LIB_NAME=mozsqlite3
diff -up mozilla-beta/security/manager/ssl/src/Makefile.in.build mozilla-beta/security/manager/ssl/src/Makefile.in
--- mozilla-beta/security/manager/ssl/src/Makefile.in.build	2012-03-09 01:39:56.000000000 +0100
+++ mozilla-beta/security/manager/ssl/src/Makefile.in	2012-03-09 13:57:37.055657208 +0100
@@ -116,7 +116,6 @@ CSRCS += md4.c
 EXTRA_DEPS = $(NSS_DEP_LIBS)
 
 DEFINES += \
-  -DNSS_ENABLE_ECC \
   -DDLL_PREFIX=\"$(DLL_PREFIX)\" \
   -DDLL_SUFFIX=\"$(DLL_SUFFIX)\" \
   $(NULL)
