diff -up xulrunner-21.0/mozilla-release/config/Makefile.in.old xulrunner-21.0/mozilla-release/config/Makefile.in
--- xulrunner-21.0/mozilla-release/config/Makefile.in.old	2013-05-11 21:19:23.000000000 +0200
+++ xulrunner-21.0/mozilla-release/config/Makefile.in	2013-05-13 08:53:03.351969145 +0200
@@ -37,7 +37,7 @@ override NSBUILDROOT :=
 endif
 
 ifdef GNU_CC
-MODULE_OPTIMIZE_FLAGS = -O3
+MODULE_OPTIMIZE_FLAGS = -Os
 endif
 
 include $(topsrcdir)/config/config.mk
diff -up xulrunner-21.0/mozilla-release/configure.in.old xulrunner-21.0/mozilla-release/configure.in
--- xulrunner-21.0/mozilla-release/configure.in.old	2013-05-11 21:19:23.000000000 +0200
+++ xulrunner-21.0/mozilla-release/configure.in	2013-05-13 09:02:12.931823544 +0200
@@ -1791,7 +1791,7 @@ case "$host" in
 *-linux*|*-kfreebsd*-gnu|*-gnu*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_linux.cfg\"'
-    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
+    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-Os}"
     ;;
 
 *os2*)
@@ -2041,7 +2041,7 @@ ia64*-hpux*)
             # -Os is broken on gcc 4.5.x we need to tweak it to get good results.
             MOZ_OPTIMIZE_SIZE_TWEAK="-finline-limit=50"
         esac
-        MOZ_PGO_OPTIMIZE_FLAGS="-O3"
+        MOZ_PGO_OPTIMIZE_FLAGS="-Os"
         MOZ_OPTIMIZE_FLAGS="-Os -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
         MOZ_DEBUG_FLAGS="-g"
     fi
diff -up xulrunner-21.0/mozilla-release/js/src/config/Makefile.in.old xulrunner-21.0/mozilla-release/js/src/config/Makefile.in
--- xulrunner-21.0/mozilla-release/js/src/config/Makefile.in.old	2013-05-11 21:19:32.000000000 +0200
+++ xulrunner-21.0/mozilla-release/js/src/config/Makefile.in	2013-05-13 08:53:03.353969129 +0200
@@ -26,7 +26,7 @@ override NSBUILDROOT :=
 endif
 
 ifdef GNU_CC
-MODULE_OPTIMIZE_FLAGS = -O3
+MODULE_OPTIMIZE_FLAGS = -Os
 endif
 
 include $(topsrcdir)/config/config.mk
diff -up xulrunner-21.0/mozilla-release/js/src/configure.in.old xulrunner-21.0/mozilla-release/js/src/configure.in
--- xulrunner-21.0/mozilla-release/js/src/configure.in.old	2013-05-11 21:19:32.000000000 +0200
+++ xulrunner-21.0/mozilla-release/js/src/configure.in	2013-05-13 09:00:24.431225716 +0200
@@ -1391,7 +1391,7 @@ case "$host" in
 *-linux*|*-kfreebsd*-gnu|*-gnu*)
     HOST_CFLAGS="$HOST_CFLAGS -DXP_UNIX"
     HOST_NSPR_MDCPUCFG='\"md/_linux.cfg\"'
-    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-O3}"
+    HOST_OPTIMIZE_FLAGS="${HOST_OPTIMIZE_FLAGS=-Os}"
     ;;
 
 *os2*)
@@ -1596,8 +1596,8 @@ ia64*-hpux*)
             # -Os is broken on gcc 4.5.x we need to tweak it to get good results.
             MOZ_OPTIMIZE_SIZE_TWEAK="-finline-limit=50"
         esac
-        MOZ_PGO_OPTIMIZE_FLAGS="-O3"
-        MOZ_OPTIMIZE_FLAGS="-O3 -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
+        MOZ_PGO_OPTIMIZE_FLAGS="-Os"
+        MOZ_OPTIMIZE_FLAGS="-Os -freorder-blocks $MOZ_OPTIMIZE_SIZE_TWEAK"
         MOZ_DEBUG_FLAGS="-g"
     fi
 
diff -up xulrunner-21.0/mozilla-release/media/webrtc/trunk/Makefile.old xulrunner-21.0/mozilla-release/media/webrtc/trunk/Makefile
diff -up xulrunner-21.0/mozilla-release/memory/jemalloc/src/configure.ac.old xulrunner-21.0/mozilla-release/memory/jemalloc/src/configure.ac
--- xulrunner-21.0/mozilla-release/memory/jemalloc/src/configure.ac.old	2013-05-11 21:19:46.000000000 +0200
+++ xulrunner-21.0/mozilla-release/memory/jemalloc/src/configure.ac	2013-05-13 08:53:03.355969113 +0200
@@ -573,7 +573,7 @@ if test "x$enable_debug" = "x0" -a "x$no
   echo "$EXTRA_CFLAGS" | grep "\-O" >/dev/null || optimize="yes"
   if test "x${optimize}" = "xyes" ; then
     if test "x$GCC" = "xyes" ; then
-      JE_CFLAGS_APPEND([-O3])
+      JE_CFLAGS_APPEND([-Os])
       JE_CFLAGS_APPEND([-funroll-loops])
     elif test "x$je_cv_msvc" = "xyes" ; then
       JE_CFLAGS_APPEND([-O2])
diff -up xulrunner-21.0/mozilla-release/memory/jemalloc/src/configure.old xulrunner-21.0/mozilla-release/memory/jemalloc/src/configure
