diff -up xulrunner-25.0/mozilla-release/js/src/vm/SPSProfiler.cpp.s390-inlines xulrunner-25.0/mozilla-release/js/src/vm/SPSProfiler.cpp
--- xulrunner-25.0/mozilla-release/js/src/vm/SPSProfiler.cpp.s390-inlines	2013-10-23 16:04:01.773093979 +0200
+++ xulrunner-25.0/mozilla-release/js/src/vm/SPSProfiler.cpp	2013-10-23 16:04:44.551134551 +0200
@@ -4,6 +4,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "jscntxtinlines.h"
+
 #include "vm/SPSProfiler.h"
 
 #include "mozilla/DebugOnly.h"
