diff -up xulrunner-24.0/mozilla-release/toolkit/crashreporter/client/Makefile.in.gcc47 xulrunner-24.0/mozilla-release/toolkit/crashreporter/client/Makefile.in
--- xulrunner-24.0/mozilla-release/toolkit/crashreporter/client/Makefile.in.gcc47	2013-09-11 01:15:24.000000000 +0200
+++ xulrunner-24.0/mozilla-release/toolkit/crashreporter/client/Makefile.in	2013-09-13 13:33:49.866875578 +0200
@@ -60,6 +60,9 @@ $(DIST)/bin/crashreporter.crt: $(topsrcd
 	$(PYTHON) $(srcdir)/certdata2pem.py < $< > $@
 endif
 
+# Needed for moz_free
+LIBS += -L$(LIBXUL_DIST)/lib -lmozalloc
+
 LIBS += \
   $(DEPTH)/toolkit/crashreporter/google-breakpad/src/common/linux/$(LIB_PREFIX)breakpad_linux_common_s.$(LIB_SUFFIX) \
   $(NULL)
