diff -up mozilla-release/toolkit/crashreporter/client/Makefile.in.gcc47 mozilla-release/toolkit/crashreporter/client/Makefile.in
--- mozilla-release/toolkit/crashreporter/client/Makefile.in.gcc47	2013-12-09 15:12:50.659231760 +0100
+++ mozilla-release/toolkit/crashreporter/client/Makefile.in	2013-12-09 15:13:45.448324783 +0100
@@ -42,6 +42,9 @@ endif
 
 ifdef MOZ_WIDGET_GTK
 
+# Needed for moz_free
+LIBS += -L$(LIBXUL_DIST)/lib -lmozalloc
+
 LIBS += \
   $(DEPTH)/toolkit/crashreporter/google-breakpad/src/common/linux/$(LIB_PREFIX)breakpad_linux_common_s.$(LIB_SUFFIX) \
   $(NULL)
