From 896b1643b4295a37fba4e0fb5cb8ffb3c10da851 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Tue, 9 Jun 2015 14:34:23 +0200
Subject: [PATCH 1/2] use ['info'] instead of undefined ['key']

---
 apc.php | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/apc.php b/apc.php
index cedf096..119d74e 100644
--- a/apc.php
+++ b/apc.php
@@ -915,7 +915,6 @@ function block_sort($array1, $array2)
 	}
 	$fieldname='info';
 	$fieldheading='User Entry Label';
-	$fieldkey='key';
 
 	$cols=6;
 	echo <<<EOB
@@ -1041,7 +1040,7 @@ function block_sort($array1, $array2)
         } else if ($MYREQUEST['OB'] == OB_USER_CACHE) {
 
           echo '<td class="td-last center">';
-          echo '[<a href="', $MY_SELF, '&OB=', $MYREQUEST['OB'], '&DU=', urlencode($entry[$fieldkey]), '">Delete Now</a>]';
+          echo '[<a href="', $MY_SELF, '&OB=', $MYREQUEST['OB'], '&DU=', urlencode($entry[$fieldname]), '">Delete Now</a>]';
           echo '</td>';
         } else {
           echo '<td class="td-last center"> &nbsp; </td>';

From 20c53f1ee86a659d5316977096efe5edf03eb958 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Tue, 9 Jun 2015 14:36:25 +0200
Subject: [PATCH 2/2] silent the fetch, to hide "Warning: Class
 __PHP_Incomplete_Class has no unserializer in..."

---
 apc.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apc.php b/apc.php
index 119d74e..47debee 100644
--- a/apc.php
+++ b/apc.php
@@ -1048,7 +1048,7 @@ function block_sort($array1, $array2)
         echo '</tr>';
 		if ($sh == $MYREQUEST["SH"]) {
 			echo '<tr>';
-			echo '<td colspan="7"><pre>'.htmlentities(print_r(apcu_fetch($entry['info']), 1)).'</pre></td>';
+			echo '<td colspan="7"><pre>'.htmlentities(print_r(@apcu_fetch($entry['info']), 1)).'</pre></td>';
 			echo '</tr>';
 		}
         $i++;
From f095923ecaa46a6e3c64bee913cf527077393539 Mon Sep 17 00:00:00 2001
From: Stanislav Khromov <stanislav.khromov+github@gmail.com>
Date: Fri, 17 Oct 2014 15:44:20 +0200
Subject: [PATCH] Scroll to cache entry when clicking on it

---
 apc.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/apc.php b/apc.php
index efc7eec..cedf096 100644
--- a/apc.php
+++ b/apc.php
@@ -1020,8 +1020,8 @@ function block_sort($array1, $array2)
 		$sh=md5($entry["info"]);
         $field_value = htmlentities(strip_tags($entry[$fieldname],''), ENT_QUOTES, 'UTF-8');
         echo
-          '<tr class=tr-',$i%2,'>',
-          "<td class=td-0><a href=\"$MY_SELF&OB=",$MYREQUEST['OB'],"&SH=",$sh,"\">",$field_value,'</a></td>',
+          '<tr id="key-'. $sh .'" class=tr-',$i%2,'>',
+          "<td class=td-0><a href=\"$MY_SELF&OB=",$MYREQUEST['OB'],"&SH=",$sh,"#key-". $sh ."\">",$field_value,'</a></td>',
           '<td class="td-n center">',$entry['num_hits'],'</td>',
           '<td class="td-n right">',$entry['mem_size'],'</td>',
           '<td class="td-n center">',date(DATE_FORMAT,$entry['access_time']),'</td>',
