From db61fc22941e6179fe9bf29def99af6b16e0bb57 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Wed, 11 Jun 2014 09:49:43 +0200
Subject: [PATCH] Fix PHP 5.4 build

---
 apc_bin.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/apc_bin.c b/apc_bin.c
index a8f705c..7ae4213 100644
--- a/apc_bin.c
+++ b/apc_bin.c
@@ -34,6 +34,12 @@
 
 #include "ext/standard/md5.h"
 
+#ifdef HAVE_PTRDIFF_T
+#include <stddef.h>
+#else
+typedef long ptrdiff_t;
+#endif
+
 /* in apc_cache.c */
 extern zval* apc_copy_zval(zval* dst, const zval* src, apc_context_t* ctxt TSRMLS_DC);
 
-- 
1.9.3

