--- pecl/uuid/trunk/tests/uuid_create.phpt	2012/11/06 05:35:52	328254
+++ pecl/uuid/trunk/tests/uuid_create.phpt	2012/11/06 08:38:46	328255
@@ -10,7 +10,7 @@
 <?php
         // check basic format of generated UUIDs
         $uuid = uuid_create();
-        if (ereg("[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}", $uuid)) {
+        if (preg_match("/[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}/", $uuid)) {
                 echo "basic format ok\n";
         } else {
                 echo "basic UUID format check failed, generated UUID was $uuid\n";
