--- pecl/uuid/trunk/uuid.c	2012/11/06 09:09:21	328258
+++ pecl/uuid/trunk/uuid.c	2012/11/06 09:46:03	328259
@@ -177,7 +177,7 @@
 		  default:
 			php_error_docref(NULL TSRMLS_CC, 
 							 E_WARNING,
-							 "Unknown/invalid UUID type '%d' requested, using default type instead",
+							 "Unknown/invalid UUID type '%ld' requested, using default type instead",
 							 uuid_type);
 			uuid_generate(uuid);
 			break;        
--- pecl/uuid/trunk/config.m4	2015/03/28 06:45:58	336225
+++ pecl/uuid/trunk/config.m4	2015/03/28 07:05:04	336226
@@ -6,18 +6,28 @@
 
 
 if test "$PHP_UUID" != "no"; then
-   PHP_CHECK_FUNC_LIB(uuid_type, uuid)
-   PHP_CHECK_FUNC_LIB(uuid_variant, uuid)
-
 
+  AC_MSG_CHECKING([for the location of libuuid])
+  for dir in $PHP_UUID /usr/local /usr; do
+    if test -f "$dir/include/uuid/uuid.h"; then
+      PHP_UUID_DIR="$dir"
+    fi
+  done
+  if test -z "$PHP_UUID_DIR"; then
+    AC_MSG_ERROR(not found)
+  else
+    AC_MSG_RESULT(found in $PHP_UUID_DIR)
+  fi
   PHP_ADD_INCLUDE($PHP_UUID_DIR/include)
+  PHP_CHECK_FUNC_LIB(uuid_type, uuid)
+  PHP_CHECK_FUNC_LIB(uuid_variant, uuid)
 
   export OLD_CPPFLAGS="$CPPFLAGS"
   export CPPFLAGS="$CPPFLAGS $INCLUDES -DHAVE_UUID"
   AC_CHECK_HEADER([uuid/uuid.h], [], AC_MSG_ERROR('uuid/uuid.h' header not found))
   PHP_SUBST(UUID_SHARED_LIBADD)
 
-  PHP_ADD_LIBRARY_WITH_PATH(uuid, $PHP_UUID_DIR/lib, UUID_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(uuid, $PHP_UUID_DIR/$PHP_LIBDIR, UUID_SHARED_LIBADD)
   export CPPFLAGS="$OLD_CPPFLAGS"
 
   export OLD_CPPFLAGS="$CPPFLAGS"
