--- tests/Doctrine/Tests/ORM/Functional/Ticket/DDC2692Test.php.old	2016-06-13 18:09:07.905814745 +0200
+++ tests/Doctrine/Tests/ORM/Functional/Ticket/DDC2692Test.php	2016-06-13 18:09:59.319090970 +0200
@@ -29,7 +29,8 @@
 
     public function testIsListenerCalledOnlyOnceOnPreFlush()
     {
-        $listener = $this->getMock('Doctrine\Tests\ORM\Functional\Ticket\DDC2692Listener', array('preFlush'));
+        $listener = $this->getMockObjectGenerator()->getMock('Doctrine\Tests\ORM\Functional\Ticket\DDC2692Listener', array('preFlush'));
+        $this->registerMockObject($listener);
         $listener->expects($this->once())->method('preFlush');
 
         $this->_em->getEventManager()->addEventSubscriber($listener);
