diff -up ./config/tcpdf_config_alt.php.config ./config/tcpdf_config_alt.php
--- ./config/tcpdf_config_alt.php.config	2013-04-21 21:15:04.000000000 +0200
+++ ./config/tcpdf_config_alt.php	2013-05-09 16:47:02.868144489 +0200
@@ -61,7 +61,7 @@ if (substr($k_path_main, -1) != '/') {
  * Installation path (/var/www/tcpdf/).
  * By default it is automatically calculated but you can also set it as a fixed string to improve performances.
  */
-define ('K_PATH_MAIN', $k_path_main);
+define ('K_PATH_MAIN', '/usr/share/php/tcpdf/');
 
 // Automatic calculation for the following K_PATH_URL constant
 if (isset($_SERVER['HTTP_HOST']) AND (!empty($_SERVER['HTTP_HOST']))) {
@@ -89,7 +89,7 @@ define ('K_PATH_FONTS', K_PATH_MAIN.'fon
 /**
  * cache directory for temporary files (full path)
  */
-define ('K_PATH_CACHE', K_PATH_MAIN.'cache/');
+define ('K_PATH_CACHE', '/var/cache/php-tcpdf/');
 
 /**
  * cache directory for temporary files (url path)
diff -up ./config/tcpdf_config.php.config ./config/tcpdf_config.php
--- ./config/tcpdf_config.php.config	2013-04-21 21:15:04.000000000 +0200
+++ ./config/tcpdf_config.php	2013-05-09 16:46:46.391084498 +0200
@@ -65,7 +65,7 @@ if (!defined('K_TCPDF_EXTERNAL_CONFIG'))
 	 * Installation path (/var/www/tcpdf/).
 	 * By default it is automatically calculated but you can also set it as a fixed string to improve performances.
 	 */
-	define ('K_PATH_MAIN', $k_path_main);
+	define ('K_PATH_MAIN', '/usr/share/php/tcpdf/');
 
 	// Automatic calculation for the following K_PATH_URL constant
 	$k_path_url = $k_path_main; // default value for console mode
@@ -94,7 +94,7 @@ if (!defined('K_TCPDF_EXTERNAL_CONFIG'))
 	/**
 	 * cache directory for temporary files (full path)
 	 */
-	define ('K_PATH_CACHE', K_PATH_MAIN.'cache/');
+	define ('K_PATH_CACHE', '/var/cache/php-tcpdf/');
 
 	/**
 	 * cache directory for temporary files (url path)
diff -up ./tcpdf.php.config ./tcpdf.php
--- ./tcpdf.php.config	2013-05-09 16:43:00.199254100 +0200
+++ ./tcpdf.php	2013-05-09 16:43:00.212254149 +0200
@@ -144,7 +144,7 @@
 
 if (!defined('K_TCPDF_EXTERNAL_CONFIG')) {
 	// Main configuration file. Define the K_TCPDF_EXTERNAL_CONFIG constant to skip this file.
-	require_once(dirname(__FILE__).'/config/tcpdf_config.php');
+	require_once('/etc/php-tcpdf/tcpdf_config.php');
 }
 
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
