diff -up tcpdf/tcpdf.php.patch tcpdf/tcpdf.php
--- tcpdf/tcpdf.php.patch	2013-05-09 11:18:50.076930131 +0200
+++ tcpdf/tcpdf.php	2013-05-09 11:21:06.140921148 +0200
@@ -13427,8 +13427,7 @@ class TCPDF {
 		++$this->n; // signature object ($this->sig_obj_id + 1)
 		$this->signature_data = array();
 		if (strlen($signing_cert) == 0) {
-			$signing_cert = 'file://'.dirname(__FILE__).'/config/cert/tcpdf.crt';
-			$private_key_password = 'tcpdfdemo';
+			throw new RuntimeException('Please provide a certificate file and password!');
 		}
 		if (strlen($private_key) == 0) {
 			$private_key = $signing_cert;
@@ -18811,7 +18810,7 @@ Putting 1 is equivalent to putting 0 and
 				if (isset($tag['attribute']['action'])) {
 					$this->form_action = $tag['attribute']['action'];
 				} else {
-					$this->form_action = K_PATH_URL.$_SERVER['SCRIPT_NAME'];
+					throw new RuntimeException('Please explicitly set action attribute path!');
 				}
 				if (isset($tag['attribute']['enctype'])) {
 					$this->form_enctype = $tag['attribute']['enctype'];
