--- graphviz-2.26.0/lib/gvc/gvconfig.c.old	2009-07-08 17:14:20.000000000 +0200
+++ graphviz-2.26.0/lib/gvc/gvconfig.c	2010-10-20 11:51:21.238392202 +0200
@@ -471,6 +471,7 @@
         libdir = gvconfig_libdir(gvc);
         rc = stat(libdir, &libdir_st);
         if (rc == -1) {
+            gvtextlayout_select(gvc);   /* choose best available textlayout plugin immediately */
     	    /* if we fail to stat it then it probably doesn't exist so just fail silently */
 	    return;
         }
@@ -485,6 +486,7 @@
         if (rescan) {
     	    config_rescan(gvc, gvc->config_path);
     	    gvc->config_found = TRUE;
+            gvtextlayout_select(gvc);   /* choose best available textlayout plugin immediately */
     	    return;
         }
     
@@ -492,6 +494,7 @@
     
         rc = stat(gvc->config_path, &config_st);
         if (rc == -1) {
+            gvtextlayout_select(gvc);   /* choose best available textlayout plugin immediately */
     	    /* silently return without setting gvc->config_found = TRUE */
     	    return;
         }
