From 5ba641f5da8deada1fad51c3bce9b78d311cb5c5 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sun, 7 Aug 2016 20:07:31 -0400
Subject: [PATCH] ...

---
 tests/validate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/validate.c b/tests/validate.c
index e26d067..e0b0cd3 100644
--- a/tests/validate.c
+++ b/tests/validate.c
@@ -2394,7 +2394,7 @@ int main(int argc,char **argv)
       else
         {
           (void) FormatLocaleFile(stdout,"Version: %s\n",
-            GetMagickHomeURL();
+            GetMagickVersion((size_t *) NULL));
           (void) FormatLocaleFile(stdout,"Copyright: %s\n\n",
             GetMagickCopyright());
           (void) FormatLocaleFile(stdout,
