diff -NurEbBH ImageMagick-6.9.2-7.orig/wand/MagickWand-config.in ImageMagick-6.9.2-7/wand/MagickWand-config.in
--- ImageMagick-6.9.2-7.orig/wand/MagickWand-config.in	2015-11-26 17:14:07.000000000 +0300
+++ ImageMagick-6.9.2-7/wand/MagickWand-config.in	2015-12-05 21:53:37.350660894 +0300
@@ -38,19 +38,19 @@
       echo '@PACKAGE_VERSION@ Q@QUANTUM_DEPTH@ @MAGICK_HDRI@'
       ;;
     --cflags)
-      PKG_CONFIG_PATH="@libdir@/pkgconfig" pkg-config --cflags MagickWand
+      pkg-config --cflags MagickWand
       ;;
     --cxxflags)
-      PKG_CONFIG_PATH="@libdir@/pkgconfig" pkg-config --cflags MagickWand
+      pkg-config --cflags MagickWand
       ;;
     --cppflags)
-      PKG_CONFIG_PATH="@libdir@/pkgconfig" pkg-config --cflags MagickWand
+      pkg-config --cflags MagickWand
       ;;
     --ldflags)
-      PKG_CONFIG_PATH="@libdir@/pkgconfig" pkg-config --libs MagickWand
+      pkg-config --libs MagickWand
       ;;
     --libs)
-      PKG_CONFIG_PATH="@libdir@/pkgconfig" pkg-config --libs MagickWand
+      pkg-config --libs MagickWand
       ;;
     *)
       echo "${usage}" 1>&2
