diff -up Horde_Stringprep-1.0.2/lib/Horde/Stringprep.php.syslib Horde_Stringprep-1.0.2/lib/Horde/Stringprep.php
--- Horde_Stringprep-1.0.2/lib/Horde/Stringprep.php.syslib	2015-06-18 13:41:42.987550940 +0200
+++ Horde_Stringprep-1.0.2/lib/Horde/Stringprep.php	2015-06-18 13:42:31.127777933 +0200
@@ -34,11 +34,7 @@ class Horde_Stringprep
      */
     public static function autoload()
     {
-        if (file_exists(__DIR__ . '/Stringprep/vendor/autoload.php')) {
-            require_once __DIR__ . '/Stringprep/vendor/autoload.php';
-        } else {
-            require_once __DIR__ . '/../../bundle/vendor/autoload.php';
-        }
+        require_once('/usr/share/php/Znerol/Component/Stringprep/autoload.php');
     }
 
 }
