diff -up ./bin/php-parse.php.rpm ./bin/php-parse.php
--- ./bin/php-parse.php.rpm	2015-04-04 14:49:15.704865853 +0200
+++ ./bin/php-parse.php	2015-04-04 14:50:15.625104006 +0200
@@ -1,7 +1,8 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 
-require __DIR__ . '/../lib/bootstrap.php';
+require 'PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
 
 ini_set('xdebug.max_nesting_level', 3000);
 
