--- imagick-3.1.0RC2/config.m4	2012-05-30 22:21:12.000000000 +0200
+++ imagick-3.1.0RC2/config.m4	2013-03-13 13:55:38.292504304 +0100
@@ -52,7 +52,9 @@
   else	
     AC_MSG_CHECKING(for MagickWand.h header file)
 
-    if test -r $WAND_DIR/include/ImageMagick/wand/MagickWand.h; then
+    if test -r $WAND_DIR/include/ImageMagick-6/wand/MagickWand.h; then
+      AC_MSG_RESULT(found in $WAND_DIR/include/ImageMagick-6/wand/MagickWand.h)
+    elif test -r $WAND_DIR/include/ImageMagick/wand/MagickWand.h; then
       AC_MSG_RESULT(found in $WAND_DIR/include/ImageMagick/wand/MagickWand.h)
     else
       AC_MSG_ERROR(Cannot locate header file MagickWand.h)
