From b08a238775e998bb07b530c3cd80219e100c95cf Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Sat, 2 Aug 2014 17:50:00 +0200
Subject: [PATCH] set always_populate_raw_post_data=-1 to avoid failed test
 (because of warning)

---
 tests/message002.phpt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/message002.phpt b/tests/message002.phpt
index f782f8f..403fb26 100644
--- a/tests/message002.phpt
+++ b/tests/message002.phpt
@@ -9,6 +9,8 @@ b=c
 HTTP_X_TEST=test
 --COOKIE--
 foo=bar
+--INI--
+always_populate_raw_post_data=-1
 --FILE--
 <?php
 echo "Test\n";
-- 
1.9.2

From 400771d35709df29e48f4ca43551d40beac108e9 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Sat, 2 Aug 2014 17:50:41 +0200
Subject: [PATCH] slip online test

---
 tests/client010.phpt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/client010.phpt b/tests/client010.phpt
index 803403f..055d585 100644
--- a/tests/client010.phpt
+++ b/tests/client010.phpt
@@ -3,6 +3,7 @@ client upload
 --SKIPIF--
 <?php
 include "skipif.inc";
+skip_online_test();
 ?>
 --FILE--
 <?php
-- 
1.9.2

From b0d5cbf0fc0e1c9ac8868dab045d17648cc084d9 Mon Sep 17 00:00:00 2001
From: Michael Wallner <mike@php.net>
Date: Mon, 4 Aug 2014 10:45:19 +0200
Subject: [PATCH] =?utf8?q?fix=20PHP-5.4=20compatibility=C2=A7?=
MIME-Version: 1.0
Content-Type: text/plain; charset=utf8
Content-Transfer-Encoding: 8bit

---
 php_http.h                | 2 +-
 php_http_header_parser.c  | 2 +-
 php_http_message_parser.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/php_http_header_parser.c b/php_http_header_parser.c
index f560a85..ec41a24 100644
--- a/php_http_header_parser.c
+++ b/php_http_header_parser.c
@@ -62,7 +62,7 @@ php_http_header_parser_state_t php_http_header_parser_state_is(php_http_header_p
 	php_http_header_parser_state_t state;
 
 	if (parser->stack.top) {
-		return (php_http_header_parser_state_t) zend_ptr_stack_top(&parser->stack);
+		return (php_http_header_parser_state_t) parser->stack.elements[parser->stack.top - 1];
 	}
 
 	return PHP_HTTP_HEADER_PARSER_STATE_START;
diff --git a/php_http_message_parser.c b/php_http_message_parser.c
index 6328fa4..20ef3ac 100644
--- a/php_http_message_parser.c
+++ b/php_http_message_parser.c
@@ -80,7 +80,7 @@ php_http_message_parser_state_t php_http_message_parser_state_push(php_http_mess
 php_http_message_parser_state_t php_http_message_parser_state_is(php_http_message_parser_t *parser)
 {
 	if (parser->stack.top) {
-		return (php_http_message_parser_state_t) zend_ptr_stack_top(&parser->stack);
+		return (php_http_message_parser_state_t) parser->stack.elements[parser->stack.top - 1];
 	}
 	return PHP_HTTP_MESSAGE_PARSER_STATE_START;
 }
-- 
1.9.2

From 28e32ca3227d5a3a8b9a1f409aac4f7b6e3fa5ea Mon Sep 17 00:00:00 2001
From: Michael Wallner <mike@php.net>
Date: Tue, 5 Aug 2014 06:59:14 +0200
Subject: [PATCH] fix leak

---
 php_http_client_curl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/php_http_client_curl.c b/php_http_client_curl.c
index 15d2619..baf99ce 100644
--- a/php_http_client_curl.c
+++ b/php_http_client_curl.c
@@ -1309,7 +1309,7 @@ static void php_http_curle_options_init(php_http_options_t *registry TSRMLS_DC)
 		}
 		if ((opt = php_http_option_register(registry, ZEND_STRL("certtype"), CURLOPT_SSLCERTTYPE, IS_STRING))) {
 			opt->flags |= PHP_HTTP_CURLE_OPTION_CHECK_STRLEN;
-			ZVAL_STRING(&opt->defval, "PEM", 1);
+			ZVAL_STRING(&opt->defval, "PEM", 0);
 		}
 		if ((opt = php_http_option_register(registry, ZEND_STRL("key"), CURLOPT_SSLKEY, IS_STRING))) {
 			opt->flags |= PHP_HTTP_CURLE_OPTION_CHECK_STRLEN;
@@ -1317,7 +1317,7 @@ static void php_http_curle_options_init(php_http_options_t *registry TSRMLS_DC)
 		}
 		if ((opt = php_http_option_register(registry, ZEND_STRL("keytype"), CURLOPT_SSLKEYTYPE, IS_STRING))) {
 			opt->flags |= PHP_HTTP_CURLE_OPTION_CHECK_STRLEN;
-			ZVAL_STRING(&opt->defval, "PEM", 1);
+			ZVAL_STRING(&opt->defval, "PEM", 0);
 		}
 		if ((opt = php_http_option_register(registry, ZEND_STRL("keypasswd"), CURLOPT_SSLKEYPASSWD, IS_STRING))) {
 			opt->flags |= PHP_HTTP_CURLE_OPTION_CHECK_STRLEN;
-- 
1.9.2

From 57bfeaae7196fd754aad2150dfd1d3258ea8ce66 Mon Sep 17 00:00:00 2001
From: Michael Wallner <mike@php.net>
Date: Tue, 5 Aug 2014 08:07:26 +0200
Subject: [PATCH] fix write on stack

---
 php_http_encoding.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/php_http_encoding.c b/php_http_encoding.c
index 403d781..7f0462c 100644
--- a/php_http_encoding.c
+++ b/php_http_encoding.c
@@ -53,7 +53,6 @@ const char *php_http_encoding_dechunk(const char *encoded, size_t encoded_len, c
 				php_error_docref(NULL TSRMLS_CC, E_NOTICE, "Data does not seem to be chunked encoded");
 				memcpy(*decoded, encoded, encoded_len);
 				*decoded_len = encoded_len;
-				decoded[*decoded_len] = '\0';
 				return encoded + encoded_len;
 			} else {
 				efree(*decoded);
-- 
1.9.2

