Index: trunk/inc/common.class.php
===================================================================
--- trunk/inc/common.class.php	(revision 321)
+++ trunk/inc/common.class.php	(revision 322)
@@ -118,7 +118,7 @@
    **/
    function getTabNameForItem(CommonGLPI $item, $withtemplate=0) {
       global $LANG;
-      
+
       if(!isset($withtemplate) || empty($withtemplate)) {
          return $LANG['plugin_pdf']['title'][1];
       }
@@ -310,6 +310,7 @@
 
                if (!is_integer($itemtype)
                    && $itemtype != 'empty'
+                   && method_exists($itemtype, "displayTabContentForPdf")
                    && ($obj = getItemForItemtype($itemtype))) {
                   if ($obj->displayTabContentForPdf($this->pdf, $this->obj, $tabnum)) {
                      continue;
Index: trunk/inc/computer.class.php
===================================================================
--- trunk/inc/computer.class.php	(revision 321)
+++ trunk/inc/computer.class.php	(revision 322)
@@ -44,6 +44,7 @@
 
       $onglets = parent::defineAllTabs($options);
       unset($onglets['OcsLink$1']); // TODO add method to print OCS
+      unset($onglets['Item_Problem$1']); // TODO add method to print linked Problems
       return $onglets;
    }
 
