diff -up ./phpdcd.rpm ./phpdcd
--- ./phpdcd.rpm	2014-05-04 09:36:58.000000000 +0200
+++ ./phpdcd	2014-05-04 09:38:12.000000000 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 /**
  * phpdcd
@@ -42,23 +42,7 @@
  * @since     File available since Release 1.0.0
  */
 
-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
-}
-
-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require 'SebastianBergmann/PHPDCD/Autoload.php';
 
 $application = new SebastianBergmann\PHPDCD\CLI\Application;
 $application->run();
