diff --git a/tests/console_commandline_addargument.phpt b/tests/console_commandline_addargument.phpt
index eb2e3af..7f7f689 100644
--- a/tests/console_commandline_addargument.phpt
+++ b/tests/console_commandline_addargument.phpt
@@ -27,7 +27,7 @@ $parser->addArgument('Some invalid name');
 --EXPECTF--
 array(4) {
   ["arg1"]=>
-  object(Console_CommandLine_Argument)#5 (8) {
+  object(Console_CommandLine_Argument)#%d (8) {
     ["multiple"]=>
     bool(false)
     ["optional"]=>
@@ -48,7 +48,7 @@ array(4) {
     }
   }
   ["arg2"]=>
-  object(Console_CommandLine_Argument)#6 (8) {
+  object(Console_CommandLine_Argument)#%d (8) {
     ["multiple"]=>
     bool(true)
     ["optional"]=>
@@ -69,7 +69,7 @@ array(4) {
     }
   }
   ["arg3"]=>
-  object(Console_CommandLine_Argument)#7 (8) {
+  object(Console_CommandLine_Argument)#%d (8) {
     ["multiple"]=>
     bool(true)
     ["optional"]=>
@@ -90,7 +90,7 @@ array(4) {
     }
   }
   ["arg4"]=>
-  object(Console_CommandLine_Argument)#8 (8) {
+  object(Console_CommandLine_Argument)#%d (8) {
     ["multiple"]=>
     bool(false)
     ["optional"]=>
diff --git a/tests/console_commandline_addoption.phpt b/tests/console_commandline_addoption.phpt
index 73da0ea..1386121 100644
--- a/tests/console_commandline_addoption.phpt
+++ b/tests/console_commandline_addoption.phpt
@@ -31,7 +31,7 @@ var_dump($parser->options);
 --EXPECTF--
 array(4) {
   ["opt1"]=>
-  object(Console_CommandLine_Option)#5 (14) {
+  object(Console_CommandLine_Option)#%d (14) {
     ["short_name"]=>
     string(2) "-a"
     ["long_name"]=>
@@ -65,7 +65,7 @@ array(4) {
     }
   }
   ["opt2"]=>
-  object(Console_CommandLine_Option)#6 (14) {
+  object(Console_CommandLine_Option)#%d (14) {
     ["short_name"]=>
     string(2) "-b"
     ["long_name"]=>
@@ -105,7 +105,7 @@ array(4) {
     }
   }
   ["list_opt2"]=>
-  object(Console_CommandLine_Option)#7 (14) {
+  object(Console_CommandLine_Option)#%d (14) {
     ["short_name"]=>
     NULL
     ["long_name"]=>
@@ -148,7 +148,7 @@ array(4) {
     }
   }
   ["opt3"]=>
-  object(Console_CommandLine_Option)#8 (14) {
+  object(Console_CommandLine_Option)#%d (14) {
     ["short_name"]=>
     NULL
     ["long_name"]=>
diff --git a/tests/console_commandline_parse_17.phpt b/tests/console_commandline_parse_17.phpt
index d3f8762..23e1444 100644
--- a/tests/console_commandline_parse_17.phpt
+++ b/tests/console_commandline_parse_17.phpt
@@ -18,8 +18,8 @@ try {
 }
 
 ?>
---EXPECT--
-object(Console_CommandLine_Result)#18 (4) {
+--EXPECTF--
+object(Console_CommandLine_Result)#%d (4) {
   ["options"]=>
   array(11) {
     ["true"]=>
diff --git a/tests/console_commandline_parse_18.phpt b/tests/console_commandline_parse_18.phpt
index 6efc149..93250c4 100644
--- a/tests/console_commandline_parse_18.phpt
+++ b/tests/console_commandline_parse_18.phpt
@@ -18,8 +18,8 @@ try {
 }
 
 ?>
---EXPECT--
-object(Console_CommandLine_Result)#16 (4) {
+--EXPECTF--
+object(Console_CommandLine_Result)#%d (4) {
   ["options"]=>
   array(4) {
     ["verbose"]=>
@@ -37,7 +37,7 @@ object(Console_CommandLine_Result)#16 (4) {
   ["command_name"]=>
   string(7) "install"
   ["command"]=>
-  object(Console_CommandLine_Result)#19 (4) {
+  object(Console_CommandLine_Result)#%d (4) {
     ["options"]=>
     array(2) {
       ["force"]=>
diff --git a/tests/console_commandline_parse_7.phpt b/tests/console_commandline_parse_7.phpt
index 1bfb6fd..de5c6af 100644
--- a/tests/console_commandline_parse_7.phpt
+++ b/tests/console_commandline_parse_7.phpt
@@ -18,8 +18,8 @@ try {
 }
 
 ?>
---EXPECT--
-object(Console_CommandLine_Result)#18 (4) {
+--EXPECTF--
+object(Console_CommandLine_Result)#%d (4) {
   ["options"]=>
   array(11) {
     ["true"]=>
diff --git a/tests/console_commandline_parse_8.phpt b/tests/console_commandline_parse_8.phpt
index ceba877..0925bbe 100644
--- a/tests/console_commandline_parse_8.phpt
+++ b/tests/console_commandline_parse_8.phpt
@@ -19,8 +19,8 @@ try {
 }
 
 ?>
---EXPECT--
-object(Console_CommandLine_Result)#18 (4) {
+--EXPECTF--
+object(Console_CommandLine_Result)#%d (4) {
   ["options"]=>
   array(11) {
     ["true"]=>
