diff -up enigmail/package/install.rdf.orig enigmail/package/install.rdf
--- enigmail/package/install.rdf.orig	2011-07-26 17:53:13.000000000 +0200
+++ enigmail/package/install.rdf	2011-07-30 19:10:04.392345142 +0200
@@ -27,14 +27,6 @@
         <em:maxVersion>2.2.*</em:maxVersion>
       </Description>
     </em:targetApplication>
-    <!-- em:targetApplication>
-      < ! - - Spicebird - - >
-      <Description>
-        <em:id>{ee53ece0-255c-4cc6-8a7e-81a8b6e5ba2c}</em:id>
-        <em:minVersion>0.7</em:minVersion>
-        <em:maxVersion>0.8</em:maxVersion>
-      </Description>
-    </em:targetApplication -->
 
 <!-- targetPlatform placeholder -->
 
