diff -up enigmail-1.1.2/package/enigmail.js.perm enigmail-1.1.2/package/enigmail.js
--- enigmail-1.1.2/package/enigmail.js.perm	2010-06-30 19:18:50.000000000 +0200
+++ enigmail-1.1.2/package/enigmail.js	2010-06-30 19:19:22.000000000 +0200
@@ -1231,7 +1231,6 @@ function () {
                .getInstallLocation(ENIGMAIL_EXTENSION_ID);
       var extensionLoc = installLoc.getItemFile(ENIGMAIL_EXTENSION_ID, "wrappers");
       extensionLoc.append("gpg-agent-wrapper.sh");
-			extensionLoc.permissions=0755;
       agentProcess = Components.classes[NS_PROCESS_UTIL_CONTRACTID].createInstance(Components.interfaces.nsIProcess);
       agentProcess.init(extensionLoc);
       agentProcess.run(true, [ "stop", this.gpgAgentProcess ], 2);
@@ -1758,7 +1757,6 @@ function (domWindow) {
                    .getInstallLocation(ENIGMAIL_EXTENSION_ID);
           var extensionLoc = installLoc.getItemFile(ENIGMAIL_EXTENSION_ID, "wrappers");
           extensionLoc.append("gpg-agent-wrapper.sh");
-          extensionLoc.permissions=0755;
           args.unshift(command.path);
           args.unshift("start");
           command = extensionLoc;
