diff -up comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc.include comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc
--- comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc.include	2009-09-16 02:57:07.000000000 +0200
+++ comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc	2009-10-12 09:47:59.000000000 +0200
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <stdio.h>
 
 #include <cassert>
 #include <cstdlib>
diff -up comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc.include comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc
--- comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc.include	2009-09-16 02:57:07.000000000 +0200
+++ comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc	2009-10-12 09:47:59.000000000 +0200
@@ -37,6 +37,7 @@
 #include <unistd.h>
 #include <sys/utsname.h>
 #include <sys/wait.h>
+#include <stdio.h>
 
 #include <cstdlib>
 #include <ctime>
diff -up comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc.include comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc
--- comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc.include	2009-09-16 02:57:07.000000000 +0200
+++ comm-central/mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc	2009-10-12 10:09:56.000000000 +0200
@@ -41,6 +41,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <algorithm>
+#include <stdio.h>
 
 #include <functional>
 #include <list>
