diff -up comm-release/config/baseconfig.mk.dir comm-release/config/baseconfig.mk
--- comm-release/config/baseconfig.mk.dir	2012-11-20 10:35:27.280236537 +0100
+++ comm-release/config/baseconfig.mk	2012-11-20 10:35:45.208242562 +0100
@@ -2,7 +2,7 @@ INCLUDED_AUTOCONF_MK = 1
 
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 MOZILLA_SRCDIR = $(topsrcdir)/mozilla
 MOZDEPTH = $(DEPTH)/mozilla
diff -up comm-release/mozilla/config/baseconfig.mk.dir comm-release/mozilla/config/baseconfig.mk
--- comm-release/mozilla/config/baseconfig.mk.dir	2012-11-16 22:16:02.000000000 +0100
+++ comm-release/mozilla/config/baseconfig.mk	2012-11-20 10:34:30.610216973 +0100
@@ -2,7 +2,7 @@ INCLUDED_AUTOCONF_MK = 1
 
 includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 idldir = $(datadir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 DIST = $(DEPTH)/dist
 
diff -up comm-release/mozilla/js/src/config/baseconfig.mk.dir comm-release/mozilla/js/src/config/baseconfig.mk
--- comm-release/mozilla/js/src/config/baseconfig.mk.dir	2012-11-16 22:16:14.000000000 +0100
+++ comm-release/mozilla/js/src/config/baseconfig.mk	2012-11-20 10:34:32.511217747 +0100
@@ -1,6 +1,6 @@
 INCLUDED_AUTOCONF_MK = 1
 
-installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)
 sdkdir = $(libdir)/$(MOZ_APP_NAME)-devel-$(MOZ_APP_VERSION)
 
 ifneq (,$(filter /%,$(TOP_DIST)))
