Modified: trunk/qelectrotech.pro
===================================================================
--- trunk/qelectrotech.pro	2015-10-03 15:41:59 UTC (rev 4223)
+++ trunk/qelectrotech.pro	2015-10-03 16:28:11 UTC (rev 4224)
@@ -203,7 +203,4 @@
 	DEFINES += QET_COMMON_COLLECTION_PATH_RELATIVE_TO_BINARY_PATH
 }
 
-# Compilers-specific options
-unix {
-	QMAKE_COPY_DIR = 'cp -f -r --preserve=timestamps'
-}
+
