diff -up libdvdcss-1.2.11/src/css.c.orig libdvdcss-1.2.11/src/css.c
--- libdvdcss-1.2.11/src/css.c.orig	2012-02-18 17:43:34.000000000 +0100
+++ libdvdcss-1.2.11/src/css.c	2012-02-18 17:45:01.000000000 +0100
@@ -2,7 +2,7 @@
  * css.c: Functions for DVD authentication and descrambling
  *****************************************************************************
  * Copyright (C) 1999-2008 VideoLAN
- * $Id: css.c 239 2011-03-23 14:59:36Z jb $
+ * $Id: css.c 249 2012-02-18 16:22:03Z jb $
  *
  * Authors: Stéphane Borel <stef@via.ecp.fr>
  *          Håkan Hjort <d95hjort@dtek.chalmers.se>
@@ -138,8 +138,8 @@ int _dvdcss_test( dvdcss_t dvdcss )
 
     if( i_ret < 0 )
     {
-        print_error( dvdcss, "css error: could not get RPC status" );
-        return -2;
+        print_error( dvdcss, "css error: could not get RPC status. Assuming RPC-I drive." );
+        i_type = i_mask = i_rpc = 0;
     }
 
     switch( i_rpc )
