--- php-5.4.8/ext/mysqli/config.m4.orig	2012-10-25 14:36:21.560016333 +0200
+++ php-5.4.8/ext/mysqli/config.m4	2012-10-25 14:37:04.706164258 +0200
@@ -81,5 +81,8 @@
   if test "$PHP_MYSQLI" = "yes" || test "$PHP_MYSQLI" = "mysqlnd"; then
     PHP_ADD_EXTENSION_DEP(mysqli, mysqlnd)
     AC_DEFINE([MYSQLI_USE_MYSQLND], 1, [Whether mysqlnd is enabled])
+    PHP_INSTALL_HEADERS([ext/mysqli/mysqli_mysqlnd.h])
+  else
+    PHP_INSTALL_HEADERS([ext/mysqli/mysqli_libmysql.h])
   fi
 fi
