--- phpcov.rpm	2015-10-05 14:23:45.648930666 +0200
+++ phpcov	2015-10-05 14:23:51.362957863 +0200
@@ -9,23 +9,16 @@
  * file that was distributed with this source code.
  */
 
-$loaded = false;
-
-foreach (array(__DIR__ . '/../../autoload.php', __DIR__ . '/vendor/autoload.php') as $file) {
-    if (file_exists($file)) {
-        require $file;
-        $loaded = true;
-        break;
-    }
+// Ensure correct include_path for RHSCL
+$inc = get_include_path();
+$dirs = explode(':', $inc);
+if (!in_array('/usr/share/php', $dirs)) {
+    $dirs[] = '/usr/share/php';
+    set_include_path(implode(':', $dirs));
 }
+unset ($inc, $dirs);
 
-if (!$loaded) {
-    die(
-        'You need to set up the project dependencies using the following commands:' . PHP_EOL .
-        'wget http://getcomposer.org/composer.phar' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-}
+require 'SebastianBergmann/PHPCOV/autoload.php';
 
 $application = new SebastianBergmann\PHPCOV\Application;
 $application->run();
