--- Symfony/Component/Filesystem/Tests/bootstrap.php	2012-09-20 03:42:09.000000000 -0400
+++ Symfony/Component/Filesystem/Tests/bootstrap.php	2012-10-06 21:54:13.360908929 -0400
@@ -10,9 +10,18 @@
  */
 
 spl_autoload_register(function ($class) {
-    if (0 === strpos(ltrim($class, '/'), 'Symfony\Component\Filesystem')) {
-        if (file_exists($file = __DIR__.'/../'.substr(str_replace('\\', '/', $class), strlen('Symfony\Component\Filesystem')).'.php')) {
-            require_once $file;
+    if (0 === strpos(ltrim($class, '/'), 'Symfony\Component')) {
+        $file = substr(str_replace('\\', '/', $class), strlen('Symfony\Component')).'.php';
+        if (file_exists(__DIR__.'/../..'.$file)) {
+            // Load from source tree
+            require_once __DIR__.'/../..'.$file;
+        } else {
+            try {
+                // Try loading from incude path
+                require_once 'Symfony/Component'.$file;
+            } catch (Exception $e) {
+                // Fail silently so class not found fatal error still raised
+            }
         }
     }
 });
