commit 9252e8e3e4d3659dd54a4ea1632ec72d194c3ed2
Author: David Durieux <d.durieux@siprossii.com>
Date:   Fri Sep 27 11:02:58 2013 +0200

    Fix memory linit on install and update. closes #2259

diff --git /install/install.php b/install/install.php
index 7beb14a..ed6543a 100644
--- fusioninventory/install/install.php
+++ fusioninventory/install/install.php
@@ -43,6 +43,10 @@
 function pluginFusioninventoryInstall($version, $migration='') {
    global $DB;
 
+   ini_set("memory_limit", "-1");
+   ini_set("max_execution_time", "0");
+
+   
    if ($migration == '') {
       $migration = new Migration($version);
    }
diff --git a/install/update.php b/install/update.php
index 9431228..f2c686f 100644
--- fusioninventory/install/update.php
+++ fusioninventory/install/update.php
@@ -199,6 +199,8 @@ function pluginFusioninventoryUpdate($current_version, $migrationname='Migration
    global $DB;
 
    ini_set("max_execution_time", "0");
+   ini_set("memory_limit", "-1");
+
 
    foreach (glob(GLPI_ROOT.'/plugins/fusioninventory/inc/*.php') as $file) {
       require_once($file);
