Index: trunk/inc/detail.class.php
===================================================================
--- trunk/inc/detail.class.php	(revision 111)
+++ trunk/inc/detail.class.php	(revision 112)
@@ -131,8 +131,8 @@
       global $DB;
 
       $detail = new self();
-      foreach ($DB->query("glpi_plugin_massocsimport_details", 
-                          "`plugin_massocsimport_threads_id` = '$threads_id'") as $data) {
+      foreach ($DB->request($detail->getTable(),
+                            array('plugin_massocsimport_threads_id' => $threads_id)) as $data) {
          $detail->delete($data);
       }
    }
@@ -164,4 +164,4 @@
    }
 
 }
-?>
\ No newline at end of file
+?>
