diff --git a/src/PHPDocsMD/Console/CLI.php b/src/PHPDocsMD/Console/CLI.php
index 6befd28..8f4586b 100644
--- a/src/PHPDocsMD/Console/CLI.php
+++ b/src/PHPDocsMD/Console/CLI.php
@@ -14,8 +14,7 @@ class CLI extends Application {

     public function __construct()
     {
-        $json = json_decode(file_get_contents(__DIR__.'/../../../composer.json'));
-        parent::__construct('PHP Markdown Documentation Generator', $json->version);
+        parent::__construct('PHP Markdown Documentation Generator', '__VERSION__');
     }

     /**
