
Upstream-HEAD: vendor
Upstream-2.0: vendor
Upstream-Status: vendor-specific change

--- httpd-2.0.48/server/core.c.release
+++ httpd-2.0.48/server/core.c
@@ -2758,7 +2758,7 @@
         ap_add_version_component(pconf, AP_SERVER_BASEPRODUCT "/" AP_SERVER_MAJORVERSION);
     }
     else {
-        ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (" PLATFORM ")");
+        ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (@RELEASE@)");
     }
 
     /*
