
Upstream-HEAD: vendor
Upstream-2.0: vendor
Upstream-Status: vendor-specific change

--- httpd-2.4.3/server/core.c.release
+++ httpd-2.4.3/server/core.c
@@ -3189,7 +3189,7 @@ static void set_banner(apr_pool_t *pconf
         ap_add_version_component(pconf, AP_SERVER_BASEPRODUCT "/" AP_SERVER_MAJORVERSION);
     }
     else {
-        ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (" PLATFORM ")");
+        ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (@RELEASE@)");
     }
 
     /*
