--- httpd-2.2.9/configure.in.deplibs
+++ httpd-2.2.9/configure.in
@@ -588,9 +588,8 @@ APACHE_HELP_STRING(--with-suexec-umask,u
   AC_DEFINE_UNQUOTED(AP_SUEXEC_UMASK, 0$withval, [umask for suexec'd process] ) ] )
 
 dnl APR should go after the other libs, so the right symbols can be picked up
-apulinklibs="`$apu_config --avoid-ldap --link-libtool --libs`" \
- || apulinklibs="`$apu_config --link-libtool --libs`"
-AP_LIBS="$AP_LIBS $apulinklibs `$apr_config --link-libtool --libs`" 
+apulinklibs="`$apu_config --link-libtool`"
+AP_LIBS="$AP_LIBS $apulinklibs `$apr_config --link-libtool`" 
 APACHE_SUBST(AP_LIBS)
 APACHE_SUBST(AP_BUILD_SRCLIB_DIRS)
 APACHE_SUBST(AP_CLEAN_SRCLIB_DIRS)
