diff -up pecl_http-2.0.0dev7/php_http_message.h.ORIG pecl_http-2.0.0dev7/php_http_message.h
--- pecl_http-2.0.0dev7/php_http_message.h.ORIG	2012-03-09 16:56:03.000000000 +0100
+++ pecl_http-2.0.0dev7/php_http_message.h	2012-03-09 16:56:09.000000000 +0100
@@ -25,7 +25,6 @@
 		}
 
 typedef php_http_info_type_t php_http_message_type_t;
-typedef struct php_http_message php_http_message_t;
 
 struct php_http_message {
 	PHP_HTTP_INFO_IMPL(http, type)
