--- bin/phpspec.rpm	2015-04-19 06:39:34.900178171 +0200
+++ bin/phpspec	2015-04-19 06:41:59.521743839 +0200
@@ -5,22 +5,9 @@
 
-if (is_file($autoload = getcwd() . '/vendor/autoload.php')) {
-    require $autoload;
-} elseif (is_file($autoload = getcwd() . '/../../autoload.php')) {
-    require $autoload;
+if (!ini_get('date.timezone')) {
+    ini_set('date.timezone', 'UTC');
 }
 
-if (is_file($autoload = __DIR__ . '/../vendor/autoload.php')) {
-    require($autoload);
-} elseif (is_file($autoload = __DIR__ . '/../../../autoload.php')) {
-    require($autoload);
-} else {
-    fwrite(STDERR,
-        'You must set up the project dependencies, run the following commands:' . PHP_EOL .
-        'curl -s http://getcomposer.org/installer | php' . PHP_EOL .
-        'php composer.phar install' . PHP_EOL
-    );
-    exit(1);
-}
+require_once 'PhpSpec/autoload.php';
 
 $app = new PhpSpec\Console\Application(PHPSPEC_VERSION);
 $app->run();
