diff -up bin/phpspec.rpm bin/phpspec
--- bin/phpspec.rpm	2016-07-18 10:56:02.330062305 +0200
+++ bin/phpspec	2016-07-18 10:57:56.042682157 +0200
@@ -3,24 +3,20 @@
 
 call_user_func(function ($version) {
 
-    if (is_file($autoload = getcwd() . '/vendor/autoload.php')) {
-        require $autoload;
-    } elseif (is_file($autoload = getcwd() . '/../../autoload.php')) {
-        require $autoload;
+    if (!ini_get('date.timezone')) {
+        ini_set('date.timezone', 'UTC');
     }
 
-    if (is_file($autoload = __DIR__ . '/../vendor/autoload.php')) {
-        require($autoload);
-    } elseif (is_file($autoload = __DIR__ . '/../../../autoload.php')) {
-        require($autoload);
-    } else {
-        fwrite(STDERR,
-            'You must set up the project dependencies, run the following commands:' . PHP_EOL .
-            'curl -s http://getcomposer.org/installer | php' . PHP_EOL .
-            'php composer.phar install' . PHP_EOL
-        );
-        exit(1);
+    // Ensure correct include_path for SCL
+    $inc = get_include_path();
+    $dirs = explode(':', $inc);
+    if (!in_array('/usr/share/php', $dirs)) {
+        $dirs[] = '/usr/share/php';
+        set_include_path(implode(':', $dirs));
     }
+    unset ($inc, $dirs);
+
+    require_once 'PhpSpec/autoload.php';
 
     $app = new PhpSpec\Console\Application($version);
     $app->run();
