From 725b11f7e588e1860c6df866e1fe6df9c9378290 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@famillecollet.com>
Date: Thu, 2 Jul 2015 14:17:54 +0200
Subject: [PATCH] Fix autoloader

---
 resources/autoload.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/resources/autoload.php b/resources/autoload.php
index 022c1ac..aac276a 100644
--- a/resources/autoload.php
+++ b/resources/autoload.php
@@ -22,7 +22,7 @@ function ($class) {
             return;
         }
         $relative_class = substr($class, $len);
-        $file = dirname(__DIR__).'/'.str_replace('\\', '/', $relative_class).'.php';
+        $file = dirname(dirname(__DIR__)).'/'.str_replace('\\', '/', $relative_class).'.php';
         if (file_exists($file)) {
             require $file;
         }
