From 9024139f4bf2f029a05ffaeaaf2158afef342785 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Sat, 28 Jan 2017 10:29:55 -0500
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/issues/363

---
 Magick++/tests/color.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Magick++/tests/color.cpp b/Magick++/tests/color.cpp
index 0ce2ebc..4d12324 100644
--- a/Magick++/tests/color.cpp
+++ b/Magick++/tests/color.cpp
@@ -40,7 +40,7 @@ int main( int /*argc*/, char **argv)
       struct colorStr colorMap [] =
       { 
 	{ "red", 1,0,0 },
-	{ "green", 0,0.5019607843137256,0 },
+	{ "green", 0,0.5019607843137255,0 },
 	{ "blue", 0,0,1 },
 	{ "black", 0,0,0 },
 	{ "white", 1,1,1 },
From a5c463d59268d2557d90565415b8e5047b25e7d2 Mon Sep 17 00:00:00 2001
From: Dirk Lemstra <dirk@git.imagemagick.org>
Date: Mon, 30 Jan 2017 13:28:58 +0100
Subject: [PATCH] Use lime instead of green to avoid rounding issues (fixes
 #363).

---
 Magick++/tests/color.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Magick++/tests/color.cpp b/Magick++/tests/color.cpp
index 4d12324..138ed21 100644
--- a/Magick++/tests/color.cpp
+++ b/Magick++/tests/color.cpp
@@ -40,7 +40,7 @@ int main( int /*argc*/, char **argv)
       struct colorStr colorMap [] =
       { 
 	{ "red", 1,0,0 },
-	{ "green", 0,0.5019607843137255,0 },
+	{ "lime", 0,1,0 },
 	{ "blue", 0,0,1 },
 	{ "black", 0,0,0 },
 	{ "white", 1,1,1 },
