diff -up gmagick-1.1.0RC2/gmagickpixel_methods.c.orig gmagick-1.1.0RC2/gmagickpixel_methods.c
--- gmagick-1.1.0RC2/gmagickpixel_methods.c.orig	2012-01-21 07:42:13.727957934 +0100
+++ gmagick-1.1.0RC2/gmagickpixel_methods.c	2012-01-21 07:43:09.328959265 +0100
@@ -82,7 +82,7 @@ PHP_METHOD(gmagickpixel, getcolor)
 	zend_bool as_array = 0, normalise_array = 0;
 
 	/* Parse parameters given to function */
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|b", &as_array, &normalise_array) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "|bb", &as_array, &normalise_array) == FAILURE) {
 		return;
 	}
 
