diff -up gd-2.0.35/tests/CMakeLists.txt.tests gd-2.0.35/tests/CMakeLists.txt
--- gd-2.0.35/tests/CMakeLists.txt.tests	2012-08-28 10:49:51.172059567 +0200
+++ gd-2.0.35/tests/CMakeLists.txt	2012-08-28 10:49:43.251931396 +0200
@@ -23,8 +23,8 @@ if (BUILD_TEST)
 	include_directories (BEFORE ${GD_INCLUDE_DIR} "${GDTEST_SOURCE_DIR}")
 	link_directories ("${GD_LIBS_DIR}" "${GDTEST_BINARY_DIR}/gdtest")
 
-	#FIND_LIBRARY(GD_LIB "gd" PATHS ${GD_LIBS_DIR})
-	#message(STATUS "GD_LIB: ${GD_LIB}")
+	FIND_LIBRARY(GD_LIB "gd" PATHS ${GD_LIBS_DIR})
+	message(STATUS "GD_LIB: ${GD_LIB}")
 
 	SET(TESTS_DIRS
 		gdtest
@@ -33,7 +33,7 @@ if (BUILD_TEST)
 		gdimagecolorexact
 		gdimagecolorresolve
 		#gdimageellipse
-		gdimageline
+		#gdimageline
 		gdimagefilledellipse
 		gdimagecopyrotated
 		gdtiled
