From a184630682c25e314979b459950ec6c81fa82a5f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Sur=C3=BD?= <ondrej@sury.org>
Date: Fri, 20 May 2016 10:15:57 +0200
Subject: [PATCH] Initialize full_filename in tests/gdimagefile/gdnametest.c

---
 tests/gdimagefile/gdnametest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/gdimagefile/gdnametest.c b/tests/gdimagefile/gdnametest.c
index 2be648e..a339e69 100644
--- a/tests/gdimagefile/gdnametest.c
+++ b/tests/gdimagefile/gdnametest.c
@@ -84,7 +84,7 @@ do_test() {
     for (n = 0; names[n].nm; n++) {
         gdImagePtr orig, copy;
         int status;
-        char *full_filename;
+        char *full_filename = NULL;
 
         /* Some image readers are buggy and crash the program so we
          * skip them.  Bug fixers should remove these from the list of
