diff -up src/Generator.php.rpm src/Generator.php
--- src/Generator.php.rpm	2017-02-07 15:28:34.953786524 +0100
+++ src/Generator.php	2017-02-07 15:28:38.772803856 +0100
@@ -13,6 +13,9 @@ use Doctrine\Instantiator\Exception\Inva
 use Doctrine\Instantiator\Exception\UnexpectedValueException as InstantiatorUnexpectedValueException;
 use PHPUnit\Util\InvalidArgumentHelper;
 
+# temporary workaround
+require_once 'Text/Template.php';
+
 /**
  * Mock Object Code Generator
  */
