From bb70609b26b363821d21617bb49274d512836ee9 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 28 Nov 2016 15:13:53 +0100
Subject: [PATCH 1/4] better value for include_path (other path can be useful,
 e.g. phpunit autoloader)

---
 tests/AllTests.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/AllTests.php b/tests/AllTests.php
index 001008a..d96aa4d 100644
--- a/tests/AllTests.php
+++ b/tests/AllTests.php
@@ -60,7 +60,7 @@
 require_once 'Net_DNS2_ResolverTest.php';
 require_once 'Net_DNS2_DNSSECTest.php';
 
-set_include_path('..:.');
+set_include_path('..:' . get_include_path());
 
 /**
  * This test suite assumes that Net_DNS2 will be in the include path, otherwise it


