diff --git a/tests/Sabre/CalDAV/GetEventsByTimerangeTest.php b/tests/Sabre/CalDAV/GetEventsByTimerangeTest.php
index 45531f2..2e9637d 100644
--- a/tests/Sabre/CalDAV/GetEventsByTimerangeTest.php
+++ b/tests/Sabre/CalDAV/GetEventsByTimerangeTest.php
@@ -32,7 +32,7 @@ VERSION:2.0
 BEGIN:VEVENT
 CREATED:20120313T142342Z
 UID:171EBEFC-C951-499D-B234-7BA7D677B45D
-DTEND;TZID=Europe/Berlin:20120227T000000
+DTEND;TZID=Europe/Berlin:20120227T010000
 TRANSP:OPAQUE
 SUMMARY:Monday 0h
 DTSTART;TZID=Europe/Berlin:20120227T000000
