From e81dd7342121398de1c6a55380a36dad80a74875 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sun, 9 Mar 2014 07:43:59 +0100
Subject: [PATCH 1/2] fix build: synchro.c:133:2: error: 'else' without a
 previous 'if'

---
 src/synchro.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/synchro.c b/src/synchro.c
index 07360be..54efd46 100644
--- a/src/synchro.c
+++ b/src/synchro.c
@@ -128,8 +128,9 @@ void pthreads_synchro_block(zval *this_ptr, zend_fcall_info *info, zend_fcall_in
 	zend_fcall_info_args_clear(info, 1);
 	
 	/* return the result */
-	if (!failed && retval)
+	if (!failed && retval) {
 		ZVAL_ZVAL(return_value, retval, 1, 1);
+	}
 	else ZVAL_NULL(return_value);
 } /* }}} */
 
-- 
1.8.3.1

From a45e52eafdbc6e3a394d8f72868f9104a7099f31 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sun, 9 Mar 2014 07:45:18 +0100
Subject: [PATCH 2/2] unused variables

---
 classes/pool.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/classes/pool.h b/classes/pool.h
index 144f717..d7e2f0d 100644
--- a/classes/pool.h
+++ b/classes/pool.h
@@ -278,7 +278,6 @@ static inline int pool_collect_function(void *bucket, void *argument TSRMLS_DC)
 	zval **task = (zval**) bucket;
 	pool_call_t *call = (pool_call_t*) argument;
 	zval *remove = NULL;
-	zend_ulong index = 0L;
 	int result = ZEND_HASH_APPLY_KEEP;
 	
 	call->fci.retval_ptr_ptr = &remove;
@@ -320,7 +319,6 @@ PHP_METHOD(Pool, collect) {
 	work = zend_read_property(Z_OBJCE_P(getThis()), getThis(), ZEND_STRL("work"), 1 TSRMLS_CC);
 	
 	if (Z_TYPE_P(work) == IS_ARRAY && zend_hash_num_elements(Z_ARRVAL_P(work))) {
-		zval **task = NULL;
 		pool_call_t call;
 		
 		call.fci = fci;
-- 
1.8.3.1

