diff --git a/src/ssl/ssl_common.c b/src/ssl/ssl_common.c
index 3a5327b..4b18d74 100644
--- a/src/ssl/ssl_common.c
+++ b/src/ssl/ssl_common.c
@@ -156,7 +156,9 @@ iotssl_log_errors(lcbio_XSSL *xs)
         if (ERR_GET_LIB(curerr) == ERR_LIB_SSL) {
             switch (ERR_GET_REASON(curerr)) {
             case SSL_R_CERTIFICATE_VERIFY_FAILED:
+#ifdef SSL_R_MISSING_VERIFY_MESSAGE
             case SSL_R_MISSING_VERIFY_MESSAGE:
+#endif
                 xs->errcode = LCB_SSL_CANTVERIFY;
                 break;
 
