--- MagickWandForPHP-1.0.9/magickwand_inc.h.php54	2011-11-14 01:26:39.000000000 +0100
+++ MagickWandForPHP-1.0.9/magickwand_inc.h	2011-11-26 14:37:53.999059934 +0100
@@ -61,7 +61,7 @@
 
 static MagickBooleanType
 		   MW_zend_register_resource(
-			   const MagickBooleanType verify, const void *obj_ptr, zval *zval_p, const int le_zend_rsrc_type, int *return_rsrc_id )
+			   const MagickBooleanType verify, const void *obj_ptr, zval *zval_p, const int le_zend_rsrc_type, int *return_rsrc_id TSRMLS_DC)
 {
 	if ( verify == MagickFalse ) { return MagickFalse; }
 
@@ -188,7 +188,7 @@
 /* ************************************************************************************************************** */
 
 #define MW_ZEND_REGISTER_RESOURCE( ResType, obj, zval_p, return_rsrc_id )  \
-	MW_zend_register_resource( ResType ## _IS_GOOD( obj ), (void *) obj, zval_p, le_ ## ResType, return_rsrc_id )
+	MW_zend_register_resource( ResType ## _IS_GOOD( obj ), (void *) obj, zval_p, le_ ## ResType, return_rsrc_id TSRMLS_CC)
 
 /* ************************************************************************************************************** */
 
