From 18fccbd309f250fd5fe34cd9c6d6f5da67aed958 Mon Sep 17 00:00:00 2001
From: Anthon Pang <apang@softwaredevelopment.ca>
Date: Fri, 11 Sep 2015 23:11:09 -0400
Subject: [PATCH] Bump version to 0.3.1

---
 src/Version.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Version.php b/src/Version.php
index c650a26..efd45ca 100644
--- a/src/Version.php
+++ b/src/Version.php
@@ -18,5 +18,5 @@
  */
 class Version
 {
-    const VERSION = 'v0.3.0';
+    const VERSION = 'v0.3.1';
 }
