diff --git a/bin/pscss b/bin/pscss
index 451f0fb..38f0a8e 100755
--- a/bin/pscss
+++ b/bin/pscss
@@ -1,9 +1,14 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 
 error_reporting(E_ALL);
 
-include 'scss.inc.php';
+spl_autoload_register(function ($class) {
+    if (0 === strpos($class, 'Leafo\\ScssPhp\\')) {
+        $src = str_replace('\\', '/', $class) . '.php';
+        @include_once $src;
+    }
+});
 
 use Leafo\ScssPhp\Compiler;
 use Leafo\ScssPhp\Parser;
