diff -up mozilla-1.9.2/browser/installer/Makefile.in.version mozilla-1.9.2/browser/installer/Makefile.in
--- mozilla-1.9.2/browser/installer/Makefile.in.version	2009-11-09 01:33:57.000000000 +0100
+++ mozilla-1.9.2/browser/installer/Makefile.in	2009-11-13 13:48:15.000000000 +0100
@@ -43,6 +43,8 @@ VPATH		= @srcdir@
 
 include $(DEPTH)/config/autoconf.mk
 
+MOZ_APP_VERSION="__RPM_VERSION_INTERNAL__"
+
 NO_PKG_FILES = \
 	$(MOZ_APP_NAME)-config \
 	$(MOZ_APP_NAME)-bin.elf \
