diff -up mozilla-central/memory/mozalloc/mozalloc.h.throw mozilla-central/memory/mozalloc/mozalloc.h
--- mozilla-central/memory/mozalloc/mozalloc.h.throw	2010-07-28 09:25:12.596150733 +0200
+++ mozilla-central/memory/mozalloc/mozalloc.h	2010-07-28 09:25:28.325150319 +0200
@@ -214,7 +214,7 @@ MOZALLOC_EXPORT void* moz_valloc(size_t 
 #define MOZALLOC_THROW_IF_HAS_EXCEPTIONS throw()
 #endif
 
-#ifdef MOZ_CPP_EXCEPTIONS
+#if 1
 #define MOZALLOC_THROW_BAD_ALLOC throw(std::bad_alloc)
 #else
 #define MOZALLOC_THROW_BAD_ALLOC MOZALLOC_THROW_IF_HAS_EXCEPTIONS
