diff -up mozilla-1.9.2/browser/base/content/utilityOverlay.js.checkForUpdates mozilla-1.9.2/browser/base/content/utilityOverlay.js
--- mozilla-1.9.2/browser/base/content/utilityOverlay.js.checkForUpdates	2010-06-13 15:36:16.617677299 -0700
+++ mozilla-1.9.2/browser/base/content/utilityOverlay.js	2010-06-13 15:39:54.485387418 -0700
@@ -510,7 +510,7 @@ function buildHelpMenu()
   // Disable the UI if the update enabled pref has been locked by the 
   // administrator or if we cannot update for some other reason
   var checkForUpdates = document.getElementById("checkForUpdates");
-  var canCheckForUpdates = updates.canCheckForUpdates;
+  var canCheckForUpdates = updates.canCheckForUpdates && updates.canApplyUpdates;
   checkForUpdates.setAttribute("disabled", !canCheckForUpdates);
   if (!canCheckForUpdates)
     return; 
