diff -up firefox-8.0/mozilla-release/browser/app/profile/firefox.js.addons firefox-8.0/mozilla-release/browser/app/profile/firefox.js
--- firefox-8.0/mozilla-release/browser/app/profile/firefox.js.addons	2011-11-04 22:33:52.000000000 +0100
+++ firefox-8.0/mozilla-release/browser/app/profile/firefox.js	2011-11-15 10:35:31.446869396 +0100
@@ -77,7 +77,7 @@ pref("extensions.update.autoUpdateDefaul
 // Disable add-ons installed into the shared user and shared system areas by
 // default. This does not include the application directory. See the SCOPE
 // constants in AddonManager.jsm for values to use here
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 0);
 
 // Dictionary download preference
 pref("browser.dictionaries.download.url", "https://addons.mozilla.org/%LOCALE%/firefox/dictionaries/");
diff -up firefox-8.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm.addons firefox-8.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm
--- firefox-8.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm.addons	2011-11-04 22:34:18.000000000 +0100
+++ firefox-8.0/mozilla-release/toolkit/mozapps/extensions/XPIProvider.jsm	2011-11-15 10:42:17.866426274 +0100
@@ -1591,8 +1591,8 @@ var XPIProvider = {
       this.showUpgradeUI();
       flushCaches = true;
     }
-    else if (aAppChanged === undefined) {
-      // For new profiles we will never need to show the add-on selection UI
+    else {
+      // For all profiles we will never need to show the add-on selection UI
       Services.prefs.setBoolPref(PREF_SHOWN_SELECTION_UI, true);
     }
 
