diff -up mozilla-central/js/src/shell/js.cpp.sbrk mozilla-central/js/src/shell/js.cpp
--- mozilla-central/js/src/shell/js.cpp.sbrk	2010-07-21 02:52:15.000000000 +0200
+++ mozilla-central/js/src/shell/js.cpp	2010-07-28 08:57:39.423150378 +0200
@@ -1118,7 +1118,7 @@ GC(JSContext *cx, uintN argc, jsval *vp)
     char buf[256];
     JS_snprintf(buf, sizeof(buf), "before %lu, after %lu, break %08lx\n",
                 (unsigned long)preBytes, (unsigned long)cx->runtime->gcBytes,
-#ifdef HAVE_SBRK
+#if 0
                 (unsigned long)sbrk(0)
 #else
                 0
diff -up mozilla-central/js/src/xpconnect/shell/xpcshell.cpp.sbrk mozilla-central/js/src/xpconnect/shell/xpcshell.cpp
--- mozilla-central/js/src/xpconnect/shell/xpcshell.cpp.sbrk	2010-07-21 02:52:20.000000000 +0200
+++ mozilla-central/js/src/xpconnect/shell/xpcshell.cpp	2010-07-28 08:55:56.761535989 +0200
@@ -540,7 +540,7 @@ GC(JSContext *cx, JSObject *obj, uintN a
     JS_GC(cx);
     fprintf(gOutFile, "before %lu, after %lu, break %08lx\n",
            (unsigned long)preBytes, (unsigned long)rt->gcBytes,
-#if defined(XP_UNIX) && !defined(__SYMBIAN32__)
+#if 0 //defined(XP_UNIX) && !defined(__SYMBIAN32__)
            (unsigned long)sbrk(0)
 #else
            0
