diff -up mozilla-1.9.2/toolkit/crashreporter/google-breakpad/src/tools/linux/dump_syms/Makefile.in.static mozilla-1.9.2/toolkit/crashreporter/google-breakpad/src/tools/linux/dump_syms/Makefile.in
--- mozilla-1.9.2/toolkit/crashreporter/google-breakpad/src/tools/linux/dump_syms/Makefile.in.static	2010-09-14 20:53:09.000000000 +0200
+++ mozilla-1.9.2/toolkit/crashreporter/google-breakpad/src/tools/linux/dump_syms/Makefile.in	2010-10-13 14:21:14.000000000 +0200
@@ -64,6 +64,6 @@ FORCE_USE_PIC = 1
 
 #XXX: bug 554854 causes us to be unable to run binaries on the build slaves
 # due to them having an older libstdc++
-HOST_LDFLAGS += -static
+#HOST_LDFLAGS += -static
 
 include $(topsrcdir)/config/rules.mk
