diff --git a/ipc/chromium/src/base/basictypes.h b/ipc/chromium/src/base/basictypes.h
--- a/ipc/chromium/src/base/basictypes.h
+++ b/ipc/chromium/src/base/basictypes.h
@@ -14,29 +14,29 @@
 #error You_must_include_basictypes.h_before_prtypes.h!
 #endif
 
 #ifndef NO_NSPR_10_SUPPORT
 #define NO_NSPR_10_SUPPORT
 #define NO_NSPR_10_SUPPORT_SAVE
 #endif
 
-#include "nspr/prtypes.h"
+#include "prtypes.h"
 
 #ifdef NO_NSPR_10_SUPPORT_SAVE
 #undef NO_NSPR_10_SUPPORT_SAVE
 #undef NO_NSPR_10_SUPPORT
 #endif
 
 #ifdef _WIN32
 #undef _WIN32
 #define _WIN32_SAVE
 #endif
 
-#include "nspr/obsolete/protypes.h"
+#include "obsolete/protypes.h"
 
 #define _INT32
 #define _UINT32
 
 #ifdef _WIN32_SAVE
 #undef _WIN32_SAVE
 #define _WIN32
 #endif
