diff -up ./php-cs-fixer.rpm ./php-cs-fixer
--- ./php-cs-fixer.rpm	2017-02-09 09:20:40.000000000 +0100
+++ ./php-cs-fixer	2017-02-09 09:22:43.400071075 +0100
@@ -32,12 +32,7 @@ set_error_handler(function ($severity, $
     }
 });
 
-// installed via composer?
-if (file_exists($a = __DIR__.'/../../autoload.php')) {
-    require_once $a;
-} else {
-    require_once __DIR__.'/vendor/autoload.php';
-}
+require_once '/usr/share/php/PhpCsFixer/autoload.php';
 
 use PhpCsFixer\Console\Application;
 
diff -up ./tests/ProjectCodeTest.php.rpm ./tests/ProjectCodeTest.php
--- ./tests/ProjectCodeTest.php.rpm	2017-02-09 09:44:35.627516496 +0100
+++ ./tests/ProjectCodeTest.php	2017-02-09 09:44:39.935537666 +0100
@@ -234,6 +234,7 @@ final class ProjectCodeTest extends \PHP
             ->files()
             ->name('*.php')
             ->in(__DIR__.'/../src')
+            ->notName('autoload.php')
             ->exclude(array(
                 'Resources',
             ))
