diff -up Horde_Test-2.4.1/lib/Horde/Test/Autoload.php.rpm Horde_Test-2.4.1/lib/Horde/Test/Autoload.php
--- Horde_Test-2.4.1/lib/Horde/Test/Autoload.php.rpm	2014-05-06 14:15:13.781403553 +0200
+++ Horde_Test-2.4.1/lib/Horde/Test/Autoload.php	2014-05-06 14:16:25.522649055 +0200
@@ -41,11 +41,7 @@ class Horde_Test_Autoload
             return;
         }
 
-        if (file_exists(__DIR__ . '/vendor/autoload.php')) {
-            require __DIR__ . '/vendor/autoload.php';
-        } else {
-            require __DIR__ . '/../../../bundle/vendor/autoload.php';
-        }
+        require_once '/usr/share/php/PHPUnit/Autoload.php';
 
         spl_autoload_register(
             function($class) {
