diff --git a/bin/jsonlint b/bin/jsonlint
index 1b9272a..c15cfa6 100755
--- a/bin/jsonlint
+++ b/bin/jsonlint
@@ -10,20 +10,7 @@
  * file that was distributed with this source code.
  */
 
-function includeIfExists($file)
-{
-    if (file_exists($file)) {
-        return include $file;
-    }
-}
-
-if ((!$loader = includeIfExists(__DIR__.'/../vendor/autoload.php')) && (!$loader = includeIfExists(__DIR__.'/../../../autoload.php'))) {
-    $msg = 'You must set up the project dependencies, run the following commands:'.PHP_EOL.
-           'curl -sS https://getcomposer.org/installer | php'.PHP_EOL.
-           'php composer.phar install'.PHP_EOL;
-    fwrite(STDERR, $msg);
-    exit(1);
-}
+require_once '/usr/share/php/Seld/JsonLint/autoload.php';
 
 use Seld\JsonLint\JsonParser;
 
