Index: trunk/front/config.form.php
===================================================================
--- trunk/front/config.form.php	(révision 523)
+++ trunk/front/config.form.php	(révision 524)
@@ -47,7 +47,7 @@
 if ($plugin->isActivated("ocsinventoryng")) {
    Html::header(__("Automatic synchronization's configuration", 'ocsinventoryng'), '', "tools", "pluginocsinventoryngmenu", "config");
 
-   $config->showForm($_SERVER['PHP_SELF']);
+   $config->display(array('id' => 1));
 } else {
    Html::header(__('Setup'), '', "tools", "pluginocsinventoryngmenu", "config");
    echo "<div class='center'><br><br>";
Index: trunk/inc/config.class.php
===================================================================
--- trunk/inc/config.class.php	(révision 523)
+++ trunk/inc/config.class.php	(révision 524)
@@ -41,7 +41,7 @@
          switch ($item->getType()) {
             case __CLASS__ :
                //If connection to the OCS DB  is ok, and all rights are ok too
-               return array('1' => self::getTypeName());
+               return array('1' => __('Check OCSNG import script', 'ocsinventoryng'));
 
          }
       }
@@ -65,6 +65,7 @@
 
       $ong = array();
       $this->addDefaultFormTab($ong);
+      $this->addStandardTab(__CLASS__, $ong, $options);
       return $ong;
    }
 
