Crude workaround for bug #587170.  There are probably better ways ...


diff -Naur mysql-5.1.50.orig/storage/federated/Makefile.am mysql-5.1.50/storage/federated/Makefile.am
--- mysql-5.1.50.orig/storage/federated/Makefile.am	2010-08-03 13:24:29.000000000 -0400
+++ mysql-5.1.50/storage/federated/Makefile.am	2010-08-27 23:49:22.280394828 -0400
@@ -38,6 +38,7 @@
 ha_federated_la_CXXFLAGS=	$(AM_CXXFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_federated_la_CFLAGS =	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_federated_la_SOURCES =	ha_federated.cc
+ha_federated_la_LIBADD = ../../mysys/string.o
 
 
 EXTRA_LIBRARIES =	libfederated.a
