--- mysql-5.5.6-rc/mysql-test/r/mysqld--help-notwin.result.orig	2010-09-25 08:11:15.655515395 +0200
+++ mysql-5.5.6-rc/mysql-test/r/mysqld--help-notwin.result	2010-09-25 08:11:58.507519135 +0200
@@ -913,7 +913,7 @@
 slave-type-conversions 
 slow-launch-time 2
 slow-query-log FALSE
-socket /tmp/mysql.sock
+socket /var/lib/mysql/mysql.sock
 sort-buffer-size 2097152
 sporadic-binlog-dump-fail FALSE
 sql-mode 
