--- mysql-5.5.7-rc/Makefile.am.orig	2010-11-10 17:13:07.506287855 +0100
+++ mysql-5.5.7-rc/Makefile.am	2010-11-10 17:13:27.982746445 +0100
@@ -269,7 +269,6 @@
 			$(top_srcdir)/include/mysql.h \
 			$(top_srcdir)/include/mysql/psi/psi_abi_v1.h \
 			$(top_srcdir)/include/mysql/psi/psi_abi_v2.h \
-                        $(top_srcdir)/include/mysql/client_plugin.h \
                         $(top_srcdir)/include/mysql/plugin_auth.h
 
 #
