--- mysql-5.1.47/mysql-test/t/disabled.def.orig	2010-05-06 17:58:11.000000000 +0200
+++ mysql-5.1.47/mysql-test/t/disabled.def	2010-05-21 07:12:00.270542964 +0200
@@ -13,3 +13,6 @@
 query_cache_28249        : Bug#43861 2009-03-25 main.query_cache_28249 fails sporadically
 partition_innodb_plugin  : Bug#53307 2010-04-30 VasilDimov valgrind warnings
 ps_3innodb               : Bug#53309 2010-04-30 VasilDimov valgrind warnings
+#
+plugin_load         : gives wrong answer on PPC64
+outfile_loaddata    : gives different results on different platforms
