--- mysql-5.1.49/mysql-test/t/disabled.def.orig	2010-07-09 15:02:24.000000000 +0200
+++ mysql-5.1.49/mysql-test/t/disabled.def	2010-07-23 10:31:09.431784888 +0200
@@ -14,3 +14,6 @@
 partition_innodb_plugin  : Bug#53307 2010-04-30 VasilDimov valgrind warnings
 main.mysqlhotcopy_myisam : bug#54129 2010-06-04 Horst
 main.mysqlhotcopy_archive: bug#54129 2010-06-04 Horst
+#
+plugin_load         : gives wrong answer on PPC64
+outfile_loaddata    : gives different results on different platforms
