--- mysql-5.5.8/libmysqld/CMakeLists.txt.orig	2010-12-18 08:38:39.647131283 +0100
+++ mysql-5.5.8/libmysqld/CMakeLists.txt	2010-12-18 08:38:42.916733537 +0100
@@ -83,6 +83,7 @@
            ../sql/scheduler.cc ../sql/sql_audit.cc
            ../sql/sql_alter.cc ../sql/sql_partition_admin.cc
            ../sql/event_parse_data.cc
+           ../sql/rpl_reporting.cc
            ../sql/sql_signal.cc ../sql/rpl_handler.cc
 	       ../sql/rpl_utility.cc
            ../sql/sys_vars.cc
