diff -up mysql-5.5.6-rc/libmysqld/Makefile.am.report mysql-5.5.6-rc/libmysqld/Makefile.am
--- mysql-5.5.6-rc/libmysqld/Makefile.am.report	2010-09-25 10:12:21.974211826 +0200
+++ mysql-5.5.6-rc/libmysqld/Makefile.am	2010-09-25 10:13:22.497467790 +0200
@@ -82,6 +82,7 @@ sqlsources = derror.cc field.cc field_co
 	rpl_injector.cc my_user.c partition_info.cc sql_alter.cc \
 	sql_servers.cc event_parse_data.cc sql_signal.cc \
 	rpl_handler.cc mdl.cc keycaches.cc sql_audit.cc \
+	rpl_reporting.cc \
 	sql_partition_admin.cc
 
 libmysqld_int_a_SOURCES= $(libmysqld_sources)
