"extern int errno" is just a really bad idea.


diff -Naur mysql-5.5.8.orig/include/my_sys.h mysql-5.5.8/include/my_sys.h
--- mysql-5.5.8.orig/include/my_sys.h	2010-12-03 12:58:24.000000000 -0500
+++ mysql-5.5.8/include/my_sys.h	2010-12-20 21:20:12.622190325 -0500
@@ -201,13 +201,8 @@
 #define my_afree(PTR) my_free(PTR)
 #endif /* HAVE_ALLOCA */
 
-#ifndef errno				/* did we already get it? */
-#ifdef HAVE_ERRNO_AS_DEFINE
 #include <errno.h>			/* errno is a define */
-#else
-extern int errno;			/* declare errno */
-#endif
-#endif					/* #ifndef errno */
+
 extern char *home_dir;			/* Home directory for user */
 extern const char *my_progname;		/* program-name (printed in errors) */
 extern char curr_dir[];		/* Current directory for user */
