diff -up mysql-5.5.6-rc/include/my_sys.h.orig mysql-5.5.6-rc/include/my_sys.h
--- mysql-5.5.6-rc/include/my_sys.h.orig	2010-09-17 22:48:10.000000000 +0200
+++ mysql-5.5.6-rc/include/my_sys.h	2010-09-24 19:01:43.250648886 +0200
@@ -201,13 +201,8 @@ extern void my_large_free(uchar *ptr);
 #define my_afree(PTR) my_free(PTR)
 #endif /* HAVE_ALLOCA */
 
-#ifndef errno				/* did we already get it? */
-#ifdef HAVE_ERRNO_AS_DEFINE
 #include <errno.h>			/* errno is a define */
-#else
-extern int errno;			/* declare errno */
-#endif
-#endif					/* #ifndef errno */
+
 extern char *home_dir;			/* Home directory for user */
 extern const char *my_progname;		/* program-name (printed in errors) */
 extern char curr_dir[];		/* Current directory for user */
