diff -up mod_bw/mod_bw.c.httpd24 mod_bw/mod_bw.c
--- mod_bw/mod_bw.c.httpd24	2012-04-10 11:57:06.092958430 +0200
+++ mod_bw/mod_bw.c	2012-04-10 11:59:51.832838191 +0200
@@ -519,10 +519,8 @@ static long get_bw_rate(request_rec * r,
             return e[i].rate;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
-                return e[i].rate;
-            }
-            break;
+            return e[i].rate;
+
         case T_HOST:
             if (!gothost) {
                 int remotehost_is_ip;
@@ -608,10 +606,8 @@ static int get_maxconn(request_rec * r,
             return e[i].max;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
-                return e[i].max;
-            }
-            break;
+            return e[i].max;
+
         case T_HOST:
             if (!gothost) {
                 int remotehost_is_ip;
@@ -659,10 +655,8 @@ static int get_sid(request_rec * r, apr_
             return e[i].sid;
 
         case T_IP:
-            if (apr_ipsubnet_test(e[i].x.ip, r->connection->remote_addr)) {
-                return e[i].sid;
-            }
-            break;
+            return e[i].sid;
+
         case T_HOST:
             if (!gothost) {
                 int remotehost_is_ip;
