From e927cddb40786cbc4a27645e095f8f1da6cb8e6c Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sun, 26 Jun 2016 17:08:05 +0200
Subject: [PATCH] fix BC, re-add INTERLACE_NO

---
 gmagick_helpers.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gmagick_helpers.c b/gmagick_helpers.c
index 88aab56..4cc4893 100644
--- a/gmagick_helpers.c
+++ b/gmagick_helpers.c
@@ -185,6 +185,8 @@ void php_gmagick_initialize_constants()
 	GMAGICK_REGISTER_CONST_LONG("COMPRESSION_JBIG2", JBIG2Compression);
 #endif
  
+	GMAGICK_REGISTER_CONST_LONG("INTERLACE_UNDEFINED", UndefinedInterlace);
+	GMAGICK_REGISTER_CONST_LONG("INTERLACE_NO", NoInterlace); //Don't interlace image (RGBRGBRGBRGBRGBRGB...)
 	GMAGICK_REGISTER_CONST_LONG("INTERLACE_NONE", NoInterlace); //Don't interlace image (RGBRGBRGBRGBRGBRGB...)
 	GMAGICK_REGISTER_CONST_LONG("INTERLACE_LINE", LineInterlace); //Use scanline interlacing (RRR...GGG...BBB...RRR...GGG...BBB...)
 	GMAGICK_REGISTER_CONST_LONG("INTERLACE_PLANE", PlaneInterlace);//Use plane interlacing (RRRRRR...GGGGGG...BBBBBB...)
@@ -340,7 +342,6 @@ void php_gmagick_initialize_constants()
 	GMAGICK_REGISTER_CONST_LONG("RENDERINGINTENT_PERCEPTUAL", PerceptualIntent);
 	GMAGICK_REGISTER_CONST_LONG("RENDERINGINTENT_ABSOLUTE", AbsoluteIntent);
 	GMAGICK_REGISTER_CONST_LONG("RENDERINGINTENT_RELATIVE", RelativeIntent);
-	GMAGICK_REGISTER_CONST_LONG("INTERLACE_UNDEFINED", UndefinedInterlace);
 	GMAGICK_REGISTER_CONST_LONG("FILLRULE_UNDEFINED", UndefinedRule);
 	GMAGICK_REGISTER_CONST_LONG("FILLRULE_EVENODD", EvenOddRule);
 	GMAGICK_REGISTER_CONST_LONG("FILLRULE_NONZERO", NonZeroRule);
