From cbc919f4f7224e7217bffa89472b5d092cd30764 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 24 Jun 2016 12:54:29 +0200
Subject: [PATCH] clean duplicate constant definition, fix #40

---
 gmagick_helpers.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/gmagick_helpers.c b/gmagick_helpers.c
index 9c8ed06..88aab56 100644
--- a/gmagick_helpers.c
+++ b/gmagick_helpers.c
@@ -341,10 +341,6 @@ void php_gmagick_initialize_constants()
 	GMAGICK_REGISTER_CONST_LONG("RENDERINGINTENT_ABSOLUTE", AbsoluteIntent);
 	GMAGICK_REGISTER_CONST_LONG("RENDERINGINTENT_RELATIVE", RelativeIntent);
 	GMAGICK_REGISTER_CONST_LONG("INTERLACE_UNDEFINED", UndefinedInterlace);
-	GMAGICK_REGISTER_CONST_LONG("INTERLACE_NO", NoInterlace);
-	GMAGICK_REGISTER_CONST_LONG("INTERLACE_LINE", LineInterlace);
-	GMAGICK_REGISTER_CONST_LONG("INTERLACE_PLANE", PlaneInterlace);
-	GMAGICK_REGISTER_CONST_LONG("INTERLACE_PARTITION", PartitionInterlace);
 	GMAGICK_REGISTER_CONST_LONG("FILLRULE_UNDEFINED", UndefinedRule);
 	GMAGICK_REGISTER_CONST_LONG("FILLRULE_EVENODD", EvenOddRule);
 	GMAGICK_REGISTER_CONST_LONG("FILLRULE_NONZERO", NonZeroRule);
