Index: gmagick.c
===================================================================
--- gmagick.c	(révision 332363)
+++ gmagick.c	(copie de travail)
@@ -1082,7 +1082,7 @@
         ZEND_ARG_INFO(0, miterLimit)
 ZEND_END_ARG_INFO()
 
-#if GMAGICK_LIB_MASK >= 1004000 
+#if GMAGICK_LIB_MASK >= 1003000 
 ZEND_BEGIN_ARG_INFO_EX(gmagickdraw_setstrokedasharray_args, 0, 0, 1)
         ZEND_ARG_INFO(0, dashArray)
 ZEND_END_ARG_INFO()
@@ -1137,7 +1137,7 @@
 	PHP_ME(gmagickdraw, getstrokelinecap,  gmagick_empty_args, ZEND_ACC_PUBLIC)
 	PHP_ME(gmagickdraw, getstrokelinejoin,  gmagick_empty_args, ZEND_ACC_PUBLIC)
 	PHP_ME(gmagickdraw, getstrokemiterlimit, gmagick_empty_args, ZEND_ACC_PUBLIC)
-#if GMAGICK_LIB_MASK >= 1004000 	
+#if GMAGICK_LIB_MASK >= 1003000 	
     PHP_ME(gmagickdraw, getstrokedasharray, gmagick_empty_args, ZEND_ACC_PUBLIC)
     PHP_ME(gmagickdraw, setstrokedasharray, gmagickdraw_setstrokedasharray_args, ZEND_ACC_PUBLIC)
 #endif
Index: gmagick_helpers.c
===================================================================
--- gmagick_helpers.c	(révision 332363)
+++ gmagick_helpers.c	(copie de travail)
@@ -632,7 +632,8 @@
                 return NULL;
         }
 
-        double_array = (double *)emalloc(sizeof(double) * elements);
+        /* Number of elements + 1 for final 0 */
+        double_array = (double *)emalloc(sizeof(double) * (elements+1));
 
         for (zend_hash_internal_pointer_reset(Z_ARRVAL_P(param_array));
                         zend_hash_get_current_data(Z_ARRVAL_P(param_array), (void **) &ppzval) == SUCCESS;
@@ -648,6 +649,6 @@
                 double_array[i] = Z_DVAL_P(tmp_pzval);
                 tmp_pzval = NULL;
         }
-        *num_elements = elements;
+        double_array[elements] = 0.0;
         return double_array;
-}
\ Pas de retour chariot à la fin du fichier
+}
Index: gmagickdraw_methods.c
===================================================================
--- gmagickdraw_methods.c	(révision 332363)
+++ gmagickdraw_methods.c	(copie de travail)
@@ -1098,7 +1098,7 @@
 /* }}} */
 
 
-#if GMAGICK_LIB_MASK >= 1004000 
+#if GMAGICK_LIB_MASK >= 1003000 
 /* {{{ proto array GmagickDraw::getStrokeDashArray()
         Returns an array representing the pattern of dashes and gaps used to stroke paths (see DrawSetStrokeDashArray). The array must be freed once it is no longer required by the user.
 */
@@ -1157,4 +1157,4 @@
         RETURN_TRUE;
 }
 /* }}} */
-#endif
\ Pas de retour chariot à la fin du fichier
+#endif
Index: php_gmagick.h
===================================================================
--- php_gmagick.h	(révision 332363)
+++ php_gmagick.h	(copie de travail)
@@ -335,7 +335,7 @@
 PHP_METHOD(gmagickdraw, getstrokelinecap);
 PHP_METHOD(gmagickdraw, getstrokelinejoin);
 PHP_METHOD(gmagickdraw, getstrokemiterlimit);
-#if GMAGICK_LIB_MASK >= 1004000 
+#if GMAGICK_LIB_MASK >= 1003000 
 PHP_METHOD(gmagickdraw, setstrokedasharray);
 PHP_METHOD(gmagickdraw, getstrokedasharray);
 #endif
