diff -up bin/validate-json.rpm bin/validate-json
--- bin/validate-json.rpm	2016-04-15 11:02:32.629960716 +0200
+++ bin/validate-json	2016-04-15 11:07:01.583136406 +0200
@@ -6,28 +6,7 @@
  * @author Christian Weiske <christian.weiske@netresearch.de>
  */
 
-/**
- * Dead simple autoloader
- *
- * @param string $className Name of class to load
- *
- * @return void
- */
-function __autoload($className)
-{
-    $className = ltrim($className, '\\');
-    $fileName  = '';
-    $namespace = '';
-    if ($lastNsPos = strrpos($className, '\\')) {
-        $namespace = substr($className, 0, $lastNsPos);
-        $className = substr($className, $lastNsPos + 1);
-        $fileName  = str_replace('\\', DIRECTORY_SEPARATOR, $namespace) . DIRECTORY_SEPARATOR;
-    }
-    $fileName .= str_replace('_', DIRECTORY_SEPARATOR, $className) . '.php';
-    if (stream_resolve_include_path($fileName)) {
-        require_once $fileName;
-    }
-}
+require_once '/usr/share/php/JsonSchema5/autoload.php';
 
 /**
  * Show the json parse error that happened last
@@ -85,11 +64,6 @@ function parseHeaderValue($headerValue)
 }
 
 
-// support running this tool from git checkout
-if (is_dir(__DIR__ . '/../src/JsonSchema')) {
-    set_include_path(__DIR__ . '/../src' . PATH_SEPARATOR . get_include_path());
-}
-
 $arOptions = array();
 $arArgs = array();
 array_shift($argv);//script itself
