--- mod_wsgi-3.4/mod_wsgi.c.connsbh
+++ mod_wsgi-3.4/mod_wsgi.c
@@ -10600,7 +10600,13 @@ static void wsgi_process_socket(apr_pool
      * will add their own input/output filters to the chain.
      */
 
+#if AP_MODULE_MAGIC_AT_LEAST(20110619,0)
+    /* For 2.4 a NULL sbh pointer should work. */
+    sbh = NULL;
+#else
+    /* For 2.2 a dummy sbh pointer is needed. */
     ap_create_sb_handle(&sbh, p, -1, 0);
+#endif
 
     c = (conn_rec *)apr_pcalloc(p, sizeof(conn_rec));
 
