diff -up geoip-1.0.7/geoip.c.orig geoip-1.0.7/geoip.c
--- geoip-1.0.7/geoip.c.orig	2011-10-15 16:25:26.612833903 +0200
+++ geoip-1.0.7/geoip.c	2011-10-15 16:26:14.279997257 +0200
@@ -229,7 +229,7 @@ PHP_FUNCTION(geoip_db_get_all_info)
 
 	for (i=0; i < NUM_DB_TYPES; i++)
 	{
-		if (NULL != GeoIPDBDescription[i])
+		if (GeoIPDBDescription[i] && GeoIPDBFileName[i])
 		{
 			zval *row;
 			ALLOC_INIT_ZVAL(row);
