--- pecl/geoip/trunk/geoip.c	2010/03/31 20:39:48	297236
+++ pecl/geoip/trunk/geoip.c	2011/06/05 19:00:52	311837
@@ -238,7 +238,9 @@
 
 			add_assoc_bool(row, "available", GeoIP_db_avail(i));
 			add_assoc_string(row, "description", (char *)GeoIPDBDescription[i], 1);
-			add_assoc_string(row, "filename", GeoIPDBFileName[i], 1);
+			if (GeoIPDBFileName[i]) {
+				add_assoc_string(row, "filename", GeoIPDBFileName[i], 1);
+			}
 
 			add_index_zval(return_value, i, row);
 		}
