diff -up bin/zephir.rpm bin/zephir
--- bin/zephir.rpm	2016-07-30 18:49:46.000000000 +0200
+++ bin/zephir	2016-07-30 18:52:16.000000000 +0200
@@ -1,27 +1,9 @@
 #!/bin/bash
-if [ -z "$ZEPHIRDIR" ]; then
-  ZEPHIRDIR="%ZEPHIRDIR%"
-  if [[ "$ZEPHIRDIR" == "%""ZEPHIRDIR""%" ]]; then
-    if [ -h $0 ]; then
-        REALPATH="`dirname $0`/`readlink $0`"
-        CURDIR=`dirname $REALPATH`
-    else
-        CURDIR=`dirname $0`
-    fi
-    PARENTDIR=`cd $CURDIR/..; pwd`
-
-    if [ -f $PARENTDIR/compiler.php ]; then
-      export ZEPHIRDIR=$PARENTDIR
-    else
-      echo "Environment variable ZEPHIRDIR is not set"
-      exit 1
-    fi
-  fi
-fi
+export ZEPHIRDIR="@DATADIR@/zephir"
 
 if [[ $1 && $2 && $3 && "$1"=="-c" ]]; then
-    php -d safe_mode=Off -d enable_dl=On -c $2 $ZEPHIRDIR/compiler.php ${*:3}
+    @BINDIR@/php -d memory_limit=1G -d extension=zephir_parser.so -c $2 $ZEPHIRDIR/compiler.php ${*:3}
 else
-    php -d safe_mode=Off -d enable_dl=On $ZEPHIRDIR/compiler.php $*
+    @BINDIR@/php -d memory_limit=1G -d extension=zephir_parser.so $ZEPHIRDIR/compiler.php $*
 fi
 
