diff -up ./config.m4.old ./config.m4
--- ./config.m4.old	2017-03-14 08:58:13.000000000 +0100
+++ ./config.m4	2017-03-14 08:58:19.000000000 +0100
@@ -22,6 +22,9 @@ PHP_ARG_ENABLE(mysql-xdevapi-message-cla
 	[  --enable-mysql-xdevapi-message-classes
 						Enable support for the experimental message classes in mysql-xdevapi], yes, no)
 
+PHP_ARG_WITH(openssl, for OpenSSL support,
+[  --with-openssl[=DIR]      Include OpenSSL support (requires OpenSSL >= 1.0.1)])
+
 dnl If some extension uses mysql-xdevapi it will get compiled in PHP core
 if test "$PHP_MYSQL_XDEVAPI" != "no" || test "$PHP_MYSQL_XDEVAPI_ENABLED" = "yes"; then
 	PHP_REQUIRE_CXX
