--- php-5.5.14/ext/intl/locale/locale_methods.c	2014-06-25 15:06:23.000000000 +0200
+++ php-5.5.15RC1/ext/intl/locale/locale_methods.c	2014-07-08 23:55:31.000000000 +0200
@@ -279,7 +279,7 @@
 	if( fromParseLocale==1 ){
 		/* Handle singletons */
 		if( strcmp(tag_name , LOC_LANG_TAG)==0 ){
-			if( strlen(loc_name)>1 && isIDPrefix(loc_name) ){
+			if( strlen(loc_name)>1 && (isIDPrefix(loc_name) == 1) ){
 				return estrdup(loc_name);
 			}
 		}
@@ -506,7 +506,7 @@
     }
 
 	if(loc_name_len == 0) {
-		loc_name = INTL_G(default_locale);
+		loc_name = intl_locale_get_default(TSRMLS_C);
 	}
 
 	if( strcmp(tag_name, DISP_NAME) != 0 ){
