--- acinclude.m4.old	2012-12-10 12:30:11.684122918 +0100
+++ acinclude.m4	2012-12-10 12:31:49.933474452 +0100
@@ -2963,12 +2963,12 @@
   cat>>Makefile.objects<<EOF
 
 $ac_bdir[$]ac_hdrobj: $abs_srcdir/$ac_provsrc
-	dtrace -h -C -s $ac_srcdir[$]ac_provsrc -o \$[]@ && \$(SED) -ibak 's,PHP_,DTRACE_,g' \$[]@
+	CFLAGS="\$(CFLAGS_CLEAN)" dtrace -h -C -s $ac_srcdir[$]ac_provsrc -o \$[]@ && \$(SED) -ibak 's,PHP_,DTRACE_,g' \$[]@
 
 \$(PHP_DTRACE_OBJS): $ac_bdir[$]ac_hdrobj
 
 $ac_bdir[$]ac_provsrc.o: \$(PHP_DTRACE_OBJS)
-	dtrace -G -o \$[]@ -s $abs_srcdir/$ac_provsrc $dtrace_objs
+	CFLAGS="\$(CFLAGS_CLEAN)" dtrace -G -o \$[]@ -s $abs_srcdir/$ac_provsrc $dtrace_objs
 
 EOF
 ])
