diff -up ext/readline/readline.c.old ext/readline/readline.c
--- ext/readline/readline.c.old	2014-01-20 08:19:13.009830341 +0100
+++ ext/readline/readline.c	2014-01-20 08:20:56.463158818 +0100
@@ -354,6 +354,11 @@ PHP_FUNCTION(readline_clear_history)
 		return;
 	}
 
+#if HAVE_LIBEDIT
+	/* clear_history is the only function where rl_initialize
+	   is not call to ensure correct allocation */
+	using_history();
+#endif
 	clear_history();
 
 	RETURN_TRUE;
