diff --git a/driver-testsuite/bootstrap.php b/driver-testsuite/bootstrap.php
index f69a45b..ff6c7d1 100644
--- a/driver-testsuite/bootstrap.php
+++ b/driver-testsuite/bootstrap.php
@@ -1,20 +1,6 @@
 <?php
 
-$file = __DIR__.'/../../../autoload.php';
-
-if (!file_exists($file)) {
-    echo PHP_EOL.'The Mink driver testsuite expects Mink to be installed as a composer dependency of your project'.PHP_EOL;
-    exit(1);
-}
-
-/** @var \Composer\Autoload\ClassLoader $loader */
-$loader = require $file;
-
-$loader->addPsr4('Behat\Mink\Tests\Driver\\', __DIR__.'/tests');
-
-// Clean the global variables
-unset($file);
-unset($loader);
+require_once __DIR__ . '/autoload.php';
 
 // Check the definition of the driverLoaderFactory
 
