From 6f0959bceb5f4f07971d74fe0f2d47e5113e09ad Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Tue, 20 Jan 2015 09:42:46 +0100
Subject: [PATCH] fix "composer only" bug, rely in include_path

---
 src/Bartlett/Reflect/Api/V3/Analyser.php | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Bartlett/Reflect/Api/V3/Analyser.php b/src/Bartlett/Reflect/Api/V3/Analyser.php
index 373070b..5a78942 100644
--- a/src/Bartlett/Reflect/Api/V3/Analyser.php
+++ b/src/Bartlett/Reflect/Api/V3/Analyser.php
@@ -65,6 +65,7 @@ public function run($source, array $analysers, $alias)
 
     protected function registerAnalysers()
     {
+        $file = 'Bartlett/CompatInfo/Analyser/CompatibilityAnalyser.php';
         $reflectBaseDir = dirname(dirname(dirname(dirname(dirname(__DIR__)))));
 
         $baseDir   = dirname(dirname(dirname($reflectBaseDir)));
@@ -77,7 +78,11 @@ protected function registerAnalysers()
             $namespaces['Bartlett\CompatInfo\Analyser']
                 = $baseDir . '/src/Bartlett/CompatInfo/Analyser'
             ;
+        } else if ($path = stream_resolve_include_path($file)) {
+            // CompatInfo only, without composer
+            $namespaces['Bartlett\CompatInfo\Analyser'] = dirname($path);
         }
+
         $am = new AnalyserManager($namespaces);
         $am->registerAnalysers();
 
