diff -up ./bin/phpreflect.json.dist.rpm ./bin/phpreflect.json.dist
--- ./bin/phpreflect.json.dist.rpm	2014-10-16 14:13:07.000000000 +0200
+++ ./bin/phpreflect.json.dist	2014-10-16 16:49:30.538850254 +0200
@@ -9,6 +9,19 @@
         {
             "name": "Analyser",
             "class": "Bartlett\\Reflect\\Plugin\\Analyser\\AnalyserPlugin"
+        },
+        {
+            "name": "Cache",
+            "class": "Bartlett\\Reflect\\Plugin\\Cache\\CachePlugin",
+            "options": {
+                "adapter": "DoctrineCacheAdapter",
+                "backend": {
+                    "class": "Doctrine\\Common\\Cache\\FilesystemCache",
+                    "args": [
+                        "%{HOME}/.cache/phpreflect"
+                    ]
+                }
+            }
         }
     ],
     "analysers" : [
diff -up ./bin/phpreflect.rpm ./bin/phpreflect
--- ./bin/phpreflect.rpm	2014-10-16 14:13:07.000000000 +0200
+++ ./bin/phpreflect	2014-10-16 16:50:40.458190967 +0200
@@ -1,15 +1,11 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 
 if (\Phar::running()) {
     $vendorDir = 'phar://phpreflect.phar/vendor';
 } else {
     $baseDir   = dirname(__DIR__);
-    $vendorDir = $baseDir . '/vendor';
-
-    if (!file_exists($vendorDir . '/autoload.php')) {
-        $vendorDir = dirname(dirname($baseDir));
-    }
+    $vendorDir = '/usr/share/php';
 
     if (!getenv("REFLECT")) {
         $home  = defined('PHP_WINDOWS_VERSION_BUILD') ? 'USERPROFILE' : 'HOME';
@@ -26,8 +22,28 @@ if (\Phar::running()) {
         }
     }
 }
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->setUseIncludePath(true);
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+require_once $vendorDir . '/PHP/Timer.php';
+require_once $vendorDir . '/SebastianBergmann/Version/Version.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Symfony\\Component\\Finder'          => $vendorDir,
+    'Symfony\\Component\\EventDispatcher' => $vendorDir,
+    'Symfony\\Component\\Console'         => $vendorDir,
+    'Symfony\\Component\\ClassLoader'     => $vendorDir,
+    'Doctrine\\Common\\Cache'             => $vendorDir,
+    'Seld\\JsonLint'                      => $vendorDir,
+    'Bartlett'                            => $vendorDir,
+));
+$loader->useIncludePath(true);
+$loader->register();
 
 if (PHP_SAPI !== 'cli') {
     return;
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2014-10-16 14:13:07.000000000 +0200
+++ ./tests/bootstrap.php	2014-10-16 16:49:30.538850254 +0200
@@ -1,6 +1,19 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-require_once $vendorDir . '/autoload.php';
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Bartlett\\Tests\\Reflect'            => $baseDir . '/tests',
+    'Bartlett'                            => $baseDir . '/src',
+));
+$loader->useIncludePath(true);
+$loader->register();
