diff -up ./bin/phpreflect.json.dist.rpm ./bin/phpreflect.json.dist
--- ./bin/phpreflect.json.dist.rpm	2015-06-18 13:30:46.000000000 +0200
+++ ./bin/phpreflect.json.dist	2015-06-26 09:18:15.577663962 +0200
@@ -6,6 +6,19 @@
         }
     ],
     "plugins": [
+        {
+            "name": "Cache",
+            "class": "Bartlett\\Reflect\\Plugin\\CachePlugin",
+            "options": {
+                "adapter": "DoctrineCacheAdapter",
+                "backend": {
+                    "class": "Doctrine\\Common\\Cache\\FilesystemCache",
+                    "args": [
+                        "%{HOME}/.cache/php-reflect"
+                    ]
+                }
+            }
+        }
     ],
     "analysers": [
     ],
diff -up ./bin/phpreflect.rpm ./bin/phpreflect
--- ./bin/phpreflect.rpm	2015-06-18 13:30:46.000000000 +0200
+++ ./bin/phpreflect	2015-06-26 10:06:59.277781055 +0200
@@ -2,35 +2,16 @@
 <?php
 $appName = 'phpReflect';
 
-if (\Phar::running()) {
-    $vendorDir = 'phar://' . strtolower($appName) . '.phar/vendor';
-} else {
-    $baseDir   = dirname(__DIR__);
-    $vendorDir = $baseDir . '/vendor';
-
-    if (!file_exists($vendorDir . '/autoload.php')) {
-        $vendorDir = dirname(dirname($baseDir));
-    }
-
-    if (!getenv("BARTLETTRC")) {
-        putenv("BARTLETTRC=" . strtolower($appName) . '.json');
-    }
+$vendorDir = '/usr/share/php';
+if (is_file('/usr/share/php-bartlett-PHP-CompatInfo/compatinfo.sqlite')) {
+    putenv('BARTLETT_COMPATINFO_DB=/usr/share/php-bartlett-PHP-CompatInfo/compatinfo.sqlite');
 }
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->setUseIncludePath(true);
-
-if (\Phar::running()) {
-    if ($home) {
-        // when optional resources are not embedded in phar distribution,
-        // avoid PHP Warnings, and lookup first in global vendor dir (COMPOSER_HOME) if exists
-        $classMapfiltered = $fallbackClassMap($loader->getClassMap(), 'Net_Growl');
-        $loader->addClassMap($classMapfiltered);
-
-        // try to find PEAR Net_Growl classes package in folder $fallbackNetGrowlDir (see phar-stub.php)
-        $loader->add(false, $fallbackNetGrowlDir);
-    }
+if (!getenv("BARTLETTRC")) {
+    putenv("BARTLETTRC=" . strtolower($appName) . '.json');
 }
 
+require_once $vendorDir . '/Bartlett/Reflect/autoload.php';
+
 if (PHP_SAPI !== 'cli') {
     return;
 }
@@ -40,5 +21,5 @@ use Bartlett\Reflect\Console\Application
 
 Environment::setScanDir();
 
-$application = new Application($appName, '4.0');
+$application = new Application($appName, '@package_version@');
 $application->run();
diff -up ./src/Bartlett/Reflect/Console/Application.php.rpm ./src/Bartlett/Reflect/Console/Application.php
--- ./src/Bartlett/Reflect/Console/Application.php.rpm	2015-06-18 13:30:46.000000000 +0200
+++ ./src/Bartlett/Reflect/Console/Application.php	2015-06-26 09:18:15.584663991 +0200
@@ -42,7 +42,7 @@ use Symfony\Component\EventDispatcher\Ev
  * @package  PHP_Reflect
  * @author   Laurent Laville <pear@laurent-laville.org>
  * @license  http://www.opensource.org/licenses/bsd-license.php  BSD License
- * @version  Release: @package_version@
+ * @version  Release: 3.1.1
  * @link     http://php5.laurent-laville.org/reflect/
  * @since    Class available since Release 3.0.0-alpha1
  */
@@ -82,7 +82,7 @@ class Application extends BaseApplicatio
             $jsonFile =  './' . getenv("BARTLETTRC");
         }
 
-        parent::__construct($appName, '@package_version@');
+        parent::__construct($appName, $appVersion);
         $this->release = $appVersion;
 
         $defaultAnalyser = (strcasecmp($appName, 'phpcompatinfo') === 0)
diff -up tests/bootstrap.php.rpm tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2016-09-22 20:49:42.000000000 +0200
+++ ./tests/bootstrap.php	2016-10-31 18:03:22.837465194 +0100
@@ -1,11 +1,10 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->addClassMap(
-    array(
+require_once 'Bartlett/Reflect/autoload.php';
+$tab = array(
         'Bartlett\Tests\Reflect\Analyser\FooAnalyser'
             => __DIR__ . '/Analyser/FooAnalyser.php',
         'Bartlett\Tests\Reflect\Analyser\BarAnalyser'
@@ -14,5 +13,13 @@ $loader->addClassMap(
             => __DIR__ . '/Model/GenericModelTest.php',
         'Bartlett\Tests\Reflect\Environment\YourLogger'
             => __DIR__ . '/Environment/YourLogger.php',
-    )
 );
+
+\Fedora\Autoloader\Autoload::addClassMap($tab, '/');
+\Fedora\Autoloader\Autoload::addPsr4('Bartlett\\Tests\\Reflect', $baseDir . '/tests');
+
+// Needed when installed for 'Bartlett\CompatInfo\Analyser\CompatibilityAnalyser'
+if (is_file('/usr/share/php-bartlett-PHP-CompatInfo/compatinfo.sqlite')) {
+   putenv("BARTLETT_COMPATINFO_DB=/usr/share/php-bartlett-PHP-CompatInfo/compatinfo.sqlite");
+}
+
