diff -up php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect.json.rpm php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect.json
--- php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect.json.rpm	2014-04-04 23:06:38.000000000 +0200
+++ php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect.json	2014-05-13 08:23:54.982741367 +0200
@@ -1,30 +1,8 @@
 {
     "source-providers": [
         {
-            "in": "phar:///var/compatinfo/branches/v3/examples/archives/PHP_CompatInfo-2.0.0.zip as CompatInfo",
-            "name": "*.php",
-            "path": "PHP/"
-        },
-        {
-            "in": "phar:///var/compatinfo/branches/v3/examples/archives/PHP_Reflect-1.0.0.tar as Reflect",
-            "path": "PHP/",
-            "name": "*.php"
-        },
-        {
-            "in": "phar:///var/compatinfo/branches/v3/examples/archives/PHP_Reflect-1.0.0.tar as ReflectAL",
-            "path": "PHP/",
-            "name": "Autoload.php"
-        },
-        {
-            "in": "phar:///var/compatinfo/branches/v3/examples/archives/phing-2.6.1.tgz as phing",
-            "name": "*.php",
-            "path": "tasks/system",
-            "notPath": "contrib/",
-            "size": "< 4K"
-        },
-        {
-            "in": "phar:///var/compatinfo/branches/v3/examples/archives/Zend-1.12.3.tgz as ZF1",
-            "name": "*.php"
+                "in": ". as current",
+                "name": "/\\.(php|inc|phtml|module|install)$/"
         }
     ],
     "plugins": [
diff -up php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect.rpm php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect
--- php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect.rpm	2014-04-04 23:06:38.000000000 +0200
+++ php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/bin/reflect	2014-05-13 08:40:44.969643786 +0200
@@ -1,31 +1,39 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 // CLI version
-//putenv("REFLECT=/path/to/reflect.json");
+if (!getenv("REFLECT")) {
+    $files = array(
+        realpath('./reflect.json'),
+        getenv('HOME').'/.config/phpreflect.json',
+        '/etc/phpreflect.json',
+    );
+    foreach ($files as $file) {
+        if (file_exists($file)) {
+            putenv("REFLECT=$file");
+            echo "Configuration read from $file\n";
+            break;
+        }
+    }
+}
 
-$baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-if (\Phar::running()) {
-    require_once 'phar://phpreflect.phar/vendor/nikic/php-parser/lib/bootstrap.php';
-    require_once 'phar://phpreflect.phar/vendor/symfony/class-loader/Symfony/Component/ClassLoader/UniversalClassLoader.php';
-    require_once 'phar://phpreflect.phar/vendor/phpunit/php-timer/PHP/Timer.php';
-} else {
-    require_once $vendorDir . '/nikic/php-parser/lib/bootstrap.php';
-    require_once $vendorDir . '/symfony/class-loader/Symfony/Component/ClassLoader/UniversalClassLoader.php';
-    require_once $vendorDir . '/phpunit/php-timer/PHP/Timer.php';
-}
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+require_once $vendorDir . '/PHP/Timer.php';
 
 use Symfony\Component\ClassLoader\UniversalClassLoader;
 
 $loader = new UniversalClassLoader();
 $loader->registerNamespaces(array(
-    'Symfony\\Component\\Finder'          => $vendorDir . '/symfony/finder',
-    'Symfony\\Component\\EventDispatcher' => $vendorDir . '/symfony/event-dispatcher',
-    'Symfony\\Component\\Console'         => $vendorDir . '/symfony/console',
-    'Symfony\\Component\\ClassLoader'     => $vendorDir . '/symfony/class-loader',
-    'Bartlett\\Tests\\Reflect'            => $baseDir . '/tests',
-    'Bartlett'                            => $baseDir . '/src',
+    'Symfony\\Component\\Finder'          => $vendorDir,
+    'Symfony\\Component\\EventDispatcher' => $vendorDir,
+    'Symfony\\Component\\Console'         => $vendorDir,
+    'Symfony\\Component\\ClassLoader'     => $vendorDir,
+    'Bartlett'                            => $vendorDir,
 ));
 $loader->register();
 
diff -up php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/tests/bootstrap.php.rpm php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/tests/bootstrap.php
--- php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/tests/bootstrap.php.rpm	2014-04-04 23:06:38.000000000 +0200
+++ php-reflect-b45be1d0d1b5a46253f9cda894dd7717f0a64abd/tests/bootstrap.php	2014-05-13 08:23:54.982741367 +0200
@@ -1,20 +1,19 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-require_once $vendorDir . '/nikic/php-parser/lib/bootstrap.php';
-require_once $vendorDir . '/symfony/class-loader/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
 
 use Symfony\Component\ClassLoader\UniversalClassLoader;
 
 $loader = new UniversalClassLoader();
 $loader->registerNamespaces(array(
-    'Symfony\\Component\\Finder'          => $vendorDir . '/symfony/finder',
-    'Symfony\\Component\\EventDispatcher' => $vendorDir . '/symfony/event-dispatcher',
-    'Symfony\\Component\\Console'         => $vendorDir . '/symfony/console',
-    'Symfony\\Component\\ClassLoader'     => $vendorDir . '/symfony/class-loader',
     'Bartlett\\Tests\\Reflect'            => $baseDir . '/tests',
     'Bartlett'                            => $baseDir . '/src',
 ));
+$loader->useIncludePath(true);
 $loader->register();
