diff -up ./bin/phpreflect.json.dist.rpm ./bin/phpreflect.json.dist
--- ./bin/phpreflect.json.dist.rpm	2015-05-11 12:55:49.000000000 +0200
+++ ./bin/phpreflect.json.dist	2015-05-11 13:22:14.660514268 +0200
@@ -6,6 +6,19 @@
         }
     ],
     "plugins": [
+        {
+            "name": "Cache",
+            "class": "Bartlett\\Reflect\\Plugin\\CachePlugin",
+            "options": {
+                "adapter": "DoctrineCacheAdapter",
+                "backend": {
+                    "class": "Doctrine\\Common\\Cache\\FilesystemCache",
+                    "args": [
+                        "%{HOME}/.cache/php-reflect"
+                    ]
+                }
+            }
+        }
     ],
     "analysers": [
     ],
diff -up ./bin/phpreflect.rpm ./bin/phpreflect
--- ./bin/phpreflect.rpm	2015-05-11 12:55:49.000000000 +0200
+++ ./bin/phpreflect	2015-05-11 13:24:26.130135886 +0200
@@ -5,19 +5,35 @@ $appName = 'phpReflect';
 if (\Phar::running()) {
     $vendorDir = 'phar://' . strtolower($appName) . '.phar/vendor';
 } else {
-    $baseDir   = dirname(__DIR__);
-    $vendorDir = $baseDir . '/vendor';
+    putenv('BARTLETT_COMPATINFO_DB=/usr/share/php-bartlett-PHP-CompatInfo/compatinfo.sqlite');
 
-    if (!file_exists($vendorDir . '/autoload.php')) {
-        $vendorDir = dirname(dirname($baseDir));
-    }
+    $vendorDir = '/usr/share/php';
 
     if (!getenv("BARTLETTRC")) {
         putenv("BARTLETTRC=" . strtolower($appName) . '.json');
     }
 }
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->setUseIncludePath(true);
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+require_once $vendorDir . '/SebastianBergmann/Version/Version.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Symfony\\Component'                  => $vendorDir,
+    'Doctrine\\Common'                    => $vendorDir,
+    'phpDocumentor\\Reflection'           => $vendorDir,
+    'Seld\\JsonLint'                      => $vendorDir,
+    'JsonSchema'                          => $vendorDir,
+    'Psr\\Log'                            => $vendorDir,
+    'Monolog'                             => $vendorDir,
+    'Bartlett'                            => $vendorDir,
+));
+$loader->useIncludePath(true);
+$loader->register();
 
 if (PHP_SAPI !== 'cli') {
     return;
@@ -28,5 +44,5 @@ use Bartlett\Reflect\Console\Application
 
 Environment::setScanDir();
 
-$application = new Application($appName, '3.1');
+$application = new Application($appName, '@package_version@');
 $application->run();
diff -up ./src/Bartlett/Reflect/Console/Application.php.rpm ./src/Bartlett/Reflect/Console/Application.php
--- ./src/Bartlett/Reflect/Console/Application.php.rpm	2015-05-11 12:55:49.000000000 +0200
+++ ./src/Bartlett/Reflect/Console/Application.php	2015-05-11 13:22:14.661514273 +0200
@@ -82,7 +82,7 @@ class Application extends BaseApplicatio
             $jsonFile =  './' . getenv("BARTLETTRC");
         }
 
-        parent::__construct($appName, '@package_version@');
+        parent::__construct($appName, $appVersion);
         $this->release = $appVersion;
 
         $defaultAnalyser = (strcasecmp($appName, 'phpcompatinfo') === 0)
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2015-05-11 12:55:49.000000000 +0200
+++ ./tests/bootstrap.php	2015-05-11 13:22:14.661514273 +0200
@@ -1,11 +1,9 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->addClassMap(
-    array(
+$tab = array(
         'Bartlett\Tests\Reflect\Analyser\FooAnalyser'
             => __DIR__ . '/Analyser/FooAnalyser.php',
         'Bartlett\Tests\Reflect\Analyser\BarAnalyser'
@@ -14,5 +12,24 @@ $loader->addClassMap(
             => __DIR__ . '/Model/GenericModelTest.php',
         'Bartlett\Tests\Reflect\Environment\YourLogger'
             => __DIR__ . '/Environment/YourLogger.php',
-    )
 );
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Bartlett\\Tests\\Reflect'            => $baseDir . '/tests',
+    'Bartlett\\UmlWriter'                 => $vendorDir,
+    'Bartlett'                            => $baseDir . '/src',
+    'Monolog'                             => $vendorDir,
+));
+$loader->useIncludePath(true);
+$loader->register();
+
+foreach ($tab as $class => $file) {
+    require_once $file;
+}
diff -up ./tests/Environment/YourLogger.php.rpm ./tests/Environment/YourLogger.php
--- ./tests/Environment/YourLogger.php.rpm	2015-05-11 12:55:49.000000000 +0200
+++ ./tests/Environment/YourLogger.php	2015-05-11 13:22:14.661514273 +0200
@@ -14,6 +14,7 @@ class YourLogger extends Logger
         if (!file_exists($tempDir)) {
             mkdir($tempDir, 0755, true);
         }
+        date_default_timezone_set('UTC');
         $filename = sprintf('phpreflect-%s.log', date('Ymd'));
 
         $stream = new RotatingFileHandler("$tempDir/$filename", 30);
