diff -up ./bin/phpreflect.rpm ./bin/phpreflect
--- ./bin/phpreflect.rpm	2015-02-03 11:38:30.000000000 +0100
+++ ./bin/phpreflect	2015-02-04 14:45:45.995075976 +0100
@@ -1,23 +1,35 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 $appName = 'phpReflect';
 
 if (\Phar::running()) {
     $vendorDir = 'phar://' . strtolower($appName) . '.phar/vendor';
 } else {
-    $baseDir   = dirname(__DIR__);
-    $vendorDir = $baseDir . '/vendor';
-
-    if (!file_exists($vendorDir . '/autoload.php')) {
-        $vendorDir = dirname(dirname($baseDir));
-    }
+    $vendorDir = '/usr/share/php';
 
     if (!getenv("BARTLETTRC")) {
         putenv("BARTLETTRC=" . strtolower($appName) . '.json');
     }
 }
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->setUseIncludePath(true);
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+require_once $vendorDir . '/SebastianBergmann/Version/Version.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Symfony\\Component'                  => $vendorDir,
+    'Doctrine\\Common'                    => $vendorDir,
+    'phpDocumentor\\Reflection'           => $vendorDir,
+    'Seld\\JsonLint'                      => $vendorDir,
+    'JsonSchema'                          => $vendorDir,
+    'Bartlett'                            => $vendorDir,
+));
+$loader->useIncludePath(true);
+$loader->register();
 
 if (PHP_SAPI !== 'cli') {
     return;
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2015-02-03 11:38:30.000000000 +0100
+++ ./tests/bootstrap.php	2015-02-04 14:45:45.995075976 +0100
@@ -1,11 +1,9 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-$loader = require_once $vendorDir . '/autoload.php';
-$loader->addClassMap(
-    array(
+$tab = array(
         'Bartlett\Tests\Reflect\Analyser\FooAnalyser'
             => __DIR__ . '/Analyser/FooAnalyser.php',
         'Bartlett\Tests\Reflect\Analyser\BarAnalyser'
@@ -14,5 +12,23 @@ $loader->addClassMap(
             => __DIR__ . '/Model/GenericModelTest.php',
         'Bartlett\Tests\Reflect\Environment\YourLogger'
             => __DIR__ . '/Environment/YourLogger.php',
-    )
 );
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Bartlett\\Tests\\Reflect'            => $baseDir . '/tests',
+    'Bartlett'                            => $baseDir . '/src',
+    'Monolog'                             => $vendorDir,
+));
+$loader->useIncludePath(true);
+$loader->register();
+
+foreach ($tab as $class => $file) {
+    require_once $file;
+}
diff -up ./tests/Environment/YourLogger.php.rpm ./tests/Environment/YourLogger.php
--- ./tests/Environment/YourLogger.php.rpm	2015-02-03 11:38:30.000000000 +0100
+++ ./tests/Environment/YourLogger.php	2015-02-04 14:48:07.609164546 +0100
@@ -14,6 +14,7 @@ class YourLogger extends Logger
         if (!file_exists($tempDir)) {
             mkdir($tempDir, 0755, true);
         }
+        date_default_timezone_set('UTC');
         $filename = sprintf('phpreflect-%s.log', date('Ymd'));
 
         $stream = new RotatingFileHandler("$tempDir/$filename", 30);
