diff -up ./bin/phpreflect.rpm ./bin/phpreflect
--- ./bin/phpreflect.rpm	2014-07-23 17:49:46.000000000 +0200
+++ ./bin/phpreflect	2014-07-24 07:32:27.586541797 +0200
@@ -1,11 +1,11 @@
-#!/usr/bin/env php
+#!/usr/bin/php
 <?php
 
 if (\Phar::running()) {
     $vendorDir = 'phar://phpreflect.phar/vendor';
 } else {
     $baseDir   = dirname(__DIR__);
-    $vendorDir = $baseDir . '/vendor';
+    $vendorDir = '/usr/share/php';
 
     if (!getenv("REFLECT")) {
         $files = array(
@@ -21,7 +21,24 @@ if (\Phar::running()) {
         }
     }
 }
-require_once $vendorDir . '/autoload.php';
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+require_once $vendorDir . '/PHP/Timer.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Symfony\\Component\\Finder'          => $vendorDir,
+    'Symfony\\Component\\EventDispatcher' => $vendorDir,
+    'Symfony\\Component\\Console'         => $vendorDir,
+    'Symfony\\Component\\ClassLoader'     => $vendorDir,
+    'Bartlett'                            => $vendorDir,
+));
+$loader->register();
 
 if (PHP_SAPI !== 'cli') {
     return;
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2014-07-23 17:49:46.000000000 +0200
+++ ./tests/bootstrap.php	2014-07-24 07:32:12.151469115 +0200
@@ -1,6 +1,19 @@
 <?php
 
 $baseDir   = dirname(__DIR__);
-$vendorDir = $baseDir . '/vendor';
+$vendorDir = '/usr/share/php';
 
-require_once $vendorDir . '/autoload.php';
+require_once $vendorDir . '/PhpParser/Autoloader.php';
+PhpParser\Autoloader::register();
+
+require_once $vendorDir . '/Symfony/Component/ClassLoader/UniversalClassLoader.php';
+
+use Symfony\Component\ClassLoader\UniversalClassLoader;
+
+$loader = new UniversalClassLoader();
+$loader->registerNamespaces(array(
+    'Bartlett\\Tests\\Reflect'            => $baseDir . '/tests',
+    'Bartlett'                            => $baseDir . '/src',
+));
+$loader->useIncludePath(true);
+$loader->register();
