diff -up ./bin/composer.rpm ./bin/composer
--- ./bin/composer.rpm	2015-07-14 14:37:15.000000000 +0200
+++ ./bin/composer	2015-07-16 08:58:33.415021590 +0200
@@ -5,7 +5,7 @@ if (PHP_SAPI !== 'cli') {
     echo 'Warning: Composer should be invoked via the CLI version of PHP, not the '.PHP_SAPI.' SAPI'.PHP_EOL;
 }
 
-require __DIR__.'/../src/bootstrap.php';
+require '/usr/share/php/Composer/autoload.php';
 
 use Composer\Console\Application;
 
diff -up ./src/Composer/Json/JsonFile.php.rpm ./src/Composer/Json/JsonFile.php
--- ./src/Composer/Json/JsonFile.php.rpm	2015-07-14 14:37:15.000000000 +0200
+++ ./src/Composer/Json/JsonFile.php	2015-07-16 08:58:33.415021590 +0200
@@ -149,7 +149,7 @@ class JsonFile
             self::validateSyntax($content, $this->path);
         }
 
-        $schemaFile = __DIR__ . '/../../../res/composer-schema.json';
+        $schemaFile = (getenv('BUILDROOT')?:'') . '/usr/share/composer/res/composer-schema.json';
         $schemaData = json_decode(file_get_contents($schemaFile));
 
         if ($schema === self::LAX_SCHEMA) {
