diff -up ./bin/composer.rpm ./bin/composer
--- ./bin/composer.rpm	2016-12-12 10:30:58.174009561 +0100
+++ ./bin/composer	2016-12-12 10:32:17.465427241 +0100
@@ -5,7 +5,7 @@ if (PHP_SAPI !== 'cli') {
     echo 'Warning: Composer should be invoked via the CLI version of PHP, not the '.PHP_SAPI.' SAPI'.PHP_EOL;
 }
 
-require __DIR__.'/../src/bootstrap.php';
+require '/usr/share/php/Composer/autoload.php';
 
 use Composer\Factory;
 use Composer\XdebugHandler;
diff -up ./src/Composer/Autoload/AutoloadGenerator.php.rpm ./src/Composer/Autoload/AutoloadGenerator.php
--- ./src/Composer/Autoload/AutoloadGenerator.php.rpm	2016-12-11 17:04:50.000000000 +0100
+++ ./src/Composer/Autoload/AutoloadGenerator.php	2016-12-12 10:30:58.175009567 +0100
@@ -305,7 +305,7 @@ EOF;
         file_put_contents($targetDir.'/autoload_real.php', $this->getAutoloadRealFile(true, (bool) $includePathFileContents, $targetDirLoader, (bool) $includeFilesFileContents, $vendorPathCode, $appBaseDirCode, $suffix, $useGlobalIncludePath, $prependAutoloader, $staticPhpVersion));
 
         $this->safeCopy(__DIR__.'/ClassLoader.php', $targetDir.'/ClassLoader.php');
-        $this->safeCopy(__DIR__.'/../../../LICENSE', $targetDir.'/LICENSE');
+        $this->safeCopy((getenv('BUILDROOT')?:'') . '/usr/share/composer/LICENSE', $targetDir.'/LICENSE');
 
         if ($this->runScripts) {
             $this->eventDispatcher->dispatchScript(ScriptEvents::POST_AUTOLOAD_DUMP, $this->devMode, array(), array(
diff -up ./src/Composer/Compiler.php.rpm ./src/Composer/Compiler.php
--- ./src/Composer/Compiler.php.rpm	2016-12-11 17:04:50.000000000 +0100
+++ ./src/Composer/Compiler.php	2016-12-12 10:30:58.175009567 +0100
@@ -97,7 +97,7 @@ class Compiler
         $finder = new Finder();
         $finder->files()
             ->name('*.json')
-            ->in(__DIR__.'/../../res')
+            ->in((getenv('BUILDROOT')?:'') . '/usr/share/composer/res')
             ->in(SpdxLicenses::getResourcesDir())
             ->sort($finderSort)
         ;
diff -up ./src/Composer/Json/JsonFile.php.rpm ./src/Composer/Json/JsonFile.php
--- ./src/Composer/Json/JsonFile.php.rpm	2016-12-11 17:04:50.000000000 +0100
+++ ./src/Composer/Json/JsonFile.php	2016-12-12 10:30:58.175009567 +0100
@@ -156,7 +156,7 @@ class JsonFile
             self::validateSyntax($content, $this->path);
         }
 
-        $schemaFile = __DIR__ . '/../../../res/composer-schema.json';
+        $schemaFile = (getenv('BUILDROOT')?:'') . '/usr/share/composer/res/composer-schema.json';
         $schemaData = json_decode(file_get_contents($schemaFile));
 
         if ($schema === self::LAX_SCHEMA) {
diff -up ./tests/Composer/Test/Json/ComposerSchemaTest.php.rpm ./tests/Composer/Test/Json/ComposerSchemaTest.php
--- ./tests/Composer/Test/Json/ComposerSchemaTest.php.rpm	2016-12-11 17:04:50.000000000 +0100
+++ ./tests/Composer/Test/Json/ComposerSchemaTest.php	2016-12-12 10:30:58.175009567 +0100
@@ -95,7 +95,7 @@ class ComposerSchemaTest extends \PHPUni
 
     private function check($json)
     {
-        $schema = json_decode(file_get_contents(__DIR__ . '/../../../../res/composer-schema.json'));
+        $schema = json_decode(file_get_contents((getenv('BUILDROOT')?:'') . '/usr/share/composer/res/composer-schema.json'));
         $validator = new Validator();
         $validator->check(json_decode($json), $schema);
 
