--- mysql-workbench-oss-5.2.22/res/scripts/snippets/shell_snippets.py.orig	2010-06-04 21:38:58.639945233 +0200
+++ mysql-workbench-oss-5.2.22/res/scripts/snippets/shell_snippets.py	2010-06-04 21:39:15.835195627 +0200
@@ -1,19 +1,19 @@
- # iterate through all schemas
- for schema in grt.root.wb.doc.physicalModels[0].catalog.schemata:
+# iterate through all schemas
+for schema in grt.root.wb.doc.physicalModels[0].catalog.schemata:
      print schema.name
 
- # iterate through all tables from schema
- schema = grt.root.wb.doc.physicalModels[0].catalog.schemata[0]
- for table in schema.tables:
+# iterate through all tables from schema
+schema = grt.root.wb.doc.physicalModels[0].catalog.schemata[0]
+for table in schema.tables:
      print table.name
 
- # iterate through columns from schema
- schema = grt.root.wb.doc.physicalModels[0].catalog.schemata[0]
- for table in schema.tables:
+# iterate through columns from schema
+schema = grt.root.wb.doc.physicalModels[0].catalog.schemata[0]
+for table in schema.tables:
      for column in table.columns:
          print table.name, column.name
 
- # iterate through all figures of a diagram
- diagram = grt.root.wb.doc.physicalModels[0].diagram
- for figure in diagram.figures:
+# iterate through all figures of a diagram
+diagram = grt.root.wb.doc.physicalModels[0].diagram
+for figure in diagram.figures:
      print figure.__grtclassname__, figure.left, figure.top
--- mysql-workbench-gpl-5.2.28-src/res/scripts/snippets/shell_snippets.py.orig	2010-09-21 17:38:45.243356701 +0200
+++ mysql-workbench-gpl-5.2.28-src/res/scripts/snippets/shell_snippets.py	2010-09-21 17:38:59.850746929 +0200
@@ -18,14 +18,14 @@
 for figure in diagram.figures:
      print figure.__grtclassname__, figure.left, figure.top
 
- # iterate through all resultset rows
- resultset = grt.root.wb.sqlEditors[0].activeResultset
- flag = resultset.goToFirst()
- while flag:
+# iterate through all resultset rows
+resultset = grt.root.wb.sqlEditors[0].activeResultset
+flag = resultset.goToFirst()
+while flag:
    print resultset.stringFieldValue(0)
    flag = resultset.nextRow()
 
- # replace sql editor contents
- editor = grt.root.wb.sqlEditors[0].activeQueryBuffer
- new_text = editor.sql
- editor.replaceContents(new_text)
+# replace sql editor contents
+editor = grt.root.wb.sqlEditors[0].activeQueryBuffer
+new_text = editor.sql
+editor.replaceContents(new_text)
