--- mysql-workbench-gpl-5.2.44-src/ext/Makefile.am.orig	2012-10-20 08:30:57.754850970 +0200
+++ mysql-workbench-gpl-5.2.44-src/ext/Makefile.am	2012-10-20 08:31:19.548923647 +0200
@@ -3,7 +3,7 @@
 antlr_dirs=antlr-runtime
 endif
 
-SUBDIRS=scintilla/gtk vsqlite++ $(antlr_dirs)
+SUBDIRS=scintilla/gtk $(antlr_dirs)
 
 if BUNDLE_UTILITIES
 all-am:
--- mysql-workbench-gpl-5.2.44-src/configure.in.orig	2012-10-20 08:38:28.584365925 +0200
+++ mysql-workbench-gpl-5.2.44-src/configure.in	2012-10-20 08:38:50.359439239 +0200
@@ -487,8 +487,6 @@
 res/sqlidedata/Makefile
 ext/Makefile
 ext/scintilla/gtk/Makefile
-ext/vsqlite++/Makefile
-ext/vsqlite++/src/sqlite/Makefile
 backend/Makefile
 backend/wbpublic/Makefile
 backend/wbprivate/Makefile
