--- mysql-workbench-gpl-5.2.32-src/library/forms/Makefile.am.tinyxml	2011-03-09 17:29:40.830185016 +0100
+++ mysql-workbench-gpl-5.2.32-src/library/forms/Makefile.am	2011-03-09 17:29:58.392510545 +0100
@@ -67,7 +67,7 @@
 gtk/src/lf_menu.cpp\
 gtk/src/lf_menubar.cpp
 
-libmforms_la_LIBADD=@GNOME_LIBS@ @GNOME_KEYRING_LIBS@
+libmforms_la_LIBADD=@GNOME_LIBS@ @GNOME_KEYRING_LIBS@ -ltinyxml
 
 INCLUDES=@GNOME_CFLAGS@ @SIGC_CFLAGS@ @CAIRO_CFLAGS@ @GNOME_KEYRING_CFLAGS@ \
 -I$(top_srcdir)/library/forms\
@@ -79,7 +79,6 @@
 -I$(top_srcdir)/ext/vsqlite\
 -I$(top_srcdir)/library/\
 -I$(top_srcdir)/ext\
--I$(top_srcdir)/ext/tinyxml\
 -I$(top_srcdir)/ext/scintilla/include
 
 
--- mysql-workbench-gpl-5.2.32-src/modules/wb.mysql.import/Makefile.am.tinyxml	2011-03-04 01:54:41.000000000 +0100
+++ mysql-workbench-gpl-5.2.32-src/modules/wb.mysql.import/Makefile.am	2011-03-09 17:21:41.863136166 +0100
@@ -17,7 +17,6 @@
 	-I$(top_srcdir)/library/sql-parser/source/linux\
 	-I$(top_srcdir)/ext/vsqlite\
 	-I$(top_srcdir)/ext/vsqlite++/include\
-	-I$(top_srcdir)/ext/tinyxml\
 	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
 
 
@@ -26,8 +25,4 @@
 
 wb_mysql_import_grt_la_SOURCES=\
 	src/wb_mysql_import.cpp\
-	src/wb_mysql_import_dbd4.cpp\
-	$(top_srcdir)/ext/tinyxml/tinyxmlparser.cpp\
-	$(top_srcdir)/ext/tinyxml/tinyxml.cpp\
-	$(top_srcdir)/ext/tinyxml/tinystr.cpp\
-	$(top_srcdir)/ext/tinyxml/tinyxmlerror.cpp
+	src/wb_mysql_import_dbd4.cpp
--- mysql-workbench-gpl-5.2.32-src/backend/wbprivate/Makefile.am.tinyxml	2011-03-04 01:54:50.000000000 +0100
+++ mysql-workbench-gpl-5.2.32-src/backend/wbprivate/Makefile.am	2011-03-09 17:21:41.860136279 +0100
@@ -13,8 +13,7 @@
   -I$(top_srcdir)/ext/cppconn/ -I$(top_srcdir)/library/dbc/src\
   -I$(top_srcdir)/ext/scintilla/include\
   -I$(top_srcdir)/ext/vsqlite++/include\
-  -I$(top_srcdir)/ext/vsqlite\
-  -I$(top_srcdir)/ext/tinyxml
+  -I$(top_srcdir)/ext/vsqlite
 
 libwbprivate_a_SOURCES=\
 	model/canvas_floater.cpp\
@@ -74,11 +73,7 @@
   $(top_srcdir)/frontend/common/grt_shell_window.cpp\
   $(top_srcdir)/frontend/common/grt_code_editor.cpp\
   $(top_srcdir)/frontend/common/grt_python_debugger.cpp\
-  $(top_srcdir)/frontend/common/plugin_manager_window.cpp\
-  $(top_srcdir)/ext/tinyxml/tinyxmlparser.cpp\
-  $(top_srcdir)/ext/tinyxml/tinyxml.cpp\
-  $(top_srcdir)/ext/tinyxml/tinystr.cpp\
-  $(top_srcdir)/ext/tinyxml/tinyxmlerror.cpp
+  $(top_srcdir)/frontend/common/plugin_manager_window.cpp
 
 #BUILT_SOURCES=stdafx.h.gch
 #stdafx.h.gch: workbench/stdafx.h
