diff -up mysql-workbench-gpl-5.2.41-src/configure.in.antlr mysql-workbench-gpl-5.2.41-src/configure.in
--- mysql-workbench-gpl-5.2.41-src/configure.in.antlr	2012-08-11 09:16:26.180997691 +0200
+++ mysql-workbench-gpl-5.2.41-src/configure.in	2012-08-11 09:16:26.201997764 +0200
@@ -377,7 +377,8 @@ AC_SUBST(ODBC_LIBS)
 AC_CHECK_HEADER([antlr3.h], ac_system_antlr_runtime=yes, ac_system_antlr_runtime=no)
 if test "$ac_system_antlr_runtime" = yes; then
     AC_RUN_IFELSE([AC_LANG_SOURCE([#include <antlr3config.h>
-int main() { return strcmp(PACKAGE_VERSION, "3.4") >= 0; }
+#include <string.h>
+int main() { return strcmp(PACKAGE_VERSION, "3.4") < 0; }
 ])],
            AC_MSG_RESULT([antlr runtimes for C >= 3.4 found]),
            [AC_MSG_RESULT([required antlr runtimes not found, using bundled])
diff -up mysql-workbench-gpl-5.2.41-src/ext/Makefile.am.antlr mysql-workbench-gpl-5.2.41-src/ext/Makefile.am
--- mysql-workbench-gpl-5.2.41-src/ext/Makefile.am.antlr	2012-08-11 09:17:06.014135012 +0200
+++ mysql-workbench-gpl-5.2.41-src/ext/Makefile.am	2012-08-11 09:17:18.452177547 +0200
@@ -1,5 +1,5 @@
 
-SUBDIRS=scintilla/gtk vsqlite++ antlr-runtime
+SUBDIRS=scintilla/gtk vsqlite++
 
 if BUNDLE_UTILITIES
 all-am:
