diff -up mysql-workbench-gpl-5.2.27/modules/db/Makefile.am.dnl mysql-workbench-gpl-5.2.27/modules/db/Makefile.am
--- mysql-workbench-gpl-5.2.27/modules/db/Makefile.am.dnl	2010-09-18 17:06:10.582256133 +0200
+++ mysql-workbench-gpl-5.2.27/modules/db/Makefile.am	2010-09-18 17:07:37.070380972 +0200
@@ -11,4 +11,4 @@ INCLUDES=-I$(top_srcdir)/generated -I$(t
 
 dbutils_grt_la_SOURCES=src/dbutils.cpp
 dbutils_grt_la_LDFLAGS=-module
-dnl dbutils_grt_la_LIBADD=$(top_srcdir)/backend/grtdb/libgrtdbbe.la $(top_srcdir)/backend/grt/libgrtbe.la
+#dbutils_grt_la_LIBADD=$(top_srcdir)/backend/grtdb/libgrtdbbe.la $(top_srcdir)/backend/grt/libgrtbe.la
diff -up mysql-workbench-gpl-5.2.27/modules/db/Makefile.in.dnl mysql-workbench-gpl-5.2.27/modules/db/Makefile.in
--- mysql-workbench-gpl-5.2.27/modules/db/Makefile.in.dnl	2010-09-18 17:06:15.708005899 +0200
+++ mysql-workbench-gpl-5.2.27/modules/db/Makefile.in	2010-09-18 17:07:07.737130492 +0200
@@ -523,7 +523,7 @@ uninstall-am: uninstall-pkglibLTLIBRARIE
 	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
 	tags uninstall uninstall-am uninstall-pkglibLTLIBRARIES
 
-dnl dbutils_grt_la_LIBADD=$(top_srcdir)/backend/grtdb/libgrtdbbe.la $(top_srcdir)/backend/grt/libgrtbe.la
+#dbutils_grt_la_LIBADD=$(top_srcdir)/backend/grtdb/libgrtdbbe.la $(top_srcdir)/backend/grt/libgrtbe.la
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
