diff -up mysql-workbench-gpl-5.2.36-src/backend/wbprivate/Makefile.am.tinyxml mysql-workbench-gpl-5.2.36-src/backend/wbprivate/Makefile.am
--- mysql-workbench-gpl-5.2.36-src/backend/wbprivate/Makefile.am.tinyxml	2011-12-02 12:37:03.000000000 +0100
+++ mysql-workbench-gpl-5.2.36-src/backend/wbprivate/Makefile.am	2011-12-10 07:49:14.230858719 +0100
@@ -13,8 +13,7 @@ INCLUDES=@GLIB_CFLAGS@ @SIGC_CFLAGS@ @GR
   -I$(top_srcdir)/ext/cppconn/ -I$(top_srcdir)/library/dbc/src\
   -I$(top_srcdir)/ext/scintilla/include\
   -I$(top_srcdir)/ext/vsqlite++/include\
-  -I$(top_srcdir)/ext/vsqlite\
-  -I$(top_srcdir)/ext/tinyxml
+  -I$(top_srcdir)/ext/vsqlite
 
 libwbprivate_a_SOURCES=\
 	model/canvas_floater.cpp\
@@ -76,11 +75,7 @@ libwbprivate_a_SOURCES=\
   $(top_srcdir)/frontend/common/grt_python_debugger.cpp\
   $(top_srcdir)/frontend/common/plugin_manager_window.cpp\
   $(top_srcdir)/frontend/common/plugin_install_window.cpp\
-  $(top_srcdir)/frontend/common/select_option_dialog.cpp\
-  $(top_srcdir)/ext/tinyxml/tinyxmlparser.cpp\
-  $(top_srcdir)/ext/tinyxml/tinyxml.cpp\
-  $(top_srcdir)/ext/tinyxml/tinystr.cpp\
-  $(top_srcdir)/ext/tinyxml/tinyxmlerror.cpp
+  $(top_srcdir)/frontend/common/select_option_dialog.cpp
 
 #BUILT_SOURCES=stdafx.h.gch
 #stdafx.h.gch: workbench/stdafx.h
diff -up mysql-workbench-gpl-5.2.36-src/library/forms/Makefile.am.tinyxml mysql-workbench-gpl-5.2.36-src/library/forms/Makefile.am
--- mysql-workbench-gpl-5.2.36-src/library/forms/Makefile.am.tinyxml	2011-12-02 12:37:01.000000000 +0100
+++ mysql-workbench-gpl-5.2.36-src/library/forms/Makefile.am	2011-12-10 07:49:59.540022494 +0100
@@ -74,7 +74,7 @@ gtk/src/lf_menu.cpp\
 gtk/src/lf_menubar.cpp\
 gtk/src/lf_hypertext.cpp
 
-libmforms_la_LIBADD=@GNOME_LIBS@ @GNOME_KEYRING_LIBS@
+libmforms_la_LIBADD=@GNOME_LIBS@ @GNOME_KEYRING_LIBS@ -ltinyxml
 
 INCLUDES=@GLIB_CFLAGS@ @GNOME_CFLAGS@ @SIGC_CFLAGS@ @CAIRO_CFLAGS@ @GNOME_KEYRING_CFLAGS@ @PCRE_CFLAGS@ \
 -I$(top_srcdir)/library/forms\
@@ -86,7 +86,6 @@ INCLUDES=@GLIB_CFLAGS@ @GNOME_CFLAGS@ @S
 -I$(top_srcdir)/ext/vsqlite\
 -I$(top_srcdir)/library/\
 -I$(top_srcdir)/ext\
--I$(top_srcdir)/ext/tinyxml\
 -I$(top_srcdir)/ext/scintilla/include
 
 
diff -up mysql-workbench-gpl-5.2.36-src/modules/wb.mysql.import/Makefile.am.tinyxml mysql-workbench-gpl-5.2.36-src/modules/wb.mysql.import/Makefile.am
--- mysql-workbench-gpl-5.2.36-src/modules/wb.mysql.import/Makefile.am.tinyxml	2011-12-02 12:37:03.000000000 +0100
+++ mysql-workbench-gpl-5.2.36-src/modules/wb.mysql.import/Makefile.am	2011-12-10 07:44:49.683838383 +0100
@@ -17,7 +17,6 @@ INCLUDES=\
 	-I$(top_srcdir)/library/sql-parser/source/linux\
 	-I$(top_srcdir)/ext/vsqlite\
 	-I$(top_srcdir)/ext/vsqlite++/include\
-	-I$(top_srcdir)/ext/tinyxml\
 	@MYSQL_INCLUDE@ @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
 
 
@@ -26,8 +25,4 @@ wb_mysql_import_grt_la_LDFLAGS=-module
 
 wb_mysql_import_grt_la_SOURCES=\
 	src/wb_mysql_import.cpp\
-	src/wb_mysql_import_dbd4.cpp\
-	$(top_srcdir)/ext/tinyxml/tinyxmlparser.cpp\
-	$(top_srcdir)/ext/tinyxml/tinyxml.cpp\
-	$(top_srcdir)/ext/tinyxml/tinystr.cpp\
-	$(top_srcdir)/ext/tinyxml/tinyxmlerror.cpp
+	src/wb_mysql_import_dbd4.cpp
