--- a/ext/date/tests/bug66985.phpt	2014-10-30 07:32:03.297693403 +0100
+++ b/ext/date/tests/bug66985.phpt	2014-10-30 07:32:45.138877977 +0100
@@ -3,7 +3,7 @@
 --FILE--
 <?php
 $zones = array(
-	"CST6CDT", "Cuba", "Egypt", "Eire", "EST5EDT", "Factory", "GB-Eire",
+	"CST6CDT", "Cuba", "Egypt", "Eire", "EST5EDT", "GB-Eire",
 	"GMT0", "Greenwich", "Hongkong", "Iceland", "Iran", "Israel", "Jamaica",
 	"Japan", "Kwajalein", "Libya", "MST7MDT", "Navajo", "NZ-CHAT", "Poland",
 	"Portugal", "PST8PDT", "Singapore", "Turkey", "Universal", "W-SU",
@@ -45,11 +45,6 @@
 )
 DateTimeZone Object
 (
-    [timezone_type] => 3
-    [timezone] => Factory
-)
-DateTimeZone Object
-(
     [timezone_type] => 3
     [timezone] => GB-Eire
 )
diff -up ./ext/date/tests/strtotime3-64bit.phpt.datetests ./ext/date/tests/strtotime3-64bit.phpt
--- ./ext/date/tests/strtotime3-64bit.phpt.datetests	2016-07-21 02:23:03.000000000 +0200
+++ ./ext/date/tests/strtotime3-64bit.phpt	2016-07-26 07:39:45.713272263 +0200
@@ -44,7 +44,7 @@ foreach ($strs as $str) {
 }
 
 ?>
---EXPECT--
+--EXPECTF--
 bool(false)
 bool(false)
 string(31) "Thu, 15 Jun 2006 00:00:00 +0100"
@@ -53,7 +53,7 @@ bool(false)
 string(31) "Fri, 16 Jun 2006 23:49:12 +0100"
 bool(false)
 string(31) "Fri, 16 Jun 2006 02:22:00 +0100"
-string(31) "Sun, 16 Jun 0222 02:22:00 -0036"
+string(31) "Sun, 16 Jun 0222 02:22:00 %s"
 string(31) "Fri, 16 Jun 2006 02:22:33 +0100"
 bool(false)
 string(31) "Tue, 02 Mar 2004 00:00:00 +0000"
diff -up ./ext/date/tests/bug33414-2.phpt.datetests ./ext/date/tests/bug33414-2.phpt
--- ./ext/date/tests/bug33414-2.phpt.datetests	2016-07-21 02:23:03.000000000 +0200
+++ ./ext/date/tests/bug33414-2.phpt	2016-07-26 07:28:10.323598643 +0200
@@ -74,7 +74,7 @@ $strtotime_tstamp = strtotime("next Frid
 print "result=".date("l Y-m-d H:i:s T I", $strtotime_tstamp)."\n";
 print "wanted=Friday            00:00:00\n\n";
 ?>
---EXPECT--
+--EXPECTF--
 TZ=Pacific/Rarotonga - wrong day.
 tStamp=Thursday 1970-01-01 17:17:17 CKT 0
 result=Tuesday 1970-01-06 00:00:00 CKT 0
@@ -106,8 +106,8 @@ result=Tuesday 2005-04-12 00:00:00 CDT 1
 wanted=Tuesday            00:00:00
 
 TZ=Pacific/Pitcairn - wrong day.
-tStamp=Thursday 1970-01-01 17:17:17 PNT 0
-result=Wednesday 1970-01-07 00:00:00 PNT 0
+tStamp=Thursday 1970-%s
+result=Wednesday 1970-%s
 wanted=Wednesday            00:00:00
 
 TZ=Pacific/Fakaofo - wrong day.

