
Make generated php_config.h constant across rebuilds.

--- php-7.0.0/configure.in.fixheader
+++ php-7.0.0/configure.in
@@ -1278,7 +1278,7 @@ fi
 EXTRA_LDFLAGS="$EXTRA_LDFLAGS $PHP_LDFLAGS"
 EXTRA_LDFLAGS_PROGRAM="$EXTRA_LDFLAGS_PROGRAM $PHP_LDFLAGS"
 
-PHP_BUILD_DATE=`date '+%Y-%m-%d'`
+PHP_BUILD_DATE=`date '+%Y-%m-%d' -r $srcdir/NEWS`
 AC_DEFINE_UNQUOTED(PHP_BUILD_DATE,"$PHP_BUILD_DATE",[PHP build date])
 
 case $host_alias in
@@ -1289,7 +1289,7 @@ case $host_alias in
   AC_DEFINE_UNQUOTED(PHP_UNAME,"$PHP_UNAME",[hardcode for each of the cross compiler host])
   ;;
 *)
-  PHP_UNAME=`uname -a | xargs`
+  PHP_UNAME=`uname | xargs`
   AC_DEFINE_UNQUOTED(PHP_UNAME,"$PHP_UNAME",[uname -a output])
   PHP_OS=`uname | xargs`
   AC_DEFINE_UNQUOTED(PHP_OS,"$PHP_OS",[uname output])
