From 8cfc819265183c57e5b24528d766768f55ae76df Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 25 Mar 2016 17:25:05 +0100
Subject: [PATCH] missing property

---
 src/Monolog/Handler/SocketHandler.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Monolog/Handler/SocketHandler.php b/src/Monolog/Handler/SocketHandler.php
index c458b1c..7a61bf4 100644
--- a/src/Monolog/Handler/SocketHandler.php
+++ b/src/Monolog/Handler/SocketHandler.php
@@ -30,6 +30,7 @@ class SocketHandler extends AbstractProcessingHandler
     private $persistent = false;
     private $errno;
     private $errstr;
+    private $lastWritingAt;
 
     /**
      * @param string  $connectionString Socket connection string
