diff -ru bin.old/classmap_generator.php bin/classmap_generator.php
--- bin.old/classmap_generator.php	2015-08-03 17:13:58.000000000 +0200
+++ bin/classmap_generator.php	2015-08-06 11:42:29.000000000 +0200
@@ -29,19 +29,7 @@
  */
 
 // Setup/verify autoloading
-if (file_exists(__DIR__ . '/../vendor/autoload.php')) {
-    // Local install
-    require __DIR__ . '/../vendor/autoload.php';
-} elseif (file_exists(getcwd() . '/vendor/autoload.php')) {
-    // Root project is current working directory
-    require getcwd() . '/vendor/autoload.php';
-} elseif (file_exists(__DIR__ . '/../../../autoload.php')) {
-    // Relative to composer install
-    require __DIR__ . '/../../../autoload.php';
-} else {
-    fwrite(STDERR, "Unable to setup autoloading; aborting\n");
-    exit(2);
-}
+require '/usr/share/php/Zend/autoload.php';
 
 $libraryPath = getcwd();
 
diff -ru bin.old/pluginmap_generator.php bin/pluginmap_generator.php
--- bin.old/pluginmap_generator.php	2015-08-03 17:13:58.000000000 +0200
+++ bin/pluginmap_generator.php	2015-08-06 11:42:22.000000000 +0200
@@ -29,19 +29,7 @@
  */
 
 // Setup/verify autoloading
-if (file_exists(__DIR__ . '/../vendor/autoload.php')) {
-    // Local install
-    require __DIR__ . '/../vendor/autoload.php';
-} elseif (file_exists(getcwd() . '/vendor/autoload.php')) {
-    // Root project is current working directory
-    require getcwd() . '/vendor/autoload.php';
-} elseif (file_exists(__DIR__ . '/../../../autoload.php')) {
-    // Relative to composer install
-    require __DIR__ . '/../../../autoload.php';
-} else {
-    fwrite(STDERR, "Unable to setup autoloading; aborting\n");
-    exit(2);
-}
+require '/usr/share/php/Zend/autoload.php';
 
 $libPath = getenv('LIB_PATH') ? getenv('LIB_PATH') : __DIR__ . '/../library';
 if (!is_dir($libPath)) {
diff -ru bin.old/templatemap_generator.php bin/templatemap_generator.php
--- bin.old/templatemap_generator.php	2015-08-03 17:13:58.000000000 +0200
+++ bin/templatemap_generator.php	2015-08-06 11:42:12.000000000 +0200
@@ -29,19 +29,7 @@
  */
 
 // Setup/verify autoloading
-if (file_exists(__DIR__ . '/../vendor/autoload.php')) {
-    // Local install
-    require __DIR__ . '/../vendor/autoload.php';
-} elseif (file_exists(getcwd() . '/vendor/autoload.php')) {
-    // Root project is current working directory
-    require getcwd() . '/vendor/autoload.php';
-} elseif (file_exists(__DIR__ . '/../../../autoload.php')) {
-    // Relative to composer install
-    require __DIR__ . '/../../../autoload.php';
-} else {
-    fwrite(STDERR, "Unable to setup autoloading; aborting\n");
-    exit(2);
-}
+require '/usr/share/php/Zend/autoload.php';
 
 $libraryPath = getcwd();
 $viewPath    = getcwd() . '/view';
