diff -up ext/assets/filters/cssmin.c.rpm ext/assets/filters/cssmin.c
--- ext/assets/filters/cssmin.c.rpm	2014-09-07 09:53:18.975164612 +0200
+++ ext/assets/filters/cssmin.c	2014-09-07 09:53:23.879188187 +0200
@@ -18,11 +18,7 @@
 */
 
 #include "assets/filters/cssmin.h"
-#ifdef PHALCON_NON_FREE
-#include "assets/filters/cssminifier.h"
-#else
 #include "assets/filters/nocssminifier.h"
-#endif
 #include "assets/filterinterface.h"
 
 #include "kernel/main.h"
diff -up ext/assets/filters/jsmin.c.rpm ext/assets/filters/jsmin.c
--- ext/assets/filters/jsmin.c.rpm	2014-09-07 09:53:10.325123035 +0200
+++ ext/assets/filters/jsmin.c	2014-09-07 09:53:29.054213063 +0200
@@ -18,11 +18,7 @@
 */
 
 #include "assets/filters/jsmin.h"
-#ifdef PHALCON_NON_FREE
-#include "assets/filters/jsminifier.h"
-#else
 #include "assets/filters/nojsminifier.h"
-#endif
 #include "assets/filterinterface.h"
 
 #include "kernel/main.h"
diff -up build/_resource/config/phalcon_c_priority_files.php.rpm build/_resource/config/phalcon_c_priority_files.php
--- build/_resource/config/phalcon_c_priority_files.php.rpm	2014-09-07 09:58:44.109729141 +0200
+++ build/_resource/config/phalcon_c_priority_files.php	2014-09-07 09:58:48.428749943 +0200
@@ -3,8 +3,8 @@
 
 return array(
     // Header files
-    'assets/filters/jsminifier.h',
-    'assets/filters/cssminifier.h',
+    'assets/filters/nojsminifier.h',
+    'assets/filters/nocssminifier.h',
     'mvc/model/query/parser.h',
     'mvc/model/query/scanner.h',
     'mvc/model/query/phql.h',
