--- phpab.php.rpm	2014-11-25 11:32:11.000000000 +0100
+++ phpab.php	2014-11-25 14:40:59.288514790 +0100
@@ -42,12 +42,15 @@
  *   4 - Lint Error
  */
 
-define('PHPAB_VERSION', '%development%');
+define('PHPAB_VERSION', '@VERSION@');
 
 if (!ini_get('date.timezone')) {
      ini_set('date.timezone', 'UTC');
 }
-require __DIR__ . '/src/autoload.php';
+require '/usr/share/php/TheSeer/DirectoryScanner/autoload.php';
+require '/usr/share/php/TheSeer/Autoload/autoload.php';
+require '/usr/share/php/ezc/Base/base.php';
+spl_autoload_register(array('\ezcBase','autoload'));
 
 $factory = new \TheSeer\Autoload\Factory();
 $factory->getCLI()->run();
