--- phpab.php.rpm	2015-06-30 22:02:53.000000000 +0200
+++ phpab.php	2015-07-01 10:13:32.443481551 +0200
@@ -42,12 +42,12 @@
  *   4 - Lint Error
  */
 
-define('PHPAB_VERSION', '%development%');
+define('PHPAB_VERSION', '@VERSION@');
 
 if (!ini_get('date.timezone')) {
      ini_set('date.timezone', 'UTC');
 }
-require __DIR__ . '/src/autoload.php';
+require '/usr/share/php/TheSeer/Autoload/autoload.php';
 
 $factory = new \TheSeer\Autoload\Factory();
 $factory->getCLI()->run();
