diff -up ./bin/generate_vcards.rpm ./bin/generate_vcards
--- ./bin/generate_vcards.rpm	2015-07-20 15:36:00.466610252 +0200
+++ ./bin/generate_vcards	2015-07-20 15:36:24.434738474 +0200
@@ -6,8 +6,7 @@ namespace Sabre\VObject;
 // This sucks.. we have to try to find the composer autoloader. But chances
 // are, we can't find it this way. So we'll do our bestest
 $paths = array(
-    __DIR__ . '/../vendor/autoload.php',  // In case vobject is cloned directly
-    __DIR__ . '/../../../autoload.php',   // In case vobject is a composer dependency.
+    '/usr/share/php/Sabre/VObject/autoload.php',  // RPM installation
 );
 
 foreach($paths as $path) {
diff -up ./bin/vobject.rpm ./bin/vobject
--- ./bin/vobject.rpm	2015-07-20 15:34:43.499198500 +0200
+++ ./bin/vobject	2015-07-20 15:36:20.645718203 +0200
@@ -6,8 +6,7 @@ namespace Sabre\VObject;
 // This sucks.. we have to try to find the composer autoloader. But chances
 // are, we can't find it this way. So we'll do our bestest
 $paths = array(
-    __DIR__ . '/../vendor/autoload.php',  // In case vobject is cloned directly
-    __DIR__ . '/../../../autoload.php',   // In case vobject is a composer dependency.
+    '/usr/share/php/Sabre/VObject/autoload.php',  // RPM installation
 );
 
 foreach($paths as $path) {
diff -up ./tests/bootstrap.php.rpm ./tests/bootstrap.php
--- ./tests/bootstrap.php.rpm	2015-07-20 15:36:46.774857986 +0200
+++ ./tests/bootstrap.php	2015-07-20 15:40:20.007998716 +0200
@@ -3,8 +3,7 @@
 date_default_timezone_set('UTC');
 
 $try = array(
-    __DIR__ . '/../vendor/autoload.php',
-    __DIR__ . '/../../../autoload.php',
+    '@BUILDROOT@/usr/share/php/Sabre/VObject/autoload.php',  // RPM installation
 );
 
 foreach($try as $path) {
@@ -14,8 +13,6 @@ foreach($try as $path) {
     }
 }
 
-$autoLoader->addPsr4('Sabre\\VObject\\',__DIR__ . '/VObject');
-
 if (!defined('SABRE_TEMPDIR')) {
   define('SABRE_TEMPDIR', __DIR__ . '/temp/');
 }
@@ -23,3 +21,7 @@ if (!defined('SABRE_TEMPDIR')) {
 if (!file_exists(SABRE_TEMPDIR)) {
   mkdir(SABRE_TEMPDIR);
 }
+
+// Not catched by autoloader
+require_once __DIR__ . '/VObject/TestCase.php';
+require_once __DIR__ . '/VObject/ITip/BrokerTester.php';
