--- php/php-src/branches/PHP_5_3/ext/standard/php_crypt_r.c	2011/08/19 22:45:43	315217
+++ php/php-src/branches/PHP_5_3/ext/standard/php_crypt_r.c	2011/08/19 22:49:18	315218
@@ -382,7 +382,7 @@
 	/* Now make the output string */
 	memcpy(passwd, MD5_MAGIC, MD5_MAGIC_LEN);
 	strlcpy(passwd + MD5_MAGIC_LEN, sp, sl + 1);
-	strlcat(passwd, "$", 1);
+	strcat(passwd, "$");
 
 	PHP_MD5Final(final, &ctx);
 
