diff -up php-5.3.7RC1/configure.in.gnusrc php-5.3.7RC1/configure.in
--- php-5.3.7RC1/configure.in.gnusrc	2011-06-16 14:40:37.000000000 +0200
+++ php-5.3.7RC1/configure.in	2011-06-25 16:27:25.384248242 +0200
@@ -58,6 +58,8 @@ AC_DEFUN([PHP_EXT_DIR],[ext/$1])dnl
 AC_DEFUN([PHP_EXT_SRCDIR],[$abs_srcdir/ext/$1])dnl
 AC_DEFUN([PHP_ALWAYS_SHARED],[])dnl
 
+AC_DEFINE([_GNU_SOURCE], 1, [Define to enable GNU C Library extensions])
+
 dnl Setting up the PHP version based on the information above.
 dnl -------------------------------------------------------------------------
 
diff -up php-5.3.7RC1/ext/interbase/interbase.c.gnusrc php-5.3.7RC1/ext/interbase/interbase.c
--- php-5.3.7RC1/ext/interbase/interbase.c.gnusrc	2011-02-24 03:42:38.000000000 +0100
+++ php-5.3.7RC1/ext/interbase/interbase.c	2011-06-25 16:27:25.385248245 +0200
@@ -24,7 +24,6 @@
 #include "config.h"
 #endif
 
-#define _GNU_SOURCE
 
 #include "php.h"
 
diff -up php-5.3.7RC1/ext/pdo_firebird/firebird_driver.c.gnusrc php-5.3.7RC1/ext/pdo_firebird/firebird_driver.c
--- php-5.3.7RC1/ext/pdo_firebird/firebird_driver.c.gnusrc	2011-01-01 03:19:59.000000000 +0100
+++ php-5.3.7RC1/ext/pdo_firebird/firebird_driver.c	2011-06-25 16:27:25.385248245 +0200
@@ -22,7 +22,6 @@
 #include "config.h"
 #endif
 
-#define _GNU_SOURCE
 
 #include "php.h"
 #ifdef ZEND_ENGINE_2
diff -up php-5.3.7RC1/ext/standard/file.c.gnusrc php-5.3.7RC1/ext/standard/file.c
--- php-5.3.7RC1/ext/standard/file.c.gnusrc	2011-05-29 12:23:06.000000000 +0200
+++ php-5.3.7RC1/ext/standard/file.c	2011-06-25 16:27:25.386248248 +0200
@@ -113,9 +113,6 @@ php_file_globals file_globals;
 #endif
 
 #if defined(HAVE_FNMATCH) && !defined(PHP_WIN32)
-# ifndef _GNU_SOURCE
-#  define _GNU_SOURCE
-# endif
 # include <fnmatch.h>
 #endif
 
diff -up php-5.3.7RC1/ext/zlib/zlib_fopen_wrapper.c.gnusrc php-5.3.7RC1/ext/zlib/zlib_fopen_wrapper.c
--- php-5.3.7RC1/ext/zlib/zlib_fopen_wrapper.c.gnusrc	2011-01-01 03:19:59.000000000 +0100
+++ php-5.3.7RC1/ext/zlib/zlib_fopen_wrapper.c	2011-06-25 16:27:25.387248251 +0200
@@ -19,8 +19,6 @@
 
 /* $Id: zlib_fopen_wrapper.c 306939 2011-01-01 02:19:59Z felipe $ */
 
-#define _GNU_SOURCE
-
 #include "php.h"
 #include "php_zlib.h"
 #include "fopen_wrappers.h"
diff -up php-5.3.7RC1/main/php.h.gnusrc php-5.3.7RC1/main/php.h
--- php-5.3.7RC1/main/php.h.gnusrc	2011-01-01 03:19:59.000000000 +0100
+++ php-5.3.7RC1/main/php.h	2011-06-25 16:27:25.388248254 +0200
@@ -30,6 +30,7 @@
 #define PHP_HAVE_STREAMS
 #define YYDEBUG 0
 
+#include "php_config.h"
 #include "php_version.h"
 #include "zend.h"
 #include "zend_qsort.h"
diff -up php-5.3.7RC1/main/streams/cast.c.gnusrc php-5.3.7RC1/main/streams/cast.c
--- php-5.3.7RC1/main/streams/cast.c.gnusrc	2011-01-20 07:32:59.000000000 +0100
+++ php-5.3.7RC1/main/streams/cast.c	2011-06-25 16:27:25.388248254 +0200
@@ -18,7 +18,6 @@
 
 /* $Id: cast.c 307611 2011-01-20 06:32:59Z pajoye $ */
 
-#define _GNU_SOURCE
 #include "php.h"
 #include "php_globals.h"
 #include "php_network.h"
diff -up php-5.3.7RC1/main/streams/memory.c.gnusrc php-5.3.7RC1/main/streams/memory.c
--- php-5.3.7RC1/main/streams/memory.c.gnusrc	2011-01-01 03:19:59.000000000 +0100
+++ php-5.3.7RC1/main/streams/memory.c	2011-06-25 16:27:25.389248257 +0200
@@ -18,7 +18,6 @@
 
 /* $Id: memory.c 306939 2011-01-01 02:19:59Z felipe $ */
 
-#define _GNU_SOURCE
 #include "php.h"
 
 PHPAPI int php_url_decode(char *str, int len);
diff -up php-5.3.7RC1/main/streams/streams.c.gnusrc php-5.3.7RC1/main/streams/streams.c
--- php-5.3.7RC1/main/streams/streams.c.gnusrc	2011-06-05 23:57:01.000000000 +0200
+++ php-5.3.7RC1/main/streams/streams.c	2011-06-25 16:28:00.615353649 +0200
@@ -21,7 +21,6 @@
 
 /* $Id: streams.c 311849 2011-06-05 21:57:01Z iliaa $ */
 
-#define _GNU_SOURCE
 #include "php.h"
 #include "php_globals.h"
 #include "php_network.h"
diff -up php-5.3.7RC1/Zend/zend_language_parser.c.gnusrc php-5.3.7RC1/Zend/zend_language_parser.c
--- php-5.3.7RC1/Zend/zend_language_parser.c.gnusrc	2011-06-16 14:46:49.000000000 +0200
+++ php-5.3.7RC1/Zend/zend_language_parser.c	2011-06-25 16:27:25.392248266 +0200
@@ -366,6 +366,8 @@
 #include "zend_API.h"
 #include "zend_constants.h"
 
+#include <string.h>
+
 
 #define YYERROR_VERBOSE
 #define YYSTYPE znode
