From 1c991263923d22dba109af22ba3b51dd46d90ccb Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 7 Oct 2013 09:21:43 +0200
Subject: [PATCH] report LIBEVENT_VERSION in phpinfo()

---
 php_event.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/php_event.c b/php_event.c
index f440734..8885781 100644
--- a/php_event.c
+++ b/php_event.c
@@ -1352,7 +1352,8 @@ PHP_MINFO_FUNCTION(event)
 	php_info_print_table_row(2, "Thread safety support", "disabled");
 #endif
 
-	php_info_print_table_row(2, "Version", PHP_EVENT_VERSION);
+	php_info_print_table_row(2, "Extension version", PHP_EVENT_VERSION);
+	php_info_print_table_row(2, "libevent2 headers version", LIBEVENT_VERSION);
 	php_info_print_table_end();
 }
 /* }}} */
-- 
1.8.2.3

