diff --git a/src/util.h b/src/util.h
index 57ef991..8c3b652 100644
--- a/src/util.h
+++ b/src/util.h
@@ -96,7 +96,7 @@ int _php_event_getsockname(evutil_socket_t fd, zval **ppzaddress, zval **ppzport
 
 #define PHP_EVENT_REQUIRE_BASE_BY_REF(zbase)                  \
 	do {                                                      \
-		if (!Z_ISREF_P((zbase))) {                            \
+		if (!Z_ISREF_P((zbase)) || Z_REFCOUNT_P(zbase) < 2) { \
 			php_error_docref(NULL TSRMLS_CC, E_ERROR,         \
 					"EventBase must be passed by reference"); \
 		}                                                     \
