diff -up tests/bootstrap.php.rpm tests/bootstrap.php
--- tests/bootstrap.php.rpm	2015-01-05 13:27:31.069437796 +0100
+++ tests/bootstrap.php	2015-01-05 13:27:40.291470405 +0100
@@ -1,6 +1,7 @@
 <?php
 error_reporting(E_ALL | E_STRICT);
-require __DIR__ . '/../vendor/autoload.php';
+require __DIR__ . '/../hamcrest/Hamcrest/autoload.php';
+require __DIR__ . '/autoload.php';
 
 if (defined('E_DEPRECATED')) {
     error_reporting(error_reporting() | E_DEPRECATED);
@@ -14,5 +15,3 @@ set_include_path(implode(PATH_SEPARATOR,
     HAMCREST_BASE . '/hamcrest',
     get_include_path()
 )));
-
-require_once 'Hamcrest.php';
