diff -up jsonc-1.3.5/json-c/linkhash.c.el5 jsonc-1.3.5/json-c/linkhash.c
--- jsonc-1.3.5/json-c/linkhash.c.el5	2014-04-10 18:36:50.000000000 +0200
+++ jsonc-1.3.5/json-c/linkhash.c	2014-04-10 18:37:07.000000000 +0200
@@ -405,7 +405,7 @@ unsigned long lh_char_hash(const void *k
 		int seed;
 		/* we can't use -1 as it is the unitialized sentinel */
 		while ((seed = json_c_get_random_seed()) == -1);
-#if defined __GNUC__
+#if 0
 		__sync_val_compare_and_swap(&random_seed, -1, seed);
 #elif defined _MSC_VER
 		InterlockedCompareExchange(&random_seed, seed, -1);
