diff -up redis-2.6.0/src/config.h.el5 redis-2.6.0/src/config.h
--- redis-2.6.0/src/config.h.el5	2012-10-24 17:51:51.000000000 +0200
+++ redis-2.6.0/src/config.h	2012-10-24 17:52:35.000000000 +0200
@@ -52,13 +52,7 @@
 #define aof_fsync fsync
 #endif
 
-/* Define rdb_fsync_range to sync_file_range() on Linux, otherwise we use
- * the plain fsync() call. */
-#ifdef __linux__
-#define rdb_fsync_range(fd,off,size) sync_file_range(fd,off,size,SYNC_FILE_RANGE_WAIT_BEFORE|SYNC_FILE_RANGE_WRITE)
-#else
 #define rdb_fsync_range(fd,off,size) fsync(fd)
-#endif
 
 /* Byte ordering detection */
 #include <sys/types.h> /* This will likely define BYTE_ORDER */
