--- mysql-5.5.8/cmake/readline.cmake.readline	2010-12-03 18:58:26.000000000 +0100
+++ mysql-5.5.8/cmake/readline.cmake	2010-12-22 23:11:19.641789020 +0100
@@ -139,7 +139,7 @@
 
   IF(${name}_LIBRARY AND ${name}_INCLUDE_DIR)
     SET(SYSTEM_READLINE_FOUND 1)
-    SET(CMAKE_REQUIRED_LIBRARIES ${${name}_LIBRARY})
+    SET(CMAKE_REQUIRED_LIBRARIES ${${name}_LIBRARY} ncurses)
     CHECK_CXX_SOURCE_COMPILES("
     #include <stdio.h>
     #include <readline/readline.h>
@@ -172,7 +172,7 @@
     ${name}_USE_NEW_READLINE_INTERFACE)
     
     IF(${name}_USE_LIBEDIT_INTERFACE  OR ${name}_USE_NEW_READLINE_INTERFACE)
-      SET(READLINE_LIBRARY ${${name}_LIBRARY})
+      SET(READLINE_LIBRARY ${${name}_LIBRARY} ncurses)
       SET(READLINE_INCLUDE_DIR ${${name}_INCLUDE_DIR})
       SET(HAVE_HIST_ENTRY ${${name}_HAVE_HIST_ENTRY})
       SET(USE_LIBEDIT_INTERFACE ${${name}_USE_LIBEDIT_INTERFACE})
