From cddddd0fa49f1b0eeb19be18389711222b1a0aee Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Mon, 9 Nov 2015 16:52:13 +0100
Subject: [PATCH] allow to skip online tests

---
 tests/Utils/FileSystem.phpt | 24 ++++++++++++++++++------
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/tests/Utils/FileSystem.phpt b/tests/Utils/FileSystem.phpt
index 68871f2..2a0c96c 100644
--- a/tests/Utils/FileSystem.phpt
+++ b/tests/Utils/FileSystem.phpt
@@ -41,8 +41,13 @@ test(function () { // copy
 	FileSystem::copy(TEMP_DIR . '/3/file', TEMP_DIR . '/3/x/file');
 	Assert::same('Hello', file_get_contents(TEMP_DIR . '/3/x/file'));
 
-	FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/y/file');
-	Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
+	if (getenv('SKIP_ONLINE_TESTS') ?: false) {
+		FileSystem::copy(__FILE__, TEMP_DIR . '/3/x/y/file');
+		Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
+	} else {
+		FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/y/file');
+		Assert::true(is_file(TEMP_DIR . '/3/x/y/file'));
+	}
 
 	FileSystem::write(TEMP_DIR . '/5/newfile', 'World');
 
@@ -51,10 +56,17 @@ test(function () { // copy
 	}, 'Nette\InvalidStateException', "File or directory '%a%' already exists.");
 	Assert::same('Hello', file_get_contents(TEMP_DIR . '/3/x/file'));
 
-	Assert::exception(function () {
-		FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/file', FALSE);
-	}, 'Nette\InvalidStateException', "File or directory '%a%' already exists.");
-	Assert::same('Hello', file_get_contents(TEMP_DIR . '/3/x/file'));
+	if (getenv('SKIP_ONLINE_TESTS') ?: false) {
+		Assert::exception(function () {
+			FileSystem::copy(__FILE__, TEMP_DIR . '/3/x/file', FALSE);
+		}, 'Nette\InvalidStateException', "File or directory '%a%' already exists.");
+		Assert::same('Hello', file_get_contents(TEMP_DIR . '/3/x/file'));
+	} else {
+		Assert::exception(function () {
+			FileSystem::copy('http://example.com', TEMP_DIR . '/3/x/file', FALSE);
+		}, 'Nette\InvalidStateException', "File or directory '%a%' already exists.");
+		Assert::same('Hello', file_get_contents(TEMP_DIR . '/3/x/file'));
+	}
 
 	FileSystem::copy(TEMP_DIR . '/5/newfile', TEMP_DIR . '/3/x/file');
 	Assert::same('World', file_get_contents(TEMP_DIR . '/3/x/file'));
