--- bluegriffon/app/profile/bluegriffon-prefs.js.rpmprefs	2011-06-01 17:10:02.000000000 +0200
+++ bluegriffon/app/profile/bluegriffon-prefs.js	2011-06-10 22:11:11.582111789 +0200
@@ -108,8 +108,8 @@
 pref("bluegriffon.structurebar.role.show", true);
 
 // updates
-pref("bluegriffon.updates.check.enabled", true);
-pref("bluegriffon.updates.frequency", "launch");
+pref("bluegriffon.updates.check.enabled", false);
+pref("bluegriffon.updates.frequency", "manually");
 
 pref("html5.enable", true);
 pref("bluegriffon.defaults.doctype", "kXHTML5");
@@ -121,3 +121,7 @@
 
 pref("bluegriffon.osx.dock-integration", true);
 pref("extensions.getAddons.cache.enabled", false);
+
+// Locales
+pref("general.useragent.locale", "");
+pref("intl.locale.matchOS", true);
