From de4135de41a32e593119e1feed8f0416470f7721 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Sun, 28 Feb 2016 17:53:09 +0100
Subject: [PATCH] remove uneeded AutoloaderFactory::unregisterAutoloaders()

---
 test/ResetAutoloadFunctionsTrait.php | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/test/ResetAutoloadFunctionsTrait.php b/test/ResetAutoloadFunctionsTrait.php
index 24e5e37..25b4a4c 100644
--- a/test/ResetAutoloadFunctionsTrait.php
+++ b/test/ResetAutoloadFunctionsTrait.php
@@ -9,8 +9,6 @@
 
 namespace ZendTest\ModuleManager;
 
-use Zend\Loader\AutoloaderFactory;
-
 /**
  * Offer common setUp/tearDown methods for preserve current autoload functions and include paths.
  */
@@ -52,7 +50,6 @@ protected function preserveIncludePath()
      */
     protected function restoreAutoloadFunctions()
     {
-        AutoloaderFactory::unregisterAutoloaders();
         $loaders = spl_autoload_functions();
         if (is_array($loaders)) {
             foreach ($loaders as $loader) {
