From 0205ec625f6d585848933d710a2ea267c8c72071 Mon Sep 17 00:00:00 2001
From: Remi Collet <fedora@famillecollet.com>
Date: Fri, 18 Sep 2015 11:41:58 +0200
Subject: [PATCH] honour --with-libdir option

---
 config.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.m4 b/config.m4
index d5b5999..c219769 100644
--- a/config.m4
+++ b/config.m4
@@ -80,7 +80,7 @@ if test "$PHP_AMQP" != "no"; then
 		fi
 	fi
 
-	PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $AMQP_DIR/lib, AMQP_SHARED_LIBADD)
+	PHP_ADD_LIBRARY_WITH_PATH($LIBNAME, $AMQP_DIR/$PHP_LIBDIR, AMQP_SHARED_LIBADD)
 	PHP_SUBST(AMQP_SHARED_LIBADD)
 
 	AMQP_SOURCES="amqp.c amqp_exchange.c amqp_queue.c amqp_connection.c amqp_connection_resource.c amqp_channel.c amqp_envelope.c amqp_object_store.c"
