diff -up php5.3-201104170830/ext/ldap/php_ldap.h.remi-oci8 php5.3-201104170830/ext/ldap/php_ldap.h
--- php5.3-201104170830/ext/ldap/php_ldap.h.remi-oci8	2011-01-01 04:37:23.000000000 +0100
+++ php5.3-201104170830/ext/ldap/php_ldap.h	2011-04-17 11:55:25.630871609 +0200
@@ -27,7 +27,7 @@
 #include <lber.h>
 #endif
 
-#include <ldap.h>
+#include "/usr/include/ldap.h"
 
 extern zend_module_entry ldap_module_entry;
 #define ldap_module_ptr &ldap_module_entry
diff -up php5.3-201104170830/ext/oci8/config.m4.remi-oci8 php5.3-201104170830/ext/oci8/config.m4
--- php5.3-201104170830/ext/oci8/config.m4.remi-oci8	2011-03-30 00:35:22.000000000 +0200
+++ php5.3-201104170830/ext/oci8/config.m4	2011-04-17 11:55:25.628871315 +0200
@@ -291,6 +291,7 @@ if test "$PHP_OCI8" != "no"; then 
 
     dnl Header directory for Instant Client SDK RPM install
     OCISDKRPMINC=`echo "$PHP_OCI8_INSTANT_CLIENT" | $PHP_OCI8_SED -e 's!^/usr/lib/oracle/\(.*\)/client\('${PHP_OCI8_IC_LIBDIR_SUFFIX}'\)*/lib[/]*$!/usr/include/oracle/\1/client\2!'`
+    OCISDKRPMINC=`echo "$PHP_OCI8_INSTANT_CLIENT" | $PHP_OCI8_SED -e 's!^/usr/\(lib64\|lib\)/oracle/\(.*\)/\(client64\|client\)/lib[/]*$!/usr/include/oracle/\2/\3!'`
 
     dnl Header directory for Instant Client SDK zip file install
     OCISDKZIPINC=$PHP_OCI8_INSTANT_CLIENT/sdk/include
diff -up php5.3-201104170830/ext/pdo_oci/config.m4.remi-oci8 php5.3-201104170830/ext/pdo_oci/config.m4
--- php5.3-201104170830/ext/pdo_oci/config.m4.remi-oci8	2011-04-02 04:35:24.000000000 +0200
+++ php5.3-201104170830/ext/pdo_oci/config.m4	2011-04-17 12:02:42.837194120 +0200
@@ -104,8 +104,10 @@ You need to tell me where to find your O
     else
       AC_MSG_ERROR([I'm too dumb to figure out where the include dir is in your Instant Client install])
     fi
-    if test -f "$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
-    PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib"
+    if test -f "$PDO_OCI_IC_PREFIX/lib64/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
+      PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/lib64/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib"
+    elif test -f "$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
+      PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/lib/oracle/$PDO_OCI_IC_VERS/$PDO_OCI_CLIENT_DIR/lib"
     elif test -f "$PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/lib/libclntsh.$SHLIB_SUFFIX_NAME" ; then
       PDO_OCI_LIB_DIR="$PDO_OCI_IC_PREFIX/$PDO_OCI_CLIENT_DIR/lib"
     elif test -f "$PDO_OCI_IC_PREFIX/libclntsh.$SHLIB_SUFFIX_NAME" ; then
