From 6d1885342c1504a8697b41e075844b082f7b335f Mon Sep 17 00:00:00 2001
From: Nikita Popov <nikic@php.net>
Date: Fri, 6 Jun 2014 17:09:49 +0200
Subject: [PATCH] Fix bug #67392: dtrace breaks argument unpack

---
 NEWS                | 3 +++
 Zend/zend_execute.c | 3 ++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/Zend/zend_execute.c b/Zend/zend_execute.c
index bdf8b6e..ad92c5f 100644
--- a/Zend/zend_execute.c
+++ b/Zend/zend_execute.c
@@ -1490,7 +1490,8 @@ ZEND_API void execute_internal(zend_execute_data *execute_data_ptr, zend_fcall_i
 	} else {
 		zval **return_value_ptr = &EX_TMP_VAR(execute_data_ptr, execute_data_ptr->opline->result.var)->var.ptr;
 		execute_data_ptr->function_state.function->internal_function.handler(
-			execute_data_ptr->opline->extended_value, *return_value_ptr, return_value_ptr,
+			execute_data_ptr->opline->extended_value + execute_data_ptr->call->num_additional_args,
+			*return_value_ptr, return_value_ptr,
 			execute_data_ptr->object, return_value_used TSRMLS_CC
 		);
 	}
-- 
1.9.2

