From 55237fe153694f5501636a49e387f36297037e4a Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Thu, 15 Sep 2016 13:27:20 +0200
Subject: [PATCH] fix ZTS build

---
 ext/mysqlnd/mysqlnd_wireprotocol.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/mysqlnd/mysqlnd_wireprotocol.c b/ext/mysqlnd/mysqlnd_wireprotocol.c
index 855a25c..844330f 100644
--- a/ext/mysqlnd/mysqlnd_wireprotocol.c
+++ b/ext/mysqlnd/mysqlnd_wireprotocol.c
@@ -1610,7 +1610,7 @@ php_mysqlnd_rowp_read_text_protocol_aux(MYSQLND_MEMORY_POOL_CHUNK * row_buffer,
 		const unsigned long len = php_mysqlnd_net_field_length(&p);
 
 		if (len != MYSQLND_NULL_LENGTH && ((p + len) > packet_end)) {
-			php_error_docref(NULL, E_WARNING, "Malformed server packet. Field length pointing "MYSQLND_SZ_T_SPEC
+			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Malformed server packet. Field length pointing "MYSQLND_SZ_T_SPEC
 											  " bytes after end of packet", (p + len) - packet_end - 1);
 			DBG_RETURN(FAIL);
 		}
-- 
2.1.4

From 3c117d4136d26c8d6e4127c89b2160c477f45e11 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@php.net>
Date: Thu, 15 Sep 2016 15:32:39 +0200
Subject: [PATCH] fix test (32bits)

---
 ext/standard/tests/strings/str_pad_variation5.phpt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/standard/tests/strings/str_pad_variation5.phpt b/ext/standard/tests/strings/str_pad_variation5.phpt
index cd979a1..4e1b1a9 100644
--- a/ext/standard/tests/strings/str_pad_variation5.phpt
+++ b/ext/standard/tests/strings/str_pad_variation5.phpt
@@ -31,4 +31,4 @@ var_dump( str_pad($input, $pad_length) );
 --EXPECTF--
 *** Testing str_pad() function: with large value for for 'pad_length' argument ***
 
-Fatal error: Allowed memory size of 134217728 bytes exhausted%s(tried to allocate 2147483648 bytes) in %s on line %d
+Fatal error: String allocation overflow, max size is 2147483647 in %s on line %d
-- 
2.1.4

